diff --git a/gni/v8.gni b/gni/v8.gni
index 3dd9fde0fa..f071edaa5d 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -243,6 +243,15 @@ template("v8_header_set") {
   }
 }
 
+template("v8_static_lib") {
+  static_library(target_name) {
+    forward_variables_from(invoker, "*", [ "configs" ])
+    configs += invoker.configs
+    configs -= v8_remove_configs
+    configs += v8_add_configs
+  }
+}
+
 template("v8_executable") {
   executable(target_name) {
     forward_variables_from(invoker,
diff --git a/gypfiles/detect_host_arch.py b/gypfiles/detect_host_arch.py
new file mode 100755
index 0000000000..8d6a1ba917
