diff --git a/include/v8-isolate.h b/include/v8-isolate.h
index 90a184d2b5..e4492b0bd3 100644
--- a/include/v8-isolate.h
+++ b/include/v8-isolate.h
@@ -204,6 +204,7 @@ enum class MemoryPressureLevel { kNone, kModerate, kCritical };
  */
 class V8_EXPORT Isolate {
  public:
+  ArrayBuffer::Allocator* array_buffer_allocator();
   /**
    * Initial configuration parameters for a new Isolate.
    */
@@ -297,6 +298,7 @@ class V8_EXPORT Isolate {
     Isolate* experimental_attach_to_shared_isolate = nullptr;
   };
 
+  void SetArrayBufferAllocatorShared(std::shared_ptr<ArrayBuffer::Allocator> allocator);
   /**
    * Stack-allocated class which sets the isolate for all operations
    * executed within a local scope.
