diff --git a/include/v8-primitive-object.h b/include/v8-primitive-object.h
index 573932d078..84c20af271 100644
--- a/include/v8-primitive-object.h
+++ b/include/v8-primitive-object.h
@@ -78,6 +78,8 @@ class V8_EXPORT BooleanObject : public Object {
  */
 class V8_EXPORT StringObject : public Object {
  public:
+  V8_DEPRECATED("Use Isolate* version") static
+                       Local<Value> New(Local<String> value);
   static Local<Value> New(Isolate* isolate, Local<String> value);
 
   Local<String> ValueOf() const;
