diff --git a/src/debug/debug-scopes.cc b/src/debug/debug-scopes.cc
index f92e0ac9c8..1715175a41 100644
--- a/src/debug/debug-scopes.cc
+++ b/src/debug/debug-scopes.cc
@@ -255,7 +255,10 @@ void ScopeIterator::TryParseAndRetrieveScopes(ReparseStrategy strategy) {
   info_ = std::make_unique<ParseInfo>(isolate_, flags, &compile_state,
                                       reusable_compile_state_.get());
 
-  const bool parse_result =
+  bool has_source_code = shared_info->HasSourceCode();
+  bool parse_result = false;
+  if (has_source_code)
+    parse_result =
       flags.is_toplevel()
           ? parsing::ParseProgram(info_.get(), script, maybe_outer_scope,
                                   isolate_, parsing::ReportStatisticsMode::kNo)
