diff --git a/src/objects/js-function.cc b/src/objects/js-function.cc
index dcba799475..ccbb1f6b53 100644
--- a/src/objects/js-function.cc
+++ b/src/objects/js-function.cc
@@ -1229,6 +1229,13 @@ Handle<String> JSFunction::ToString(Handle<JSFunction> function) {
     return NativeCodeFunctionSourceString(shared_info);
   }
 
+  //NWJS#6061: moved here or it will crash when trying to print
+  //function as a class
+  // Check if we have source code for the {function}.
+  if (!shared_info->HasSourceCode()) {
+    return NativeCodeFunctionSourceString(shared_info);
+  }
+
   // Check if we should print {function} as a class.
   Handle<Object> maybe_class_positions = JSReceiver::GetDataProperty(
       isolate, function, isolate->factory()->class_positions_symbol());
@@ -1243,11 +1250,6 @@ Handle<String> JSFunction::ToString(Handle<JSFunction> function) {
                                             end_position);
   }
 
-  // Check if we have source code for the {function}.
-  if (!shared_info->HasSourceCode()) {
-    return NativeCodeFunctionSourceString(shared_info);
-  }
-
   // If this function was compiled from asm.js, use the recorded offset
   // information.
 #if V8_ENABLE_WEBASSEMBLY
