diff --git a/src/objects/shared-function-info-inl.h b/src/objects/shared-function-info-inl.h
index 3cfea73f04..b5ac1702df 100644
--- a/src/objects/shared-function-info-inl.h
+++ b/src/objects/shared-function-info-inl.h
@@ -637,6 +637,13 @@ bool SharedFunctionInfo::ShouldFlushCode(
   }
   if (!data.IsBytecodeArray()) return false;
 
+  Object script_obj = script();
+  if (!script_obj.IsUndefined()) {
+    Script script = Script::cast(script_obj);
+    if (script.source().IsUndefined())
+      return false;
+  }
+
   if (IsStressFlushingEnabled(code_flush_mode)) return true;
 
   BytecodeArray bytecode = BytecodeArray::cast(data);
