diff -r -u --color up/chromium/base/message_loop/message_pump_mac.h nw/chromium/base/message_loop/message_pump_mac.h
--- up/chromium/base/message_loop/message_pump_mac.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/base/message_loop/message_pump_mac.h	2023-02-24 20:12:01.316791401 +0000
@@ -156,7 +156,7 @@
   // Get the current mode mask from |enabled_modes_|.
   int GetModeMask() const;
 
- private:
+ protected:
   class ScopedModeEnabler;
 
   // The maximum number of run loop modes that can be monitored.
@@ -172,14 +172,16 @@
   // the instance method; the instance method returns true if it resignalled
   // |work_source_| to be called again from the loop.
   static void RunWorkSource(void* info);
-  bool RunWork();
+ protected:
+  virtual bool RunWork();
 
   // Perform idle-priority work.  This is normally called by PreWaitObserver,
   // but is also associated with |idle_work_source_|.  When this function
   // actually does perform idle work, it will resignal that source.  The
   // static method calls the instance method.
   static void RunIdleWorkSource(void* info);
-  void RunIdleWork();
+  virtual void RunIdleWork();
+  virtual void PreWaitObserverHook();
 
   // Perform work that may have been deferred because it was not runnable
   // within a nested run loop.  This is associated with
