diff -r -u --color up/chromium/base/BUILD.gn nw/chromium/base/BUILD.gn
--- up/chromium/base/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/base/BUILD.gn	2023-02-24 06:16:18.475918139 +0000
@@ -182,6 +182,7 @@
 # This does not include test code (test support and anything in the test
 # directory) which should use source_set as is recommended for GN targets).
 component("base") {
+  include_dirs = [ "//v8/include" ]
   sources = [
     "allocator/allocator_check.cc",
     "allocator/allocator_check.h",
@@ -473,6 +474,8 @@
     "message_loop/message_pump_glib.cc",
     "message_loop/message_pump_glib.h",
     "message_loop/message_pump_type.h",
+    "message_loop/message_pump_uv.cc",
+    "message_loop/message_pump_uv.h",
     "message_loop/timer_slack.h",
     "message_loop/work_id_provider.cc",
     "message_loop/work_id_provider.h",
@@ -1489,7 +1492,10 @@
 
   configs += [
     "//build/config:precompiled_headers",
-    "//build/config/compiler:prevent_unsafe_narrowing",
+
+    # TODO(crbug.com/1292951): Enable.
+    # "//build/config/compiler:prevent_unsafe_narrowing",
+
     "//build/config/compiler:wexit_time_destructors",
     "//build/config/compiler:wglobal_constructors",
   ]
