diff -r -u --color up/chromium/base/process/launch_posix.cc nw/chromium/base/process/launch_posix.cc
--- up/chromium/base/process/launch_posix.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/base/process/launch_posix.cc	2023-02-24 20:54:05.212528585 +0000
@@ -459,6 +459,7 @@
 #ifndef PR_SET_NO_NEW_PRIVS
 #define PR_SET_NO_NEW_PRIVS 38
 #endif
+#if 0
     if (!options.allow_new_privs) {
       if (prctl(PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0) && errno != EINVAL) {
         // Only log if the error is not EINVAL (i.e. not supported).
@@ -473,6 +474,7 @@
       }
     }
 #endif
+#endif
 
     if (current_directory != nullptr) {
       RAW_CHECK(chdir(current_directory) == 0);
