diff -r -u --color up/chromium/build/toolchain/apple/toolchain.gni nw/chromium/build/toolchain/apple/toolchain.gni
--- up/chromium/build/toolchain/apple/toolchain.gni	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/build/toolchain/apple/toolchain.gni	2023-02-24 06:16:18.787936616 +0000
@@ -558,7 +558,9 @@
       # hardlink but ensure the file have distinct metadata (thus avoid the
       # error with ditto, see https://crbug.com/1042182).
       if (host_os == "mac") {
-        command = "rm -rf {{output}} && /bin/cp -Rc {{source}} {{output}}"
+        command =
+          "rm -rf {{output}} && cp -fRc {{source}} {{output}} 2>/dev/null " +
+          "|| (rm -rf {{output}} && cp -fR {{source}} {{output}})"
       } else {
         command = "rm -rf {{output}} && /bin/cp -Rl {{source}} {{output}}"
       }
