diff -r -u --color up/chromium/build_overrides/partition_alloc.gni nw/chromium/build_overrides/partition_alloc.gni
--- up/chromium/build_overrides/partition_alloc.gni	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/build_overrides/partition_alloc.gni	2023-02-24 20:54:05.376527856 +0000
@@ -42,7 +42,7 @@
 
 # - NaCl: No plans to support it.
 # - iOS: not done yet.
-_is_partition_alloc_everywhere_platform = !is_nacl && !is_ios
+_is_partition_alloc_everywhere_platform = is_win
 
 # Under Windows debug build, the allocator shim is not compatible with CRT.
 #     NaCl in particular does seem to link some binaries statically
