diff -r -u --color up/chromium/build/config/nacl/rules.gni nw/chromium/build/config/nacl/rules.gni
--- up/chromium/build/config/nacl/rules.gni	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/build/config/nacl/rules.gni	2023-02-24 06:16:18.763935195 +0000
@@ -69,8 +69,13 @@
       # Starts empty so the code below can use += everywhere.
       data = []
 
-      nmfflags +=
+      if (target_os == "linux") {
+        nmfflags +=
+          [ "--library-path=" + rebase_path("${root_out_dir}/lib", root_build_dir) ]
+      } else {
+        nmfflags +=
           [ "--library-path=" + rebase_path(root_out_dir, root_build_dir) ]
+      }
 
       # NOTE: There is no explicit dependency for the lib directory
       # (lib32 and lib64 for x86/x64) created in the product directory.
