diff -r -u --color up/chromium/chrome/browser/ui/views/chrome_views_delegate_linux.cc nw/chromium/chrome/browser/ui/views/chrome_views_delegate_linux.cc
--- up/chromium/chrome/browser/ui/views/chrome_views_delegate_linux.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/chrome_views_delegate_linux.cc	2023-02-24 20:54:08.388514164 +0000
@@ -7,6 +7,8 @@
 #include "base/environment.h"
 #include "base/nix/xdg_util.h"
 #include "build/branding_buildflags.h"
+#include "content/nw/src/nw_content.h"
+
 #include "chrome/browser/ui/views/native_widget_factory.h"
 #include "chrome/common/channel_info.h"
 #include "chrome/grit/chrome_unscaled_resources.h"
@@ -69,6 +71,9 @@
 
 #if BUILDFLAG(IS_LINUX)
 gfx::ImageSkia* ChromeViewsDelegate::GetDefaultWindowIcon() const {
+  gfx::ImageSkia* ret = nw::GetAppIcon();
+  if (ret)
+    return ret;
   ui::ResourceBundle& rb = ui::ResourceBundle::GetSharedInstance();
   return rb.GetImageSkiaNamed(GetWindowIconResourceId());
 }
