diff -r -u --color up/chromium/chrome/browser/page_load_metrics/page_load_metrics_initialize.cc nw/chromium/chrome/browser/page_load_metrics/page_load_metrics_initialize.cc
--- up/chromium/chrome/browser/page_load_metrics/page_load_metrics_initialize.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/page_load_metrics/page_load_metrics_initialize.cc	2023-02-24 06:16:21.308085855 +0000
@@ -192,11 +192,14 @@
           web_contents()->GetBrowserContext()));
   tracker->AddObserver(
       std::make_unique<PageAnchorsMetricsObserver>(tracker->GetWebContents()));
+#if 0
   std::unique_ptr<TranslatePageLoadMetricsObserver> translate_observer =
       TranslatePageLoadMetricsObserver::CreateIfNeeded(
           tracker->GetWebContents());
   if (translate_observer)
     tracker->AddObserver(std::move(translate_observer));
+#endif
+
 }
 
 bool PageLoadMetricsEmbedder::IsNewTabPageUrl(const GURL& url) {
