diff -r -u --color up/chromium/chrome/browser/extensions/extension_webkit_preferences.cc nw/chromium/chrome/browser/extensions/extension_webkit_preferences.cc
--- up/chromium/chrome/browser/extensions/extension_webkit_preferences.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/extension_webkit_preferences.cc	2023-02-24 20:12:03.312811725 +0000
@@ -33,12 +33,21 @@
   }
 
   if (extension->is_platform_app()) {
-    webkit_prefs->databases_enabled = false;
-    webkit_prefs->local_storage_enabled = false;
-    webkit_prefs->sync_xhr_in_documents_enabled = false;
-    webkit_prefs->cookie_enabled = false;
-    webkit_prefs->target_blank_implies_no_opener_enabled_will_be_removed =
+    if (extension->is_nwjs_app()) {
+      webkit_prefs->databases_enabled = true;
+      webkit_prefs->local_storage_enabled = true;
+      webkit_prefs->sync_xhr_in_documents_enabled = true;
+      webkit_prefs->cookie_enabled = true;
+      webkit_prefs->target_blank_implies_no_opener_enabled_will_be_removed =
+        true;
+    }else{
+      webkit_prefs->databases_enabled = false;
+      webkit_prefs->local_storage_enabled = false;
+      webkit_prefs->sync_xhr_in_documents_enabled = false;
+      webkit_prefs->cookie_enabled = false;
+      webkit_prefs->target_blank_implies_no_opener_enabled_will_be_removed =
         false;
+    }
   }
 
   // Prevent font size preferences from affecting the PDF Viewer extension.
