diff -r -u --color up/chromium/chrome/browser/ui/views/fullscreen_control/fullscreen_control_host.cc nw/chromium/chrome/browser/ui/views/fullscreen_control/fullscreen_control_host.cc
--- up/chromium/chrome/browser/ui/views/fullscreen_control/fullscreen_control_host.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/fullscreen_control/fullscreen_control_host.cc	2023-02-24 06:16:22.488155736 +0000
@@ -289,9 +289,13 @@
 }
 
 bool FullscreenControlHost::IsExitUiNeeded() {
+#if 1
+  return false;
+#else
   return browser_view_->IsFullscreen() &&
          browser_view_->CanUserExitFullscreen() &&
          browser_view_->ShouldHideUIForFullscreen();
+#endif
 }
 
 float FullscreenControlHost::CalculateCursorBufferHeight() const {
