diff -r -u --color up/chromium/chrome/browser/ui/views/overlay/video_overlay_window_views.h nw/chromium/chrome/browser/ui/views/overlay/video_overlay_window_views.h
--- up/chromium/chrome/browser/ui/views/overlay/video_overlay_window_views.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/overlay/video_overlay_window_views.h	2023-02-24 20:12:04.552824351 +0000
@@ -17,6 +17,9 @@
 #include "ui/gfx/geometry/size.h"
 #include "ui/views/widget/widget.h"
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Woverloaded-virtual"
+
 namespace viz {
 class FrameSinkId;
 }  // namespace viz
@@ -330,4 +333,6 @@
   bool has_registered_frame_sink_hierarchy_ = false;
 };
 
+#pragma clang diagnostic pop
+
 #endif  // CHROME_BROWSER_UI_VIEWS_OVERLAY_VIDEO_OVERLAY_WINDOW_VIEWS_H_
