diff -r -u --color up/chromium/chrome/browser/chrome_browser_main_mac.mm nw/chromium/chrome/browser/chrome_browser_main_mac.mm
--- up/chromium/chrome/browser/chrome_browser_main_mac.mm	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/chrome_browser_main_mac.mm	2023-02-24 06:16:20.768053875 +0000
@@ -4,6 +4,9 @@
 
 #include "chrome/browser/chrome_browser_main_mac.h"
 
+#include "content/nw/src/nw_base.h"
+#include "base/strings/utf_string_conversions.h"
+
 #import <Cocoa/Cocoa.h>
 
 #include "base/bind.h"
@@ -89,6 +92,7 @@
   }
   updater::SchedulePeriodicTasks();
 
+#if 0
   // Disk image installation is sort of a first-run task, so it shares the
   // no first run switches.
   //
@@ -109,6 +113,7 @@
       exit(0);
     }
   }
+#endif
 #endif  // BUILDFLAG(ENABLE_UPDATER)
 
   // Create the app delegate. This object is intentionally leaked as a global
@@ -117,7 +122,7 @@
   [NSApp setDelegate:app_controller];
 
   chrome::BuildMainMenu(NSApp, app_controller,
-                        l10n_util::GetStringUTF16(IDS_PRODUCT_NAME), false);
+                        base::UTF8ToUTF16(nw::package()->GetName()), true);
   [app_controller mainMenuCreated];
 
   ui::WarmScreenCapture();
