diff -r -u --color up/chromium/chrome/browser/extensions/browser_context_keyed_service_factories.cc nw/chromium/chrome/browser/extensions/browser_context_keyed_service_factories.cc
--- up/chromium/chrome/browser/extensions/browser_context_keyed_service_factories.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/browser_context_keyed_service_factories.cc	2023-02-24 20:54:07.208519605 +0000
@@ -7,6 +7,8 @@
 #include "chrome/browser/extensions/api/api_browser_context_keyed_service_factories.h"
 #include "chrome/browser/extensions/chrome_browser_context_keyed_service_factories.h"
 
+#include "content/nw/src/api/object_manager_factory.h"
+
 namespace chrome_extensions {
 
 void EnsureBrowserContextKeyedServiceFactoriesBuilt() {
