diff -r -u --color up/chromium/chrome/browser/ui/browser_list.cc nw/chromium/chrome/browser/ui/browser_list.cc
--- up/chromium/chrome/browser/ui/browser_list.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/browser_list.cc	2023-02-24 06:16:22.324146024 +0000
@@ -262,7 +262,8 @@
 // static
 void BrowserList::MoveBrowsersInWorkspaceToFront(
     const std::string& new_workspace) {
-  DCHECK(!new_workspace.empty());
+  if(new_workspace.empty())
+    return;
 
   BrowserList* instance = GetInstance();
 
