diff -r -u --color up/chromium/chrome/test/chromedriver/chrome/chrome_impl.cc nw/chromium/chrome/test/chromedriver/chrome/chrome_impl.cc
--- up/chromium/chrome/test/chromedriver/chrome/chrome_impl.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/test/chromedriver/chrome/chrome_impl.cc	2023-02-24 06:16:23.084191032 +0000
@@ -19,6 +19,7 @@
 #include "base/values.h"
 #include "chrome/test/chromedriver/chrome/chrome.h"
 #include "chrome/test/chromedriver/chrome/device_metrics.h"
+#include "base/strings/string_util.h"
 #include "chrome/test/chromedriver/chrome/devtools_client.h"
 #include "chrome/test/chromedriver/chrome/devtools_client_impl.h"
 #include "chrome/test/chromedriver/chrome/devtools_event_listener.h"
@@ -66,13 +67,25 @@
   status = UpdateWebViews(views_info, w3c_compliant);
   if (status.IsError())
     return status;
+  std::string ret;
   for (int i = views_info.GetSize() - 1; i >= 0; --i) {
     const WebViewInfo& view = views_info.Get(i);
-    if (view.type == WebViewInfo::kPage) {
-      *web_view_id = view.id;
-      return Status(kOk);
+    if (view.type == WebViewInfo::kPage ||
+        view.type == WebViewInfo::kApp ||
+        (view.type == WebViewInfo::kOther &&
+         !base::StartsWith(view.url, "chrome-extension://", base::CompareCase::SENSITIVE) &&
+         !base::StartsWith(view.url, "about:blank", base::CompareCase::SENSITIVE))) {
+      ret = view.id;
+      if (view.type != WebViewInfo::kOther) {
+        *web_view_id = view.id;
+        return Status(kOk);
+      }
     }
   }
+  if (!ret.empty()) {
+    *web_view_id = ret;
+    return Status(kOk);
+  }
   return Status(kUnknownError, "unable to discover open window in chrome");
 }
 
