diff -r -u --color up/chromium/chrome/browser/resources/print_preview/ui/destination_settings.ts nw/chromium/chrome/browser/resources/print_preview/ui/destination_settings.ts
--- up/chromium/chrome/browser/resources/print_preview/ui/destination_settings.ts	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/resources/print_preview/ui/destination_settings.ts	2023-02-24 20:12:03.932818038 +0000
@@ -255,7 +255,7 @@
         0, this.getRecentDestinationsDisplayCount_(recentDestinations));
     this.destinationStore_!.init(
         this.pdfPrinterDisabled_, isDriveMounted, defaultPrinter,
-        serializedDefaultDestinationRulesStr, recentDestinations);
+        serializedDefaultDestinationRulesStr, recentDestinations, false);
   }
 
   /**
