diff -r -u --color up/chromium/chrome/browser/ui/dialogs/outdated_upgrade_bubble.cc nw/chromium/chrome/browser/ui/dialogs/outdated_upgrade_bubble.cc
--- up/chromium/chrome/browser/ui/dialogs/outdated_upgrade_bubble.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/dialogs/outdated_upgrade_bubble.cc	2023-02-24 20:12:04.424823047 +0000
@@ -33,7 +33,9 @@
 
 // For ChromeOS Lacros, browser updates are done via system services, thus
 // we redirect to the safetyCheck page that interacts with these. On other
-// platforms it may be possible to download an updated browser via a site.
+// platforms it may be possible to download an updated browser via a
+// site.
+#if 0
 const char* kUpdateBrowserRedirectUrl =
 #if BUILDFLAG(IS_CHROMEOS_LACROS)
     // The URL to be used to update Lacros-Chrome when auto-update failed
@@ -44,7 +46,7 @@
     // too long.
     "https://www.google.com/chrome";
 #endif
-
+#endif
 bool g_upgrade_bubble_is_showing = false;
 
 void OnWindowClosing() {
@@ -87,6 +89,7 @@
 }  // namespace
 
 void ShowOutdatedUpgradeBubble(Browser* browser, bool auto_update_enabled) {
+#if 0
   if (g_upgrade_bubble_is_showing)
     return;
 
@@ -116,4 +119,5 @@
       auto_update_enabled
           ? base::UserMetricsAction("OutdatedUpgradeBubble.Show")
           : base::UserMetricsAction("OutdatedUpgradeBubble.ShowNoAU"));
+#endif
 }
