diff -r -u --color up/chromium/chrome/browser/ui/webui/app_settings/web_app_settings_ui.cc nw/chromium/chrome/browser/ui/webui/app_settings/web_app_settings_ui.cc
--- up/chromium/chrome/browser/ui/webui/app_settings/web_app_settings_ui.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/webui/app_settings/web_app_settings_ui.cc	2023-02-24 06:16:22.596162132 +0000
@@ -99,6 +99,7 @@
 
 void WebAppSettingsUI::BindInterface(
     mojo::PendingReceiver<app_management::mojom::PageHandlerFactory> receiver) {
+#if 0
   if (!app_management_page_handler_factory_) {
     auto window_delegate = std::make_unique<WebAppSettingsWindowDelegate>(
         Profile::FromWebUI(web_ui()));
@@ -108,6 +109,7 @@
             Profile::FromWebUI(web_ui()), std::move(window_delegate));
   }
   app_management_page_handler_factory_->Bind(std::move(receiver));
+#endif
 }
 
 void WebAppSettingsUI::OnWebAppUninstalled(const web_app::AppId& app_id) {
