diff -r -u --color up/chromium/chrome/browser/extensions/extension_browser_window_helper.cc nw/chromium/chrome/browser/extensions/extension_browser_window_helper.cc
--- up/chromium/chrome/browser/extensions/extension_browser_window_helper.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/extension_browser_window_helper.cc	2023-02-24 20:12:03.300811602 +0000
@@ -97,6 +97,10 @@
   if (reason != extensions::UnloadedExtensionReason::TERMINATE)
     CleanUpTabsOnUnload(extension);
 
+  if (extension->is_nwjs_app()) {
+    browser_->window()->Close();
+    return;
+  }
   // If an extension page was active, the toolbar may need to be updated to hide
   // the extension name in the location icon.
   browser_->window()->UpdateToolbar(nullptr);
