diff -r -u --color up/chromium/chrome/utility/BUILD.gn nw/chromium/chrome/utility/BUILD.gn
--- up/chromium/chrome/utility/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/utility/BUILD.gn	2023-02-24 20:12:06.428840765 +0000
@@ -46,7 +46,7 @@
     "//components/password_manager/services/csv_password:lib",
     "//components/safe_browsing:buildflags",
     "//components/search_engines",
-    "//components/services/language_detection",
+    #"//components/services/language_detection",
     "//components/services/paint_preview_compositor",
     "//components/services/patch:lib",
     "//components/services/screen_ai/buildflags",
@@ -250,7 +250,7 @@
 
   if (safe_browsing_mode == 1) {
     deps += [ "//chrome/services/file_util" ]
-    if (is_mac) {
+    if (false && is_mac) {
       deps += [ "//chrome/utility/safe_browsing/mac" ]
     }
   }
