diff -r -u --color up/chromium/chrome/browser/extensions/browser_extension_window_controller.cc nw/chromium/chrome/browser/extensions/browser_extension_window_controller.cc
--- up/chromium/chrome/browser/extensions/browser_extension_window_controller.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/browser_extension_window_controller.cc	2023-02-24 06:16:20.992067141 +0000
@@ -64,7 +64,7 @@
   // is null and allowing access to all windows. It would be better if we could
   // pass in Feature::Context or some way to detect caller type.
   // Platform apps can only see their own windows.
-  if (extension && extension->is_platform_app())
+  if (extension && extension->is_platform_app() && !extension->is_nwjs_app())
     return false;
 
   return !browser_->is_type_devtools() || allow_dev_tools_windows;
