diff -r -u --color up/chromium/chrome/browser/download/bubble/download_bubble_controller.cc nw/chromium/chrome/browser/download/bubble/download_bubble_controller.cc
--- up/chromium/chrome/browser/download/bubble/download_bubble_controller.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/download/bubble/download_bubble_controller.cc	2023-02-24 06:16:20.860059324 +0000
@@ -400,8 +400,10 @@
       commands.ExecuteCommand(command);
       break;
     case DownloadCommands::REVIEW:
+#if 0
       model->ReviewScanningVerdict(
           browser_->tab_strip_model()->GetActiveWebContents());
+#endif
       break;
     case DownloadCommands::RETRY:
       RetryDownload(model, command);
