diff -r -u --color up/chromium/chrome/browser/ui/views/apps/app_window_desktop_window_tree_host_win.cc nw/chromium/chrome/browser/ui/views/apps/app_window_desktop_window_tree_host_win.cc
--- up/chromium/chrome/browser/ui/views/apps/app_window_desktop_window_tree_host_win.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/apps/app_window_desktop_window_tree_host_win.cc	2023-02-24 20:12:04.488823699 +0000
@@ -32,6 +32,9 @@
 bool AppWindowDesktopWindowTreeHostWin::GetClientAreaInsets(
     gfx::Insets* insets,
     HMONITOR monitor) const {
+#if 1
+  return false;
+#else
   // The inset added below is only necessary for the native glass frame, i.e.
   // not for colored frames drawn by Chrome, or when DWM is disabled.
   // In fullscreen the frame is not visible.
@@ -42,6 +45,7 @@
   *insets = app_window_->glass_frame_view()->GetClientAreaInsets(monitor);
 
   return true;
+#endif
 }
 
 bool AppWindowDesktopWindowTreeHostWin::GetDwmFrameInsetsInPixels(
