diff -r -u --color up/chromium/chrome/browser/web_applications/extensions/web_app_extension_shortcut.cc nw/chromium/chrome/browser/web_applications/extensions/web_app_extension_shortcut.cc
--- up/chromium/chrome/browser/web_applications/extensions/web_app_extension_shortcut.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/web_applications/extensions/web_app_extension_shortcut.cc	2023-02-24 20:54:08.688512780 +0000
@@ -224,6 +224,7 @@
   // cannot be shown in the launcher.
   if (extension->location() ==
           extensions::mojom::ManifestLocation::kComponent ||
+      extension->is_nwjs_app() ||
       !extensions::ui_util::CanDisplayInAppLauncher(extension, profile)) {
     return false;
   }
