diff -r -u --color up/chromium/chrome/services/file_util/BUILD.gn nw/chromium/chrome/services/file_util/BUILD.gn
--- up/chromium/chrome/services/file_util/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/services/file_util/BUILD.gn	2023-02-24 06:16:22.992185584 +0000
@@ -38,11 +38,11 @@
     deps += [ "//components/services/filesystem/public/mojom" ]
   }
 
-  if (is_mac) {
+  if (false && is_mac) {
     deps += [ "//chrome/utility/safe_browsing/mac" ]
   }
 
-  if (safe_browsing_mode == 1) {
+  if (false && safe_browsing_mode == 1) {
     sources += [
       "safe_archive_analyzer.cc",
       "safe_archive_analyzer.h",
