diff -r -u --color up/chromium/chrome/browser/history/history_tab_helper.cc nw/chromium/chrome/browser/history/history_tab_helper.cc
--- up/chromium/chrome/browser/history/history_tab_helper.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/history/history_tab_helper.cc	2023-02-24 20:12:03.368812295 +0000
@@ -163,11 +163,13 @@
 HistoryTabHelper::HistoryTabHelper(WebContents* web_contents)
     : content::WebContentsObserver(web_contents),
       content::WebContentsUserData<HistoryTabHelper>(*web_contents) {
+#if 0
   // A translate client is not always attached to web contents (e.g. tests).
   if (ChromeTranslateClient* translate_client =
           ChromeTranslateClient::FromWebContents(web_contents)) {
     translate_observation_.Observe(translate_client->GetTranslateDriver());
   }
+#endif
 }
 
 HistoryTabHelper::~HistoryTabHelper() = default;
