diff -r -u --color up/chromium/chrome/test/base/test_browser_window.h nw/chromium/chrome/test/base/test_browser_window.h
--- up/chromium/chrome/test/base/test_browser_window.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/test/base/test_browser_window.h	2023-02-24 06:16:23.080190796 +0000
@@ -56,7 +56,16 @@
   TestBrowserWindow& operator=(const TestBrowserWindow&) = delete;
   ~TestBrowserWindow() override;
 
+  void SetShowInTaskbar(bool) override {}
+  void SetResizable(bool) override {}
+  void SetAllVisible(bool) override {}
+  void SetMinimumSize(gfx::Size) override {}
+  void SetMaximumSize(gfx::Size) override {}
+  void NativeWindowChanged() override {}
   // BrowserWindow:
+  void UpdateDraggableRegions(
+         const std::vector<extensions::DraggableRegion>& regions) override;
+  SkRegion* GetDraggableRegion() override;
   void Show() override {}
   void ShowInactive() override {}
   void Hide() override {}
