diff -r -u --color up/chromium/chrome/browser/profiles/profile_io_data.cc nw/chromium/chrome/browser/profiles/profile_io_data.cc
--- up/chromium/chrome/browser/profiles/profile_io_data.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/profiles/profile_io_data.cc	2023-02-24 20:12:03.668815350 +0000
@@ -5,6 +5,8 @@
 #include "chrome/browser/profiles/profile_io_data.h"
 
 #include "base/feature_list.h"
+#include "content/nw/src/policy_cert_verifier.h"
+
 #include "base/logging.h"
 #include "base/strings/string_util.h"
 #include "build/build_config.h"
