diff -r -u --color up/chromium/chrome/browser/status_icons/status_icon.h nw/chromium/chrome/browser/status_icons/status_icon.h
--- up/chromium/chrome/browser/status_icons/status_icon.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/status_icons/status_icon.h	2023-02-24 20:12:04.112819871 +0000
@@ -75,11 +75,11 @@
   // thread to do it.  Use sparingly.
   virtual void ForceVisible();
 
- protected:
+ public:
   // Invoked after a call to SetContextMenu() to let the platform-specific
   // subclass update the native context menu based on the new model. If NULL is
   // passed, subclass should destroy the native context menu.
-  virtual void UpdatePlatformContextMenu(StatusIconMenuModel* model) = 0;
+  virtual void UpdatePlatformContextMenu(ui::MenuModel* model) = 0;
 
  private:
   base::ObserverList<StatusIconObserver>::Unchecked observers_;
