diff -r -u --color up/chromium/chrome/browser/component_updater/widevine_cdm_component_installer.h nw/chromium/chrome/browser/component_updater/widevine_cdm_component_installer.h
--- up/chromium/chrome/browser/component_updater/widevine_cdm_component_installer.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/component_updater/widevine_cdm_component_installer.h	2023-02-24 20:12:03.140809973 +0000
@@ -5,6 +5,8 @@
 #ifndef CHROME_BROWSER_COMPONENT_UPDATER_WIDEVINE_CDM_COMPONENT_INSTALLER_H_
 #define CHROME_BROWSER_COMPONENT_UPDATER_WIDEVINE_CDM_COMPONENT_INSTALLER_H_
 
+#include "base/callback_forward.h"
+
 namespace component_updater {
 
 class ComponentUpdateService;
@@ -17,6 +19,7 @@
 // 4) Register the Widevine CDM (via the adapter) with Chrome.
 // The first part is IO intensive so we do it asynchronously in the file thread.
 void RegisterWidevineCdmComponent(ComponentUpdateService* cus);
+void RegisterWidevineCdmComponent(ComponentUpdateService* cus, base::OnceClosure callback);
 
 }  // namespace component_updater
 
