diff -r -u --color up/chromium/chrome/browser/new_tab_page/modules/safe_browsing/safe_browsing_handler.cc nw/chromium/chrome/browser/new_tab_page/modules/safe_browsing/safe_browsing_handler.cc
--- up/chromium/chrome/browser/new_tab_page/modules/safe_browsing/safe_browsing_handler.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/new_tab_page/modules/safe_browsing/safe_browsing_handler.cc	2023-02-24 20:54:07.412518664 +0000
@@ -54,8 +54,8 @@
         handler,
     Profile* profile)
     : handler_(this, std::move(handler)),
-      metrics_collector_(
-          SafeBrowsingMetricsCollectorFactory::GetForProfile(profile)),
+      metrics_collector_(nullptr),
+      //          SafeBrowsingMetricsCollectorFactory::GetForProfile(profile)),
       pref_service_(profile->GetPrefs()),
       saved_last_cooldown_start_time_(0),
       saved_module_shown_count_(0) {}
