diff -r -u --color up/chromium/chrome/browser/extensions/api/debugger/debugger_api.cc nw/chromium/chrome/browser/extensions/api/debugger/debugger_api.cc
--- up/chromium/chrome/browser/extensions/api/debugger/debugger_api.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/api/debugger/debugger_api.cc	2023-02-24 20:12:03.236810951 +0000
@@ -362,10 +362,12 @@
   if (Manifest::IsPolicyLocation(extension_->location()))
     return true;
 
+#if 0
   subscription_ = ExtensionDevToolsInfoBarDelegate::Create(
       extension_id(), extension_->name(),
       base::BindOnce(&ExtensionDevToolsClientHost::InfoBarDestroyed,
                      base::Unretained(this)));
+#endif
   return true;
 }
 
