diff -r -u --color up/chromium/chrome/test/data/webui/print_preview/destination_dialog_test.ts nw/chromium/chrome/test/data/webui/print_preview/destination_dialog_test.ts
--- up/chromium/chrome/test/data/webui/print_preview/destination_dialog_test.ts	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/test/data/webui/print_preview/destination_dialog_test.ts	2023-02-24 06:16:24.376267506 +0000
@@ -113,7 +113,7 @@
         false /* pdfPrinterDisabled */, true /* isDriveMounted */,
         'FooDevice' /* printerName */,
         '' /* serializedDefaultDestinationSelectionRulesStr */,
-        [makeRecentDestination(destinations[4]!)] /* recentDestinations */);
+        [makeRecentDestination(destinations[4]!)] /* recentDestinations */, false);
     await whenPrinterListReady;
     whenPrinterListReady = nativeLayer.waitForGetPrinters(1);
     // This should trigger 1 new getPrinters() call, for extension printers.
@@ -137,7 +137,7 @@
             '' /* serializedDefaultDestinationSelectionRulesStr */, [
               makeRecentDestination(destinations[4]!),
               makeRecentDestination(extensionDestinations[0]!),
-            ] /* recentDestinations */);
+            ] /* recentDestinations */, false);
         await whenAllPreloaded;
         finishSetup();
         flush();
