diff -r -u --color up/chromium/chrome/chrome_elf/chrome_elf.ver nw/chromium/chrome/chrome_elf/chrome_elf.ver
--- up/chromium/chrome/chrome_elf/chrome_elf.ver	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/chrome_elf/chrome_elf.ver	2023-02-24 06:16:22.860177767 +0000
@@ -1,2 +1,2 @@
-INTERNAL_NAME=chrome_elf_dll
-ORIGINAL_FILENAME=chrome_elf.dll
+INTERNAL_NAME=nw_elf_dll
+ORIGINAL_FILENAME=nw_elf.dll
