diff -r -u --color up/chromium/chrome/test/base/test_browser_window_aura.h nw/chromium/chrome/test/base/test_browser_window_aura.h
--- up/chromium/chrome/test/base/test_browser_window_aura.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/test/base/test_browser_window_aura.h	2023-02-24 20:12:05.004828851 +0000
@@ -27,6 +27,9 @@
   ~TestBrowserWindowAura() override;
 
   // TestBrowserWindow overrides:
+  void UpdateDraggableRegions(
+         const std::vector<extensions::DraggableRegion>& regions) override;
+  SkRegion* GetDraggableRegion() override;
   gfx::NativeWindow GetNativeWindow() const override;
   void Show() override;
   void Hide() override;
