diff -r -u --color up/chromium/chrome/browser/ui/media_router/media_router_ui_service.h nw/chromium/chrome/browser/ui/media_router/media_router_ui_service.h
--- up/chromium/chrome/browser/ui/media_router/media_router_ui_service.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/media_router/media_router_ui_service.h	2023-02-24 20:12:04.440823210 +0000
@@ -53,7 +53,9 @@
   void DisableService();
 
   raw_ptr<Profile> profile_;
+#if defined(NWJS_SDK)
   std::unique_ptr<MediaRouterActionController> action_controller_;
+#endif
   std::unique_ptr<PrefChangeRegistrar> profile_pref_registrar_;
 
   base::ObserverList<Observer>::Unchecked observers_;
