diff -r -u --color up/chromium/chrome/browser/chrome_content_browser_client.h nw/chromium/chrome/browser/chrome_content_browser_client.h
--- up/chromium/chrome/browser/chrome_content_browser_client.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/chrome_content_browser_client.h	2023-02-24 06:16:20.772054112 +0000
@@ -170,12 +170,14 @@
   bool ShouldTreatURLSchemeAsFirstPartyWhenTopLevel(
       base::StringPiece scheme,
       bool is_embedded_origin_secure) override;
+  bool IsNWOrigin(const url::Origin& origin, content::BrowserContext* context) override;
   bool ShouldIgnoreSameSiteCookieRestrictionsWhenTopLevel(
       base::StringPiece scheme,
       bool is_embedded_origin_secure) override;
   std::string GetSiteDisplayNameForCdmProcess(
       content::BrowserContext* browser_context,
       const GURL& site_url) override;
+  static bool IsNWURL(const GURL& origin, content::BrowserContext* context);
   void OverrideURLLoaderFactoryParams(
       content::BrowserContext* browser_context,
       const url::Origin& origin,
@@ -284,6 +286,7 @@
       const GURL& destination_url,
       base::OnceCallback<void(file_access::ScopedFileAccess)>
           continuation_callback) override;
+  base::FilePath GetRootPath() override;
   void AllowWorkerFileSystem(
       const GURL& url,
       content::BrowserContext* browser_context,
