diff -r -u --color up/chromium/chrome/common/chrome_paths.cc nw/chromium/chrome/common/chrome_paths.cc
--- up/chromium/chrome/common/chrome_paths.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/common/chrome_paths.cc	2023-02-24 20:12:04.860827487 +0000
@@ -294,7 +294,7 @@
     // was shipped along with chrome.  The value can be overridden
     // if it is installed via component updater.
     case chrome::DIR_PNACL_COMPONENT:
-#if BUILDFLAG(IS_MAC)
+#if 0
       // PNaCl really belongs in the InternalPluginsDirectory but actually
       // copying it there would result in the files also being shipped, which
       // we don't want yet. So for now, just find them in the directory where
