diff -r -u --color up/chromium/chrome/browser/enterprise/connectors/reporting/reporting_service_settings.h nw/chromium/chrome/browser/enterprise/connectors/reporting/reporting_service_settings.h
--- up/chromium/chrome/browser/enterprise/connectors/reporting/reporting_service_settings.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/enterprise/connectors/reporting/reporting_service_settings.h	2023-02-24 20:12:03.216810747 +0000
@@ -43,6 +43,7 @@
 
   // All events that the reporting connector supports.
   static const constexpr char* kAllReportingEvents[] = {
+#if 0
       extensions::SafeBrowsingPrivateEventRouter::kKeyPasswordReuseEvent,
       extensions::SafeBrowsingPrivateEventRouter::kKeyPasswordChangedEvent,
       extensions::SafeBrowsingPrivateEventRouter::kKeyDangerousDownloadEvent,
@@ -55,6 +56,7 @@
           kKeyUrlFilteringInterstitialEvent,
       kExtensionInstallEvent,
       kBrowserCrashEvent,
+#endif
   };
 
  private:
