diff -r -u --color up/chromium/chrome/browser/chrome_browser_application_mac.mm nw/chromium/chrome/browser/chrome_browser_application_mac.mm
--- up/chromium/chrome/browser/chrome_browser_application_mac.mm	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/chrome_browser_application_mac.mm	2023-02-24 06:16:20.768053875 +0000
@@ -7,6 +7,11 @@
 #include <Carbon/Carbon.h>  // for <HIToolbox/Events.h>
 
 #include "base/check.h"
+#include "chrome/browser/apps/platform_apps/app_window_registry_util.h"
+#include "chrome/browser/lifetime/browser_close_manager.h"
+#include "chrome/browser/lifetime/application_lifetime_desktop.h"
+#include "content/public/common/content_features.h"
+
 #include "base/command_line.h"
 #include "base/mac/call_with_eh_frame.h"
 #include "base/observer_list.h"
@@ -219,6 +224,15 @@
 // the NSApplicationWillTerminateNotification to be posted, which ends the
 // NSApplication event loop, so final post- MessageLoop::Run() work is done
 // before exiting.
+
+- (void)closeAllWindowsQuit:(id)sender {
+  if (base::FeatureList::IsEnabled(::features::kNWNewWin)) {
+    chrome::CloseAllBrowsers(false, true);
+    return;
+  }
+  AppWindowRegistryUtil::CloseAllAppWindows(true);
+}
+
 - (void)terminate:(id)sender {
   AppController* appController = static_cast<AppController*>([NSApp delegate]);
   [appController tryToTerminateApplication:self];
