diff -r -u --color up/chromium/chrome/browser/ui/window_sizer/window_sizer.h nw/chromium/chrome/browser/ui/window_sizer/window_sizer.h
--- up/chromium/chrome/browser/ui/window_sizer/window_sizer.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/window_sizer/window_sizer.h	2023-02-24 20:54:08.604513168 +0000
@@ -72,14 +72,14 @@
   //
   // |show_state| will be overwritten and return the initial visual state of
   // the window to use.
-  static void GetBrowserWindowBoundsAndShowState(
+  static bool GetBrowserWindowBoundsAndShowState(
       const gfx::Rect& specified_bounds,
       const Browser* browser,
       gfx::Rect* window_bounds,
       ui::WindowShowState* show_state);
 
   // As above, but takes a state provider for testing.
-  static void GetBrowserWindowBoundsAndShowState(
+  static bool GetBrowserWindowBoundsAndShowState(
       std::unique_ptr<StateProvider> state_provider,
       const gfx::Rect& specified_bounds,
       const Browser* browser,
@@ -106,7 +106,7 @@
   virtual ~WindowSizer();
 
   // See GetBrowserWindowBoundsAndShowState() above.
-  virtual void DetermineWindowBoundsAndShowState(
+  virtual bool DetermineWindowBoundsAndShowState(
       const gfx::Rect& specified_bounds,
       gfx::Rect* bounds,
       ui::WindowShowState* show_state);
