diff -r -u --color up/chromium/chrome/browser/download/download_status_updater_linux.cc nw/chromium/chrome/browser/download/download_status_updater_linux.cc
--- up/chromium/chrome/browser/download/download_status_updater_linux.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/download/download_status_updater_linux.cc	2023-02-24 06:16:20.868059797 +0000
@@ -118,6 +118,7 @@
 bool IsRunning() {
   return inspector && get_unity_running && get_unity_running(inspector);
 }
+}  // namespace
 
 NO_SANITIZE("cfi-icall")
 void SetDownloadCount(int count) {
@@ -136,7 +137,6 @@
   }
 }
 
-}  // namespace
 
 void DownloadStatusUpdater::UpdateAppIconDownloadProgress(
     download::DownloadItem* download) {
