diff -r -u --color up/chromium/chrome/browser/sessions/session_service.cc nw/chromium/chrome/browser/sessions/session_service.cc
--- up/chromium/chrome/browser/sessions/session_service.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/sessions/session_service.cc	2023-02-24 20:54:07.960516137 +0000
@@ -506,7 +506,8 @@
 bool SessionService::RestoreIfNecessary(const StartupTabs& startup_tabs,
                                         Browser* browser,
                                         bool restore_apps) {
-  if (ShouldRestore(browser)) {
+  //nwjs#7516
+  if (false && ShouldRestore(browser)) {
     // We're going from no tabbed browsers to a tabbed browser (and not in
     // process startup), restore the last session.
     if (move_on_new_browser_ && is_saving_enabled()) {
