diff -r -u --color up/chromium/chrome/app/theme/chromium/BRANDING nw/chromium/chrome/app/theme/chromium/BRANDING
--- up/chromium/chrome/app/theme/chromium/BRANDING	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/app/theme/chromium/BRANDING	2023-02-24 20:12:02.564804108 +0000
@@ -1,10 +1,10 @@
-COMPANY_FULLNAME=The Chromium Authors
-COMPANY_SHORTNAME=The Chromium Authors
-PRODUCT_FULLNAME=Chromium
-PRODUCT_SHORTNAME=Chromium
-PRODUCT_INSTALLER_FULLNAME=Chromium Installer
-PRODUCT_INSTALLER_SHORTNAME=Chromium Installer
-COPYRIGHT=Copyright @LASTCHANGE_YEAR@ The Chromium Authors. All rights reserved.
-MAC_BUNDLE_ID=org.chromium.Chromium
-MAC_CREATOR_CODE=Cr24
+COMPANY_FULLNAME=The NW.js Community
+COMPANY_SHORTNAME=nwjs.io
+PRODUCT_FULLNAME=nwjs
+PRODUCT_SHORTNAME=nwjs
+PRODUCT_INSTALLER_FULLNAME=NWJS Installer
+PRODUCT_INSTALLER_SHORTNAME=NWJS Installer
+COPYRIGHT=Copyright @LASTCHANGE_YEAR@, The NW.js community and The Chromium Authors. All rights reserved.
+MAC_BUNDLE_ID=io.nwjs.nwjs
+MAC_CREATOR_CODE=NWJS
 MAC_TEAM_ID=
Binary files up/chromium/chrome/app/theme/chromium/mac/app.icns and nw/chromium/chrome/app/theme/chromium/mac/app.icns differ
Binary files up/chromium/chrome/app/theme/chromium/mac/document.icns and nw/chromium/chrome/app/theme/chromium/mac/document.icns differ
Binary files up/chromium/chrome/app/theme/chromium/product_logo_128.png and nw/chromium/chrome/app/theme/chromium/product_logo_128.png differ
Binary files up/chromium/chrome/app/theme/chromium/product_logo_22_mono.png and nw/chromium/chrome/app/theme/chromium/product_logo_22_mono.png differ
Binary files up/chromium/chrome/app/theme/chromium/product_logo_24.png and nw/chromium/chrome/app/theme/chromium/product_logo_24.png differ
Binary files up/chromium/chrome/app/theme/chromium/product_logo_256.png and nw/chromium/chrome/app/theme/chromium/product_logo_256.png differ
Only in nw/chromium/chrome/app/theme/chromium: product_logo_32.xpm
Binary files up/chromium/chrome/app/theme/chromium/product_logo_64.png and nw/chromium/chrome/app/theme/chromium/product_logo_64.png differ
Binary files up/chromium/chrome/app/theme/chromium/win/chromium.ico and nw/chromium/chrome/app/theme/chromium/win/chromium.ico differ
