diff -r -u --color up/chromium/chrome/browser/media/webrtc/display_media_access_handler.cc nw/chromium/chrome/browser/media/webrtc/display_media_access_handler.cc
--- up/chromium/chrome/browser/media/webrtc/display_media_access_handler.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/media/webrtc/display_media_access_handler.cc	2023-02-24 20:54:07.352518941 +0000
@@ -112,6 +112,7 @@
     return;
   }
 
+#if 0
   // SafeBrowsing Delayed Warnings experiment can delay some SafeBrowsing
   // warnings until user interaction. If the current page has a delayed warning,
   // it'll have a user interaction observer attached. Show the warning
@@ -127,6 +128,7 @@
     observer->OnDesktopCaptureRequest();
     return;
   }
+#endif
 
 #if BUILDFLAG(IS_MAC)
   // Do not allow picker UI to be shown on a page that isn't in the foreground
