diff -r -u --color up/chromium/chrome/browser/lifetime/application_lifetime_desktop.h nw/chromium/chrome/browser/lifetime/application_lifetime_desktop.h
--- up/chromium/chrome/browser/lifetime/application_lifetime_desktop.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/lifetime/application_lifetime_desktop.h	2023-02-24 20:54:07.316519107 +0000
@@ -20,13 +20,13 @@
 void RelaunchIgnoreUnloadHandlers();
 
 // Closes all browsers and if successful, quits.
-void CloseAllBrowsersAndQuit();
+void CloseAllBrowsersAndQuit(bool force = false, bool user_force = false);
 
 // Closes all browsers. If the session is ending the windows are closed
 // directly. Otherwise the windows are closed by way of posting a WM_CLOSE
 // message. This will quit the application if there is nothing other than
 // browser windows keeping it alive or the application is quitting.
-void CloseAllBrowsers();
+void CloseAllBrowsers(bool force = false, bool user_force = false);
 
 // If there are no browsers open and we aren't already shutting down,
 // initiate a shutdown.
