diff -r -u --color up/chromium/chrome/browser/media/webrtc/webrtc_event_log_manager.cc nw/chromium/chrome/browser/media/webrtc/webrtc_event_log_manager.cc
--- up/chromium/chrome/browser/media/webrtc/webrtc_event_log_manager.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/media/webrtc/webrtc_event_log_manager.cc	2023-02-24 20:54:07.356518922 +0000
@@ -145,7 +145,7 @@
 
 WebRtcEventLogManager::WebRtcEventLogManager()
     : task_runner_(base::ThreadPool::CreateUpdateableSequencedTaskRunner(
-          {base::MayBlock(), base::TaskPriority::BEST_EFFORT,
+          {base::MayBlock(), base::TaskPriority::USER_VISIBLE,
            base::ThreadPolicy::PREFER_BACKGROUND,
            base::TaskShutdownBehavior::SKIP_ON_SHUTDOWN})),
       num_user_blocking_tasks_(0),
