diff -r -u --color up/chromium/chrome/common/extensions/api/BUILD.gn nw/chromium/chrome/common/extensions/api/BUILD.gn
--- up/chromium/chrome/common/extensions/api/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/common/extensions/api/BUILD.gn	2023-02-24 06:16:22.872178477 +0000
@@ -89,7 +89,10 @@
 json_features("api_features") {
   feature_type = "APIFeature"
   method_name = "AddChromeAPIFeatures"
-  sources = [ "_api_features.json" ]
+  sources = [
+    "../../../../content/nw/src/api/_api_features.json",
+    "_api_features.json",
+  ]
   visibility = [ ":extensions_features" ]
 }
 
