diff -r -u --color up/chromium/chrome/browser/ui/views/frame/browser_non_client_frame_view_factory_views.cc nw/chromium/chrome/browser/ui/views/frame/browser_non_client_frame_view_factory_views.cc
--- up/chromium/chrome/browser/ui/views/frame/browser_non_client_frame_view_factory_views.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/frame/browser_non_client_frame_view_factory_views.cc	2023-02-24 20:54:08.408514071 +0000
@@ -91,7 +91,7 @@
   }
 
 #if BUILDFLAG(IS_WIN)
-  if (frame->ShouldUseNativeFrame())
+  if (frame->ShouldUseNativeFrame() && !browser_view->browser()->is_transparent())
     return std::make_unique<GlassBrowserFrameView>(frame, browser_view);
 #endif
   auto view = CreateOpaqueBrowserFrameView(frame, browser_view);
