diff -r -u --color up/chromium/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.cc nw/chromium/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.cc
--- up/chromium/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.cc	2023-02-24 06:16:22.472154789 +0000
@@ -259,7 +259,7 @@
     const views::Widget::InitParams& params) {
   DesktopWindowTreeHostLinux::Init(std::move(params));
 
-#if defined(USE_DBUS_MENU)
+#if 0 //defined(USE_DBUS_MENU)
   // We have now created our backing X11 window.  We now need to (possibly)
   // alert the desktop environment that there's a menu bar attached to it.
   if (CreateGlobalMenuBar()) {
@@ -276,7 +276,7 @@
 }
 
 void BrowserDesktopWindowTreeHostLinux::CloseNow() {
-#if defined(USE_DBUS_MENU)
+#if 0//defined(USE_DBUS_MENU)
   dbus_appmenu_.reset();
 #endif
   DesktopWindowTreeHostLinux::CloseNow();
