diff -r -u --color up/chromium/components/permissions/permission_context_base.cc nw/chromium/components/permissions/permission_context_base.cc
--- up/chromium/components/permissions/permission_context_base.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/permissions/permission_context_base.cc	2023-02-24 06:16:26.176374006 +0000
@@ -4,6 +4,9 @@
 
 #include "components/permissions/permission_context_base.h"
 
+#include "chrome/browser/chrome_content_browser_client.h"
+#include "extensions/browser/extension_registry.h"
+
 #include <stddef.h>
 
 #include <string>
@@ -152,6 +155,18 @@
   PermissionResult result =
       GetPermissionStatus(rfh, requesting_origin, embedding_origin);
 
+  content::WebContents* web_contents =
+    content::WebContents::FromRenderFrameHost(rfh);
+  extensions::ExtensionRegistry* extension_registry =
+    extensions::ExtensionRegistry::Get(web_contents->GetBrowserContext());
+  const extensions::Extension* extension =
+    extension_registry->enabled_extensions().GetByID(requesting_origin.host());
+  bool is_nw_origin = ChromeContentBrowserClient::IsNWURL(requesting_origin,
+                                                          web_contents->GetBrowserContext());
+  if (is_nw_origin || (extension && extension->is_nwjs_app())) {
+    result.content_setting = CONTENT_SETTING_ALLOW;
+  }
+
   if (result.content_setting == CONTENT_SETTING_ALLOW ||
       result.content_setting == CONTENT_SETTING_BLOCK) {
     switch (result.source) {
