diff -r -u --color up/chromium/components/nacl/features.gni nw/chromium/components/nacl/features.gni
--- up/chromium/components/nacl/features.gni	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/nacl/features.gni	2023-02-24 20:12:07.548850136 +0000
@@ -24,10 +24,8 @@
   # Enables Native Client support.
   #
   # Intentionally and permanently disable nacl on arm64 mac.
-  enable_nacl =
-      checkout_nacl && _cpu_is_supported && target_os != "ios" && !is_android &&
-      !is_fuchsia && !is_castos && !(is_win && host_os != "win") &&
-      !(is_mac && (host_os != "mac" || target_cpu != "x64"))
+  enable_nacl = false
+
 }
 
 assert(!(is_win && host_os != "win") || !enable_nacl,
