diff -r -u --color up/chromium/components/password_manager/core/browser/password_form_manager.h nw/chromium/components/password_manager/core/browser/password_form_manager.h
--- up/chromium/components/password_manager/core/browser/password_form_manager.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/password_manager/core/browser/password_form_manager.h	2023-02-24 06:16:26.076368089 +0000
@@ -174,6 +174,7 @@
   bool IsNewLogin() const;
   FormFetcher* GetFormFetcher();
   bool IsPendingCredentialsPublicSuffixMatch() const;
+  bool IsPendingCredentialsOriginExtension() const;
   void PresaveGeneratedPassword(const autofill::FormData& form_data,
                                 const std::u16string& generated_password);
   void PasswordNoLongerGenerated();
