diff -r -u --color up/chromium/components/translate/core/language_detection/BUILD.gn nw/chromium/components/translate/core/language_detection/BUILD.gn
--- up/chromium/components/translate/core/language_detection/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/translate/core/language_detection/BUILD.gn	2023-02-24 06:16:27.344443113 +0000
@@ -40,7 +40,7 @@
     "//components/language/core/common",
     "//components/optimization_guide/core:features",
     "//components/translate/core/common",
-    "//third_party/cld_3/src/src:cld_3",
+    #"//third_party/cld_3/src/src:cld_3",
     "//third_party/flatbuffers",
     "//third_party/icu",
     "//third_party/smhasher:murmurhash2",
