diff -r -u --color up/chromium/components/remote_cocoa/app_shim/select_file_dialog_bridge.mm nw/chromium/components/remote_cocoa/app_shim/select_file_dialog_bridge.mm
--- up/chromium/components/remote_cocoa/app_shim/select_file_dialog_bridge.mm	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/remote_cocoa/app_shim/select_file_dialog_bridge.mm	2023-02-24 20:54:11.628499223 +0000
@@ -1,7 +1,8 @@
 // Copyright 2019 The Chromium Authors
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
-
+#pragma clang diagnostic ignored "-Wunreachable-code"
+#pragma clang diagnostic ignored "-Wunused-variable"
 #include "components/remote_cocoa/app_shim/select_file_dialog_bridge.h"
 
 #include <AppKit/AppKit.h>
@@ -282,9 +283,10 @@
           base::SysUTF8ToNSString(default_path.BaseName().value());
     }
   }
-
+#if 0
   const bool keep_extension_visible =
       file_types ? file_types->keep_extension_visible : false;
+#endif
   if (type_ != SelectFileDialogType::kFolder &&
       type_ != SelectFileDialogType::kUploadFolder &&
       type_ != SelectFileDialogType::kExistingFolder) {
@@ -299,6 +301,9 @@
   }
 
   if (type_ == SelectFileDialogType::kSaveAsFile) {
+#if 1 //NWJS#6091: extension was hidden
+    [dialog setExtensionHidden:NO];
+#else
     // When file extensions are hidden and removing the extension from
     // the default filename gives one which still has an extension
     // that OS X recognizes, it will get confused and think the user
@@ -313,6 +318,7 @@
       [dialog setExtensionHidden:YES];
       [dialog setCanSelectHiddenExtension:YES];
     }
+#endif
   } else {
     // This does not use ObjCCast because the underlying object could be a
     // non-exported AppKit type (https://crbug.com/995476).
@@ -335,7 +341,7 @@
         [open_dialog setCanCreateDirectories:NO];
 
       NSString* prompt =
-          (type_ == SelectFileDialogType::kUploadFolder)
+          (false && type_ == SelectFileDialogType::kUploadFolder)
               ? l10n_util::GetNSString(IDS_SELECT_UPLOAD_FOLDER_BUTTON_TITLE)
               : l10n_util::GetNSString(IDS_SELECT_FOLDER_BUTTON_TITLE);
       [open_dialog setPrompt:prompt];
