diff -r -u --color up/chromium/content/public/common/content_switches.cc nw/chromium/content/public/common/content_switches.cc
--- up/chromium/content/public/common/content_switches.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/public/common/content_switches.cc	2023-02-24 06:16:28.168491866 +0000
@@ -136,9 +136,17 @@
 // then the GPU process won't launch.
 const char kDisableGpu[]                    = "disable-gpu";
 
+// Disable apps transparency support
+const char kDisableTransparency[]           = "disable-transparency";
+
+// Force CPU drawing, needed to enable click through on transparent window
+const char kForceCpuDraw[]                  = "force-cpu-draw";
+
 // Prevent the compositor from using its GPU implementation.
 const char kDisableGpuCompositing[]         = "disable-gpu-compositing";
-
+const char kDisableRAFThrottling[]          = "disable-raf-throttling";
+const char kEnableNodeWorker[]              = "enable-node-worker";
+const char kNWAppPath[]                     = "nwapp-path";
 // Disable proactive early init of GPU process.
 const char kDisableGpuEarlyInit[]           = "disable-gpu-early-init";
 
