diff -r -u --color up/chromium/content/app_shim_remote_cocoa/render_widget_host_ns_view_bridge.h nw/chromium/content/app_shim_remote_cocoa/render_widget_host_ns_view_bridge.h
--- up/chromium/content/app_shim_remote_cocoa/render_widget_host_ns_view_bridge.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/app_shim_remote_cocoa/render_widget_host_ns_view_bridge.h	2023-02-24 06:16:27.692463703 +0000
@@ -20,6 +20,8 @@
 #include "ui/base/mojom/attributed_string.mojom-forward.h"
 #include "ui/display/display_observer.h"
 
+@class CALayer;
+
 namespace remote_cocoa {
 
 // Mojo bridge for a RenderWidgetHostViewMac's NSView. This class may be
@@ -90,6 +92,7 @@
       bool consumed) override;
   void DidOverscroll(blink::mojom::DidOverscrollParamsPtr params) override;
 
+  CALayer* GetBackgroundLayer();
  private:
   bool IsPopup() const { return !!popup_window_; }
 
