diff -r -u --color up/chromium/content/test/test_web_contents.h nw/chromium/content/test/test_web_contents.h
--- up/chromium/content/test/test_web_contents.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/test/test_web_contents.h	2023-02-24 06:16:28.636519556 +0000
@@ -192,7 +192,7 @@
                          int route_id,
                          WindowOpenDisposition disposition,
                          const blink::mojom::WindowFeatures& window_features,
-                         bool user_gesture) override;
+                         bool user_gesture, std::string) override;
   void ShowCreatedWidget(int process_id,
                          int route_id,
                          const gfx::Rect& initial_rect,
