diff -r -u --color up/chromium/content/public/common/content_features.h nw/chromium/content/public/common/content_features.h
--- up/chromium/content/public/common/content_features.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/public/common/content_features.h	2023-02-24 20:54:12.988492952 +0000
@@ -17,6 +17,7 @@
 
 namespace features {
 
+CONTENT_EXPORT extern const base::Feature kNWNewWin;
 // All features in alphabetical order. The features should be documented
 // alongside the definition of their values in the .cc file.
 CONTENT_EXPORT BASE_DECLARE_FEATURE(kAllowContentInitiatedDataUrlNavigations);
