diff -r -u --color up/chromium/content/browser/media/capture/desktop_streams_registry_impl.cc nw/chromium/content/browser/media/capture/desktop_streams_registry_impl.cc
--- up/chromium/content/browser/media/capture/desktop_streams_registry_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/browser/media/capture/desktop_streams_registry_impl.cc	2023-02-24 20:54:12.712494225 +0000
@@ -79,7 +79,7 @@
     int render_frame_id,
     const url::Origin& origin,
     std::string* extension_name,
-    const DesktopStreamRegistryType type) {
+    const DesktopStreamRegistryType type, bool nodejs) {
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
 
   auto it = approved_streams_.find(id);
@@ -88,8 +88,8 @@
   // the same origin and the same renderer.
   if (it == approved_streams_.end() ||
       render_process_id != it->second.render_process_id ||
-      render_frame_id != it->second.render_frame_id ||
-      origin != it->second.origin || type != it->second.type) {
+      (!nodejs && render_frame_id != it->second.render_frame_id) ||
+      (!nodejs && origin != it->second.origin) || type != it->second.type) {
     return DesktopMediaID();
   }
 
