diff -r -u --color up/chromium/content/public/common/content_client.h nw/chromium/content/public/common/content_client.h
--- up/chromium/content/public/common/content_client.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/public/common/content_client.h	2023-02-24 06:16:28.164491629 +0000
@@ -5,6 +5,9 @@
 #ifndef CONTENT_PUBLIC_COMMON_CONTENT_CLIENT_H_
 #define CONTENT_PUBLIC_COMMON_CONTENT_CLIENT_H_
 
+#include "base/files/file_path.h"
+#include "base/values.h"
+
 #include <set>
 #include <string>
 #include <vector>
@@ -24,6 +27,7 @@
 class FilePath;
 class RefCountedMemory;
 class SequencedTaskRunner;
+class DictionaryValue;
 }
 
 namespace blink {
@@ -61,10 +65,10 @@
 // content code is called.
 CONTENT_EXPORT void SetContentClient(ContentClient* client);
 
-#if defined(CONTENT_IMPLEMENTATION)
+//#if defined(CONTENT_IMPLEMENTATION)
 // Content's embedder API should only be used by content.
-ContentClient* GetContentClient();
-#endif
+CONTENT_EXPORT ContentClient* GetContentClient();
+//#endif
 
 // Used for tests to override the relevant embedder interfaces. Each method
 // returns the old value. In browser tests it seems safest to call these in
@@ -84,6 +88,9 @@
   ContentClient();
   virtual ~ContentClient();
 
+  virtual void LoadNWAppAsExtension(base::Value::Dict* manifest,
+                                    const base::FilePath& path,
+                                    std::string* error) {}
   ContentBrowserClient* browser() { return browser_; }
   ContentGpuClient* gpu() { return gpu_; }
   ContentRendererClient* renderer() { return renderer_; }
@@ -97,6 +104,7 @@
   // WebSecurityOrigin or a full URL (if called from the browser process) and a
   // string is the lowest-common-denominator.
   virtual void SetActiveURL(const GURL& url, std::string top_origin) {}
+  virtual void SetNWReportURL(const GURL& url) {}
 
   // Sets the data on the current gpu.
   virtual void SetGpuInfo(const gpu::GPUInfo& gpu_info) {}
