diff -r -u --color up/chromium/content/browser/renderer_host/navigation_controller_impl.cc nw/chromium/content/browser/renderer_host/navigation_controller_impl.cc
--- up/chromium/content/browser/renderer_host/navigation_controller_impl.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/browser/renderer_host/navigation_controller_impl.cc	2023-02-24 06:16:27.936478139 +0000
@@ -719,7 +719,7 @@
     BrowserContext* browser_context,
     FrameTree& frame_tree,
     NavigationControllerDelegate* delegate)
-    : frame_tree_(frame_tree),
+  : history_initiator_(nullptr), frame_tree_(frame_tree),
       browser_context_(browser_context),
       delegate_(delegate),
       ssl_manager_(this),
@@ -3209,13 +3209,23 @@
   // function.
   std::unique_ptr<PendingEntryRef> pending_entry_ref = ReferencePendingEntry();
 
+  FrameTreeNode* nwfaketop_node = nullptr;
+  if (history_initiator_) {
+    if (history_initiator_->frame_tree_node()->frame_owner_properties().nwfaketop)
+      nwfaketop_node = history_initiator_->frame_tree_node();
+  }
+
   // Send all the same document frame loads before the different document loads.
   for (auto& item : same_document_loads) {
     FrameTreeNode* frame = item->frame_tree_node();
+    if (nwfaketop_node && nwfaketop_node->current_frame_host()->IsDescendantOfWithinFrameTree(frame->current_frame_host()))
+      continue;
     frame->navigator().Navigate(std::move(item), reload_type);
   }
   for (auto& item : different_document_loads) {
     FrameTreeNode* frame = item->frame_tree_node();
+    if (nwfaketop_node && nwfaketop_node->current_frame_host()->IsDescendantOfWithinFrameTree(frame->current_frame_host()))
+      continue;
     frame->navigator().Navigate(std::move(item), reload_type);
   }
 
