diff -r -u --color up/chromium/content/common/zygote/zygote_communication_linux.cc nw/chromium/content/common/zygote/zygote_communication_linux.cc
--- up/chromium/content/common/zygote/zygote_communication_linux.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/common/zygote/zygote_communication_linux.cc	2023-02-24 20:12:09.428865866 +0000
@@ -4,6 +4,7 @@
 
 #include "content/common/zygote/zygote_communication_linux.h"
 
+#include "content/nw/src/common/shell_switches.h"
 #include <string.h>
 #include <sys/socket.h>
 
@@ -256,6 +257,7 @@
   static const char* const kForwardSwitches[] = {
       sandbox::policy::switches::kAllowSandboxDebugging,
       switches::kDisableInProcessStackTraces,
+      "enable-spell-checking",
       sandbox::policy::switches::kDisableSeccompFilterSandbox,
       sandbox::policy::switches::kNoSandbox,
   };
