diff -r -u --color up/chromium/content/browser/renderer_host/render_view_host_impl.cc nw/chromium/content/browser/renderer_host/render_view_host_impl.cc
--- up/chromium/content/browser/renderer_host/render_view_host_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/browser/renderer_host/render_view_host_impl.cc	2023-02-24 20:54:12.828493690 +0000
@@ -502,6 +502,7 @@
       frame_tree_->controller()
           .GetSessionStorageNamespace(storage_partition_config_)
           ->id();
+  params->skip_blocking_parser = delegate_->GetSkipBlockingParser();
   params->hidden = frame_tree_->delegate()->IsHidden();
   params->never_composited = delegate_->IsNeverComposited();
   params->window_was_opened_by_another_window =
