diff -r -u --color up/chromium/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm nw/chromium/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm
--- up/chromium/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm	2023-02-24 20:54:12.580494834 +0000
@@ -59,6 +59,10 @@
 using remote_cocoa::mojom::RenderWidgetHostNSViewHost;
 using remote_cocoa::RenderWidgetHostNSViewHostHelper;
 
+namespace content {
+  extern bool g_force_cpu_draw;
+}
+
 namespace {
 
 constexpr NSString* WebAutomaticQuoteSubstitutionEnabled =
@@ -81,6 +85,7 @@
   DummyHostHelper& operator=(const DummyHostHelper&) = delete;
 
  private:
+  content::RenderWidgetHostViewMac* GetRenderWidgetHostViewMac() override { return nullptr; }
   // RenderWidgetHostNSViewHostHelper implementation.
   id GetAccessibilityElement() override { return nil; }
   id GetRootBrowserAccessibilityElement() override { return nil; }
@@ -588,6 +593,19 @@
   return [super forwardingTargetForSelector:selector];
 }
 
+- (void)drawRect:(NSRect)dirty {
+  if (content::g_force_cpu_draw) {
+    CGContextRef ctx = (CGContextRef)[[NSGraphicsContext currentContext] graphicsPort];
+    CGContextClipToRect(ctx, NSRectToCGRect(dirty));
+    //High Sierra 10.13 fix, previously we use [self layer],
+    //since we have set the layer to nil in AcceleratedWidgetMac::GotSoftwareFrame,
+    //we access the layer "directly" which is the "background_layer()" (see RenderWidgetHostViewMac constructor)
+    [_hostHelper->GetRenderWidgetHostViewMac()->background_layer() renderInContext:ctx];
+  } else {
+    [super drawRect:dirty];
+  }
+}
+
 - (void)setCanBeKeyView:(BOOL)can {
   _canBeKeyView = can;
 }
@@ -1510,8 +1528,17 @@
 }
 
 - (void)setFrameSize:(NSSize)newSize {
+  //High Sierra 10.13 fix, RenderWidgetHostViewCocoa CALayer must be nil
+  //so we can do drawRect "manually"
+  //here, we temporarily assign back the layer during resize, so the background_layer() can be resized properly
+  if (content::g_force_cpu_draw)
+    [self setLayer:_hostHelper->GetRenderWidgetHostViewMac()->background_layer()];
+
   [super setFrameSize:newSize];
   [self sendViewBoundsInWindowToHost];
+
+  if (content::g_force_cpu_draw)
+    [self setLayer:nil];
 }
 
 - (BOOL)canBecomeKeyView {
