diff -r -u --color up/chromium/content/browser/gpu/gpu_process_host.h nw/chromium/content/browser/gpu/gpu_process_host.h
--- up/chromium/content/browser/gpu/gpu_process_host.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/browser/gpu/gpu_process_host.h	2023-02-24 20:54:12.680494373 +0000
@@ -239,6 +239,8 @@
   // The GPU process reported failure to initialize.
   bool did_fail_initialize_ = false;
 
+  bool closing_;
+
   // The total number of GPU process crashes.
   static base::subtle::Atomic32 gpu_crash_count_;
   static bool crashed_before_;
