diff -r -u --color up/chromium/content/renderer/render_thread_impl.cc nw/chromium/content/renderer/render_thread_impl.cc
--- up/chromium/content/renderer/render_thread_impl.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/renderer/render_thread_impl.cc	2023-02-24 06:16:28.216494706 +0000
@@ -996,6 +996,13 @@
       std::move(closure));
 }
 
+bool RenderThreadImpl::ResolveProxy(const GURL& url, std::string* proxy_list) {
+  absl::optional<std::string> result;
+  GetRendererHost()->ResolveProxy(url, &result);
+  *proxy_list = result.value_or(std::string());
+  return result.has_value();
+}
+
 media::GpuVideoAcceleratorFactories* RenderThreadImpl::GetGpuFactories() {
   DCHECK(IsMainThread());
 
