diff -r -u --color up/chromium/content/test/BUILD.gn nw/chromium/content/test/BUILD.gn
--- up/chromium/content/test/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/test/BUILD.gn	2023-02-24 06:16:28.252496836 +0000
@@ -2451,6 +2451,7 @@
     "../browser/renderer_host/renderer_sandboxed_process_launcher_delegate_unittest.cc",
     "../browser/renderer_host/transient_allow_popup_unittest.cc",
     "../browser/renderer_host/visible_time_request_trigger_unittest.cc",
+    "../browser/resolve_proxy_helper_unittest.cc",
     "../browser/scheduler/browser_io_thread_delegate_unittest.cc",
     "../browser/scheduler/browser_task_executor_unittest.cc",
     "../browser/scheduler/browser_task_queues_unittest.cc",
@@ -3061,7 +3062,7 @@
       "//device/base",
       "//device/fido",
       "//device/fido:cablev2_authenticator",
-      "//device/fido:cablev2_test_util",
+      #"//device/fido:cablev2_test_util",
       "//device/fido:mocks",
       "//device/fido:test_support",
     ]
