diff -r -u --color up/chromium/content/app/content_main_runner_impl.cc nw/chromium/content/app/content_main_runner_impl.cc
--- up/chromium/content/app/content_main_runner_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/app/content_main_runner_impl.cc	2023-02-24 20:12:09.084862987 +0000
@@ -1034,10 +1034,10 @@
   DCHECK(is_initialized_);
   DCHECK(content_main_params_);
   DCHECK(!is_shutdown_);
-  const base::CommandLine* command_line =
-      base::CommandLine::ForCurrentProcess();
+  base::CommandLine& command_line =
+      *base::CommandLine::ForCurrentProcess();
   std::string process_type =
-      command_line->GetSwitchValueASCII(switches::kProcessType);
+      command_line.GetSwitchValueASCII(switches::kProcessType);
 
   // Run this logic on all child processes.
   if (!process_type.empty()) {
@@ -1073,7 +1073,11 @@
 #endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
   }
 
-  MainFunctionParams main_params(command_line);
+  if (process_type.empty()) {
+    command_line.AppendSwitch(sandbox::policy::switches::kNoSandbox);
+    command_line.AppendSwitch(switches::kNoZygote);
+  }
+  MainFunctionParams main_params(&command_line);
   main_params.ui_task = std::move(content_main_params_->ui_task);
   main_params.created_main_parts_closure =
       std::move(content_main_params_->created_main_parts_closure);
