diff -r -u --color up/chromium/content/public/common/user_agent.h nw/chromium/content/public/common/user_agent.h
--- up/chromium/content/public/common/user_agent.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/public/common/user_agent.h	2023-02-24 20:54:12.988492952 +0000
@@ -54,6 +54,7 @@
     IncludeAndroidBuildNumber include_android_build_number,
     IncludeAndroidModel include_android_model);
 
+CONTENT_EXPORT std::string BuildOSInfo();
 // Returns the reduced User-agent string for
 // https://github.com/WICG/ua-client-hints.
 CONTENT_EXPORT std::string GetReducedUserAgent(bool mobile,
@@ -78,7 +79,6 @@
 // if on a codenamed (i.e. not a release) build of an Android.
 CONTENT_EXPORT std::string BuildModelInfo();
 
-#if BUILDFLAG(IS_ANDROID)
 // Helper function to generate a full user agent string given a short
 // product name and some extra text to be added to the OS info.
 // This is currently only used for Android Web View.
@@ -91,7 +91,6 @@
 CONTENT_EXPORT std::string GetAndroidOSInfo(
     IncludeAndroidBuildNumber include_android_build_number,
     IncludeAndroidModel include_android_model);
-#endif
 
 // Builds a full user agent string given a string describing the OS and a
 // product name.
