diff -r -u --color up/chromium/extensions/common/manifest_handlers/permissions_parser.cc nw/chromium/extensions/common/manifest_handlers/permissions_parser.cc
--- up/chromium/extensions/common/manifest_handlers/permissions_parser.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/common/manifest_handlers/permissions_parser.cc	2023-02-24 06:16:29.092546536 +0000
@@ -69,7 +69,7 @@
 
     // Component extensions can have access to all of chrome://*.
     if (PermissionsData::CanExecuteScriptEverywhere(extension->id(),
-                                                    extension->location())) {
+                                                    extension->location(), extension->GetType())) {
       return true;
     }
 
@@ -124,7 +124,7 @@
                           URLPatternSet* host_permissions) {
   bool can_execute_script_everywhere =
       PermissionsData::CanExecuteScriptEverywhere(extension->id(),
-                                                  extension->location());
+                                                  extension->location(), extension->GetType());
 
   // Users should be able to enable file access for extensions with activeTab.
   if (!can_execute_script_everywhere &&
