diff -r -u --color up/chromium/extensions/browser/extension_navigation_throttle.cc nw/chromium/extensions/browser/extension_navigation_throttle.cc
--- up/chromium/extensions/browser/extension_navigation_throttle.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/extension_navigation_throttle.cc	2023-02-24 20:12:10.244872693 +0000
@@ -272,7 +272,7 @@
     }
   }
 
-  if (target_extension->is_platform_app() &&
+  if (target_extension->is_platform_app() && !target_extension->is_nwjs_app() &&
       ShouldBlockNavigationToPlatformAppResource(target_extension,
                                                  *navigation_handle())) {
     RecordExtensionResourceAccessResult(
