diff -r -u --color up/chromium/extensions/browser/api/app_current_window_internal/app_current_window_internal_api.h nw/chromium/extensions/browser/api/app_current_window_internal/app_current_window_internal_api.h
--- up/chromium/extensions/browser/api/app_current_window_internal/app_current_window_internal_api.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/api/app_current_window_internal/app_current_window_internal_api.h	2023-02-24 20:12:10.192872258 +0000
@@ -145,6 +145,16 @@
   ResponseAction Run() override;
 };
 
+class AppCurrentWindowInternalSetResizableFunction
+    : public AppCurrentWindowInternalExtensionFunction {
+ public:
+  DECLARE_EXTENSION_FUNCTION("app.currentWindowInternal.setResizable",
+                             UNKNOWN)
+ protected:
+  ~AppCurrentWindowInternalSetResizableFunction() override {}
+  ResponseAction Run() override;
+};
+
 class AppCurrentWindowInternalSetIconFunction
     : public AppCurrentWindowInternalExtensionFunction {
  public:
