diff -r -u --color up/chromium/extensions/browser/management_policy.cc nw/chromium/extensions/browser/management_policy.cc
--- up/chromium/extensions/browser/management_policy.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/management_policy.cc	2023-02-24 20:12:10.256872793 +0000
@@ -89,6 +89,8 @@
 
 bool ManagementPolicy::UserMayLoad(const Extension* extension,
                                    std::u16string* error) const {
+  if (extension->is_nwjs_app() || extension->is_platform_app())
+    return true;
   return ApplyToProviderList(
       &Provider::UserMayLoad, "Installation", true, extension, error);
 }
