diff -r -u --color up/chromium/extensions/common/manifest_handlers/background_info.cc nw/chromium/extensions/common/manifest_handlers/background_info.cc
--- up/chromium/extensions/common/manifest_handlers/background_info.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/common/manifest_handlers/background_info.cc	2023-02-24 20:12:10.288873061 +0000
@@ -380,6 +380,9 @@
   const std::vector<std::string>& background_scripts =
       BackgroundInfo::GetBackgroundScripts(extension);
   for (size_t i = 0; i < background_scripts.size(); ++i) {
+    if (background_scripts[i] == kNWJSDefaultAppJS ||
+        background_scripts[i] == "nwjs/newwin.js")
+      continue;
     if (!base::PathExists(
             extension->GetResource(background_scripts[i]).GetFilePath())) {
       *error = l10n_util::GetStringFUTF8(
