diff -r -u --color up/chromium/extensions/common/features/manifest_feature.cc nw/chromium/extensions/common/features/manifest_feature.cc
--- up/chromium/extensions/common/features/manifest_feature.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/common/features/manifest_feature.cc	2023-02-24 20:54:13.744489466 +0000
@@ -22,6 +22,8 @@
     Feature::Platform platform,
     int context_id,
     bool check_developer_mode) const {
+  if (extension && extension->is_nwjs_app())
+    return CreateAvailability(IS_AVAILABLE);
   Availability availability = SimpleFeature::IsAvailableToContextImpl(
       extension, context, url, platform, context_id, check_developer_mode);
   if (!availability.is_available())
