diff -r -u --color up/chromium/extensions/browser/extension_function_dispatcher.cc nw/chromium/extensions/browser/extension_function_dispatcher.cc
--- up/chromium/extensions/browser/extension_function_dispatcher.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/browser/extension_function_dispatcher.cc	2023-02-24 06:16:29.040543460 +0000
@@ -214,6 +214,14 @@
   base::debug::ScopedCrashKeyString extension_id_;
 };
 
+void DummyCallback(
+                   ExtensionFunction::ResponseType type,
+                   base::Value::List results,
+                   const std::string& error,
+		   mojom::ExtraResponseDataPtr response_data
+                   ) {
+}
+
 }  // namespace
 
 class ExtensionFunctionDispatcher::ResponseCallbackWrapper
@@ -380,6 +388,19 @@
 ExtensionFunctionDispatcher::~ExtensionFunctionDispatcher() {
 }
 
+void ExtensionFunctionDispatcher::DispatchSync(
+                    mojom::RequestParamsPtr params,
+                    bool* success,
+                    base::Value::List* response,
+                    std::string* error,
+                    content::RenderFrameHost* render_frame_host,
+                    int render_process_id) {
+  base::OnceCallback<decltype(DummyCallback)> dummy;
+  DispatchWithCallbackInternal(
+                               *params, render_frame_host, render_process_id, std::move(dummy), true,
+                               success, response, error);
+}
+
 void ExtensionFunctionDispatcher::Dispatch(
     mojom::RequestParamsPtr params,
     content::RenderFrameHost& frame,
@@ -460,7 +481,12 @@
     const mojom::RequestParams& params,
     content::RenderFrameHost* render_frame_host,
     int render_process_id,
-    ExtensionFunction::ResponseCallback callback) {
+    ExtensionFunction::ResponseCallback callback,
+    bool sync,
+    bool* success,
+    base::Value::List* response,
+    std::string* error
+                                                               ) {
   ProcessMap* process_map = ProcessMap::Get(browser_context_);
   if (!process_map) {
     constexpr char kProcessNotFound[] =
@@ -520,7 +546,12 @@
 
     // Skip the quota, event page, activity logging stuff if there
     // isn't an extension, e.g. if the function call was from WebUI.
-    function->RunWithValidation()->Execute();
+    if (!sync)
+      function->RunWithValidation()->Execute();
+    else {
+      *success = function->RunNWSync(response, error);
+      function->did_respond_ = true;
+    }
     return;
   }
 
@@ -562,7 +593,12 @@
     }
 
     base::ElapsedTimer timer;
-    function->RunWithValidation()->Execute();
+    if (!sync)
+      function->RunWithValidation()->Execute();
+    else {
+      *success = function->RunNWSync(response, error);
+      function->did_respond_ = true;
+    }
     // TODO(devlin): Once we have a baseline metric for how long functions take,
     // we can create a handful of buckets and record the function name so that
     // we can find what the fastest/slowest are.
