diff -r -u --color up/chromium/extensions/renderer/script_injector.h nw/chromium/extensions/renderer/script_injector.h
--- up/chromium/extensions/renderer/script_injector.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/script_injector.h	2023-02-24 20:12:10.320873329 +0000
@@ -28,8 +28,8 @@
 namespace extensions {
 
 // The pseudo-delegate class for a ScriptInjection that provides all necessary
-// information about how to inject the script, including what code to inject and
-// when (run location), but without any injection logic.
+// information about how to inject the script, including what code to inject,
+// when (run location), and where (world), but without any injection logic.
 class ScriptInjector {
  public:
   // The possible reasons for not injecting the script.
@@ -53,6 +53,9 @@
   // Returns the associated `UserActivationOption` for script evaluation.
   virtual blink::mojom::UserActivationOption IsUserGesture() const = 0;
 
+  // Returns true if the script should execute in the main world.
+  virtual bool ShouldExecuteInMainWorld() const = 0;
+
   // Returns the world in which to execute the javascript code.
   virtual mojom::ExecutionWorld GetExecutionWorld() const = 0;
 
