diff -r -u --color up/chromium/extensions/renderer/bindings/api_signature.cc nw/chromium/extensions/renderer/bindings/api_signature.cc
--- up/chromium/extensions/renderer/bindings/api_signature.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/bindings/api_signature.cc	2023-02-24 20:54:13.764489374 +0000
@@ -30,6 +30,7 @@
     "test.callbackFail",
     "automation.addTreeChangeObserver",
     "automation.removeTreeChangeObserver",
+    "nw.App.on",
 };
 
 std::vector<std::unique_ptr<ArgumentSpec>> ValueListToArgumentSpecs(
