diff -r -u --color up/chromium/extensions/common/api/app_window.idl nw/chromium/extensions/common/api/app_window.idl
--- up/chromium/extensions/common/api/app_window.idl	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/common/api/app_window.idl	2023-02-24 20:12:10.272872927 +0000
@@ -135,6 +135,8 @@
   // State of a window: normal, fullscreen, maximized, minimized.
   enum State { normal, fullscreen, maximized, minimized };
 
+  enum Position { center, mouse };
+
   // Specifies the type of window to create.
   enum WindowType {
     // Default window type.
@@ -298,6 +300,15 @@
     //   </p>
     // <p>This is <b>Chrome OS only</b>.</p>
     [nodoc] app.runtime.ActionType? lockScreenAction;
+    boolean? kiosk;
+
+    Position? position;
+    DOMString? title;
+    DOMString? icon;
+    boolean? show_in_taskbar;
+    boolean? new_instance;
+    DOMString? inject_js_start;
+    DOMString? inject_js_end;
   };
 
   // Called in the creating window (parent) before the load event is called in
@@ -340,6 +351,9 @@
     // <code>AppWindow</code> or HTML5 fullscreen APIs.
     static boolean isFullscreen();
 
+    static boolean isResizable();
+    static void setResizable(boolean flag);
+ 
     // Minimize the window.
     static void minimize();
 
@@ -474,6 +488,8 @@
   interface Events {
     // Fired when the window is resized.
     [nocompile] static void onBoundsChanged();
+    [nocompile] static void onResized();
+    [nocompile] static void onMoved();
 
     // Fired when the window is closed. Note, this should be listened to from
     // a window other than the window being closed, for example from the
