diff -r -u --color up/chromium/extensions/browser/extension_navigation_ui_data.cc nw/chromium/extensions/browser/extension_navigation_ui_data.cc
--- up/chromium/extensions/browser/extension_navigation_ui_data.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/extension_navigation_ui_data.cc	2023-02-24 20:54:13.708489632 +0000
@@ -4,6 +4,8 @@
 
 #include "extensions/browser/extension_navigation_ui_data.h"
 
+#include "content/public/browser/render_process_host.h"
+
 #include "base/memory/ptr_util.h"
 #include "content/public/browser/navigation_handle.h"
 #include "content/public/browser/render_process_host.h"
@@ -95,6 +97,7 @@
   copy->web_view_instance_id_ = web_view_instance_id_;
   copy->web_view_rules_registry_id_ = web_view_rules_registry_id_;
   copy->parent_routing_id_ = parent_routing_id_;
+  copy->web_view_embedder_process_id_ = web_view_embedder_process_id_;
   return copy;
 }
 
@@ -123,9 +126,12 @@
     is_web_view_ = true;
     web_view_instance_id_ = web_view->view_instance_id();
     web_view_rules_registry_id_ = web_view->rules_registry_id();
+    web_view_embedder_process_id_ =
+        web_view->owner_web_contents()->GetPrimaryMainFrame()->GetProcess()->GetID();
   } else {
     is_web_view_ = false;
-    web_view_instance_id_ = web_view_rules_registry_id_ = 0;
+    web_view_instance_id_ = web_view_rules_registry_id_ =
+        web_view_embedder_process_id_ = 0;
   }
 }
 
