diff -r -u --color up/chromium/extensions/browser/verified_contents.cc nw/chromium/extensions/browser/verified_contents.cc
--- up/chromium/extensions/browser/verified_contents.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/verified_contents.cc	2023-02-24 20:12:10.268872894 +0000
@@ -43,6 +43,7 @@
 const char kTreeHashPerFile[] = "treehash per file";
 const char kTreeHash[] = "treehash";
 const char kWebstoreKId[] = "webstore";
+const char kNWJSKId[] = "nwjs";
 
 // Helper function to iterate over a list of dictionaries, returning the
 // dictionary that has |key| -> |value| in it, if any, or null.
@@ -107,7 +108,21 @@
     base::StringPiece contents) {
   // Note: VerifiedContents constructor is private.
   auto verified_contents = base::WrapUnique(new VerifiedContents(public_key));
-  std::string payload;
+  std::string payload, manifest;
+#if 0 //NWJS: fixme
+  std::string manifest_contents;
+  base::FilePath manifest_path = path.DirName().AppendASCII("package.json");
+  if (!base::ReadFileToString(manifest_path, &manifest_contents))
+    return nullptr;
+#endif
+  if (!verified_contents->GetPayload(contents, &manifest, "manifest"))
+    return nullptr;
+#if 0
+  if (manifest != manifest_contents) {
+    LOG(FATAL) << "manifest mismatch: " << manifest;
+    return nullptr;
+  }
+#endif
   if (!verified_contents->GetPayload(contents, &payload))
     return nullptr;
 
@@ -242,7 +257,8 @@
 // the extension's key too (eg for non-webstore hosted extensions such as
 // enterprise installs).
 bool VerifiedContents::GetPayload(base::StringPiece contents,
-                                  std::string* payload) {
+                                  std::string* payload,
+                                  const char* manifest) {
   absl::optional<base::Value> top_list = base::JSONReader::Read(contents);
   if (!top_list || !top_list->is_list())
     return false;
@@ -274,6 +290,9 @@
   const base::Value* signature_dict =
       FindDictionaryWithValue(*signatures, kHeaderKidKey, kWebstoreKId);
   if (!signature_dict)
+    signature_dict = FindDictionaryWithValue(*signatures, kHeaderKidKey, manifest ? "manifest" : kNWJSKId);
+
+  if (!signature_dict)
     return false;
 
   const std::string* protected_value =
@@ -288,7 +307,7 @@
     return false;
 
   const std::string* encoded_payload =
-      signed_content->FindStringKey(kPayloadKey);
+    signed_content->FindStringKey(manifest ? "manifest" : kPayloadKey);
   if (!encoded_payload)
     return false;
 
