diff -r -u --color up/chromium/extensions/renderer/resources/guest_view/web_view/web_view_api_methods.js nw/chromium/extensions/renderer/resources/guest_view/web_view/web_view_api_methods.js
--- up/chromium/extensions/renderer/resources/guest_view/web_view/web_view_api_methods.js	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/resources/guest_view/web_view/web_view_api_methods.js	2023-02-24 20:12:10.316873295 +0000
@@ -34,11 +34,13 @@
 
   // Returns audio state.
   'getAudioState',
+  // Return storeId which can be used in chrome.cookies API
+  'getCookieStoreId',
 
   // Returns Chrome's internal process ID for the guest web page's current
   // process.
   'getProcessId',
-
+  'getGuestId',
   // Returns the user agent string used by the webview for guest page requests.
   'getUserAgent',
 
@@ -70,6 +72,8 @@
   // of the data URL.
   'loadDataWithBaseUrl',
 
+  'showDevTools',
+  'inspectElementAt',
   // Prints the contents of the webview.
   'print',
 
