diff -r -u --color up/chromium/extensions/renderer/resources/guest_view/guest_view_container_element.js nw/chromium/extensions/renderer/resources/guest_view/guest_view_container_element.js
--- up/chromium/extensions/renderer/resources/guest_view/guest_view_container_element.js	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/resources/guest_view/guest_view_container_element.js	2023-02-24 20:12:10.316873295 +0000
@@ -64,12 +64,17 @@
     GuestViewContainerElement.prototype.attributeChangedCallback =
         customElementCallbacks.attributeChangedCallback;
 
+    try {
     $CustomElementRegistry.define(
         window.customElements, $String.toLowerCase(elementName),
         containerElementType);
     $Object.defineProperty(window, elementName, {
       value: containerElementType,
     });
+    } catch (e) {
+       // GuestView is being registered in isolated world in
+       // content script sometimes //NWJS#7087
+    }
 
     delete GuestViewContainerElement.prototype.connectedCallback;
     delete GuestViewContainerElement.prototype.disconnectedCallback;
