diff -r -u --color up/v8/include/v8-debug.h nw/v8/include/v8-debug.h
--- up/v8/include/v8-debug.h	2023-02-10 19:02:29.000000000 +0000
+++ nw/v8/include/v8-debug.h	2023-02-19 17:37:43.239622249 +0000
@@ -130,6 +130,8 @@
     kDetailed = kOverview | kIsEval | kIsConstructor | kScriptNameOrSourceURL
   };
 
+  V8_DEPRECATED("Use Isolate version")
+                Local<StackFrame> GetFrame(uint32_t index) const;
   /**
    * Returns a StackFrame at a particular index.
    */
