diff -r -u --color up/chromium/net/cookies/cookie_monster.cc nw/chromium/net/cookies/cookie_monster.cc
--- up/chromium/net/cookies/cookie_monster.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/net/cookies/cookie_monster.cc	2023-02-24 06:16:31.320678360 +0000
@@ -581,6 +581,7 @@
 }
 
 const char* const CookieMonster::kDefaultCookieableSchemes[] = {"http", "https",
+                                                                "chrome-extension",
                                                                 "ws", "wss"};
 const int CookieMonster::kDefaultCookieableSchemesCount =
     std::size(kDefaultCookieableSchemes);
