diff -r -u --color up/chromium/ppapi/shared_impl/media_stream_buffer_manager.h nw/chromium/ppapi/shared_impl/media_stream_buffer_manager.h
--- up/chromium/ppapi/shared_impl/media_stream_buffer_manager.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/ppapi/shared_impl/media_stream_buffer_manager.h	2023-02-24 06:16:31.896712416 +0000
@@ -9,6 +9,7 @@
 
 #include <memory>
 #include <vector>
+#include <ostream>
 
 #include "base/compiler_specific.h"
 #include "base/containers/circular_deque.h"
