diff -r -u --color up/chromium/printing/buildflags/buildflags.gni nw/chromium/printing/buildflags/buildflags.gni
--- up/chromium/printing/buildflags/buildflags.gni	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/printing/buildflags/buildflags.gni	2023-02-24 20:54:16.368475487 +0000
@@ -44,7 +44,7 @@
 
   # Enable snapshotting a page when printing for its content to be analyzed for
   # sensitive content by enterprise users.
-  enable_print_content_analysis = is_chromeos || is_win || is_linux || is_mac
+  enable_print_content_analysis = false #is_chromeos || is_win || is_linux || is_mac
 }
 
 declare_args() {
