diff -r -u --color up/chromium/services/network/BUILD.gn nw/chromium/services/network/BUILD.gn
--- up/chromium/services/network/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/services/network/BUILD.gn	2023-02-24 06:16:32.228732029 +0000
@@ -185,6 +185,8 @@
     "web_bundle/web_bundle_url_loader_factory.h",
     "web_transport.cc",
     "web_transport.h",
+    "../../content/nw/src/policy_cert_verifier.cc",
+    "../../content/nw/src/policy_cert_verifier.h",
   ]
 
   if (is_p2p_enabled) {
