diff -r -u --color up/chromium/services/network/network_service.h nw/chromium/services/network/network_service.h
--- up/chromium/services/network/network_service.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/services/network/network_service.h	2023-02-24 06:16:32.240732737 +0000
@@ -141,6 +141,7 @@
 
   // mojom::NetworkService implementation:
   void SetParams(mojom::NetworkServiceParamsPtr params) override;
+  void SetAdditionalTrustAnchors(const net::CertificateList& anchors) override;
   void StartNetLog(base::File file,
                    net::NetLogCaptureMode capture_mode,
                    base::Value::Dict constants) override;
