diff -r -u --color up/chromium/services/network/public/mojom/network_service.mojom nw/chromium/services/network/public/mojom/network_service.mojom
--- up/chromium/services/network/public/mojom/network_service.mojom	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/services/network/public/mojom/network_service.mojom	2023-02-24 06:16:32.272734628 +0000
@@ -165,6 +165,8 @@
   // Sets the parameters and initializes the service.
   SetParams(NetworkServiceParams params);
 
+  SetAdditionalTrustAnchors(array<X509Certificate> trust_anchors);
+
   // Starts observing the NetLog event stream and writing entries to |file|.
   // |constants| is a legend used for decoding constant values in the log; it
   // will be merged with the |GetNetConstants()| dictionary before being passed
