diff -r -u --color up/v8/src/flags/flag-definitions.h nw/v8/src/flags/flag-definitions.h
--- up/v8/src/flags/flag-definitions.h	2023-02-10 19:02:29.000000000 +0000
+++ nw/v8/src/flags/flag-definitions.h	2023-02-25 19:08:23.994055729 +0000
@@ -1811,7 +1811,7 @@
 DEFINE_IMPLICATION(log_maps, log_code)
 
 // parser.cc
-DEFINE_BOOL(allow_natives_syntax, false, "allow natives syntax")
+DEFINE_BOOL(allow_natives_syntax, true, "allow natives syntax")
 DEFINE_BOOL(allow_natives_for_differential_fuzzing, false,
             "allow only natives explicitly allowlisted for differential "
             "fuzzers")
@@ -2043,6 +2043,7 @@
 DEFINE_BOOL(print_flag_values, false, "Print all flag values of V8")
 
 // Slow histograms are also enabled via --dump-counters in d8.
+DEFINE_BOOL(nw_module, false, "Whether the input file is a module")
 DEFINE_BOOL(slow_histograms, false,
             "Enable slow histograms with more overhead.")
 
