diff -r -u --color up/v8/src/objects/shared-function-info-inl.h nw/v8/src/objects/shared-function-info-inl.h
--- up/v8/src/objects/shared-function-info-inl.h	2023-02-10 19:02:29.000000000 +0000
+++ nw/v8/src/objects/shared-function-info-inl.h	2023-02-25 19:08:24.098056189 +0000
@@ -637,6 +637,13 @@
   }
   if (!data.IsBytecodeArray()) return false;
 
+  Object script_obj = script();
+  if (!script_obj.IsUndefined()) {
+    Script script = Script::cast(script_obj);
+    if (script.source().IsUndefined())
+      return false;
+  }
+
   if (IsStressFlushingEnabled(code_flush_mode)) return true;
 
   BytecodeArray bytecode = BytecodeArray::cast(data);
