diff -r -u --color up/v8/src/snapshot/deserializer.h nw/v8/src/snapshot/deserializer.h
--- up/v8/src/snapshot/deserializer.h	2023-02-10 19:02:29.000000000 +0000
+++ nw/v8/src/snapshot/deserializer.h	2023-02-19 17:37:43.639623216 +0000
@@ -112,6 +112,8 @@
 
   Handle<HeapObject> ReadObject();
 
+  bool valid_ = true;
+
  private:
   friend class DeserializerRelocInfoVisitor;
   // A circular queue of hot objects. This is added to in the same order as in
