diff -r -u --color up/v8/src/snapshot/object-deserializer.cc nw/v8/src/snapshot/object-deserializer.cc
--- up/v8/src/snapshot/object-deserializer.cc	2023-02-10 19:02:29.000000000 +0000
+++ nw/v8/src/snapshot/object-deserializer.cc	2023-02-19 17:37:43.643623226 +0000
@@ -34,6 +34,8 @@
 
 MaybeHandle<HeapObject> ObjectDeserializer::Deserialize() {
   DCHECK(deserializing_user_code());
+  if (!valid_)
+    return MaybeHandle<HeapObject>();
   HandleScope scope(isolate());
   Handle<HeapObject> result;
   {
Only in nw/v8/src: v8.gyp
