diff -r -u --color up/chromium/third_party/blink/public/web/web_navigation_policy.h nw/chromium/third_party/blink/public/web/web_navigation_policy.h
--- up/chromium/third_party/blink/public/web/web_navigation_policy.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/public/web/web_navigation_policy.h	2023-02-24 20:54:17.504469071 +0000
@@ -34,6 +34,7 @@
 namespace blink {
 
 enum WebNavigationPolicy {
+  kWebNavigationPolicyIgnore,
   kWebNavigationPolicyDownload,
   kWebNavigationPolicyCurrentTab,
   kWebNavigationPolicyNewBackgroundTab,
