diff -r -u --color up/chromium/third_party/blink/renderer/core/html/forms/chooser_only_temporal_input_type_view.cc nw/chromium/third_party/blink/renderer/core/html/forms/chooser_only_temporal_input_type_view.cc
--- up/chromium/third_party/blink/renderer/core/html/forms/chooser_only_temporal_input_type_view.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/forms/chooser_only_temporal_input_type_view.cc	2023-02-24 06:16:33.560810716 +0000
@@ -24,6 +24,7 @@
  */
 
 #include "third_party/blink/renderer/core/html/forms/chooser_only_temporal_input_type_view.h"
+#include "third_party/blink/renderer/core/frame/local_frame.h"
 
 #include "third_party/blink/renderer/core/dom/document.h"
 #include "third_party/blink/renderer/core/dom/events/event.h"
@@ -57,7 +58,7 @@
 void ChooserOnlyTemporalInputTypeView::HandleDOMActivateEvent(Event& event) {
   Document& document = GetElement().GetDocument();
   if (GetElement().IsDisabledOrReadOnly() || !GetElement().GetLayoutObject() ||
-      !LocalFrame::HasTransientUserActivation(document.GetFrame()) ||
+      (!LocalFrame::HasTransientUserActivation(document.GetFrame()) && !document.GetFrame()->isNodeJS()) ||
       GetElement().OpenShadowRoot())
     return;
 
