diff -r -u --color up/chromium/third_party/blink/renderer/core/frame/local_frame_client_impl.cc nw/chromium/third_party/blink/renderer/core/frame/local_frame_client_impl.cc
--- up/chromium/third_party/blink/renderer/core/frame/local_frame_client_impl.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/frame/local_frame_client_impl.cc	2023-02-24 06:16:33.524808589 +0000
@@ -170,6 +170,14 @@
 LocalFrameClientImpl::LocalFrameClientImpl(WebLocalFrameImpl* frame)
     : web_frame_(frame) {}
 
+void LocalFrameClientImpl::willHandleNavigationPolicy(const ResourceRequest& request, NavigationPolicy* policy, WebString* manifest, bool new_win)
+{
+  if (web_frame_->Client()) {
+    WrappedResourceRequest webreq(request);
+    web_frame_->Client()->willHandleNavigationPolicy(web_frame_, webreq, (WebNavigationPolicy*)policy, manifest, new_win);
+  }
+}
+
 LocalFrameClientImpl::~LocalFrameClientImpl() = default;
 
 void LocalFrameClientImpl::Trace(Visitor* visitor) const {
