diff -r -u --color up/chromium/third_party/blink/renderer/core/dom/document.cc nw/chromium/third_party/blink/renderer/core/dom/document.cc
--- up/chromium/third_party/blink/renderer/core/dom/document.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/dom/document.cc	2023-02-24 20:12:14.096900474 +0000
@@ -6043,14 +6043,14 @@
   if (!GetFrame())
     return scoped_refptr<const SecurityOrigin>();
 
-  return GetFrame()->Tree().Top().GetSecurityContext()->GetSecurityOrigin();
+  return GetFrame()->Tree().FindFrameByName("_top", true)->GetSecurityContext()->GetSecurityOrigin();
 }
 
 net::SiteForCookies Document::SiteForCookies() const {
   if (!GetFrame())
     return net::SiteForCookies();
 
-  Frame& top = GetFrame()->Tree().Top();
+  Frame& top = *GetFrame()->Tree().FindFrameByName("_top", true);
   const SecurityOrigin* origin = top.GetSecurityContext()->GetSecurityOrigin();
   // TODO(yhirano): Ideally |origin| should not be null here.
   if (!origin)
@@ -6074,6 +6074,8 @@
   }
 
   while (current_frame) {
+    if (current_frame->isNwFakeTop())
+      break;
     const url::Origin cur_security_origin =
         current_frame->GetSecurityContext()->GetSecurityOrigin()->ToUrlOrigin();
     if (!candidate.CompareWithFrameTreeOriginAndRevise(cur_security_origin))
