diff -r -u --color up/chromium/third_party/blink/renderer/core/html/html_frame_owner_element.h nw/chromium/third_party/blink/renderer/core/html/html_frame_owner_element.h
--- up/chromium/third_party/blink/renderer/core/html/html_frame_owner_element.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/html_frame_owner_element.h	2023-02-24 06:16:33.588812370 +0000
@@ -113,6 +113,15 @@
   mojom::blink::ScrollbarMode ScrollbarMode() const override {
     return mojom::blink::ScrollbarMode::kAuto;
   }
+
+  AtomicString nwuseragent() const override {
+    return getAttribute(html_names::kNwuseragentAttr);
+  }
+
+  bool nwfaketop() const override {
+    return hasAttribute(html_names::kNwfaketopAttr);
+  }
+
   int MarginWidth() const override { return -1; }
   int MarginHeight() const override { return -1; }
   bool AllowFullscreen() const override { return false; }
