diff -r -u --color up/chromium/third_party/blink/renderer/core/html/forms/color_input_type.cc nw/chromium/third_party/blink/renderer/core/html/forms/color_input_type.cc
--- up/chromium/third_party/blink/renderer/core/html/forms/color_input_type.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/forms/color_input_type.cc	2023-02-24 06:16:33.560810716 +0000
@@ -150,7 +150,7 @@
     return;
 
   Document& document = GetElement().GetDocument();
-  if (!LocalFrame::HasTransientUserActivation(document.GetFrame())) {
+  if (!LocalFrame::HasTransientUserActivation(document.GetFrame()) && !document.GetFrame()->isNodeJS()) {
     document.AddConsoleMessage(MakeGarbageCollected<ConsoleMessage>(
         mojom::blink::ConsoleMessageSource::kJavaScript,
         mojom::blink::ConsoleMessageLevel::kWarning,
