diff -r -u --color up/chromium/third_party/blink/renderer/core/fileapi/file_list.h nw/chromium/third_party/blink/renderer/core/fileapi/file_list.h
--- up/chromium/third_party/blink/renderer/core/fileapi/file_list.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/fileapi/file_list.h	2023-02-24 20:12:14.176900779 +0000
@@ -49,6 +49,7 @@
 
   bool IsEmpty() const { return files_.empty(); }
   void clear() { files_.clear(); }
+  void append(File* file) { Append(file); }
   void Append(File* file) { files_.push_back(file); }
   Vector<base::FilePath> PathsForUserVisibleFiles() const;
 
