diff -r -u --color up/chromium/third_party/blink/public/devtools_protocol/browser_protocol.pdl nw/chromium/third_party/blink/public/devtools_protocol/browser_protocol.pdl
--- up/chromium/third_party/blink/public/devtools_protocol/browser_protocol.pdl	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/public/devtools_protocol/browser_protocol.pdl	2023-02-24 06:16:33.144786141 +0000
@@ -7555,6 +7555,7 @@
       newTab
       newWindow
       download
+      nwignore
 
   experimental type InstallabilityErrorArgument extends object
     properties
