diff -r -u --color up/chromium/third_party/blink/renderer/core/workers/threaded_worklet_messaging_proxy.cc nw/chromium/third_party/blink/renderer/core/workers/threaded_worklet_messaging_proxy.cc
--- up/chromium/third_party/blink/renderer/core/workers/threaded_worklet_messaging_proxy.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/workers/threaded_worklet_messaging_proxy.cc	2023-02-24 20:54:18.084465795 +0000
@@ -72,7 +72,7 @@
                  : frame_client->UserAgent());
 
   auto global_scope_creation_params =
-      std::make_unique<GlobalScopeCreationParams>(
+    std::make_unique<GlobalScopeCreationParams>(false, std::string(),
           window->Url(), mojom::blink::ScriptType::kModule, global_scope_name,
           user_agent, frame_client->UserAgentMetadata(),
           frame_client->CreateWorkerFetchContext(),
