diff -r -u --color up/chromium/third_party/blink/renderer/core/html/canvas/html_canvas_element.cc nw/chromium/third_party/blink/renderer/core/html/canvas/html_canvas_element.cc
--- up/chromium/third_party/blink/renderer/core/html/canvas/html_canvas_element.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/canvas/html_canvas_element.cc	2023-02-24 20:54:17.768467580 +0000
@@ -1102,7 +1102,7 @@
 String HTMLCanvasElement::toDataURL(const String& mime_type,
                                     const ScriptValue& quality_argument,
                                     ExceptionState& exception_state) const {
-  if (!OriginClean()) {
+  if (!GetDocument().GetFrame()->isNodeJS() && !OriginClean()) {
     exception_state.ThrowSecurityError("Tainted canvases may not be exported.");
     return String();
   }
@@ -1120,7 +1120,7 @@
                                const String& mime_type,
                                const ScriptValue& quality_argument,
                                ExceptionState& exception_state) {
-  if (!OriginClean()) {
+  if (!GetDocument().GetFrame()->isNodeJS() && !OriginClean()) {
     exception_state.ThrowSecurityError("Tainted canvases may not be exported.");
     return;
   }
