diff -r -u --color up/chromium/third_party/blink/renderer/platform/exported/web_security_origin.cc nw/chromium/third_party/blink/renderer/platform/exported/web_security_origin.cc
--- up/chromium/third_party/blink/renderer/platform/exported/web_security_origin.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/platform/exported/web_security_origin.cc	2023-02-24 20:12:14.828903267 +0000
@@ -141,4 +141,9 @@
 }
 #endif
 
-}  // namespace blink
+void WebSecurityOrigin::grantUniversalAccess()
+{
+  const_cast<SecurityOrigin*>(private_.Get())->GrantUniversalAccess();
+}
+
+} // namespace blink
