diff -r -u --color up/chromium/third_party/zlib/google/zip.cc nw/chromium/third_party/zlib/google/zip.cc
--- up/chromium/third_party/zlib/google/zip.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/zlib/google/zip.cc	2023-02-24 06:16:53.641965948 +0000
@@ -199,7 +199,7 @@
   reader.SetPassword(std::move(options.password));
 
   if (!reader.OpenFromPlatformFile(src_file)) {
-    LOG(ERROR) << "Cannot open ZIP from file handle " << src_file;
+    VLOG(1) << "Cannot open ZIP from file handle " << src_file;
     return false;
   }
 
