diff -r -u --color up/chromium/third_party/blink/renderer/core/page/frame_tree.h nw/chromium/third_party/blink/renderer/core/page/frame_tree.h
--- up/chromium/third_party/blink/renderer/core/page/frame_tree.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/page/frame_tree.h	2023-02-24 20:12:14.408901664 +0000
@@ -68,7 +68,7 @@
   Frame* TraverseNext(const Frame* stay_within = nullptr) const;
 
   // For plugins and tests only.
-  Frame* FindFrameByName(const AtomicString& name) const;
+  Frame* FindFrameByName(const AtomicString& name, bool nw = false) const;
 
   // https://html.spec.whatwg.org/#the-rules-for-choosing-a-browsing-context-given-a-browsing-context-name
   struct FindResult {
