diff -r -u --color up/chromium/third_party/blink/renderer/core/html/forms/html_input_element.idl nw/chromium/third_party/blink/renderer/core/html/forms/html_input_element.idl
--- up/chromium/third_party/blink/renderer/core/html/forms/html_input_element.idl	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/forms/html_input_element.idl	2023-02-24 06:16:33.568811188 +0000
@@ -107,6 +107,9 @@
     // https://html.spec.whatwg.org/multipage/input.html#dom-input-showpicker
     [Measure, RaisesException] void showPicker();
 
+    attribute DOMString nwworkingdir;
+    [Reflect] attribute boolean nwdirectory;
+    attribute DOMString nwsaveas;
     // Non-standard APIs
     [Reflect, MeasureAs=PrefixedDirectoryAttribute] attribute boolean webkitdirectory;
     [Reflect, MeasureAs=IncrementalAttribute] attribute boolean incremental;
