diff -r -u --color up/chromium/third_party/blink/renderer/core/frame/window.idl nw/chromium/third_party/blink/renderer/core/frame/window.idl
--- up/chromium/third_party/blink/renderer/core/frame/window.idl	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/frame/window.idl	2023-02-24 06:16:33.544809771 +0000
@@ -59,10 +59,10 @@
     // other browsing contexts
     [Replaceable, CrossOrigin, CachedAccessor=kWindowProxy] readonly attribute Window frames;
     [Replaceable, CrossOrigin] readonly attribute unsigned long length;
-    [LegacyUnforgeable, CrossOrigin] readonly attribute Window? top;
+    [LegacyUnforgeable, CrossOrigin, Custom=Getter] readonly attribute Window? top;
     // FIXME: opener should be of type any.
     [CrossOrigin, Custom=Setter, ImplementedAs=OpenerWithMetrics] attribute Window opener;
-    [Replaceable, CrossOrigin] readonly attribute Window? parent;
+    [Replaceable, CrossOrigin, Custom=Getter] readonly attribute Window? parent;
     [CheckSecurity=ReturnValue, Custom=Getter] readonly attribute Element? frameElement;
 
     [CallWith=Isolate, RaisesException] Window? open(optional USVString url="", optional DOMString target = "_blank", optional [TreatNullAs=EmptyString] DOMString features = "");
