diff -r -u --color up/chromium/third_party/blink/public/mojom/frame/frame_owner_properties.mojom nw/chromium/third_party/blink/public/mojom/frame/frame_owner_properties.mojom
--- up/chromium/third_party/blink/public/mojom/frame/frame_owner_properties.mojom	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/public/mojom/frame/frame_owner_properties.mojom	2023-02-24 20:12:13.896899710 +0000
@@ -18,8 +18,10 @@
   int32 margin_height;
   // True if the frame allows fullscreen.
   bool allow_fullscreen;
+  bool nwfaketop;
   // True if the frame allows payment request.
   bool allow_payment_request;
+  string nwuseragent;
   // True if the frame is display:none.
   bool is_display_none;
   // The used value of color-scheme for the frame element.
