diff -r -u --color up/chromium/third_party/widevine/cdm/widevine.gni nw/chromium/third_party/widevine/cdm/widevine.gni
--- up/chromium/third_party/widevine/cdm/widevine.gni	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/widevine/cdm/widevine.gni	2023-02-24 06:16:53.509958893 +0000
@@ -12,7 +12,7 @@
   # on Android. For Fuchsia, see crbug.com/1354347.
   # Can be optionally enabled in Chromium on non-Android platforms. Please see
   # //src/third_party/widevine/LICENSE file for details.
-  enable_widevine = (is_chrome_branded && !is_fuchsia) || is_android
+  enable_widevine = true #(is_chrome_branded && !is_fuchsia) || is_android
 }
 
 # Disable Widevine when running linux-chromeos.
