diff -r -u --color up/chromium/third_party/blink/renderer/bindings/core/v8/window_proxy_manager.h nw/chromium/third_party/blink/renderer/bindings/core/v8/window_proxy_manager.h
--- up/chromium/third_party/blink/renderer/bindings/core/v8/window_proxy_manager.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/bindings/core/v8/window_proxy_manager.h	2023-02-24 20:54:17.528468935 +0000
@@ -42,6 +42,11 @@
   void ReleaseGlobalProxies(GlobalProxyVector&);
   void SetGlobalProxies(const GlobalProxyVector&);
 
+  bool ContextNotReady(DOMWrapperWorld& world) {
+    WindowProxy* window_proxy = WindowProxyMaybeUninitialized(world);
+    return window_proxy->ContextNotReady();
+  }
+
   WindowProxy* GetWindowProxy(DOMWrapperWorld& world) {
     WindowProxy* window_proxy = WindowProxyMaybeUninitialized(world);
     window_proxy->InitializeIfNeeded();
