diff -r -u --color up/chromium/third_party/blink/renderer/core/frame/web_local_frame_impl.h nw/chromium/third_party/blink/renderer/core/frame/web_local_frame_impl.h
--- up/chromium/third_party/blink/renderer/core/frame/web_local_frame_impl.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/frame/web_local_frame_impl.h	2023-02-24 06:16:33.544809771 +0000
@@ -510,6 +510,16 @@
 
   WebFrameWidgetImpl* FrameWidgetImpl() { return frame_widget_; }
 
+
+  void setNodeJS(bool node) override { GetFrame()->setNodeJS(node); }
+  bool isNodeJS() const override { return GetFrame()->isNodeJS(); }
+  bool isNwDisabledChildFrame() const override { return GetFrame()->isNwDisabledChildFrame(); }
+  bool isNwFakeTop() const override { return GetFrame()->isNwFakeTop(); }
+  void setDevtoolsJail(WebFrame* iframe) override {
+    GetFrame()->setDevtoolsJail(iframe ? static_cast<const WebLocalFrameImpl*>(iframe)->GetFrame() : nullptr);
+  }
+  WebFrame* getDevtoolsJail() override { return FromFrame((blink::LocalFrame*)GetFrame()->getDevtoolsJail()); }
+
   WebTextCheckClient* GetTextCheckerClient() const {
     return text_check_client_;
   }
