diff -r -u --color up/chromium/ui/accessibility/platform/atk_util_auralinux.cc nw/chromium/ui/accessibility/platform/atk_util_auralinux.cc
--- up/chromium/ui/accessibility/platform/atk_util_auralinux.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/accessibility/platform/atk_util_auralinux.cc	2023-02-24 20:54:37.164357410 +0000
@@ -1,6 +1,7 @@
 // Copyright 2015 The Chromium Authors
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
+#pragma clang diagnostic ignored "-Wunused-function"
 
 #include <atk/atk.h>
 #include <map>
@@ -122,6 +123,7 @@
 }
 
 void AtkUtilAuraLinux::InitializeAsync() {
+#if 0
   static bool initialized = false;
 
   if (initialized || !ShouldEnableAccessibility())
@@ -133,6 +135,7 @@
   g_type_class_unref(g_type_class_ref(atk_util_auralinux_get_type()));
 
   PlatformInitializeAsync();
+#endif
 }
 
 void AtkUtilAuraLinux::InitializeForTesting() {
