diff -r -u --color up/chromium/ui/accelerated_widget_mac/BUILD.gn nw/chromium/ui/accelerated_widget_mac/BUILD.gn
--- up/chromium/ui/accelerated_widget_mac/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/accelerated_widget_mac/BUILD.gn	2023-02-24 20:54:37.064357981 +0000
@@ -36,6 +36,7 @@
     "//media",
     "//skia",
     "//ui/base",
+    "//ui/display",
     "//ui/events",
     "//ui/events:events_base",
     "//ui/gfx",
