diff -r -u --color up/chromium/ui/gfx/BUILD.gn nw/chromium/ui/gfx/BUILD.gn
--- up/chromium/ui/gfx/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/ui/gfx/BUILD.gn	2023-02-24 06:16:55.982092739 +0000
@@ -393,6 +393,7 @@
 
   if (is_mac) {
     frameworks = [
+      "ApplicationServices.framework",
       "AppKit.framework",
       "CoreFoundation.framework",
       "CoreGraphics.framework",
