diff -r -u --color up/chromium/ui/ozone/public/platform_global_shortcut_listener.h nw/chromium/ui/ozone/public/platform_global_shortcut_listener.h
--- up/chromium/ui/ozone/public/platform_global_shortcut_listener.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/ui/ozone/public/platform_global_shortcut_listener.h	2023-02-24 06:16:56.146102453 +0000
@@ -52,11 +52,13 @@
   virtual bool RegisterAccelerator(KeyboardCode key_code,
                                    bool is_alt_down,
                                    bool is_ctrl_down,
-                                   bool is_shift_down) = 0;
+                                   bool is_shift_down,
+                                   bool is_cmd_down) = 0;
   virtual void UnregisterAccelerator(KeyboardCode key_code,
                                      bool is_alt_down,
                                      bool is_ctrl_down,
-                                     bool is_shift_down) = 0;
+                                     bool is_shift_down,
+                                     bool is_cmd_down) = 0;
 
  protected:
   PlatformGlobalShortcutListenerDelegate* delegate() { return delegate_; }
