diff -r -u --color up/chromium/ui/ozone/platform/x11/x11_global_shortcut_listener_ozone.h nw/chromium/ui/ozone/platform/x11/x11_global_shortcut_listener_ozone.h
--- up/chromium/ui/ozone/platform/x11/x11_global_shortcut_listener_ozone.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/ozone/platform/x11/x11_global_shortcut_listener_ozone.h	2023-02-24 20:54:39.068346453 +0000
@@ -28,11 +28,13 @@
   bool RegisterAccelerator(KeyboardCode key_code,
                            bool is_alt_down,
                            bool is_ctrl_down,
-                           bool is_shift_down) override;
+                           bool is_shift_down,
+                           bool is_cmd_down) override;
   void UnregisterAccelerator(KeyboardCode key_code,
                              bool is_alt_down,
                              bool is_ctrl_down,
-                             bool is_shift_down) override;
+                             bool is_shift_down,
+                             bool is_cmd_down) override;
 
   // ui::XGlobalShortcutListener:
   void OnKeyPressed(KeyboardCode key_code,
