diff -r -u --color up/chromium/ui/views/win/hwnd_message_handler_delegate.h nw/chromium/ui/views/win/hwnd_message_handler_delegate.h
--- up/chromium/ui/views/win/hwnd_message_handler_delegate.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/views/win/hwnd_message_handler_delegate.h	2023-02-24 20:54:39.224345539 +0000
@@ -113,6 +113,10 @@
 
   virtual gfx::NativeViewAccessible GetNativeViewAccessible() = 0;
 
+  // on windows, maximizing sometime is sent through WM_SIZE, not
+  // WM_SYSCOMMAND, see node-webkit#753
+  virtual bool ShouldHandleOnSize() const = 0;
+
   // TODO(beng): Investigate migrating these methods to On* prefixes once
   // HWNDMessageHandler is the WindowImpl.
 
@@ -137,6 +141,8 @@
   // true if the command was handled.
   virtual bool HandleCommand(int command) = 0;
 
+  virtual bool HandleSize(UINT param, const gfx::Size& size) = 0;
+
   // Called when an accelerator is invoked.
   virtual void HandleAccelerator(const ui::Accelerator& accelerator) = 0;
 
