diff --git a/deps/uv/src/unix/getaddrinfo.c b/deps/uv/src/unix/getaddrinfo.c
index 77337ace..8ddb6fca 100644
--- a/deps/uv/src/unix/getaddrinfo.c
+++ b/deps/uv/src/unix/getaddrinfo.c
@@ -87,7 +87,7 @@ int uv__getaddrinfo_translate_error(int sys_err) {
   case EAI_SYSTEM: return UV__ERR(errno);
 #endif
   }
-  assert(!"unknown EAI_* error code");
+  assert(!(int)("unknown EAI_* error code"));
   abort();
 #ifndef __SUNPRO_C
   return 0;  /* Pacify compiler. */
