diff --git a/deps/uv/src/unix/fs.c b/deps/uv/src/unix/fs.c
index 362c36c6..6c794352 100644
--- a/deps/uv/src/unix/fs.c
+++ b/deps/uv/src/unix/fs.c
@@ -25,6 +25,8 @@
  * return -1 with errno set. The dispatcher in uv__fs_work() takes care of
  * getting the errno to the right place (req->result or as the return value.)
  */
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
 
 #include "uv.h"
 #include "internal.h"
@@ -294,15 +296,13 @@ static void uv__mkostemp_initonce(void) {
   /* z/os doesn't have RTLD_DEFAULT but that's okay
    * because it doesn't have mkostemp(O_CLOEXEC) either.
    */
-#ifdef RTLD_DEFAULT
-  uv__mkostemp = (int (*)(char*, int)) dlsym(RTLD_DEFAULT, "mkostemp");
+  uv__mkostemp = (int (*)(char*, int)) dlsym(((void *) 0), "mkostemp");
 
   /* We don't care about errors, but we do want to clean them up.
    * If there has been no error, then dlerror() will just return
    * NULL.
    */
   dlerror();
-#endif  /* RTLD_DEFAULT */
 }
 
 
@@ -2243,3 +2243,6 @@ int uv_fs_statfs(uv_loop_t* loop,
 int uv_fs_get_system_error(const uv_fs_t* req) {
   return -req->result;
 }
+
+#pragma clang diagnostic pop
+
