diff --git a/deps/cares/cares.gyp b/deps/cares/cares.gyp
index 74f0e78e..4de78cbe 100644
--- a/deps/cares/cares.gyp
+++ b/deps/cares/cares.gyp
@@ -1,4 +1,7 @@
 {
+  'variables': {
+    'library%': 'static_library',
+  },
   'target_defaults': {
     'conditions': [
       ['OS!="win"', {
@@ -126,6 +129,7 @@
             '_WINSOCK_DEPRECATED_NO_WARNINGS',
           ],
           'include_dirs': [ 'config/win32' ],
+          'msvs_disabled_warnings': [4133],
           'sources': [
             'src/lib/config-win32.h',
             'src/lib/windows_port.c',
