diff --git a/deps/cares/src/lib/ares_process.c b/deps/cares/src/lib/ares_process.c
index 87329e35..f7a52e01 100644
--- a/deps/cares/src/lib/ares_process.c
+++ b/deps/cares/src/lib/ares_process.c
@@ -509,7 +509,7 @@ static void read_udp_packets(ares_channel channel, fd_set *read_fds,
           else
             fromlen = sizeof(from.sa6);
           count = socket_recvfrom(channel, server->udp_socket, (void *)buf,
-                                  sizeof(buf), 0, &from.sa, &fromlen);
+                                  sizeof(buf), 0, &from.sa, (socklen_t *)&fromlen);
         }
 
         if (count == -1 && try_again(SOCKERRNO))
