diff --git a/lib/internal/source_map/source_map_cache.js b/lib/internal/source_map/source_map_cache.js
index ee8de441..aface954 100644
--- a/lib/internal/source_map/source_map_cache.js
+++ b/lib/internal/source_map/source_map_cache.js
@@ -56,6 +56,8 @@ function getSourceMapsEnabled() {
 }
 
 function setSourceMapsEnabled(val) {
+  if (val === undefined)
+    val = false;
   validateBoolean(val, 'val');
 
   const {
