diff --git a/lib/internal/modules/cjs/helpers.js b/lib/internal/modules/cjs/helpers.js
index 3c49ab0a..cb70ddf8 100644
--- a/lib/internal/modules/cjs/helpers.js
+++ b/lib/internal/modules/cjs/helpers.js
@@ -35,12 +35,7 @@ const noAddons = getOptionValue('--no-addons');
 const addonConditions = noAddons ? [] : ['node-addons'];
 
 // TODO: Use this set when resolving pkg#exports conditions in loader.js.
-const cjsConditions = new SafeSet([
-  'require',
-  'node',
-  ...addonConditions,
-  ...userConditions,
-]);
+const cjsConditions = new SafeSet(['require', 'node']);
 
 function loadBuiltinModule(filename, request) {
   const mod = BuiltinModule.map.get(filename);
