diff --git a/lib/internal/bootstrap/browser.js b/lib/internal/bootstrap/browser.js
index d0c01ca2..fd83b1c0 100644
--- a/lib/internal/bootstrap/browser.js
+++ b/lib/internal/bootstrap/browser.js
@@ -1,5 +1,7 @@
 'use strict';
 
+if (process.__nwjs) {
+} else {
 const {
   ObjectDefineProperty,
   globalThis,
@@ -178,3 +180,5 @@ exposeInterface(globalThis, 'TextEncoderStream', TextEncoderStream);
 exposeInterface(globalThis, 'TextDecoderStream', TextDecoderStream);
 exposeInterface(globalThis, 'CompressionStream', CompressionStream);
 exposeInterface(globalThis, 'DecompressionStream', DecompressionStream);
+
+} //not nwjs
