diff --git a/src/node_builtins.cc b/src/node_builtins.cc
index 4a8e5f2d..b582d618 100644
--- a/src/node_builtins.cc
+++ b/src/node_builtins.cc
@@ -255,11 +255,12 @@ MaybeLocal<Function> BuiltinLoader::LookupAndCompileInternal(
     }
   }
 
-  const bool has_cache = cached_data != nullptr;
+  const bool has_cache = false; //cached_data != nullptr;
+
   ScriptCompiler::CompileOptions options =
       has_cache ? ScriptCompiler::kConsumeCodeCache
                 : ScriptCompiler::kEagerCompile;
-  ScriptCompiler::Source script_source(source, origin, cached_data);
+  ScriptCompiler::Source script_source(source, origin, nullptr);
 
   per_process::Debug(DebugCategory::CODE_CACHE,
                      "Compiling %s %s code cache\n",
@@ -306,6 +307,7 @@ MaybeLocal<Function> BuiltinLoader::LookupAndCompileInternal(
                                                                : "is accepted");
   }
 
+#if 0
   // Generate new cache for next compilation
   std::unique_ptr<ScriptCompiler::CachedData> new_cached_data(
       ScriptCompiler::CreateCodeCacheForFunction(fun));
@@ -322,6 +324,7 @@ MaybeLocal<Function> BuiltinLoader::LookupAndCompileInternal(
       it->second.reset(new_cached_data.release());
     }
   }
+#endif
 
   return scope.Escape(fun);
 }
