diff --git a/src/node_metadata.cc b/src/node_metadata.cc
index a221dcde..f42461d3 100644
--- a/src/node_metadata.cc
+++ b/src/node_metadata.cc
@@ -16,7 +16,8 @@
 #endif
 #endif  // HAVE_OPENSSL
 
-#ifdef OPENSSL_INFO_QUIC
+#if 0
+//#ifdef OPENSSL_INFO_QUIC
 #include <ngtcp2/version.h>
 #include <nghttp3/version.h>
 #endif
@@ -99,7 +100,7 @@ Metadata::Versions::Versions() {
   unicode = U_UNICODE_VERSION;
 #endif  // NODE_HAVE_I18N_SUPPORT
 
-#ifdef OPENSSL_INFO_QUIC
+#if 0 //def OPENSSL_INFO_QUIC
   ngtcp2 = NGTCP2_VERSION;
   nghttp3 = NGHTTP3_VERSION;
 #endif
