diff --git a/src/node_options.cc b/src/node_options.cc
index bd4ad2f6..53a212d2 100644
--- a/src/node_options.cc
+++ b/src/node_options.cc
@@ -895,13 +895,14 @@ PerProcessOptionsParser::PerProcessOptionsParser(
             &PerProcessOptions::trace_sigint,
             kAllowedInEnvironment);
 
-  Insert(iop, &PerProcessOptions::get_per_isolate_options);
-
+  //Insert(iop, &PerProcessOptions::get_per_isolate_options);
+#if 0
   AddOption("--node-memory-debug",
             "Run with extra debug checks for memory leaks in Node.js itself",
             NoOp{}, kAllowedInEnvironment);
   Implies("--node-memory-debug", "--debug-arraybuffer-allocations");
   Implies("--node-memory-debug", "--verify-base-objects");
+#endif
 }
 
 inline std::string RemoveBrackets(const std::string& host) {
