diff --git a/src/inspector_js_api.cc b/src/inspector_js_api.cc
index 4d675da4..bd3a5d46 100644
--- a/src/inspector_js_api.cc
+++ b/src/inspector_js_api.cc
@@ -93,6 +93,7 @@ class JSBindingsConnection : public AsyncWrap {
                        : AsyncWrap(env, wrap, PROVIDER_INSPECTORJSBINDING),
                          callback_(env->isolate(), callback) {
     Agent* inspector = env->inspector_agent();
+    if (inspector)
     session_ = ConnectionType::Connect(
         inspector, std::make_unique<JSBindingsSessionDelegate>(env, this));
   }
