diff --git a/src/heap_utils.cc b/src/heap_utils.cc
index 8556b58f..3d003fa4 100644
--- a/src/heap_utils.cc
+++ b/src/heap_utils.cc
@@ -341,9 +341,11 @@ class HeapSnapshotStream : public AsyncWrap,
 };
 
 inline void TakeSnapshot(Environment* env, v8::OutputStream* out) {
+#if 0
   HeapSnapshotPointer snapshot {
       env->isolate()->GetHeapProfiler()->TakeHeapSnapshot() };
   snapshot->Serialize(out, HeapSnapshot::kJSON);
+#endif
 }
 
 }  // namespace
@@ -411,6 +413,7 @@ BaseObjectPtr<AsyncWrap> CreateHeapSnapshotStream(
 }
 
 void CreateHeapSnapshotStream(const FunctionCallbackInfo<Value>& args) {
+#if 0
   Environment* env = Environment::GetCurrent(args);
   HeapSnapshotPointer snapshot {
       env->isolate()->GetHeapProfiler()->TakeHeapSnapshot() };
@@ -419,6 +422,7 @@ void CreateHeapSnapshotStream(const FunctionCallbackInfo<Value>& args) {
       CreateHeapSnapshotStream(env, std::move(snapshot));
   if (stream)
     args.GetReturnValue().Set(stream->object());
+#endif
 }
 
 void TriggerHeapSnapshot(const FunctionCallbackInfo<Value>& args) {
