diff --git a/src/tracing/agent.h b/src/tracing/agent.h
index b542a849..6ee88f56 100644
--- a/src/tracing/agent.h
+++ b/src/tracing/agent.h
@@ -84,7 +84,7 @@ class AgentWriterHandle {
 class Agent {
  public:
   Agent();
-  ~Agent();
+  ~Agent() {};
 
   TracingController* GetTracingController() {
     TracingController* controller = tracing_controller_.get();
@@ -156,8 +156,8 @@ class Agent {
 };
 
 void AgentWriterHandle::reset() {
-  if (agent_ != nullptr)
-    agent_->Disconnect(id_);
+  //if (agent_ != nullptr)
+  //  agent_->Disconnect(id_);
   agent_ = nullptr;
 }
 
