diff --git a/src/node_internals.h b/src/node_internals.h
index 1449d2ac..fad0e80c 100644
--- a/src/node_internals.h
+++ b/src/node_internals.h
@@ -55,6 +55,8 @@ extern uint64_t node_start_time;
 // Forward declaration
 class Environment;
 
+NODE_EXTERN v8::Handle<v8::Value> CallTickCallback(Environment* env, const v8::Handle<v8::Value> ret);
+
 // Convert a struct sockaddr to a { address: '1.2.3.4', port: 1234 } JS object.
 // Sets address and port properties on the info object and returns it.
 // If |info| is omitted, a new object is returned.
@@ -105,6 +107,7 @@ class NodeArrayBufferAllocator : public ArrayBufferAllocator {
   void* AllocateUninitialized(size_t size) override;
   void Free(void* data, size_t size) override;
   void* Reallocate(void* data, size_t old_size, size_t size) override;
+  void Free(void* data, size_t size, AllocationMode mode) override;
   virtual void RegisterPointer(void* data, size_t size) {
     total_mem_usage_.fetch_add(size, std::memory_order_relaxed);
   }
