diff --git a/src/api/hooks.cc b/src/api/hooks.cc
index bf4176cc..a9ebb00c 100644
--- a/src/api/hooks.cc
+++ b/src/api/hooks.cc
@@ -22,6 +22,14 @@ void RunAtExit(Environment* env) {
   env->RunAtExitCallbacks();
 }
 
+void AtExit(void (*cb)(void* arg), void* arg) {
+  //auto env = Environment::GetThreadLocalEnv();
+  thread_ctx_st* tls_ctx = (struct thread_ctx_st*)uv_key_get(&node::thread_ctx_key);
+  if (tls_ctx && tls_ctx->env) {
+    AtExit(tls_ctx->env, cb, arg);
+  }
+}
+
 void AtExit(Environment* env, void (*cb)(void* arg), void* arg) {
   CHECK_NOT_NULL(env);
   env->AtExit(cb, arg);
