diff --git a/src/node_perf.cc b/src/node_perf.cc
index 4cf13a6c..71c651f8 100644
--- a/src/node_perf.cc
+++ b/src/node_perf.cc
@@ -90,10 +90,12 @@ std::ostream& operator<<(std::ostream& o,
 
 void PerformanceState::Mark(PerformanceMilestone milestone, uint64_t ts) {
   this->milestones[milestone] = static_cast<double>(ts);
+#if 0
   TRACE_EVENT_INSTANT_WITH_TIMESTAMP0(
       TRACING_CATEGORY_NODE1(bootstrap),
       GetPerformanceMilestoneName(milestone),
       TRACE_EVENT_SCOPE_THREAD, ts / 1000);
+#endif
 }
 
 // Allows specific Node.js lifecycle milestones to be set from JavaScript
@@ -197,6 +199,7 @@ void GarbageCollectionCleanupHook(void* data) {
 
 static void InstallGarbageCollectionTracking(
     const FunctionCallbackInfo<Value>& args) {
+#if 0
   Environment* env = Environment::GetCurrent(args);
   // Reset current_gc_type to 0
   env->performance_state()->current_gc_type = 0;
@@ -205,6 +208,7 @@ static void InstallGarbageCollectionTracking(
   env->isolate()->AddGCEpilogueCallback(MarkGarbageCollectionEnd,
                                         static_cast<void*>(env));
   env->AddCleanupHook(GarbageCollectionCleanupHook, env);
+#endif
 }
 
 static void RemoveGarbageCollectionTracking(
