diff --git a/src/node_union_bytes.h b/src/node_union_bytes.h
index d296e67f..81312d4d 100644
--- a/src/node_union_bytes.h
+++ b/src/node_union_bytes.h
@@ -79,13 +79,9 @@ class UnionBytes {
   }
   v8::Local<v8::String> ToStringChecked(v8::Isolate* isolate) const {
     if (is_one_byte()) {
-      NonOwningExternalOneByteResource* source =
-          new NonOwningExternalOneByteResource(one_bytes_data(), length_);
-      return v8::String::NewExternalOneByte(isolate, source).ToLocalChecked();
+      return v8::String::NewFromOneByte(isolate, one_bytes_data(), v8::NewStringType::kNormal, length_).ToLocalChecked();
     } else {
-      NonOwningExternalTwoByteResource* source =
-          new NonOwningExternalTwoByteResource(two_bytes_data(), length_);
-      return v8::String::NewExternalTwoByte(isolate, source).ToLocalChecked();
+      return v8::String::NewFromTwoByte(isolate, two_bytes_data(), v8::NewStringType::kNormal, length_).ToLocalChecked();
     }
   }
   size_t length() { return length_; }
