diff --git a/AUTHORS b/AUTHORS
index 21bd1dd57d..af37f8db25 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -60,6 +60,7 @@ Allan Sandfeld Jensen <allan.jensen@qt.io>
 Amos Lim <eui-sang.lim@samsung.com>
 Andreas Anyuru <andreas.anyuru@gmail.com>
 Andrei Kashcha <anvaka@gmail.com>
+Andreu Botella <andreu@andreubotella.com>
 Andrew Paprocki <andrew@ishiboo.com>
 Anna Henningsen <anna@addaleax.net>
 Antoine du Hamel <duhamelantoine1995@gmail.com>
