diff --git a/include/v8-debug.h b/include/v8-debug.h
index 52255f3700..3c523a18f2 100644
--- a/include/v8-debug.h
+++ b/include/v8-debug.h
@@ -130,6 +130,8 @@ class V8_EXPORT StackTrace {
     kDetailed = kOverview | kIsEval | kIsConstructor | kScriptNameOrSourceURL
   };
 
+  V8_DEPRECATED("Use Isolate version")
+                Local<StackFrame> GetFrame(uint32_t index) const;
   /**
    * Returns a StackFrame at a particular index.
    */
