diff --git a/include/v8-initialization.h b/include/v8-initialization.h
index d3e35d6ec5..c24286bae7 100644
--- a/include/v8-initialization.h
+++ b/include/v8-initialization.h
@@ -151,6 +151,7 @@ class V8_EXPORT V8 {
    */
   static bool InitializeICUDefaultLocation(const char* exec_path,
                                            const char* icu_data_file = nullptr);
+  static void* RawICUData();
 
   /**
    * Initialize the external startup data. The embedder only needs to
