diff --git a/include/v8-isolate.h b/include/v8-isolate.h
index 2f8acc8868..68a2226e77 100644
--- a/include/v8-isolate.h
+++ b/include/v8-isolate.h
@@ -209,6 +209,7 @@ using StackState = cppgc::EmbedderStackState;
  */
 class V8_EXPORT Isolate {
  public:
+  ArrayBuffer::Allocator* array_buffer_allocator();
   /**
    * Initial configuration parameters for a new Isolate.
    */
@@ -296,6 +297,7 @@ class V8_EXPORT Isolate {
     OOMErrorCallback oom_error_callback = nullptr;
   };
 
+  void SetArrayBufferAllocatorShared(std::shared_ptr<ArrayBuffer::Allocator> allocator);
   /**
    * Stack-allocated class which sets the isolate for all operations
    * executed within a local scope.
