diff --git a/src/snapshot/object-deserializer.cc b/src/snapshot/object-deserializer.cc
index 6886fb3a5b..278e22f656 100644
--- a/src/snapshot/object-deserializer.cc
+++ b/src/snapshot/object-deserializer.cc
@@ -34,6 +34,8 @@ ObjectDeserializer::DeserializeSharedFunctionInfo(
 
 MaybeHandle<HeapObject> ObjectDeserializer::Deserialize() {
   DCHECK(deserializing_user_code());
+  if (!valid_)
+    return MaybeHandle<HeapObject>();
   HandleScope scope(isolate());
   Handle<HeapObject> result;
   {
diff --git a/src/v8.gyp b/src/v8.gyp
new file mode 100644
index 0000000000..3e8abe2bab
