diff --git a/src/objects/js-array-buffer.tq b/src/objects/js-array-buffer.tq
index 415b15d5c5..c24b57316e 100644
--- a/src/objects/js-array-buffer.tq
+++ b/src/objects/js-array-buffer.tq
@@ -9,6 +9,7 @@ bitfield struct JSArrayBufferFlags extends uint32 {
   is_asm_js_memory: bool: 1 bit;
   is_shared: bool: 1 bit;
   is_resizable_by_js: bool: 1 bit;
+  is_nodejs: bool: 1 bit;
 }
 
 extern class JSArrayBuffer extends JSObjectWithEmbedderSlots {
