diff --git a/src/snapshot/deserializer.h b/src/snapshot/deserializer.h
index b19f881308..ffd4470b85 100644
--- a/src/snapshot/deserializer.h
+++ b/src/snapshot/deserializer.h
@@ -112,6 +112,8 @@ class Deserializer : public SerializerDeserializer {
 
   Handle<HeapObject> ReadObject();
 
+  bool valid_ = true;
+
  private:
   friend class DeserializerRelocInfoVisitor;
   // A circular queue of hot objects. This is added to in the same order as in
