diff --git a/src/init/icu_util.cc b/src/init/icu_util.cc
index 67d349557c..9fd611e158 100644
--- a/src/init/icu_util.cc
+++ b/src/init/icu_util.cc
@@ -36,6 +36,10 @@ void free_icu_data_ptr() { delete[] g_icu_data_ptr; }
 }  // namespace
 #endif
 
+void* RawICUData() {
+  return (void*)g_icu_data_ptr;
+}
+
 bool InitializeICUDefaultLocation(const char* exec_path,
                                   const char* icu_data_file) {
 #if !defined(V8_INTL_SUPPORT)
