diff -r -u --color up/chromium/apps/ui/views/app_window_frame_view.cc nw/chromium/apps/ui/views/app_window_frame_view.cc
--- up/chromium/apps/ui/views/app_window_frame_view.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/apps/ui/views/app_window_frame_view.cc	2023-02-25 19:07:05.073752859 +0000
@@ -53,7 +53,7 @@
   if (draw_frame_) {
     ui::ResourceBundle& rb = ui::ResourceBundle::GetSharedInstance();
     auto close_button = std::make_unique<views::ImageButton>(
-        base::BindRepeating(&views::Widget::Close, base::Unretained(widget_)));
+       base::BindRepeating(&views::Widget::Close, base::Unretained(widget_), false));
     close_button->SetImage(
         views::Button::STATE_NORMAL,
         rb.GetNativeImageNamed(IDR_APP_WINDOW_CLOSE).ToImageSkia());
