diff -r -u --color up/chromium/apps/app_lifetime_monitor.cc nw/chromium/apps/app_lifetime_monitor.cc
--- up/chromium/apps/app_lifetime_monitor.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/apps/app_lifetime_monitor.cc	2023-02-19 17:36:24.023435098 +0000
@@ -65,8 +65,10 @@
 }
 
 void AppLifetimeMonitor::OnAppWindowHidden(AppWindow* app_window) {
+#if 0
   if (!HasOtherVisibleAppWindows(app_window))
     NotifyAppDeactivated(app_window->extension_id());
+#endif
 }
 
 void AppLifetimeMonitor::OnAppWindowShown(AppWindow* app_window,
