diff -r -u --color up/chromium/base/message_loop/message_pump.cc nw/chromium/base/message_loop/message_pump.cc
--- up/chromium/base/message_loop/message_pump.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/base/message_loop/message_pump.cc	2023-02-24 20:54:05.196528656 +0000
@@ -10,6 +10,7 @@
 #include "base/message_loop/message_pump_for_ui.h"
 #include "base/notreached.h"
 #include "build/build_config.h"
+#include "base/message_loop/message_pump_uv.h"
 
 #if BUILDFLAG(IS_APPLE)
 #include "base/message_loop/message_pump_mac.h"
@@ -58,6 +59,9 @@
       return std::make_unique<MessagePumpForUI>();
 #endif
 
+    case MessagePumpType::NODE:
+      return std::unique_ptr<MessagePump>(new MessagePumpUV());
+
     case MessagePumpType::IO:
       return std::make_unique<MessagePumpForIO>();
 
