diff -r -u --color up/chromium/base/threading/thread_restrictions.h nw/chromium/base/threading/thread_restrictions.h
--- up/chromium/base/threading/thread_restrictions.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/base/threading/thread_restrictions.h	2023-02-25 19:07:06.133758902 +0000
@@ -164,6 +164,7 @@
 namespace audio {
 class OutputDevice;
 }
+
 namespace blink {
 class CategorizedWorkerPoolImpl;
 class CategorizedWorkerPoolJob;
@@ -183,6 +184,16 @@
 class NonMainThreadImpl;
 }
 }  // namespace blink
+
+namespace extensions {
+class NwAppSetProxyConfigFunction;
+class ContentVerifier;
+}
+
+namespace nw {
+class Package;
+}
+
 namespace cc {
 class CompletionEvent;
 class TileTaskManagerImpl;
@@ -640,7 +651,7 @@
   friend bool chromeos::system::IsCoreSchedulingAvailable();
   friend int chromeos::system::NumberOfPhysicalCores();
   friend bool disk_cache::CleanupDirectorySync(const base::FilePath&);
-
+ public:
   ScopedAllowBlocking(const Location& from_here = Location::Current());
   ~ScopedAllowBlocking();
 
@@ -847,6 +858,7 @@
   friend class service_manager::ServiceProcessLauncher;
   friend class ui::WindowResizeHelperMac;    // http://crbug.com/902829
   friend class content::TextInputClientMac;  // http://crbug.com/121917
+  friend class extensions::NwAppSetProxyConfigFunction;
 
   ScopedAllowBaseSyncPrimitivesOutsideBlockingScope(
       const Location& from_here = Location::Current());
@@ -963,6 +975,9 @@
   friend class content::BrowserTestBase;
   friend class web::WebMainLoop;
 
+  friend class extensions::ContentVerifier;
+  friend class nw::Package;
+
   static void AllowBlocking() EMPTY_BODY_IF_DCHECK_IS_OFF;
   static void AllowBaseSyncPrimitives() EMPTY_BODY_IF_DCHECK_IS_OFF;
 };
Only in nw/chromium/build: common.gypi
