diff -r -u --color up/chromium/base/allocator/partition_allocator/partition_bucket.cc nw/chromium/base/allocator/partition_allocator/partition_bucket.cc
--- up/chromium/base/allocator/partition_allocator/partition_bucket.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/base/allocator/partition_allocator/partition_bucket.cc	2023-02-25 19:07:05.901757475 +0000
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#pragma clang diagnostic ignored "-Wthread-safety-analysis"
 #include "base/allocator/partition_allocator/partition_bucket.h"
 
 #include <algorithm>
