diff -r -u --color up/chromium/base/message_loop/message_pump_type.h nw/chromium/base/message_loop/message_pump_type.h
--- up/chromium/base/message_loop/message_pump_type.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/base/message_loop/message_pump_type.h	2023-02-19 17:36:24.979437517 +0000
@@ -39,6 +39,7 @@
   // OSX and IOS.
   NS_RUNLOOP,
 #endif  // BUILDFLAG(IS_APPLE)
+  NODE,
 };
 
 }  // namespace base
Only in nw/chromium/base/message_loop: message_pump_uv.cc
Only in nw/chromium/base/message_loop: message_pump_uv.h
Only in nw/chromium/base/message_loop: message_pumpuv_mac.h
Only in nw/chromium/base/message_loop: message_pumpuv_mac.mm
