diff -r -u --color up/chromium/base/message_loop/message_pump_mac.mm nw/chromium/base/message_loop/message_pump_mac.mm
--- up/chromium/base/message_loop/message_pump_mac.mm	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/base/message_loop/message_pump_mac.mm	2023-02-19 17:36:24.979437517 +0000
@@ -596,12 +596,16 @@
     // nesting-deferred work may have accumulated.  Schedule it for processing
     // if appropriate.
     self->MaybeScheduleNestingDeferredWork();
+    self->PreWaitObserverHook();
 
     // Notify the delegate that the loop is about to sleep.
     self->BeforeWait();
   });
 }
 
+void MessagePumpCFRunLoopBase::PreWaitObserverHook() {
+}
+
 // Called from the run loop.
 // static
 void MessagePumpCFRunLoopBase::AfterWaitObserver(CFRunLoopObserverRef observer,
