diff -r -u --color up/chromium/base/i18n/icu_util.cc nw/chromium/base/i18n/icu_util.cc
--- up/chromium/base/i18n/icu_util.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/base/i18n/icu_util.cc	2023-02-25 19:07:05.985757992 +0000
@@ -520,6 +520,28 @@
   return g_icudtl_pf;
 }
 
+const uint8_t* GetRawIcuMemory() {
+  CHECK(g_icudtl_mapped_file);
+  return g_icudtl_mapped_file->data();
+}
+
+bool InitializeICUFromRawMemory(const uint8_t* raw_memory) {
+#if !defined(COMPONENT_BUILD)
+#if DCHECK_IS_ON()
+  DCHECK(!g_check_called_once || !g_called_once);
+  g_called_once = true;
+#endif
+
+  UErrorCode err = U_ZERO_ERROR;
+  udata_setCommonData(const_cast<uint8_t*>(raw_memory), &err);
+  // Never try to load ICU data from files.
+  udata_setFileAccess(UDATA_ONLY_PACKAGES, &err);
+  return err == U_ZERO_ERROR;
+#else
+  return true;
+#endif
+}
+
 void ResetGlobalsForTesting() {
   g_icudtl_pf = kInvalidPlatformFile;
   g_icudtl_mapped_file = nullptr;
