diff -r -u --color up/chromium/build/config/BUILD.gn nw/chromium/build/config/BUILD.gn
--- up/chromium/build/config/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/build/config/BUILD.gn	2023-02-25 19:07:06.221759444 +0000
@@ -93,6 +93,10 @@
     defines += [ "OFFICIAL_BUILD" ]
   }
 
+  if (nwjs_sdk) {
+    defines += [ "NWJS_SDK" ]
+  }
+
   # ==============================================
   #   PLEASE DO NOT ADD MORE THINGS TO THIS LIST
   # ==============================================
