diff -r -u --color up/chromium/build/config/features.gni nw/chromium/build/config/features.gni
--- up/chromium/build/config/features.gni	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/build/config/features.gni	2023-02-24 20:54:05.348527980 +0000
@@ -28,7 +28,7 @@
   #
   # TODO(crbug.com/1314528): Remove chromecast-related conditions and force
   # builds to explicitly specify this.
-  proprietary_codecs = is_chrome_branded || is_castos || is_cast_android
+  proprietary_codecs = true
 
   # libudev usage. This currently only affects the content layer.
   use_udev = (is_linux && !is_castos) || is_chromeos
