diff -r -u --color up/chromium/build/dotfile_settings.gni nw/chromium/build/dotfile_settings.gni
--- up/chromium/build/dotfile_settings.gni	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/build/dotfile_settings.gni	2023-02-24 20:54:05.352527963 +0000
@@ -7,6 +7,7 @@
 
 build_dotfile_settings = {
   exec_script_whitelist = [
+    "//content/nw/BUILD.gn",
     "//build/config/android/rules.gni",
     "//build/config/chromeos/rules.gni",
     "//build/config/compiler/BUILD.gn",
Only in nw/chromium/build: gyp_chromium
Only in nw/chromium/build: gyp_chromium.py
Only in nw/chromium/build: gyp_environment.py
