diff -r -u --color up/chromium/build/config/BUILDCONFIG.gn nw/chromium/build/config/BUILDCONFIG.gn
--- up/chromium/build/config/BUILDCONFIG.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/build/config/BUILDCONFIG.gn	2023-02-24 20:12:01.460792867 +0000
@@ -135,6 +135,8 @@
   # set "is_official_build" to true for any build intended to ship to end-users.
   is_official_build = false
 
+  nwjs_sdk = true
+
   # Set to true when compiling with the Clang compiler.
   is_clang = current_os != "linux" ||
              (current_cpu != "s390x" && current_cpu != "s390" &&
