diff -r -u --color up/chromium/buildtools/third_party/libc++/__config_site nw/chromium/buildtools/third_party/libc++/__config_site
--- up/chromium/buildtools/third_party/libc++/__config_site	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/buildtools/third_party/libc++/__config_site	2023-02-19 17:36:25.247438196 +0000
@@ -11,8 +11,9 @@
 //    on Windows, the increase is great enough that we go above the 4GB size
 //    limit for PDBs (https://crbug.com/1327710#c5). To fix this, we set
 //    _LIBCPP_ABI_NAMESPACE to a shorter value.
-#define _LIBCPP_ABI_NAMESPACE Cr
-#define _LIBCPP_ABI_VERSION 2
+
+#define _LIBCPP_ABI_NAMESPACE __1
+// #define _LIBCPP_ABI_VERSION 2
 
 /* #undef _LIBCPP_ABI_FORCE_ITANIUM */
 /* #undef _LIBCPP_ABI_FORCE_MICROSOFT */
