diff -r -u --color up/chromium/build/config/compiler/BUILD.gn nw/chromium/build/config/compiler/BUILD.gn
--- up/chromium/build/config/compiler/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/build/config/compiler/BUILD.gn	2023-02-19 17:36:25.191438054 +0000
@@ -485,7 +485,9 @@
     #}
   }
 
-  if (use_icf && (!is_apple || use_lld)) {
+  if (is_linux && current_cpu == "x86") {
+    ldflags += [ "-Wl,--icf=none" ]
+  } else if (use_icf && (!is_apple || use_lld)) {
     ldflags += [ "-Wl,--icf=all" ]
   }
 
@@ -1578,6 +1580,9 @@
       # Disables.
       "-Wno-missing-field-initializers",  # "struct foo f = {0};"
       "-Wno-unused-parameter",  # Unused function parameters.
+      "-Wno-unused-private-field",
+      "-Wno-unused-const-variable",
+      "-Wno-unreachable-code",
     ]
 
     if (!is_nacl || is_nacl_saigo) {
@@ -1598,6 +1603,7 @@
       # which we no longer use. Check if it makes sense to remove
       # this as well. http://crbug.com/316352
       "-Wno-unneeded-internal-declaration",
+      "-Wno-unused-function",
     ]
 
     if (!is_nacl || is_nacl_saigo) {
@@ -1769,6 +1775,7 @@
     cflags += [
       "/wd4800",  # Disable warning when forcing value to bool.
       "/wd4267",  # TODO(jschuh): size_t to int.
+      "/wd4275",
     ]
   } else {
     # GCC may emit unsuppressible warnings so don't add -Werror for no chromium
