diff -r -u --color up/chromium/buildtools/third_party/libc++abi/BUILD.gn nw/chromium/buildtools/third_party/libc++abi/BUILD.gn
--- up/chromium/buildtools/third_party/libc++abi/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/buildtools/third_party/libc++abi/BUILD.gn	2023-02-24 20:12:01.500793274 +0000
@@ -4,7 +4,7 @@
 
 import("//build/config/c++/c++.gni")
 
-source_set("libc++abi") {
+static_library("libc++abi") {
   if (export_libcxxabi_from_executables) {
     visibility = [ "//build/config:executable_deps" ]
   } else {
