diff -r -u --color up/chromium/chrome/browser/win/app_icon.cc nw/chromium/chrome/browser/win/app_icon.cc
--- up/chromium/chrome/browser/win/app_icon.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/win/app_icon.cc	2023-02-25 19:07:10.813787689 +0000
@@ -4,6 +4,8 @@
 
 #include "chrome/browser/win/app_icon.h"
 
+#include "content/nw/src/nw_content.h"
+
 #include "chrome/common/chrome_constants.h"
 #include "chrome/install_static/install_details.h"
 #include "third_party/skia/include/core/SkBitmap.h"
@@ -21,6 +23,11 @@
 }  // namespace
 
 HICON GetAppIcon() {
+#if 1
+  HICON ret = nw::GetWindowHIcon();
+  if (ret)
+    return ret;
+#endif
   // TODO(mgiuca): Use GetAppIconImageFamily/CreateExact instead of LoadIcon, to
   // get correct scaling. (See http://crbug.com/551256)
   const int icon_id = GetAppIconResourceId();
@@ -30,6 +37,11 @@
 }
 
 HICON GetSmallAppIcon() {
+#if 1
+  HICON ret = nw::GetAppHIcon();
+  if (ret)
+    return ret;
+#endif
   // TODO(mgiuca): Use GetAppIconImageFamily/CreateExact instead of LoadIcon, to
   // get correct scaling. (See http://crbug.com/551256)
   const int icon_id = GetAppIconResourceId();
