diff -r -u --color up/chromium/chrome/browser/ui/webui/print_preview/pdf_printer_handler.cc nw/chromium/chrome/browser/ui/webui/print_preview/pdf_printer_handler.cc
--- up/chromium/chrome/browser/ui/webui/print_preview/pdf_printer_handler.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/webui/print_preview/pdf_printer_handler.cc	2023-02-19 17:36:29.767449637 +0000
@@ -61,6 +61,8 @@
 #include "chromeos/lacros/lacros_service.h"
 #endif
 
+#include "chrome/browser/ui/webui/print_preview/print_preview_handler.h"
+
 namespace printing {
 
 namespace {
@@ -264,6 +266,10 @@
     scoped_refptr<base::RefCountedMemory> print_data,
     PrintCallback callback) {
   print_data_ = print_data;
+  if (!chrome::NWPrintGetPDFPath().empty() && chrome::NWPrintGetCustomPrinting()) {
+    print_to_pdf_path_ = chrome::NWPrintGetPDFPath();
+    print_callback_ = std::move(callback);
+  }
   if (!print_to_pdf_path_.empty()) {
     // User has already selected a path, no need to show the dialog again.
     PostPrintToPdfTask();
