diff -r -u --color up/chromium/chrome/installer/mini_installer/BUILD.gn nw/chromium/chrome/installer/mini_installer/BUILD.gn
--- up/chromium/chrome/installer/mini_installer/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/installer/mini_installer/BUILD.gn	2023-02-19 17:36:30.087450447 +0000
@@ -122,8 +122,8 @@
   release_file = "chrome.release"
 
   inputs = [
-    "$root_out_dir/chrome.dll",
-    "$root_out_dir/chrome.exe",
+    "$root_out_dir/nw.dll",
+    "$root_out_dir/nw.exe",
     "$root_out_dir/locales/en-US.pak",
     "$root_out_dir/setup.exe",
     "//chrome/tools/build/win/makecab.py",
