diff -r -u --color up/chromium/chrome/common/extensions/api/api_sources.gni nw/chromium/chrome/common/extensions/api/api_sources.gni
--- up/chromium/chrome/common/extensions/api/api_sources.gni	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/common/extensions/api/api_sources.gni	2023-02-25 19:07:10.885788131 +0000
@@ -44,7 +44,6 @@
   "passwords_private.idl",
   "permissions.json",
   "resources_private.idl",
-  "safe_browsing_private.idl",
   "scripting.idl",
   "search.idl",
   "sessions.json",
