diff -r -u --color up/chromium/chrome/test/data/webui/print_preview/print_preview_test_utils.ts nw/chromium/chrome/test/data/webui/print_preview/print_preview_test_utils.ts
--- up/chromium/chrome/test/data/webui/print_preview/print_preview_test_utils.ts	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/test/data/webui/print_preview/print_preview_test_utils.ts	2023-02-24 20:54:09.824507542 +0000
@@ -15,6 +15,7 @@
     NativeInitialSettings {
   return {
     isInKioskAutoPrintMode: false,
+    nwPrintMode: false,
     isInAppKioskMode: false,
     pdfPrinterDisabled: false,
     thousandsDelimiter: ',',
