diff -r -u --color up/chromium/chrome/browser/profiles/profile_impl.cc nw/chromium/chrome/browser/profiles/profile_impl.cc
--- up/chromium/chrome/browser/profiles/profile_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/profiles/profile_impl.cc	2023-02-25 19:07:09.241778019 +0000
@@ -628,6 +628,7 @@
 
   mojo::PendingRemote<prefs::mojom::TrackedPreferenceValidationDelegate>
       pref_validation_delegate;
+#if 0
   scoped_refptr<safe_browsing::SafeBrowsingService> safe_browsing_service(
       g_browser_process->safe_browsing_service());
   if (safe_browsing_service.get()) {
@@ -639,6 +640,7 @@
           pref_validation_delegate.InitWithNewPipeAndPassReceiver());
     }
   }
+#endif
 
   prefs_ =
       CreatePrefService(pref_registry_, CreateExtensionPrefStore(this, false),
