diff -r -u --color up/chromium/chrome/test/base/testing_browser_process.h nw/chromium/chrome/test/base/testing_browser_process.h
--- up/chromium/chrome/test/base/testing_browser_process.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/test/base/testing_browser_process.h	2023-02-19 17:36:30.207450751 +0000
@@ -129,6 +129,7 @@
   void StartAutoupdateTimer() override {}
 #endif
 
+
   component_updater::ComponentUpdateService* component_updater() override;
   MediaFileSystemRegistry* media_file_system_registry() override;
 
