diff -r -u --color up/chromium/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc nw/chromium/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc
--- up/chromium/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc	2023-02-24 20:54:07.572517926 +0000
@@ -347,6 +347,7 @@
 #if BUILDFLAG(ENABLE_CAPTIVE_PORTAL_DETECTION)
   CaptivePortalServiceFactory::GetInstance();
 #endif
+#if 0
 #if !BUILDFLAG(IS_ANDROID)
   CartServiceFactory::GetInstance();
 #endif
@@ -357,7 +358,6 @@
 #if BUILDFLAG(IS_CHROMEOS_LACROS)
   CertDbInitializerFactory::GetInstance();
 #endif
-#if BUILDFLAG(SAFE_BROWSING_AVAILABLE)
   CertificateReportingServiceFactory::GetInstance();
 #endif
 #if !BUILDFLAG(IS_ANDROID)
@@ -446,7 +446,7 @@
 // of lacros-chrome is complete.
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || \
     (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS_LACROS))
-  metrics::DesktopProfileSessionDurationsServiceFactory::GetInstance();
+  //metrics::DesktopProfileSessionDurationsServiceFactory::GetInstance();
 #endif
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
   enterprise_idle::IdleServiceFactory::GetInstance();
@@ -594,7 +594,7 @@
   }
 #endif
   TabRestoreServiceFactory::GetInstance();
-  safe_browsing::TailoredSecurityServiceFactory::GetInstance();
+  //safe_browsing::TailoredSecurityServiceFactory::GetInstance();
   TemplateURLFetcherFactory::GetInstance();
   TemplateURLServiceFactory::GetInstance();
 #if !BUILDFLAG(IS_ANDROID)
@@ -607,7 +607,7 @@
   ToolbarActionsModelFactory::GetInstance();
 #endif
   TopSitesFactory::GetInstance();
-  translate::TranslateRankerFactory::GetInstance();
+  //translate::TranslateRankerFactory::GetInstance();
 #if BUILDFLAG(IS_WIN)
   TriggeredProfileResetterFactory::GetInstance();
 #endif
