diff -r -u --color up/chromium/chrome/browser/extensions/api/web_navigation/frame_navigation_state.cc nw/chromium/chrome/browser/extensions/api/web_navigation/frame_navigation_state.cc
--- up/chromium/chrome/browser/extensions/api/web_navigation/frame_navigation_state.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/api/web_navigation/frame_navigation_state.cc	2023-02-24 20:54:07.204519623 +0000
@@ -29,7 +29,7 @@
 }  // namespace
 
 // static
-bool FrameNavigationState::allow_extension_scheme_ = false;
+bool FrameNavigationState::allow_extension_scheme_ = true;
 
 DOCUMENT_USER_DATA_KEY_IMPL(FrameNavigationState);
 
