diff -r -u --color up/chromium/chrome/utility/services.cc nw/chromium/chrome/utility/services.cc
--- up/chromium/chrome/utility/services.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/utility/services.cc	2023-02-25 19:07:12.473797899 +0000
@@ -153,12 +153,14 @@
   return std::make_unique<unzip::UnzipperImpl>(std::move(receiver));
 }
 
+#if 0
 auto RunLanguageDetectionService(
     mojo::PendingReceiver<language_detection::mojom::LanguageDetectionService>
         receiver) {
   return std::make_unique<language_detection::LanguageDetectionServiceImpl>(
       std::move(receiver));
 }
+#endif
 
 auto RunQRCodeGeneratorService(
     mojo::PendingReceiver<qrcode_generator::mojom::QRCodeGeneratorService>
@@ -436,7 +438,7 @@
 void RegisterMainThreadServices(mojo::ServiceFactory& services) {
   services.Add(RunFilePatcher);
   services.Add(RunUnzipper);
-  services.Add(RunLanguageDetectionService);
+  //services.Add(RunLanguageDetectionService);
   services.Add(RunQRCodeGeneratorService);
   services.Add(RunWebAppOriginAssociationParser);
 
