diff -r -u --color up/chromium/chrome/chrome_elf/chrome_elf_main.h nw/chromium/chrome/chrome_elf/chrome_elf_main.h
--- up/chromium/chrome/chrome_elf/chrome_elf_main.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/chrome_elf/chrome_elf_main.h	2023-02-24 20:12:04.856827446 +0000
@@ -22,7 +22,7 @@
 // This function is a temporary workaround for https://crbug.com/655788. We
 // need to come up with a better way to initialize crash reporting that can
 // happen inside DllMain().
-void SignalInitializeCrashReporting();
+  void SignalInitializeCrashReporting(void*, void*);
 void SignalChromeElf();
 
 // Sets the metrics client ID in crash keys.
@@ -37,5 +37,5 @@
 bool IsExtensionPointDisableSet();
 
 }  // extern "C"
-
+extern "C" void* ElfGetReporterClient();
 #endif  // CHROME_CHROME_ELF_CHROME_ELF_MAIN_H_
