diff -r -u --color up/chromium/chrome/app/chrome_dll.rc nw/chromium/chrome/app/chrome_dll.rc
--- up/chromium/chrome/app/chrome_dll.rc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/app/chrome_dll.rc	2023-02-25 19:07:06.813763085 +0000
@@ -69,7 +69,6 @@
     "T",            IDC_FOCUS_TOOLBAR,          VIRTKEY, SHIFT, ALT
     VK_BACK,        IDC_FORWARD,                VIRTKEY, SHIFT
     VK_RIGHT,       IDC_FORWARD,                VIRTKEY, ALT
-    VK_F11,         IDC_FULLSCREEN,             VIRTKEY
     VK_F1,          IDC_HELP_PAGE_VIA_KEYBOARD, VIRTKEY
     VK_HOME,        IDC_HOME,                   VIRTKEY, ALT
     "N",            IDC_NEW_INCOGNITO_WINDOW,   VIRTKEY, CONTROL, SHIFT
