diff -r -u --color up/chromium/chrome/test/data/webui/print_preview/destination_store_test.ts nw/chromium/chrome/test/data/webui/print_preview/destination_store_test.ts
--- up/chromium/chrome/test/data/webui/print_preview/destination_store_test.ts	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/test/data/webui/print_preview/destination_store_test.ts	2023-02-25 19:07:12.317796940 +0000
@@ -107,7 +107,7 @@
         initialSettings.pdfPrinterDisabled, !!initialSettings.isDriveMounted,
         initialSettings.printerName,
         initialSettings.serializedDefaultDestinationSelectionRulesStr,
-        recentDestinations);
+        recentDestinations, false);
     return expectPrinterFailure ? Promise.resolve() : Promise.race([
       nativeLayer.whenCalled('getPrinterCapabilities'),
       whenCapabilitiesReady,
