diff -r -u --color up/chromium/chrome/common/chrome_paths_win.cc nw/chromium/chrome/common/chrome_paths_win.cc
--- up/chromium/chrome/common/chrome_paths_win.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/common/chrome_paths_win.cc	2023-02-24 20:12:04.864827528 +0000
@@ -19,6 +19,8 @@
 #include "chrome/install_static/install_util.h"
 #include "components/nacl/common/nacl_switches.h"
 
+#include "content/nw/src/nw_base.h"
+
 namespace chrome {
 
 namespace {
@@ -45,7 +47,8 @@
 bool GetDefaultUserDataDirectory(base::FilePath* result) {
   if (!base::PathService::Get(base::DIR_LOCAL_APP_DATA, result))
     return false;
-  *result = result->Append(install_static::GetChromeInstallSubDirectory());
+  if (nw::package()) //FIXME: crashpad initialized early in cr49
+    *result = result->Append(base::FilePath::FromUTF8Unsafe(nw::package()->GetName()));
   *result = result->Append(chrome::kUserDataDirname);
   return true;
 }
