diff -r -u --color up/chromium/chrome/browser/extensions/api/webstore_private/webstore_private_api.cc nw/chromium/chrome/browser/extensions/api/webstore_private/webstore_private_api.cc
--- up/chromium/chrome/browser/extensions/api/webstore_private/webstore_private_api.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/api/webstore_private/webstore_private_api.cc	2023-02-24 20:54:07.208519605 +0000
@@ -252,7 +252,7 @@
 #endif  // BUILDFLAG(ENABLE_SUPERVISED_USERS)
 
 // The number of user gestures to trace back for the referrer chain.
-const int kExtensionReferrerUserGestureLimit = 2;
+// const int kExtensionReferrerUserGestureLimit = 2;
 
 WebstoreInstaller::Delegate* test_webstore_installer_delegate = nullptr;
 
@@ -720,6 +720,7 @@
 
 void WebstorePrivateBeginInstallWithManifest3Function::
     ReportFrictionAcceptedEvent() {
+#if 0
   if (!profile_) {
     return;
   }
@@ -732,6 +733,7 @@
         safe_browsing::SafeBrowsingMetricsCollector::EventType::
             EXTENSION_ALLOWLIST_INSTALL_BYPASS);
   }
+#endif
 }
 
 void WebstorePrivateBeginInstallWithManifest3Function::OnInstallPromptDone(
@@ -1217,6 +1219,9 @@
 
 ExtensionFunction::ResponseAction
 WebstorePrivateGetReferrerChainFunction::Run() {
+#if 1
+  return RespondNow(ArgumentList(api::webstore_private::GetReferrerChain::Results::Create("")));
+#else
   Profile* profile = Profile::FromBrowserContext(browser_context());
   if (!SafeBrowsingNavigationObserverManager::IsEnabledAndReady(
           profile->GetPrefs(), g_browser_process->safe_browsing_service()))
@@ -1266,6 +1271,7 @@
   return RespondNow(
       ArgumentList(api::webstore_private::GetReferrerChain::Results::Create(
           serialized_referrer_proto)));
+#endif
 }
 
 WebstorePrivateGetExtensionStatusFunction::
