diff -r -u --color up/chromium/chrome/browser/ui/sad_tab.cc nw/chromium/chrome/browser/ui/sad_tab.cc
--- up/chromium/chrome/browser/ui/sad_tab.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/sad_tab.cc	2023-02-25 19:07:10.321784662 +0000
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/sad_tab.h"
 
+#include "chrome/browser/ui/browser_commands.h"
 #include <vector>
 
 #include "base/metrics/histogram_macros.h"
@@ -217,8 +218,12 @@
                                          : IDS_KILLED_TAB_FEEDBACK_MESSAGE),
             std::string(kCategoryTagCrash), std::string());
       } else {
-        web_contents_->GetController().Reload(content::ReloadType::NORMAL,
-                                              true);
+        Browser* browser = chrome::FindBrowserWithWebContents(web_contents_);
+        if (browser) {
+          chrome::CloseTab(browser);
+        } else
+        web_contents_->Close(); //GetController().Reload(content::ReloadType::NORMAL,
+                                //              true);
       }
       break;
     case Action::HELP_LINK:
