diff -r -u --color up/chromium/chrome/app/chrome_main_mac.mm nw/chromium/chrome/app/chrome_main_mac.mm
--- up/chromium/chrome/app/chrome_main_mac.mm	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/app/chrome_main_mac.mm	2023-02-25 19:07:06.813763085 +0000
@@ -20,20 +20,14 @@
 #include "content/public/common/content_paths.h"
 #include "content/public/common/content_switches.h"
 
+#include "content/nw/src/nw_base.h"
+
 void SetUpBundleOverrides() {
   @autoreleasepool {
     base::mac::SetOverrideFrameworkBundlePath(chrome::GetFrameworkBundlePath());
 
     NSBundle* base_bundle = chrome::OuterAppBundle();
     base::mac::SetBaseBundleID([[base_bundle bundleIdentifier] UTF8String]);
-
-    base::FilePath child_exe_path =
-        chrome::GetFrameworkBundlePath().Append("Helpers").Append(
-            chrome::kHelperProcessExecutablePath);
-
-    // On the Mac, the child executable lives at a predefined location within
-    // the app bundle's versioned directory.
-    base::PathService::Override(content::CHILD_PROCESS_EXE, child_exe_path);
   }
 }
 
