diff -r -u --color up/chromium/chrome/browser/ui/omnibox/chrome_omnibox_client.cc nw/chromium/chrome/browser/ui/omnibox/chrome_omnibox_client.cc
--- up/chromium/chrome/browser/ui/omnibox/chrome_omnibox_client.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/omnibox/chrome_omnibox_client.cc	2023-02-24 20:12:04.444823251 +0000
@@ -253,20 +253,24 @@
 void ChromeOmniboxClient::OnInputStateChanged() {
   if (!controller_->GetWebContents())
     return;
+#if 0
   if (auto* helper =
           OmniboxTabHelper::FromWebContents(controller_->GetWebContents())) {
     helper->OnInputStateChanged();
   }
+#endif
 }
 
 void ChromeOmniboxClient::OnFocusChanged(OmniboxFocusState state,
                                          OmniboxFocusChangeReason reason) {
   if (!controller_->GetWebContents())
     return;
+#if 0
   if (auto* helper =
           OmniboxTabHelper::FromWebContents(controller_->GetWebContents())) {
     helper->OnFocusChanged(state, reason);
   }
+#endif
 }
 
 void ChromeOmniboxClient::OnResultChanged(
