diff -r -u --color up/chromium/chrome/browser/enterprise/connectors/analysis/files_request_handler.cc nw/chromium/chrome/browser/enterprise/connectors/analysis/files_request_handler.cc
--- up/chromium/chrome/browser/enterprise/connectors/analysis/files_request_handler.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/enterprise/connectors/analysis/files_request_handler.cc	2023-02-25 19:07:08.581773960 +0000
@@ -236,6 +236,7 @@
 void FilesRequestHandler::FinishRequestEarly(
     std::unique_ptr<safe_browsing::BinaryUploadService::Request> request,
     safe_browsing::BinaryUploadService::Result result) {
+#if 0
   // We add the request here in case we never actually uploaded anything, so it
   // wasn't added in OnGetRequestData
   safe_browsing::WebUIInfoSingleton::GetInstance()->AddToDeepScanRequests(
@@ -243,7 +244,7 @@
   safe_browsing::WebUIInfoSingleton::GetInstance()->AddToDeepScanResponses(
       /*token=*/"", safe_browsing::BinaryUploadService::ResultToString(result),
       enterprise_connectors::ContentAnalysisResponse());
-
+#endif
   request->FinishRequest(result,
                          enterprise_connectors::ContentAnalysisResponse());
 }
