diff -r -u --color up/chromium/chrome/renderer/extensions/chrome_extensions_renderer_client.cc nw/chromium/chrome/renderer/extensions/chrome_extensions_renderer_client.cc
--- up/chromium/chrome/renderer/extensions/chrome_extensions_renderer_client.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/renderer/extensions/chrome_extensions_renderer_client.cc	2023-02-25 19:07:10.997788820 +0000
@@ -8,6 +8,8 @@
 #include <utility>
 
 #include "base/bind.h"
+#include "content/nw/src/nw_content.h"
+
 #include "base/command_line.h"
 #include "base/features.h"
 #include "base/lazy_instance.h"
@@ -139,6 +141,7 @@
   if (!extension_dispatcher_) {
     extension_dispatcher_ = std::make_unique<extensions::Dispatcher>(
         std::make_unique<ChromeExtensionsDispatcherDelegate>());
+    nw::ExtensionDispatcherCreated(extension_dispatcher_.get());
   }
   extension_dispatcher_->OnRenderThreadStarted(thread);
   permissions_policy_delegate_ =
