diff -r -u --color up/chromium/chrome/test/chromedriver/chrome/devtools_http_client.cc nw/chromium/chrome/test/chromedriver/chrome/devtools_http_client.cc
--- up/chromium/chrome/test/chromedriver/chrome/devtools_http_client.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/test/chromedriver/chrome/devtools_http_client.cc	2023-02-25 19:07:11.069789263 +0000
@@ -109,7 +109,17 @@
 
 bool DevToolsHttpClient::IsBrowserWindow(const WebViewInfo& view) const {
   return base::Contains(*window_types_, view.type) ||
-         (view.type == WebViewInfo::kOther && view.url == "chrome://print/");
+         (view.type == WebViewInfo::kOther &&
+          (view.url == "chrome://print/" ||
+           view.url == "" || view.url == "about:blank" ||
+           //          base::StartsWith(view.url, "chrome-extension://",
+           //                 base::CompareCase::SENSITIVE) ||
+           base::StartsWith(view.url, "file://",
+                            base::CompareCase::SENSITIVE) ||
+           base::StartsWith(view.url, "http://",
+                            base::CompareCase::SENSITIVE) ||
+           base::StartsWith(view.url, "https://",
+                            base::CompareCase::SENSITIVE)));
 }
 
 const DevToolsEndpoint& DevToolsHttpClient::endpoint() const {
