diff -r -u --color up/chromium/chrome/browser/extensions/api/desktop_capture/desktop_capture_api.cc nw/chromium/chrome/browser/extensions/api/desktop_capture/desktop_capture_api.cc
--- up/chromium/chrome/browser/extensions/api/desktop_capture/desktop_capture_api.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/api/desktop_capture/desktop_capture_api.cc	2023-02-24 20:12:03.240810992 +0000
@@ -94,9 +94,14 @@
     // was requested for capture.
     target_render_frame_host = web_contents->GetPrimaryMainFrame();
   } else {
-    origin = extension()->url();
     target_name = base::UTF8ToUTF16(GetExtensionTargetName());
     target_render_frame_host = render_frame_host();
+    // NWJS fix for nwjs/nw.js#4579
+    // NWJS app allows running on origins other than `chrome-extension://*/*`.
+    // The origin should then be from the senders URL, in order not to fail
+    // the origin checking in `DesktopStreamsRegistry::RequestMediaForStreamId`.
+    origin = extension()->is_nwjs_app() ? target_render_frame_host->GetLastCommittedURL().
+      DeprecatedGetOriginAsURL() : extension()->url();
   }
 
   if (!target_render_frame_host)
