diff -r -u --color up/chromium/chrome/browser/enterprise/connectors/reporting/realtime_reporting_client.cc nw/chromium/chrome/browser/enterprise/connectors/reporting/realtime_reporting_client.cc
--- up/chromium/chrome/browser/enterprise/connectors/reporting/realtime_reporting_client.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/enterprise/connectors/reporting/realtime_reporting_client.cc	2023-02-19 17:36:27.659444301 +0000
@@ -400,7 +400,7 @@
 }
 
 std::string RealtimeReportingClient::GetProfileUserName() const {
-  return safe_browsing::GetProfileEmail(identity_manager_);
+  return std::string();
 }
 
 #if BUILDFLAG(IS_CHROMEOS_ASH)
