diff -r -u --color up/chromium/chrome/test/chromedriver/chrome/chrome_finder.cc nw/chromium/chrome/test/chromedriver/chrome/chrome_finder.cc
--- up/chromium/chrome/test/chromedriver/chrome/chrome_finder.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/test/chromedriver/chrome/chrome_finder.cc	2023-02-24 20:12:05.008828885 +0000
@@ -30,7 +30,7 @@
 
 namespace {
 
-#if BUILDFLAG(IS_WIN)
+#if 0
 void GetApplicationDirs(std::vector<base::FilePath>* locations) {
   std::vector<base::FilePath> installation_locations;
   base::FilePath local_app_data, program_files, program_files_x86,
@@ -57,7 +57,7 @@
         installation_locations[i].Append(L"Chromium\\Application"));
   }
 }
-#elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#elif 0
 void GetApplicationDirs(std::vector<base::FilePath>* locations) {
   // TODO: Respect users' PATH variables.
   // Until then, we use an approximation of the most common defaults.
@@ -134,7 +134,7 @@
 
 }  // namespace internal
 
-#if BUILDFLAG(IS_MAC)
+#if 0
 void GetApplicationDirs(std::vector<base::FilePath>* locations);
 #endif
 
@@ -143,10 +143,8 @@
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC)
     base::FilePath(chrome::kBrowserProcessExecutablePath),
 #elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
-    base::FilePath("google-chrome"),
     base::FilePath(chrome::kBrowserProcessExecutablePath),
-    base::FilePath("chromium"),
-    base::FilePath("chromium-browser")
+    base::FilePath("nw")
 #else
     // it will compile but won't work on other OSes
     base::FilePath()
@@ -176,8 +174,11 @@
   }
 
   std::vector<base::FilePath> locations;
-  GetApplicationDirs(&locations);
   GetPathsFromEnvironment(&locations);
+  base::FilePath exe_path;
+  base::PathService::Get(base::DIR_EXE, &exe_path);
+  locations.push_back(exe_path);
+
   return internal::FindExe(base::BindRepeating(&base::PathExists), browser_exes,
                            locations, browser_exe);
 }
