diff -r -u --color up/chromium/chrome/browser/ui/webui/extensions/extensions_internals_source.cc nw/chromium/chrome/browser/ui/webui/extensions/extensions_internals_source.cc
--- up/chromium/chrome/browser/ui/webui/extensions/extensions_internals_source.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/webui/extensions/extensions_internals_source.cc	2023-02-25 19:07:10.613786458 +0000
@@ -51,6 +51,8 @@
       return "TYPE_LEGACY_PACKAGED_APP";
     case extensions::Manifest::TYPE_PLATFORM_APP:
       return "TYPE_PLATFORM_APP";
+    case extensions::Manifest::TYPE_NWJS_APP:
+      return "TYPE_NWJS_APP";
     case extensions::Manifest::TYPE_SHARED_MODULE:
       return "TYPE_SHARED_MODULE";
     case extensions::Manifest::TYPE_LOGIN_SCREEN_EXTENSION:
