diff -r -u --color up/chromium/chrome/browser/ui/views/frame/system_menu_model_builder.cc nw/chromium/chrome/browser/ui/views/frame/system_menu_model_builder.cc
--- up/chromium/chrome/browser/ui/views/frame/system_menu_model_builder.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/frame/system_menu_model_builder.cc	2023-02-24 20:12:04.532824147 +0000
@@ -118,6 +118,7 @@
 
 void SystemMenuModelBuilder::BuildSystemMenuForAppOrPopupWindow(
     ui::SimpleMenuModel* model) {
+#if 0
   model->AddItemWithStringId(IDC_BACK, IDS_CONTENT_CONTEXT_BACK);
   model->AddItemWithStringId(IDC_FORWARD, IDS_CONTENT_CONTEXT_FORWARD);
   model->AddItemWithStringId(IDC_RELOAD, IDS_APP_MENU_RELOAD);
@@ -152,6 +153,7 @@
   AppendMoveToDesksMenu(model);
 #endif
   AppendTeleportMenu(model);
+#endif
 }
 
 void SystemMenuModelBuilder::AddFrameToggleItems(ui::SimpleMenuModel* model) {
