diff -r -u --color up/chromium/chrome/browser/extensions/extension_system_factory.cc nw/chromium/chrome/browser/extensions/extension_system_factory.cc
--- up/chromium/chrome/browser/extensions/extension_system_factory.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/extension_system_factory.cc	2023-02-19 17:36:27.791444635 +0000
@@ -50,7 +50,7 @@
   DependsOn(InstallVerifierFactory::GetInstance());
   DependsOn(ProcessManagerFactory::GetInstance());
   DependsOn(RendererStartupHelperFactory::GetInstance());
-  DependsOn(BlocklistFactory::GetInstance());
+  //DependsOn(BlocklistFactory::GetInstance());
   DependsOn(EventRouterFactory::GetInstance());
   // This depends on ExtensionDownloader, which depends on
   // IdentityManager for webstore authentication.
