diff -r -u --color up/chromium/chrome/browser/enterprise/connectors/analysis/content_analysis_delegate.cc nw/chromium/chrome/browser/enterprise/connectors/analysis/content_analysis_delegate.cc
--- up/chromium/chrome/browser/enterprise/connectors/analysis/content_analysis_delegate.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/enterprise/connectors/analysis/content_analysis_delegate.cc	2023-02-25 19:07:08.577773935 +0000
@@ -37,7 +37,7 @@
 #include "chrome/browser/safe_browsing/cloud_content_scanning/binary_upload_service.h"
 #include "chrome/browser/safe_browsing/cloud_content_scanning/deep_scanning_utils.h"
 #include "chrome/browser/safe_browsing/cloud_content_scanning/file_analysis_request.h"
-#include "chrome/browser/safe_browsing/download_protection/check_client_download_request.h"
+//#include "chrome/browser/safe_browsing/download_protection/check_client_download_request.h"
 #include "chrome/grit/generated_resources.h"
 #include "components/enterprise/common/proto/connectors.pb.h"
 #include "components/policy/core/common/chrome_schema.h"
