diff -r -u --color up/chromium/chrome/browser/sync/sync_service_factory.cc nw/chromium/chrome/browser/sync/sync_service_factory.cc
--- up/chromium/chrome/browser/sync/sync_service_factory.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/sync/sync_service_factory.cc	2023-02-19 17:36:28.955447581 +0000
@@ -171,6 +171,7 @@
                                      : syncer::SyncServiceImpl::MANUAL_START;
   }
 
+#if 0
   auto sync_service =
       std::make_unique<syncer::SyncServiceImpl>(std::move(init_params));
   sync_service->Initialize();
@@ -190,6 +191,9 @@
   }
 
   return sync_service;
+#endif
+
+  return nullptr;
 }
 
 }  // anonymous namespace
