diff -r -u --color up/chromium/chrome/common/features.gni nw/chromium/chrome/common/features.gni
--- up/chromium/chrome/common/features.gni	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/common/features.gni	2023-02-25 19:07:10.901788230 +0000
@@ -95,6 +95,7 @@
 # Every grit target in //chrome should apply these defines so that the
 # proper build flags can be set.
 chrome_grit_defines = [
+  "nwjs_sdk=$nwjs_sdk",
   "chrome_root_store_supported=$chrome_root_store_supported",
   "enable_arcore=$enable_arcore",
   "enable_background_mode=$enable_background_mode",
