diff -r -u --color up/chromium/chrome/test/chromedriver/chrome_launcher.cc nw/chromium/chrome/test/chromedriver/chrome_launcher.cc
--- up/chromium/chrome/test/chromedriver/chrome_launcher.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/test/chromedriver/chrome_launcher.cc	2023-02-25 19:07:11.073789288 +0000
@@ -71,6 +71,8 @@
 #include <windows.h>
 #endif
 
+#include "base/strings/string_number_conversions.h"
+
 namespace {
 
 const char* const kCommonSwitches[] = {
@@ -79,6 +81,7 @@
     "allow-pre-commit-input",
 };
 
+#if 0
 const char* const kDesktopSwitches[] = {
     "disable-hang-monitor",
     "disable-prompt-on-repost",
@@ -98,6 +101,7 @@
     "enable-blink-features=ShadowDOMV0",
     "no-service-autorun",
 };
+#endif
 
 #if BUILDFLAG(IS_WIN)
 
@@ -148,12 +152,14 @@
 
   for (auto* common_switch : kCommonSwitches)
     switches.SetUnparsedSwitch(common_switch);
+#if 0 //FIXME if enabled, chromedriver cannot find chrome on windows
   for (auto* desktop_switch : kDesktopSwitches)
     switches.SetUnparsedSwitch(desktop_switch);
 #if BUILDFLAG(IS_WIN)
   for (auto* win_desktop_switch : kWindowsDesktopSwitches)
     switches.SetUnparsedSwitch(win_desktop_switch);
 #endif
+#endif
 
   // Chrome logs are normally sent to a file (whose location can be controlled
   // via the logPath capability). We expose a flag, --enable-chrome-logs, that
@@ -196,7 +202,7 @@
       return Status(kInvalidArgument, "user data dir can not be empty");
     *user_data_dir = base::FilePath(user_data_dir_value);
   } else {
-    command.AppendArg("data:,");
+    //command.AppendArg("data:,");
     if (!user_data_dir_temp_dir->CreateUniqueTempDir())
       return Status(kUnknownError, "cannot create temp dir for user data dir");
     switches.SetSwitch("user-data-dir",
@@ -226,6 +232,10 @@
       return status;
   }
   switches.AppendToCommandLine(&command);
+
+  for (size_t i = 0; i < capabilities.arguments.size(); i++)
+    command.AppendArg(capabilities.arguments[i]);
+
   *prepared_command = command;
   return Status(kOk);
 }
@@ -247,8 +257,13 @@
     window_types = std::make_unique<std::set<WebViewInfo::Type>>();
   }
 
-  std::unique_ptr<DevToolsHttpClient> client;
   base::CommandLine* cmd_line = base::CommandLine::ForCurrentProcess();
+  if (cmd_line->HasSwitch("launch-timeout")) {
+    std::string s_timeout = cmd_line->GetSwitchValueASCII("launch-timeout");
+    base::StringToInt(s_timeout, &wait_time);
+  }
+
+  std::unique_ptr<DevToolsHttpClient> client;
   if (cmd_line->HasSwitch("devtools-replay")) {
     base::CommandLine::StringType log_path =
         cmd_line->GetSwitchValueNative("devtools-replay");
@@ -318,7 +333,10 @@
     if (status.IsError())
       return status;
     for (size_t i = 0; i < views_info.GetSize(); ++i) {
-      if (views_info.Get(i).type == WebViewInfo::kPage) {
+      if (views_info.Get(i).type == WebViewInfo::kPage ||
+          views_info.Get(i).type == WebViewInfo::kApp
+          || (views_info.Get(i).type == WebViewInfo::kOther &&
+              !base::StartsWith(views_info.Get(i).url, "chrome-extension://", base::CompareCase::SENSITIVE))) { //node-remote page
         *user_client = std::move(client);
         return Status(kOk);
       }
