diff -r -u --color up/chromium/chrome/browser/renderer_preferences_util.cc nw/chromium/chrome/browser/renderer_preferences_util.cc
--- up/chromium/chrome/browser/renderer_preferences_util.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/renderer_preferences_util.cc	2023-02-19 17:36:28.315445961 +0000
@@ -43,6 +43,8 @@
 #include "ui/linux/linux_ui.h"
 #endif
 
+#include "content/nw/src/common/nw_content_common_hooks.h"
+
 namespace {
 
 // Parses a string |range| with a port range in the form "<min>-<max>".
@@ -202,6 +204,10 @@
 #else
   prefs->focus_ring_color = SkColorSetRGB(0x10, 0x10, 0x10);
 #endif
+
+  std::string user_agent;
+  if (nw::GetUserAgentFromManifest(&user_agent))
+    prefs->user_agent_override.ua_string_override = user_agent;
 }
 
 }  // namespace renderer_preferences_util
