diff -r -u --color up/chromium/chrome/browser/resources/print_preview/ui/app.ts nw/chromium/chrome/browser/resources/print_preview/ui/app.ts
--- up/chromium/chrome/browser/resources/print_preview/ui/app.ts	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/resources/print_preview/ui/app.ts	2023-02-25 19:07:09.565780012 +0000
@@ -105,6 +105,12 @@
       },
 
       maxSheets_: Number,
+      /** @private {boolean} */
+      isInNWPrintMode_: {
+        type: Boolean,
+        notify: true,
+        value: false,
+      },
     };
   }
 
@@ -130,6 +136,7 @@
   private startPreviewWhenReady_: boolean = false;
   private showSystemDialogBeforePrint_: boolean = false;
   private openPdfInPreview_: boolean = false;
+  private isInNWPrintMode_: boolean = false;
   private isInKioskAutoPrintMode_: boolean = false;
   private whenReady_: Promise<void>|null = null;
   private openDialogs_: CrDialogElement[] = [];
@@ -295,6 +302,7 @@
           settings.pdfPrinterDisabled, settings.isDriveMounted || false);
       this.destinationsManaged_ = settings.destinationsManaged;
       this.isInKioskAutoPrintMode_ = settings.isInKioskAutoPrintMode;
+      this.isInNWPrintMode_ = settings.nwPrintMode;
 
       // This is only visible in the task manager.
       let title = document.head.querySelector('title');
