diff -r -u --color up/chromium/chrome/browser/metrics/chrome_metrics_service_client.cc nw/chromium/chrome/browser/metrics/chrome_metrics_service_client.cc
--- up/chromium/chrome/browser/metrics/chrome_metrics_service_client.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/metrics/chrome_metrics_service_client.cc	2023-02-25 19:07:08.933776125 +0000
@@ -770,13 +770,17 @@
         std::make_unique<metrics::SamplingMetricsProvider>(sample_rate));
   }
 
+#if 0
   metrics_service_->RegisterMetricsProvider(
       std::make_unique<translate::TranslateRankerMetricsProvider>());
+#endif
 
+#if 1
   metrics_service_->RegisterMetricsProvider(
       std::make_unique<metrics::ComponentMetricsProvider>(
           std::make_unique<ChromeComponentMetricsProviderDelegate>(
               g_browser_process->component_updater())));
+#endif
 
   metrics_service_->RegisterMetricsProvider(
       std::make_unique<tracing::ChromeBackgroundTracingMetricsProvider>(
@@ -904,6 +908,7 @@
   metrics_service_->RegisterMetricsProvider(
       std::make_unique<HttpsEngagementMetricsProvider>());
 
+#if 0
   metrics_service_->RegisterMetricsProvider(
       std::make_unique<CertificateReportingMetricsProvider>());
 
@@ -921,6 +926,7 @@
   metrics_service_->RegisterMetricsProvider(
       metrics::CreateDesktopSessionMetricsProvider());
 #endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || (BUILDFLAG(IS_LINUX)
+#endif
 }
 
 void ChromeMetricsServiceClient::RegisterUKMProviders() {
