diff -r -u --color up/chromium/chrome/browser/extensions/global_shortcut_listener_ozone.cc nw/chromium/chrome/browser/extensions/global_shortcut_listener_ozone.cc
--- up/chromium/chrome/browser/extensions/global_shortcut_listener_ozone.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/global_shortcut_listener_ozone.cc	2023-02-19 17:36:27.799444655 +0000
@@ -58,7 +58,8 @@
   const bool registered =
       platform_global_shortcut_listener_->RegisterAccelerator(
           accelerator.key_code(), accelerator.IsAltDown(),
-          accelerator.IsCtrlDown(), accelerator.IsShiftDown());
+          accelerator.IsCtrlDown(), accelerator.IsShiftDown(),
+          accelerator.IsCmdDown());
   if (registered)
     registered_hot_keys_.insert(accelerator);
   return registered;
@@ -72,7 +73,7 @@
 
   platform_global_shortcut_listener_->UnregisterAccelerator(
       accelerator.key_code(), accelerator.IsAltDown(), accelerator.IsCtrlDown(),
-      accelerator.IsShiftDown());
+      accelerator.IsShiftDown(), accelerator.IsCmdDown());
   registered_hot_keys_.erase(accelerator);
 }
 
