diff -r -u --color up/chromium/chrome/browser/background/background_contents_service.cc nw/chromium/chrome/browser/background/background_contents_service.cc
--- up/chromium/chrome/browser/background/background_contents_service.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/background/background_contents_service.cc	2023-02-19 17:36:27.459443795 +0000
@@ -782,6 +782,7 @@
 
 void BackgroundContentsService::HandleExtensionCrashed(
     const extensions::Extension* extension) {
+#if 0
   // When the extensions crash, notify the user about it and restart the crashed
   // contents.
   if (!extension)
@@ -796,6 +797,7 @@
     // Restart the extension.
     RestartForceInstalledExtensionOnCrash(extension);
   }
+#endif
 }
 
 BackgroundContentsService::BackgroundContentsInfo::BackgroundContentsInfo() =
