diff -r -u --color up/chromium/chrome/browser/ui/browser_window_state.h nw/chromium/chrome/browser/ui/browser_window_state.h
--- up/chromium/chrome/browser/ui/browser_window_state.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/browser_window_state.h	2023-02-24 20:12:04.408822885 +0000
@@ -65,7 +65,7 @@
 // Return the |bounds| for the browser window to be used upon creation.
 // The |show_state| variable will receive the desired initial show state for
 // the window.
-void GetSavedWindowBoundsAndShowState(const Browser* browser,
+bool GetSavedWindowBoundsAndShowState(const Browser* browser,
                                       gfx::Rect* bounds,
                                       ui::WindowShowState* show_state);
 
