diff -r -u --color up/chromium/chrome/browser/extensions/api/scripting/scripting_api.cc nw/chromium/chrome/browser/extensions/api/scripting/scripting_api.cc
--- up/chromium/chrome/browser/extensions/api/scripting/scripting_api.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/api/scripting/scripting_api.cc	2023-02-25 19:07:08.681774575 +0000
@@ -686,6 +686,7 @@
                          : blink::mojom::UserActivationOption::kDoNotActivate,
           blink::mojom::PromiseResultOption::kAwait)),
       frame_scope, frame_ids, ScriptExecutor::MATCH_ABOUT_BLANK, run_location,
+      ScriptExecutor::ISOLATED_WORLD,
       ScriptExecutor::DEFAULT_PROCESS,
       /* webview_src */ GURL(),
       base::BindOnce(&ScriptingExecuteScriptFunction::OnScriptExecuted, this));
@@ -810,7 +811,9 @@
           std::move(sources), ConvertStyleOriginToCSSOrigin(injection_.origin),
           mojom::CSSInjection::Operation::kAdd)),
       frame_scope, frame_ids, ScriptExecutor::MATCH_ABOUT_BLANK,
-      kCSSRunLocation, ScriptExecutor::DEFAULT_PROCESS,
+      kCSSRunLocation,
+      ScriptExecutor::ISOLATED_WORLD,
+      ScriptExecutor::DEFAULT_PROCESS,
       /* webview_src */ GURL(),
       base::BindOnce(&ScriptingInsertCSSFunction::OnCSSInserted, this));
 
@@ -887,7 +890,9 @@
           std::move(sources), ConvertStyleOriginToCSSOrigin(injection.origin),
           mojom::CSSInjection::Operation::kRemove)),
       frame_scope, frame_ids, ScriptExecutor::MATCH_ABOUT_BLANK,
-      kCSSRunLocation, ScriptExecutor::DEFAULT_PROCESS,
+      kCSSRunLocation,
+      ScriptExecutor::ISOLATED_WORLD,
+      ScriptExecutor::DEFAULT_PROCESS,
       /* webview_src */ GURL(),
       base::BindOnce(&ScriptingRemoveCSSFunction::OnCSSRemoved, this));
 
