diff -r -u --color up/chromium/chrome/common/extensions/chrome_extensions_client.cc nw/chromium/chrome/common/extensions/chrome_extensions_client.cc
--- up/chromium/chrome/common/extensions/chrome_extensions_client.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/common/extensions/chrome_extensions_client.cc	2023-02-19 17:36:30.019450275 +0000
@@ -39,6 +39,8 @@
 #include "ui/base/l10n/l10n_util.h"
 #include "url/gurl.h"
 
+#include "content/nw/src/api/generated_schemas.h"
+
 namespace extensions {
 
 namespace {
