diff -r -u --color up/chromium/chrome/browser/push_messaging/push_messaging_service_factory.cc nw/chromium/chrome/browser/push_messaging/push_messaging_service_factory.cc
--- up/chromium/chrome/browser/push_messaging/push_messaging_service_factory.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/push_messaging/push_messaging_service_factory.cc	2023-02-24 20:12:03.672815390 +0000
@@ -31,6 +31,12 @@
   if (context->IsOffTheRecord())
     return nullptr;
 
+  if (!instance_id::InstanceIDProfileService::IsInstanceIDEnabled()) {
+    LOG(WARNING) << "PushMessagingService could not be built because "
+                    "InstanceID is unexpectedly disabled";
+    return nullptr;
+  }
+
   return static_cast<PushMessagingServiceImpl*>(
       GetInstance()->GetServiceForBrowserContext(context, true));
 }
