diff -r -u --color up/chromium/chrome/browser/browser_switcher/browser_switcher_navigation_throttle.cc nw/chromium/chrome/browser/browser_switcher/browser_switcher_navigation_throttle.cc
--- up/chromium/chrome/browser/browser_switcher/browser_switcher_navigation_throttle.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/browser_switcher/browser_switcher_navigation_throttle.cc	2023-02-24 20:54:07.004520545 +0000
@@ -4,6 +4,8 @@
 
 #include "chrome/browser/browser_switcher/browser_switcher_navigation_throttle.h"
 
+#pragma clang diagnostic ignored "-Wunused-function"
+
 #include <memory>
 
 #include "base/bind.h"
@@ -89,7 +91,7 @@
 BrowserSwitcherNavigationThrottle::MaybeCreateThrottleFor(
     content::NavigationHandle* navigation) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
-
+#if 0
   content::BrowserContext* browser_context =
       navigation->GetWebContents()->GetBrowserContext();
   Profile* profile = Profile::FromBrowserContext(browser_context);
@@ -103,6 +105,8 @@
   return std::make_unique<navigation_interception::InterceptNavigationThrottle>(
       navigation, base::BindRepeating(&MaybeLaunchAlternativeBrowser),
       navigation_interception::SynchronyMode::kSync);
+#endif
+  return nullptr;
 }
 
 }  // namespace browser_switcher
