diff -r -u --color up/chromium/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc nw/chromium/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc
--- up/chromium/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc	2023-02-25 19:07:08.621774206 +0000
@@ -4,6 +4,8 @@
 
 #include "chrome/browser/extensions/browser_context_keyed_service_factories.h"
 
+#include "content/nw/src/api/object_manager_factory.h"
+
 #include "build/build_config.h"
 #include "build/chromeos_buildflags.h"
 #include "chrome/browser/extensions/api/activity_log_private/activity_log_private_api.h"
@@ -83,11 +85,12 @@
   extensions::NetworkingPrivateDelegateFactory::GetInstance()
       ->SetUIDelegateFactory(std::move(networking_private_ui_delegate_factory));
 #endif
+  nw::ObjectManagerFactory::GetInstance();
   extensions::OmniboxAPI::GetFactoryInstance();
   extensions::PasswordsPrivateEventRouterFactory::GetInstance();
   extensions::PreferenceAPI::GetFactoryInstance();
   extensions::ProcessesAPI::GetFactoryInstance();
-  extensions::SafeBrowsingPrivateEventRouterFactory::GetInstance();
+  //extensions::SafeBrowsingPrivateEventRouterFactory::GetInstance();
   extensions::SessionsAPI::GetFactoryInstance();
   extensions::SettingsPrivateEventRouterFactory::GetInstance();
   extensions::SettingsOverridesAPI::GetFactoryInstance();
