diff -r -u --color up/chromium/chrome/browser/safe_browsing/chrome_safe_browsing_tab_observer_delegate.cc nw/chromium/chrome/browser/safe_browsing/chrome_safe_browsing_tab_observer_delegate.cc
--- up/chromium/chrome/browser/safe_browsing/chrome_safe_browsing_tab_observer_delegate.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/safe_browsing/chrome_safe_browsing_tab_observer_delegate.cc	2023-02-25 19:07:09.721780972 +0000
@@ -43,6 +43,7 @@
   return g_browser_process->safe_browsing_service();
 }
 
+#if 0
 std::unique_ptr<ClientSideDetectionHost>
 ChromeSafeBrowsingTabObserverDelegate::CreateClientSideDetectionHost(
     content::WebContents* web_contents) {
@@ -52,5 +53,6 @@
   return nullptr;
 #endif
 }
+#endif
 
 }  // namespace safe_browsing
