diff -r -u --color up/chromium/chrome/browser/notifications/notification_display_service_impl.cc nw/chromium/chrome/browser/notifications/notification_display_service_impl.cc
--- up/chromium/chrome/browser/notifications/notification_display_service_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/notifications/notification_display_service_impl.cc	2023-02-25 19:07:09.009776592 +0000
@@ -88,8 +88,7 @@
             profile_));
 #endif
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_MAC) || \
-    BUILDFLAG(IS_WIN)
+#if 0
     AddNotificationHandler(
         NotificationHandler::Type::TAILORED_SECURITY,
         std::make_unique<safe_browsing::TailoredSecurityNotificationHandler>());
