diff -r -u --color up/chromium/chrome/browser/net/trial_comparison_cert_verifier_controller.cc nw/chromium/chrome/browser/net/trial_comparison_cert_verifier_controller.cc
--- up/chromium/chrome/browser/net/trial_comparison_cert_verifier_controller.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/net/trial_comparison_cert_verifier_controller.cc	2023-02-25 19:07:08.997776519 +0000
@@ -112,6 +112,7 @@
     const net::CertVerifyResult& primary_result,
     const net::CertVerifyResult& trial_result,
     cert_verifier::mojom::CertVerifierDebugInfoPtr debug_info) {
+#if 0
   if (!IsAllowed() || base::GetFieldTrialParamByFeatureAsBool(
                           net::features::kCertDualVerificationTrialFeature,
                           "uma_only", false)) {
@@ -135,6 +136,7 @@
 
   CertificateReportingServiceFactory::GetForBrowserContext(profile_)->Send(
       serialized_report);
+#endif
 }
 
 // static
