diff -r -u --color up/chromium/chrome/browser/ui/browser_list.cc nw/chromium/chrome/browser/ui/browser_list.cc
--- up/chromium/chrome/browser/ui/browser_list.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/browser_list.cc	2023-02-25 19:07:10.253784244 +0000
@@ -262,7 +262,8 @@
 // static
 void BrowserList::MoveBrowsersInWorkspaceToFront(
     const std::string& new_workspace) {
-  DCHECK(!new_workspace.empty());
+  if(new_workspace.empty())
+    return;
 
   BrowserList* instance = GetInstance();
 
