diff -r -u --color up/chromium/chrome/browser/net/proxy_config_monitor.h nw/chromium/chrome/browser/net/proxy_config_monitor.h
--- up/chromium/chrome/browser/net/proxy_config_monitor.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/net/proxy_config_monitor.h	2023-02-25 19:07:08.993776494 +0000
@@ -66,6 +66,7 @@
   // they're received, to allow tests to wait until all pending proxy
   // configuration changes have been applied.
   void FlushForTesting();
+  void UpdateProxyConfig(const net::ProxyConfigWithAnnotation& proxy_config);
 
  private:
   // net::ProxyConfigService::Observer implementation:
