diff -r -u --color up/chromium/chrome/browser/apps/platform_apps/browser_context_keyed_service_factories.cc nw/chromium/chrome/browser/apps/platform_apps/browser_context_keyed_service_factories.cc
--- up/chromium/chrome/browser/apps/platform_apps/browser_context_keyed_service_factories.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/apps/platform_apps/browser_context_keyed_service_factories.cc	2023-02-19 17:36:26.935442468 +0000
@@ -12,7 +12,7 @@
 
 void EnsureBrowserContextKeyedServiceFactoriesBuilt() {
   apps::AppLoadServiceFactory::GetInstance();
-  AppShortcutManagerFactory::GetInstance();
+  //AppShortcutManagerFactory::GetInstance();
   AppTerminationObserver::GetFactoryInstance();
 }
 
