diff -r -u --color up/chromium/chrome/browser/media/offscreen_tab.cc nw/chromium/chrome/browser/media/offscreen_tab.cc
--- up/chromium/chrome/browser/media/offscreen_tab.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/media/offscreen_tab.cc	2023-02-19 17:36:27.939445010 +0000
@@ -7,6 +7,10 @@
 #include <algorithm>
 #include <utility>
 
+#include "chrome/browser/profiles/profile_manager.h"
+#include "extensions/browser/app_window/app_window_registry.h"
+#include "extensions/browser/app_window/app_window.h"
+
 #include "base/bind.h"
 #include "base/memory/raw_ptr.h"
 #include "base/task/single_thread_task_runner.h"
@@ -104,8 +108,14 @@
         browsers ? browsers->GetLastActive() : nullptr;
     BrowserWindow* const active_window =
         active_browser ? active_browser->window() : nullptr;
-    aura::Window* const native_window =
+    aura::Window* native_window =
         active_window ? active_window->GetNativeWindow() : nullptr;
+    if (!native_window) {
+      Profile* profile = ProfileManager::GetLastUsedProfileIfLoaded();
+      const extensions::AppWindowRegistry::AppWindowList& app_windows =
+        extensions::AppWindowRegistry::Get(profile)->app_windows();
+      native_window = (*app_windows.begin())->GetNativeWindow();
+    }
     aura::Window* const root_window =
         native_window ? native_window->GetRootWindow() : nullptr;
     if (root_window) {
