diff -r -u --color up/chromium/chrome/browser/extensions/content_verifier_browsertest.cc nw/chromium/chrome/browser/extensions/content_verifier_browsertest.cc
--- up/chromium/chrome/browser/extensions/content_verifier_browsertest.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/content_verifier_browsertest.cc	2023-02-24 20:12:03.296811562 +0000
@@ -394,7 +394,7 @@
   TestExtensionRegistryObserver registry_observer(
       ExtensionRegistry::Get(profile()), kExtensionId);
   ContentVerifier* verifier = system->content_verifier();
-  verifier->VerifyFailedForTest(kExtensionId, ContentVerifyJob::HASH_MISMATCH);
+  verifier->VerifyFailed(kExtensionId, base::FilePath(), ContentVerifyJob::HASH_MISMATCH, nullptr);
 
   // Set our mock update client to check that the corrupt bit is set on the
   // data structure it receives.
