diff -r -u --color up/chromium/chrome/browser/resources/component_extension_resources.grd nw/chromium/chrome/browser/resources/component_extension_resources.grd
--- up/chromium/chrome/browser/resources/component_extension_resources.grd	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/resources/component_extension_resources.grd	2023-02-25 19:07:09.493779569 +0000
@@ -12,6 +12,9 @@
   </outputs>
   <release seq="1">
     <includes>
+      <include name="IDR_NWJS_DEFAULT_JS" file="nwjs/default.js" type="BINDATA" />
+      <include name="IDR_NWJS_DEFAULT_JS_NEWWIN" file="nwjs/newwin.js" type="BINDATA" />
+      
       <include name="IDR_NETWORK_SPEECH_SYNTHESIS_JS" file="network_speech_synthesis/tts_extension.js" type="BINDATA" />
 
       <if expr="chromeos_ash">
@@ -81,6 +84,12 @@
       <include name="IDS_READ_ANYTHING_DARK_PNG" file="side_panel/images/read_anything_dark.png" type="BINDATA" />
       <include name="IDS_READ_ANYTHING_YELLOW_PNG" file="side_panel/images/read_anything_yellow.png" type="BINDATA" />
       <include name="IDS_READ_ANYTHING_BLUE_PNG" file="side_panel/images/read_anything_blue.png" type="BINDATA" />
+      <include name="IDR_NWJS_DEFAPP_INDEX_HTML" file="nwjs_default_app/nw_blank.html" allowexternalscript="true" type="BINDATA" />
+      <include name="IDR_NWJS_DEFAPP_MAIN_JS"    file="nwjs_default_app/main.js" type="BINDATA" />
+      <include name="IDR_NWJS_NEWWINAPP_INDEX_HTML" file="nwjs_newwin_app/nw_blank.html" allowexternalscript="true" type="BINDATA" />
+      <include name="IDR_NWJS_NEWWINAPP_MAIN_JS"    file="nwjs_newwin_app/main.js" type="BINDATA" />
+      <include name="IDR_NWJS_NEWWINAPP_ROTATE_JS"    file="nwjs_newwin_app/rotate.js" type="BINDATA" />
+      <include name="IDR_NWJS_NEWWINAPP_ROTATE_SVG"    file="nwjs_newwin_app/rotate.svg" type="BINDATA" />
     </includes>
   </release>
 </grit>
Only in nw/chromium/chrome/browser/resources: empty.css
Only in nw/chromium/chrome/browser/resources: empty.html
Only in nw/chromium/chrome/browser/resources: empty.js
Only in nw/chromium/chrome/browser/resources: empty.json
Only in nw/chromium/chrome/browser/resources: empty.png
Only in nw/chromium/chrome/browser/resources: nwjs
Only in nw/chromium/chrome/browser/resources: nwjs_default_app
Only in nw/chromium/chrome/browser/resources: nwjs_newwin_app
