diff -r -u --color up/chromium/chrome/common/chrome_content_client.h nw/chromium/chrome/common/chrome_content_client.h
--- up/chromium/chrome/common/chrome_content_client.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/common/chrome_content_client.h	2023-02-25 19:07:10.877788082 +0000
@@ -43,6 +43,9 @@
 
   ChromeContentClient();
   ~ChromeContentClient() override;
+  void LoadNWAppAsExtension(base::Value::Dict* manifest,
+                            const base::FilePath& path,
+                            std::string* error) override;
 
   // The methods below are called by child processes to set the function
   // pointers for built-in plugins. We avoid linking these plugins into
@@ -56,6 +59,7 @@
 #endif
 
   void SetActiveURL(const GURL& url, std::string top_origin) override;
+  void SetNWReportURL(const GURL& url) override;
   void SetGpuInfo(const gpu::GPUInfo& gpu_info) override;
   void AddPlugins(std::vector<content::ContentPluginInfo>* plugins) override;
   void AddContentDecryptionModules(
