diff -r -u --color up/chromium/chrome/browser/ui/views/status_icons/status_icon_linux_wrapper.cc nw/chromium/chrome/browser/ui/views/status_icons/status_icon_linux_wrapper.cc
--- up/chromium/chrome/browser/ui/views/status_icons/status_icon_linux_wrapper.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/status_icons/status_icon_linux_wrapper.cc	2023-02-19 17:36:29.643449323 +0000
@@ -44,8 +44,7 @@
                                                const std::u16string& tool_tip)
     : status_icon_type_(status_icon_type),
       image_(GetBestImageRep(image)),
-      tool_tip_(tool_tip),
-      menu_model_(nullptr) {
+      tool_tip_(tool_tip) {
   status_icon->SetDelegate(this);
 }
 
@@ -72,8 +71,6 @@
 }
 
 StatusIconLinuxWrapper::~StatusIconLinuxWrapper() {
-  if (menu_model_)
-    menu_model_->RemoveObserver(this);
 }
 
 void StatusIconLinuxWrapper::SetImage(const gfx::ImageSkia& image) {
@@ -159,19 +156,11 @@
 }
 
 void StatusIconLinuxWrapper::UpdatePlatformContextMenu(
-    StatusIconMenuModel* model) {
+    ui::MenuModel* model) {
   if (!GetStatusIcon())
     return;
 
-  // If a menu already exists, remove ourself from its observer list.
-  if (menu_model_)
-    menu_model_->RemoveObserver(this);
-
   GetStatusIcon()->UpdatePlatformContextMenu(model);
-  menu_model_ = model;
-
-  if (model)
-    model->AddObserver(this);
 }
 
 ui::StatusIconLinux* StatusIconLinuxWrapper::GetStatusIcon() {
@@ -185,4 +174,4 @@
     case kTypeNone:
       return nullptr;
   }
-}
\ No newline at end of file
+}
