diff -r -u --color up/chromium/chrome/browser/net/profile_network_context_service.cc nw/chromium/chrome/browser/net/profile_network_context_service.cc
--- up/chromium/chrome/browser/net/profile_network_context_service.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/net/profile_network_context_service.cc	2023-02-25 19:07:08.993776494 +0000
@@ -608,6 +608,10 @@
   proxy_config_monitor_.FlushForTesting();
 }
 
+void ProfileNetworkContextService::UpdateProxyConfig(const net::ProxyConfigWithAnnotation& proxy_config) {
+  proxy_config_monitor_.UpdateProxyConfig(proxy_config);
+}
+
 void ProfileNetworkContextService::SetDiscardDomainReliabilityUploadsForTesting(
     bool value) {
   g_discard_domain_reliability_uploads_for_testing = new bool(value);
@@ -1008,6 +1012,12 @@
   // / IsolationInfos, so storage can be isolated on a per-site basis.
   network_context_params->require_network_isolation_key = true;
 
+  const base::CommandLine& cmd_line =
+    *base::CommandLine::ForCurrentProcess();
+  if (cmd_line.HasSwitch("disable-cookie-encryption")) {
+    network_context_params->enable_encrypted_cookies = false;
+  }
+
   network_context_params->block_trust_tokens =
       !PrivacySandboxSettingsFactory::GetForProfile(profile_)
            ->IsTrustTokensAllowed();
