diff -r -u --color up/chromium/chrome/browser/media/router/discovery/BUILD.gn nw/chromium/chrome/browser/media/router/discovery/BUILD.gn
--- up/chromium/chrome/browser/media/router/discovery/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/media/router/discovery/BUILD.gn	2023-02-19 17:36:27.939445010 +0000
@@ -15,7 +15,7 @@
     "//chrome/app:generated_resources",
     "//chrome/browser:browser_process",
     "//chrome/browser/media/router:data_decoder_util",
-    "//chrome/browser/media/router:logger_list",
+    #"//chrome/browser/media/router:logger_list",
     "//chrome/browser/media/router:media_router_feature",
     "//chrome/common:constants",
     "//components/media_router/browser",
