diff -r -u --color up/chromium/chrome/renderer/extensions/resource_request_policy.cc nw/chromium/chrome/renderer/extensions/resource_request_policy.cc
--- up/chromium/chrome/renderer/extensions/resource_request_policy.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/renderer/extensions/resource_request_policy.cc	2023-02-25 19:07:10.997788820 +0000
@@ -99,7 +99,10 @@
   // resources:
 
   // Empty urls (needed for some edge cases when we have empty urls).
-  if (frame_url.is_empty())
+  // NWJS: about:blank here when load iframe in document-start-end
+  // case starting from cr53
+    bool is_empty_origin = frame_url.is_empty() || frame_url == GURL(url::kAboutBlankURL);
+  if (is_empty_origin)
     return true;
 
   // Extensions requesting their own resources (frame_url check is for images,
