diff -r -u --color up/chromium/chrome/browser/lifetime/browser_close_manager.h nw/chromium/chrome/browser/lifetime/browser_close_manager.h
--- up/chromium/chrome/browser/lifetime/browser_close_manager.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/lifetime/browser_close_manager.h	2023-02-19 17:36:27.919444959 +0000
@@ -15,7 +15,7 @@
 // shutdown.
 class BrowserCloseManager : public base::RefCounted<BrowserCloseManager> {
  public:
-  BrowserCloseManager();
+  BrowserCloseManager(bool force = false, bool user_force = false);
 
   BrowserCloseManager(const BrowserCloseManager&) = delete;
   BrowserCloseManager& operator=(const BrowserCloseManager&) = delete;
@@ -60,6 +60,9 @@
   // The browser for which we are waiting for a callback to
   // OnBrowserReportCloseable.
   raw_ptr<Browser> current_browser_;
+  bool force_ = false;
+  bool user_force_ = false;
+
 };
 
 #endif  // CHROME_BROWSER_LIFETIME_BROWSER_CLOSE_MANAGER_H_
