diff -r -u --color up/chromium/chrome/browser/ui/views/sad_tab_view.cc nw/chromium/chrome/browser/ui/views/sad_tab_view.cc
--- up/chromium/chrome/browser/ui/views/sad_tab_view.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/sad_tab_view.cc	2023-02-19 17:36:29.627449282 +0000
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/views/sad_tab_view.h"
 
+#include "components/strings/grit/components_strings.h"
 #include <string>
 
 #include "base/metrics/histogram_macros.h"
@@ -564,7 +565,7 @@
       actions_container->AddChildView(std::make_unique<views::MdTextButton>(
           base::BindRepeating(&SadTabView::PerformAction,
                               base::Unretained(this), Action::BUTTON),
-          l10n_util::GetStringUTF16(GetButtonTitle())));
+          l10n_util::GetStringUTF16(IDS_CLOSE)));
   action_button_->SetProminent(true);
   action_button_->SetProperty(
       views::kFlexBehaviorKey,
