diff -r -u --color up/chromium/chrome/browser/webshare/share_service_impl.cc nw/chromium/chrome/browser/webshare/share_service_impl.cc
--- up/chromium/chrome/browser/webshare/share_service_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/webshare/share_service_impl.cc	2023-02-25 19:07:10.809787664 +0000
@@ -206,14 +206,6 @@
       return;
     }
 
-    // Check if at least one file is marked by the download protection service
-    // to send a ping to check this file type.
-    if (!should_check_url &&
-        safe_browsing::FileTypePolicies::GetInstance()->IsCheckedBinaryFile(
-            path)) {
-      should_check_url = true;
-    }
-
     // In the case where the original blob handle was to a native file (of
     // unknown size), the serialized data does not contain an accurate file
     // size. To handle this, the comparison against kMaxSharedFileBytes should
