diff -r -u --color up/chromium/chrome/browser/extensions/tab_helper.cc nw/chromium/chrome/browser/extensions/tab_helper.cc
--- up/chromium/chrome/browser/extensions/tab_helper.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/tab_helper.cc	2023-02-24 20:12:03.328811888 +0000
@@ -5,6 +5,7 @@
 #include "chrome/browser/extensions/tab_helper.h"
 
 #include <memory>
+#include "chrome/browser/ui/browser_window.h"
 
 #include "base/bind.h"
 #include "base/check_op.h"
@@ -348,11 +349,24 @@
   IPC_BEGIN_MESSAGE_MAP_WITH_PARAM(TabHelper, message, sender)
     IPC_MESSAGE_HANDLER(ExtensionHostMsg_ContentScriptsExecuting,
                         OnContentScriptsExecuting)
+    IPC_MESSAGE_HANDLER(ExtensionHostMsg_UpdateDraggableRegions,
+                        UpdateDraggableRegions)
     IPC_MESSAGE_UNHANDLED(handled = false)
   IPC_END_MESSAGE_MAP()
   return handled;
 }
 
+void TabHelper::UpdateDraggableRegions(
+    content::RenderFrameHost* sender,
+    const std::vector<DraggableRegion>& regions) {
+  if (sender->GetParent())
+    return;
+  Browser* browser = chrome::FindBrowserWithWebContents(web_contents());
+  if (!browser)
+    return;
+  browser->window()->UpdateDraggableRegions(regions);
+}
+
 void TabHelper::DidCloneToNewWebContents(WebContents* old_web_contents,
                                          WebContents* new_web_contents) {
   // When the WebContents that this is attached to is cloned, give the new clone
