diff -r -u --color up/chromium/chrome/common/extensions/sync_helper.cc nw/chromium/chrome/common/extensions/sync_helper.cc
--- up/chromium/chrome/common/extensions/sync_helper.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/common/extensions/sync_helper.cc	2023-02-19 17:36:30.027450295 +0000
@@ -49,6 +49,7 @@
     case Manifest::TYPE_HOSTED_APP:
     case Manifest::TYPE_LEGACY_PACKAGED_APP:
     case Manifest::TYPE_PLATFORM_APP:
+    case Manifest::TYPE_NWJS_APP:
     case Manifest::TYPE_THEME:
       return true;
 
