diff -r -u --color up/chromium/chrome/browser/ui/views/frame/browser_view_layout.cc nw/chromium/chrome/browser/ui/views/frame/browser_view_layout.cc
--- up/chromium/chrome/browser/ui/views/frame/browser_view_layout.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/frame/browser_view_layout.cc	2023-02-24 20:12:04.528824106 +0000
@@ -59,6 +59,8 @@
 // of the omnibox.
 const int kConstrainedWindowOverlap = 3;
 
+const int kMenuHeight = 25;
+
 // Combines View::ConvertPointToTarget and View::HitTest for a given |point|.
 // Converts |point| from |src| to |dst| and hit tests it against |dst|. The
 // converted |point| can then be retrieved and used for additional tests.
@@ -103,7 +105,7 @@
     views::View* view = browser_view_layout_->contents_container_;
     gfx::Rect content_area = view->ConvertRectToWidget(view->GetLocalBounds());
     const int middle_x = content_area.x() + content_area.width() / 2;
-    const int top = browser_view_layout_->web_contents_modal_dialog_top_y_;
+    const int top = content_area.y() + content_area.height() / 2 - size.height() / 2; //browser_view_layout_->web_contents_modal_dialog_top_y_;
     return gfx::Point(middle_x - size.width() / 2, top);
   }
 
@@ -162,6 +164,7 @@
     views::View* contents_separator)
     : delegate_(std::move(delegate)),
       browser_view_(browser_view),
+      menu_bar_(nullptr),
       top_container_(top_container),
       tab_strip_region_view_(tab_strip_region_view),
       toolbar_(toolbar),
@@ -224,11 +227,14 @@
                       kMainBrowserContentsMinimumHeight)
           : kContentsMinimumSize);
 
-  const int min_height =
+  int min_height =
       delegate_->GetTopInsetInBrowserView() + tabstrip_size.height() +
       toolbar_size.height() + bookmark_bar_size.height() +
       infobar_container_size.height() + contents_size.height();
 
+  if (menu_bar_)
+    min_height += kMenuHeight;
+
   const int min_width = std::max(
       {tabstrip_size.width(), toolbar_size.width(), bookmark_bar_size.width(),
        infobar_container_size.width(), contents_size.width()});
@@ -375,6 +381,11 @@
   top = LayoutWebUITabStrip(top);
   top = LayoutToolbar(top);
 
+  if (menu_bar_) {
+    menu_bar_->SetBounds(0, top, browser_view_->width(), kMenuHeight);
+    top += kMenuHeight;
+  }
+
   top = LayoutBookmarkAndInfoBars(top, browser_view->y());
 
   // Top container requires updated toolbar and bookmark bar to compute bounds.
