diff -r -u --color up/chromium/chrome/browser/ui/javascript_dialogs/javascript_tab_modal_dialog_manager_delegate_desktop.cc nw/chromium/chrome/browser/ui/javascript_dialogs/javascript_tab_modal_dialog_manager_delegate_desktop.cc
--- up/chromium/chrome/browser/ui/javascript_dialogs/javascript_tab_modal_dialog_manager_delegate_desktop.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/javascript_dialogs/javascript_tab_modal_dialog_manager_delegate_desktop.cc	2023-02-19 17:36:29.431448786 +0000
@@ -39,12 +39,14 @@
   // warnings until user interaction. If the current page has a delayed warning,
   // it'll have a user interaction observer attached. Show the warning
   // immediately in that case.
+#if 0
   safe_browsing::SafeBrowsingUserInteractionObserver* observer =
       safe_browsing::SafeBrowsingUserInteractionObserver::FromWebContents(
           web_contents_);
   if (observer) {
     observer->OnJavaScriptDialog();
   }
+#endif
 }
 
 void JavaScriptTabModalDialogManagerDelegateDesktop::DidCloseDialog() {
