diff -r -u --color up/chromium/chrome/browser/media/webrtc/desktop_capture_devices_util.cc nw/chromium/chrome/browser/media/webrtc/desktop_capture_devices_util.cc
--- up/chromium/chrome/browser/media/webrtc/desktop_capture_devices_util.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/media/webrtc/desktop_capture_devices_util.cc	2023-02-24 20:54:07.348518959 +0000
@@ -142,6 +142,7 @@
       display_surface, logical_surface, cursor, std::move(capture_handle));
 }
 
+#if 0
 std::u16string GetNotificationText(const std::u16string& application_title,
                                    bool capture_audio,
                                    content::DesktopMediaID::Type capture_type) {
@@ -176,6 +177,7 @@
   }
   return std::u16string();
 }
+#endif
 
 std::string DeviceNamePrefix(
     content::WebContents* web_contents,
@@ -279,6 +281,7 @@
             media_id);
   }
 
+#if 0
   // If required, register to display the notification for stream capture.
   std::unique_ptr<MediaStreamUI> notification_ui;
   if (display_notification) {
@@ -305,4 +308,7 @@
       ->GetMediaStreamCaptureIndicator()
       ->RegisterMediaStream(web_contents, out_devices,
                             std::move(notification_ui), application_title);
+#endif
+  std::unique_ptr<content::MediaStreamUI> ui;
+  return ui;
 }
