diff -r -u --color up/chromium/chrome/browser/translate/translate_frame_binder.cc nw/chromium/chrome/browser/translate/translate_frame_binder.cc
--- up/chromium/chrome/browser/translate/translate_frame_binder.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/translate/translate_frame_binder.cc	2023-02-19 17:36:29.039447794 +0000
@@ -14,6 +14,8 @@
 void BindContentTranslateDriver(
     content::RenderFrameHost* render_frame_host,
     mojo::PendingReceiver<translate::mojom::ContentTranslateDriver> receiver) {
+  return;
+#if 0
   // Only valid for the primary main frame.
   if (!render_frame_host->IsInPrimaryMainFrame())
     return;
@@ -29,6 +31,7 @@
     return;
 
   translate_client->translate_driver()->AddReceiver(std::move(receiver));
+#endif
 }
 
 }  // namespace translate
