diff -r -u --color up/chromium/chrome/browser/chrome_browser_main_win.cc nw/chromium/chrome/browser/chrome_browser_main_win.cc
--- up/chromium/chrome/browser/chrome_browser_main_win.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/chrome_browser_main_win.cc	2023-02-25 19:07:08.441773099 +0000
@@ -394,6 +394,7 @@
       l10n_util::GetStringUTF16(IDS_UNINSTALL_CLOSE_APP));
 }
 
+#if 0
 void MaybePostSettingsResetPrompt() {
   if (base::FeatureList::IsEnabled(safe_browsing::kSettingsResetPrompt)) {
     content::GetUIThreadTaskRunner({base::TaskPriority::BEST_EFFORT})
@@ -402,6 +403,7 @@
                        safe_browsing::MaybeShowSettingsResetPromptWithDelay));
   }
 }
+#endif
 
 // Updates all Progressive Web App launchers in |profile_dir| to the latest
 // version.
@@ -628,6 +630,7 @@
   // complete run of the Chrome Cleanup tool. If post-cleanup settings reset is
   // enabled, we delay checks for settings reset prompt until the scheduled
   // reset is finished.
+#if 0
   if (safe_browsing::PostCleanupSettingsResetter::IsEnabled() &&
       !base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kAppId)) {
     // Using last opened profiles, because we want to find reset the profile
@@ -641,6 +644,8 @@
   } else {
     MaybePostSettingsResetPrompt();
   }
+#endif
+
   // Record UMA data about whether the fault-tolerant heap is enabled.
   // Use a delayed task to minimize the impact on startup time.
   content::GetUIThreadTaskRunner({})->PostDelayedTask(
