diff -r -u --color up/chromium/chrome/browser/download/bubble/download_bubble_controller.cc nw/chromium/chrome/browser/download/bubble/download_bubble_controller.cc
--- up/chromium/chrome/browser/download/bubble/download_bubble_controller.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/download/bubble/download_bubble_controller.cc	2023-02-25 19:07:08.545773738 +0000
@@ -400,8 +400,10 @@
       commands.ExecuteCommand(command);
       break;
     case DownloadCommands::REVIEW:
+#if 0
       model->ReviewScanningVerdict(
           browser_->tab_strip_model()->GetActiveWebContents());
+#endif
       break;
     case DownloadCommands::RETRY:
       RetryDownload(model, command);
