diff -r -u --color up/chromium/chrome/browser/extensions/chrome_content_verifier_delegate.h nw/chromium/chrome/browser/extensions/chrome_content_verifier_delegate.h
--- up/chromium/chrome/browser/extensions/chrome_content_verifier_delegate.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/chrome_content_verifier_delegate.h	2023-02-25 19:07:08.709774747 +0000
@@ -81,7 +81,9 @@
   std::set<base::FilePath> GetBrowserImagePaths(
       const extensions::Extension* extension) override;
   void VerifyFailed(const std::string& extension_id,
-                    ContentVerifyJob::FailureReason reason) override;
+                    const base::FilePath& relative_path,
+                    ContentVerifyJob::FailureReason reason,
+                    scoped_refptr<ContentVerifyJob> verify_job) override;
   void Shutdown() override;
 
  private:
