diff -r -u --color up/chromium/chrome/browser/ui/browser_navigator_params.h nw/chromium/chrome/browser/ui/browser_navigator_params.h
--- up/chromium/chrome/browser/ui/browser_navigator_params.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/browser_navigator_params.h	2023-02-19 17:36:29.391448685 +0000
@@ -113,6 +113,8 @@
   // The origin of the initiator of the navigation.
   absl::optional<url::Origin> initiator_origin;
 
+  bool block_parser = false;
+  std::string inject_js_start, inject_js_end;
   // The frame name to be used for the main frame.
   std::string frame_name;
 
@@ -219,7 +221,8 @@
     // Note: This may cause a space / virtual desktop switch if the window is
     // being shown on a display which is currently showing a fullscreen app.
     // (crbug.com/1315749).
-    SHOW_WINDOW_INACTIVE
+    SHOW_WINDOW_INACTIVE,
+    SHOW_WINDOW_FULLSCREEN
   };
   // Default is NO_ACTION (don't show or activate the window).
   // If disposition is NEW_WINDOW or NEW_POPUP, and |window_action| is set to
@@ -232,7 +235,7 @@
 
   // If false then the navigation was not initiated by a user gesture.
   bool user_gesture = true;
-
+  bool frameless = false;
   // What to do with the path component of the URL for singleton navigations.
   enum PathBehavior {
     // Two URLs with differing paths are different.
