diff -r -u --color up/chromium/chrome/browser/browser_features.cc nw/chromium/chrome/browser/browser_features.cc
--- up/chromium/chrome/browser/browser_features.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/browser_features.cc	2023-02-24 20:12:03.080809362 +0000
@@ -31,7 +31,7 @@
 BASE_FEATURE(kDestroyProfileOnBrowserClose,
              "DestroyProfileOnBrowserClose",
 #if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 #else
              base::FEATURE_DISABLED_BY_DEFAULT);
 #endif
