diff -r -u --color up/chromium/chrome/installer/util/util_constants.cc nw/chromium/chrome/installer/util/util_constants.cc
--- up/chromium/chrome/installer/util/util_constants.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/installer/util/util_constants.cc	2023-02-24 20:54:08.852512024 +0000
@@ -216,8 +216,8 @@
 // TODO(gab): Rename setup.exe itself altogether and use the same binary for
 // Active Setup.
 const wchar_t kActiveSetupExe[] = L"chrmstp.exe";
-const wchar_t kChromeDll[] = L"chrome.dll";
-const wchar_t kChromeExe[] = L"chrome.exe";
+const wchar_t kChromeDll[] = L"nw.dll";
+const wchar_t kChromeExe[] = L"nw.exe";
 const wchar_t kChromeNewExe[] = L"new_chrome.exe";
 const wchar_t kChromeOldExe[] = L"old_chrome.exe";
 const wchar_t kChromeProxyExe[] = L"chrome_proxy.exe";
