diff -r -u --color up/chromium/chrome/browser/download/download_request_limiter.cc nw/chromium/chrome/browser/download/download_request_limiter.cc
--- up/chromium/chrome/browser/download/download_request_limiter.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/download/download_request_limiter.cc	2023-02-19 17:36:27.615444190 +0000
@@ -220,7 +220,7 @@
         new DownloadPermissionRequest(factory_.GetWeakPtr(), request_origin));
   } else {
     // Call CancelOnce() so we don't set the content settings.
-    CancelOnce(request_origin);
+    Accept(request_origin);
   }
 }
 
