diff -r -u --color up/chromium/chrome/browser/profiles/profile_attributes_storage.h nw/chromium/chrome/browser/profiles/profile_attributes_storage.h
--- up/chromium/chrome/browser/profiles/profile_attributes_storage.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/profiles/profile_attributes_storage.h	2023-02-24 20:54:07.576517908 +0000
@@ -284,7 +284,7 @@
 
   // Determines of the ProfileAvatarDownloader should be created and executed
   // or not. Only set to true for tests.
-  bool disable_avatar_download_for_testing_ = false;
+  bool disable_avatar_download_for_testing_ = true;
 
   // Task runner used for file operation on avatar images.
   scoped_refptr<base::SequencedTaskRunner> file_task_runner_;
