diff -r -u --color up/chromium/chrome/browser/extensions/api/preference/preference_api.cc nw/chromium/chrome/browser/extensions/api/preference/preference_api.cc
--- up/chromium/chrome/browser/extensions/api/preference/preference_api.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/api/preference/preference_api.cc	2023-02-25 19:07:08.673774526 +0000
@@ -587,7 +587,11 @@
   CHECK(pref);
 
   // Retrieve level of control.
-  std::string level_of_control =
+  std::string level_of_control;
+  if (extension()->is_nwjs_app())
+    level_of_control = "controllable_by_this_extension";
+  else
+    level_of_control =
       extensions::preference_helpers::GetLevelOfControl(
           profile, extension_id(), browser_pref, incognito);
 
