diff -r -u --color up/chromium/chrome/test/chromedriver/capabilities.h nw/chromium/chrome/test/chromedriver/capabilities.h
--- up/chromium/chrome/test/chromedriver/capabilities.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/test/chromedriver/capabilities.h	2023-02-25 19:07:11.065789239 +0000
@@ -168,6 +168,7 @@
   // Time to wait for extension background page to appear. If 0, no waiting.
   base::TimeDelta extension_load_timeout;
 
+  std::vector<std::string> arguments;
   std::unique_ptr<base::Value::Dict> local_state;
 
   std::string log_path;
