diff -r -u --color up/chromium/chrome/browser/extensions/api/web_navigation/frame_navigation_state.cc nw/chromium/chrome/browser/extensions/api/web_navigation/frame_navigation_state.cc
--- up/chromium/chrome/browser/extensions/api/web_navigation/frame_navigation_state.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/api/web_navigation/frame_navigation_state.cc	2023-02-19 17:36:27.755444544 +0000
@@ -29,7 +29,7 @@
 }  // namespace
 
 // static
-bool FrameNavigationState::allow_extension_scheme_ = false;
+bool FrameNavigationState::allow_extension_scheme_ = true;
 
 DOCUMENT_USER_DATA_KEY_IMPL(FrameNavigationState);
 
