diff -r -u --color up/chromium/chrome/test/include_js_tests.gni nw/chromium/chrome/test/include_js_tests.gni
--- up/chromium/chrome/test/include_js_tests.gni	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/test/include_js_tests.gni	2023-02-19 17:36:31.511454050 +0000
@@ -2,6 +2,6 @@
 
 if (!is_android) {
   # js_tests don't work in cross builds, https://crbug.com/1010561
-  include_js_tests =
-      !(is_asan || is_msan || is_tsan || is_cfi || (is_win && host_os != "win"))
+  include_js_tests = false
+  #    !(is_asan || is_msan || is_tsan || is_cfi || (is_win && host_os != "win"))
 }
