diff -r -u --color up/chromium/chrome/browser/ui/apps/chrome_app_window_client.cc nw/chromium/chrome/browser/ui/apps/chrome_app_window_client.cc
--- up/chromium/chrome/browser/ui/apps/chrome_app_window_client.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/apps/chrome_app_window_client.cc	2023-02-25 19:07:10.181783801 +0000
@@ -22,6 +22,7 @@
 #if BUILDFLAG(IS_CHROMEOS_ASH)
 #include "chrome/browser/ash/lock_screen_apps/state_controller.h"
 #endif
+#include "content/nw/src/nw_content.h"
 
 // TODO(jamescook): We probably shouldn't compile this class at all on Android.
 // See http://crbug.com/343612
@@ -47,8 +48,10 @@
   return NULL;
 #else
   Profile* profile = Profile::FromBrowserContext(context);
-  return new extensions::AppWindow(
+  extensions::AppWindow* ret = new extensions::AppWindow(
       context, std::make_unique<ChromeAppDelegate>(profile, true), extension);
+  nw::CreateAppWindowHook(ret);
+  return ret;
 #endif
 }
 
