diff -r -u --color up/chromium/chrome/chrome_elf/BUILD.gn nw/chromium/chrome/chrome_elf/BUILD.gn
--- up/chromium/chrome/chrome_elf/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/chrome_elf/BUILD.gn	2023-02-25 19:07:10.865788008 +0000
@@ -51,6 +51,7 @@
 # We should move chrome_result_codes.h to another target which does not bring
 # in the world.
 shared_library("chrome_elf") {
+  output_name = "nw_elf"
   sources = [
     "chrome_elf_main.cc",
     "chrome_elf_main.h",
