diff -r -u --color up/chromium/chrome/browser/ui/startup/startup_tab_provider.cc nw/chromium/chrome/browser/ui/startup/startup_tab_provider.cc
--- up/chromium/chrome/browser/ui/startup/startup_tab_provider.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/startup/startup_tab_provider.cc	2023-02-24 20:12:04.468823496 +0000
@@ -98,7 +98,7 @@
   const GURL reset_settings_url =
       settings_url.Resolve(chrome::kResetProfileSettingsSubPage);
   url_points_to_an_approved_settings_page = url == reset_settings_url;
-#if BUILDFLAG(IS_WIN)
+#if 0 //BUILDFLAG(IS_WIN)
   // On Windows, also allow a hash for the Chrome Cleanup Tool.
   const GURL reset_settings_url_with_cct_hash = reset_settings_url.Resolve(
       std::string("#") + settings::ResetSettingsHandler::kCctResetSettingsHash);
