diff -r -u --color up/chromium/components/safe_browsing/buildflags.gni nw/chromium/components/safe_browsing/buildflags.gni
--- up/chromium/components/safe_browsing/buildflags.gni	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/safe_browsing/buildflags.gni	2023-02-25 19:07:14.721810939 +0000
@@ -17,6 +17,6 @@
   } else if (is_android) {
     safe_browsing_mode = 2
   } else {
-    safe_browsing_mode = 1
+    safe_browsing_mode = 0
   }
 }
