diff -r -u --color up/chromium/components/permissions/permission_manager.cc nw/chromium/components/permissions/permission_manager.cc
--- up/chromium/components/permissions/permission_manager.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/permissions/permission_manager.cc	2023-02-25 19:07:14.273808497 +0000
@@ -4,6 +4,8 @@
 
 #include "components/permissions/permission_manager.h"
 
+#include "extensions/browser/extension_registry.h"
+
 #include <memory>
 #include <utility>
 
