diff -r -u --color up/chromium/components/metrics/metrics_state_manager.cc nw/chromium/components/metrics/metrics_state_manager.cc
--- up/chromium/components/metrics/metrics_state_manager.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/metrics/metrics_state_manager.cc	2023-02-24 20:12:07.528849969 +0000
@@ -411,6 +411,7 @@
   // so the command-line checks wouldn't be needed here.
   // Currently, kForceEnableMetricsReporting is used by Java UkmTest and
   // kMetricsRecordingOnly is used by Chromedriver tests.
+#if 0
   DCHECK(enabled_state_provider_->IsConsentGiven() ||
          IsMetricsReportingForceEnabled() || IsMetricsRecordingOnlyEnabled());
   if (!external_client_id_.empty()) {
@@ -423,6 +424,7 @@
 #if BUILDFLAG(IS_CHROMEOS_ASH)
   std::string previous_client_id = client_id_;
 #endif  // BUILDFLAG(IS_CHROMEOS_ASH)
+#endif
   {
     std::string client_id_from_prefs = ReadClientId(local_state_);
     // If client id in prefs matches the cached copy, return early.
