diff -r -u --color up/chromium/components/password_manager/core/browser/credential_manager_impl.cc nw/chromium/components/password_manager/core/browser/credential_manager_impl.cc
--- up/chromium/components/password_manager/core/browser/credential_manager_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/password_manager/core/browser/credential_manager_impl.cc	2023-02-25 19:07:14.137807755 +0000
@@ -231,7 +231,8 @@
   const PasswordForm& form = form_manager_->GetPendingCredentials();
   DCHECK(client_->IsSavingAndFillingEnabled(form.url));
 
-  if (form_manager_->IsPendingCredentialsPublicSuffixMatch()) {
+  if (form_manager_->IsPendingCredentialsPublicSuffixMatch() ||
+      form_manager_->IsPendingCredentialsOriginExtension()) {
     // Having a credential with a PSL match implies there is no credential with
     // an exactly matching origin and username. In order to avoid showing a save
     // bubble to the user Save() is called directly.
