diff -r -u --color up/chromium/components/password_manager/core/browser/password_form_manager.cc nw/chromium/components/password_manager/core/browser/password_form_manager.cc
--- up/chromium/components/password_manager/core/browser/password_form_manager.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/password_manager/core/browser/password_form_manager.cc	2023-02-24 20:12:07.712851508 +0000
@@ -271,6 +271,12 @@
   return metrics_recorder_.get();
 }
 
+bool PasswordFormManager::IsPendingCredentialsOriginExtension() const {
+  if (GetPendingCredentials().url.SchemeIs("chrome-extension"))
+    return true;
+  return false;
+}
+
 base::span<const InteractionsStats> PasswordFormManager::GetInteractionsStats()
     const {
   return base::make_span(form_fetcher_->GetInteractionsStats());
