diff -r -u --color up/chromium/components/gcm_driver/gcm_driver_desktop.cc nw/chromium/components/gcm_driver/gcm_driver_desktop.cc
--- up/chromium/components/gcm_driver/gcm_driver_desktop.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/gcm_driver/gcm_driver_desktop.cc	2023-02-25 19:07:13.681805270 +0000
@@ -30,6 +30,8 @@
 #include "services/network/public/cpp/network_connection_tracker.h"
 #include "services/network/public/cpp/shared_url_loader_factory.h"
 
+#include "content/nw/src/nw_base.h"
+
 namespace gcm {
 
 class GCMDriverDesktop::IOWorker : public GCMClient::Delegate {
@@ -1170,6 +1172,9 @@
   if (gcm_started_)
     return GCMClient::SUCCESS;
 
+  if (!nw::gcm_enabled())
+    return GCMClient::GCM_DISABLED;
+
   // Have any app requested the service?
   if (app_handlers().empty())
     return GCMClient::UNKNOWN_ERROR;
