diff -r -u --color up/chromium/components/resources/BUILD.gn nw/chromium/components/resources/BUILD.gn
--- up/chromium/components/resources/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/resources/BUILD.gn	2023-02-25 19:07:14.705810852 +0000
@@ -133,3 +133,9 @@
     public_deps = [ "//device/vr/buildflags" ]
   }
 }
+
+copy("about_credits_nw") {
+  sources = [ about_credits_file ]
+  outputs = [ "$root_build_dir/credits.html" ]
+  deps = [ ":about_credits" ]
+}
