diff -r -u --color up/chromium/components/pdf/renderer/internal_plugin_renderer_helpers.cc nw/chromium/components/pdf/renderer/internal_plugin_renderer_helpers.cc
--- up/chromium/components/pdf/renderer/internal_plugin_renderer_helpers.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/pdf/renderer/internal_plugin_renderer_helpers.cc	2023-02-24 20:54:11.244500994 +0000
@@ -57,7 +57,7 @@
   }
 
   // Only create the in-process plugin within a PDF renderer.
-  CHECK(IsPdfRenderer());
+  //CHECK(IsPdfRenderer());
 
   // Origins allowed to embed the internal plugin are trusted (the PDF viewer
   // and Print Preview), and should never directly create the in-process plugin.
@@ -65,7 +65,7 @@
   //
   // See crbug.com/1259635 and crbug.com/1261758 for examples of previous bugs.
   CHECK(!delegate->IsAllowedOrigin(frame->GetSecurityOrigin()));
-  CHECK(parent_frame->IsWebRemoteFrame());
+  //CHECK(parent_frame->IsWebRemoteFrame());
 
   mojo::AssociatedRemote<pdf::mojom::PdfService> pdf_service;
   render_frame->GetRemoteAssociatedInterfaces()->GetInterface(
