diff -r -u --color up/chromium/components/keep_alive_registry/keep_alive_registry.h nw/chromium/components/keep_alive_registry/keep_alive_registry.h
--- up/chromium/components/keep_alive_registry/keep_alive_registry.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/keep_alive_registry/keep_alive_registry.h	2023-02-25 19:07:13.765805728 +0000
@@ -76,10 +76,12 @@
   KeepAliveRegistry();
   ~KeepAliveRegistry();
 
+ public:
   // Add/Remove entries. Do not use directly, use ScopedKeepAlive instead.
   void Register(KeepAliveOrigin origin, KeepAliveRestartOption restart);
   void Unregister(KeepAliveOrigin origin, KeepAliveRestartOption restart);
 
+ private:
   // Methods called when a specific aspect of the state of the registry changes.
   void OnKeepAliveStateChanged(bool new_keeping_alive);
   void OnRestartAllowedChanged(bool new_restart_allowed);
