diff -r -u --color up/chromium/components/favicon/core/favicon_driver_impl.cc nw/chromium/components/favicon/core/favicon_driver_impl.cc
--- up/chromium/components/favicon/core/favicon_driver_impl.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/favicon/core/favicon_driver_impl.cc	2023-02-19 17:36:32.811457334 +0000
@@ -33,7 +33,7 @@
         favicon_service_, this, FaviconDriverObserver::TOUCH_LARGEST));
   } else {
     handlers_.push_back(std::make_unique<FaviconHandler>(
-        favicon_service_, this, FaviconDriverObserver::NON_TOUCH_16_DIP));
+        favicon_service_, this, FaviconDriverObserver::NON_TOUCH_LARGEST));
   }
 }
 
