diff -r -u --color up/chromium/components/os_crypt/os_crypt_mac.mm nw/chromium/components/os_crypt/os_crypt_mac.mm
--- up/chromium/components/os_crypt/os_crypt_mac.mm	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/os_crypt/os_crypt_mac.mm	2023-02-24 20:54:11.148501437 +0000
@@ -135,7 +135,7 @@
 
 void OSCryptImpl::SetRawEncryptionKey(const std::string& raw_key) {
   base::AutoLock auto_lock(OSCryptImpl::GetLock());
-  DCHECK(!cached_encryption_key_) << "Encryption key already set.";
+  //DCHECK(!cached_encryption_key_) << "Encryption key already set.";
   cached_encryption_key_ =
       crypto::SymmetricKey::Import(crypto::SymmetricKey::AES, raw_key);
   key_is_cached_ = true;
