diff -r -u --color up/chromium/components/sessions/core/session_service_commands.cc nw/chromium/components/sessions/core/session_service_commands.cc
--- up/chromium/components/sessions/core/session_service_commands.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/sessions/core/session_service_commands.cc	2023-02-19 17:36:34.087460558 +0000
@@ -165,9 +165,10 @@
   PERSISTED_SHOW_STATE_MAXIMIZED = 3,
   // SHOW_STATE_INACTIVE (4) never persisted.
   PERSISTED_SHOW_STATE_FULLSCREEN = 5,
-  PERSISTED_SHOW_STATE_DETACHED_DEPRECATED = 6,
-  PERSISTED_SHOW_STATE_DOCKED_DEPRECATED = 7,
-  PERSISTED_SHOW_STATE_END = 8,
+  // hidden not persisted
+  PERSISTED_SHOW_STATE_DETACHED_DEPRECATED = 7,
+  PERSISTED_SHOW_STATE_DOCKED_DEPRECATED = 8,
+  PERSISTED_SHOW_STATE_END = 9,
 };
 
 // Assert to ensure PersistedWindowShowState is updated if ui::WindowShowState
@@ -191,6 +192,7 @@
       return PERSISTED_SHOW_STATE_FULLSCREEN;
     case ui::SHOW_STATE_DEFAULT:
     case ui::SHOW_STATE_INACTIVE:
+    case ui::SHOW_STATE_HIDDEN:
       return PERSISTED_SHOW_STATE_NORMAL;
 
     case ui::SHOW_STATE_END:
