diff -r -u --color up/chromium/components/services/storage/dom_storage/local_storage_impl.cc nw/chromium/components/services/storage/dom_storage/local_storage_impl.cc
--- up/chromium/components/services/storage/dom_storage/local_storage_impl.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/services/storage/dom_storage/local_storage_impl.cc	2023-02-19 17:36:34.063460497 +0000
@@ -155,13 +155,13 @@
 
   // To avoid excessive IO we apply limits to the amount of data being written
   // and the frequency of writes.
-  static const size_t kMaxBytesPerHour = kPerStorageAreaQuota;
-  static constexpr int kMaxCommitsPerHour = 60;
+  //static const size_t kMaxBytesPerHour = std::numeric_limits<size_t>::max();
+  static constexpr int kMaxCommitsPerHour = std::numeric_limits<int32_t>::max();
 
   StorageAreaImpl::Options options;
-  options.max_size = kPerStorageAreaQuota + kPerStorageAreaOverQuotaAllowance;
+  options.max_size = std::numeric_limits<size_t>::max();
   options.default_commit_delay = kCommitDefaultDelaySecs;
-  options.max_bytes_per_hour = kMaxBytesPerHour;
+  options.max_bytes_per_hour = std::numeric_limits<int>::max();
   options.max_commits_per_hour = kMaxCommitsPerHour;
 #if BUILDFLAG(IS_ANDROID)
     options.cache_mode = StorageAreaImpl::CacheMode::KEYS_ONLY_WHEN_POSSIBLE;
