diff -r -u --color up/chromium/components/translate/translate_internals/translate_internals_handler.cc nw/chromium/components/translate/translate_internals/translate_internals_handler.cc
--- up/chromium/components/translate/translate_internals/translate_internals_handler.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/translate/translate_internals/translate_internals_handler.cc	2023-02-19 17:36:34.831462438 +0000
@@ -160,6 +160,7 @@
 
 void TranslateInternalsHandler::OnRemovePrefItem(
     const base::Value::List& args) {
+#if 0
   std::unique_ptr<translate::TranslatePrefs> translate_prefs =
       GetTranslateClient()->GetTranslatePrefs();
 
@@ -190,6 +191,7 @@
   }
 
   SendPrefsToJs();
+#endif
 }
 
 void TranslateInternalsHandler::OnSetRecentTargetLanguage(
