diff -r -u --color up/chromium/components/cookie_config/cookie_store_util.cc nw/chromium/components/cookie_config/cookie_store_util.cc
--- up/chromium/components/cookie_config/cookie_store_util.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/cookie_config/cookie_store_util.cc	2023-02-25 19:07:13.445803878 +0000
@@ -15,6 +15,8 @@
     BUILDFLAG(IS_CHROMEOS)
 namespace {
 
+bool g_should_encrypt = true;
+
 // Use the operating system's mechanisms to encrypt cookies before writing
 // them to persistent store.  Currently this only is done with desktop OS's
 // because ChromeOS and Android already protect the entire profile contents.
@@ -40,7 +42,7 @@
   // rewritten.
   return false;
 #else
-  return true;
+  return g_should_encrypt;
 #endif
 }
 
@@ -66,6 +68,11 @@
 }
 #else   // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_APPLE) || BUILDFLAG(IS_LINUX) ||
         // BUILDFLAG(IS_CHROMEOS)
+
+void SetEnableCookieCrypto(bool enable) {
+  g_should_encrypt = enable;
+}
+
 net::CookieCryptoDelegate* GetCookieCryptoDelegate() {
   return NULL;
 }
