diff -r -u --color up/chromium/components/safe_browsing/core/browser/db/v4_update_protocol_manager.cc nw/chromium/components/safe_browsing/core/browser/db/v4_update_protocol_manager.cc
--- up/chromium/components/safe_browsing/core/browser/db/v4_update_protocol_manager.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/safe_browsing/core/browser/db/v4_update_protocol_manager.cc	2023-02-24 20:54:11.664499057 +0000
@@ -315,6 +315,7 @@
     return;
   }
 
+#if 0
   net::NetworkTrafficAnnotationTag traffic_annotation =
       net::DefineNetworkTrafficAnnotation("safe_browsing_v4_update", R"(
         semantics {
@@ -358,7 +359,7 @@
                      base::Unretained(this)));
 
   request_ = std::move(loader);
-
+#endif
   // Begin the update request timeout.
   timeout_timer_.Start(FROM_HERE, base::Seconds(kV4TimerUpdateWaitSecMax), this,
                        &V4UpdateProtocolManager::HandleTimeout);
