diff -r -u --color up/chromium/components/metrics/metrics_service.cc nw/chromium/components/metrics/metrics_service.cc
--- up/chromium/components/metrics/metrics_service.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/metrics/metrics_service.cc	2023-02-19 17:36:33.095458052 +0000
@@ -262,7 +262,7 @@
       client_(client),
       local_state_(local_state),
       recording_state_(UNSET),
-      test_mode_active_(false),
+      test_mode_active_(true),
       state_(CONSTRUCTED),
       idle_since_last_transmission_(false),
       session_id_(-1) {
@@ -345,9 +345,9 @@
 }
 
 void MetricsService::Start() {
-  HandleIdleSinceLastTransmission(false);
+  test_mode_active_ = true;
   EnableRecording();
-  EnableReporting();
+  DisableReporting();
 }
 
 void MetricsService::StartRecordingForTests() {
