diff -r -u --color up/chromium/components/nacl/loader/BUILD.gn nw/chromium/components/nacl/loader/BUILD.gn
--- up/chromium/components/nacl/loader/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/nacl/loader/BUILD.gn	2023-02-24 20:12:07.548850136 +0000
@@ -153,6 +153,12 @@
     if (!export_libcxxabi_from_executables) {
       configs -= [ "//build/config/compiler:export_dynamic" ]
     }
+    if (is_linux) {
+      if (!is_component_build) {
+        # Set rpath to find our own libfreetype even in a non-component build.
+        configs += [ "//build/config/gcc:rpath_for_built_shared_libraries" ]
+      }
+    }
 
     data_deps =
         [ "//native_client/src/trusted/service_runtime/linux:bootstrap" ]
