diff -r -u --color up/chromium/components/nacl/zygote/nacl_fork_delegate_linux.cc nw/chromium/components/nacl/zygote/nacl_fork_delegate_linux.cc
--- up/chromium/components/nacl/zygote/nacl_fork_delegate_linux.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/nacl/zygote/nacl_fork_delegate_linux.cc	2023-02-25 19:07:13.873806316 +0000
@@ -403,7 +403,7 @@
   write_pickle.WriteInt(pid);
   write_pickle.WriteBool(known_dead);
 
-  const std::vector<int> empty_fds;
+  const std::vector<int> empty_fds = std::vector<int>();
   char reply_buf[kNaClMaxIPCMessageLength];
   ssize_t reply_size = 0;
   bool got_reply =
