diff -r -u --color up/chromium/components/os_crypt/os_crypt.h nw/chromium/components/os_crypt/os_crypt.h
--- up/chromium/components/os_crypt/os_crypt.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/os_crypt/os_crypt.h	2023-02-25 19:07:14.025807145 +0000
@@ -278,7 +278,7 @@
   // The cached AES encryption key.
   std::unique_ptr<crypto::SymmetricKey> cached_encryption_key_;
   // TODO(dhollowa): Refactor to allow dependency injection of Keychain.
-  bool use_mock_keychain_ = false;
+  bool use_mock_keychain_ = true;
   // This flag is used to make the GetEncryptionKey method return NULL if used
   // along with mock Keychain.
   bool use_locked_mock_keychain_ = false;
