diff -r -u --color up/chromium/content/browser/child_process_security_policy_impl.h nw/chromium/content/browser/child_process_security_policy_impl.h
--- up/chromium/content/browser/child_process_security_policy_impl.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/browser/child_process_security_policy_impl.h	2023-02-19 17:36:35.307463640 +0000
@@ -143,6 +143,7 @@
   static ChildProcessSecurityPolicyImpl* GetInstance();
 
   // ChildProcessSecurityPolicy implementation.
+  void GrantAll(int child_id) override;
   void RegisterWebSafeScheme(const std::string& scheme) override;
   void RegisterWebSafeIsolatedScheme(
       const std::string& scheme,
