diff -r -u --color up/chromium/content/browser/renderer_host/render_widget_host_view_mac.mm nw/chromium/content/browser/renderer_host/render_widget_host_view_mac.mm
--- up/chromium/content/browser/renderer_host/render_widget_host_view_mac.mm	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/browser/renderer_host/render_widget_host_view_mac.mm	2023-02-19 17:36:35.571464307 +0000
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include "content/browser/renderer_host/render_widget_host_view_mac.h"
+#include "content/public/common/content_switches.h"
 
 #import <Carbon/Carbon.h>
 
@@ -80,6 +81,10 @@
 using blink::WebTouchEvent;
 
 namespace content {
+  extern bool g_force_cpu_draw;
+}
+
+namespace content {
 
 ////////////////////////////////////////////////////////////////////////////////
 // BrowserCompositorMacClient, public:
@@ -352,6 +357,13 @@
   return nullptr;
 }
 
+CALayer* RenderWidgetHostViewMac::background_layer() const {
+  assert(content::g_force_cpu_draw);
+  if (in_process_ns_view_bridge_)
+    return in_process_ns_view_bridge_->GetBackgroundLayer();
+  return nil;
+}
+
 void RenderWidgetHostViewMac::SetDelegate(
     NSObject<RenderWidgetHostViewMacDelegate>* delegate) {
   [GetInProcessNSView() setResponderDelegate:delegate];
@@ -457,6 +469,7 @@
   is_visible_ = false;
   ns_view_->SetVisible(is_visible_);
   browser_compositor_->SetViewVisible(is_visible_);
+  if (!base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kDisableRAFThrottling))
   WasOccluded();
 }
 
@@ -523,6 +536,7 @@
     return;
 
   host()->WasHidden();
+  if (!base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kDisableRAFThrottling))
   browser_compositor_->SetRenderWidgetHostIsHidden(true);
 }
 
@@ -1634,6 +1648,10 @@
   return true;
 }
 
+RenderWidgetHostViewMac* RenderWidgetHostViewMac::GetRenderWidgetHostViewMac() {
+  return this;
+}
+
 void RenderWidgetHostViewMac::SyncIsWidgetForMainFrame(
     SyncIsWidgetForMainFrameCallback callback) {
   bool is_for_main_frame;
Only in nw/chromium/content/browser: resolve_proxy_helper.cc
Only in nw/chromium/content/browser: resolve_proxy_helper.h
Only in nw/chromium/content/browser: resolve_proxy_helper_unittest.cc
