diff -r -u --color up/chromium/content/browser/renderer_host/render_widget_host_input_event_router.cc nw/chromium/content/browser/renderer_host/render_widget_host_input_event_router.cc
--- up/chromium/content/browser/renderer_host/render_widget_host_input_event_router.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/browser/renderer_host/render_widget_host_input_event_router.cc	2023-02-19 17:36:35.563464287 +0000
@@ -7,6 +7,9 @@
 #include <algorithm>
 #include <deque>
 #include <memory>
+
+#include "content/public/browser/render_frame_host.h"
+
 #include <vector>
 
 #include "base/containers/cxx20_erase.h"
@@ -1970,8 +1973,13 @@
     touch_emulator_->SetDeviceScaleFactor(last_device_scale_factor_);
   if (auto* cursor_manager = last_mouse_move_root_view_->GetCursorManager()) {
     for (auto it : owner_map_) {
-      if (it.second)
+      if (it.second) {
+        if (touch_emulator_ && touch_emulator_->rfh_limit()) {
+          if (touch_emulator_->rfh_limit()->GetView() != it.second.get())
+            continue;
+        }
         cursor_manager->UpdateCursor(it.second.get(), cursor);
+      }
     }
   }
 }
