diff -r -u --color up/chromium/content/public/browser/web_contents.cc nw/chromium/content/public/browser/web_contents.cc
--- up/chromium/content/public/browser/web_contents.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/public/browser/web_contents.cc	2023-02-19 17:36:35.787464853 +0000
@@ -11,6 +11,14 @@
 
 namespace content {
 
+bool WebContents::is_silent_printing() {
+  return silent_printing_;
+}
+
+void WebContents::set_silent_printing(bool flag) {
+  silent_printing_ = flag;
+}
+
 WebContents::CreateParams::CreateParams(BrowserContext* context,
                                         base::Location creator_location)
     : CreateParams(context, nullptr, creator_location) {}
