diff -r -u --color up/chromium/content/public/browser/web_contents.h nw/chromium/content/public/browser/web_contents.h
--- up/chromium/content/public/browser/web_contents.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/public/browser/web_contents.h	2023-02-19 17:36:35.787464853 +0000
@@ -338,6 +338,11 @@
 
   ~WebContents() override = default;
 
+  virtual bool is_silent_printing();
+  virtual void set_silent_printing(bool);
+
+  bool silent_printing_ = false;
+
   // Intrinsic tab state -------------------------------------------------------
 
   // Gets/Sets the delegate.
