diff -r -u --color up/chromium/content/browser/worker_host/shared_worker_service_impl.cc nw/chromium/content/browser/worker_host/shared_worker_service_impl.cc
--- up/chromium/content/browser/worker_host/shared_worker_service_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/browser/worker_host/shared_worker_service_impl.cc	2023-02-25 19:07:16.453820380 +0000
@@ -191,7 +191,9 @@
     return;
   }
   auto partition_domain = site_instance->GetPartitionDomain(storage_partition_);
-  SharedWorkerInstance instance(info->url, info->options->type,
+  SharedWorkerInstance instance(
+                                info->is_node_js, info->root_path,
+                                info->url, info->options->type,
                                 info->options->credentials, info->options->name,
                                 storage_key, creation_context_type);
   host = CreateWorker(
