diff -r -u --color up/chromium/content/public/renderer/content_renderer_client.h nw/chromium/content/public/renderer/content_renderer_client.h
--- up/chromium/content/public/renderer/content_renderer_client.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/public/renderer/content_renderer_client.h	2023-02-25 19:07:16.565820990 +0000
@@ -12,6 +12,8 @@
 #include <string>
 #include <vector>
 
+#include "base/files/file_path.h"
+
 #include "base/callback_forward.h"
 #include "base/files/file_path.h"
 #include "base/memory/ref_counted.h"
@@ -48,6 +50,7 @@
 
 namespace blink {
 class WebElement;
+class WebString;
 class WebFrame;
 class WebLocalFrame;
 class WebPlugin;
@@ -82,10 +85,18 @@
 
 namespace content {
 class RenderFrame;
+class RenderView;
 
 // Embedder API for participating in renderer logic.
 class CONTENT_EXPORT ContentRendererClient {
  public:
+  virtual base::FilePath GetRootPath();
+  virtual void willHandleNavigationPolicy(RenderFrame* rv,
+                                          blink::WebFrame* frame,
+                                          const blink::WebURLRequest& request,
+                                          blink::WebNavigationPolicy* policy,
+                                          blink::WebString* manifest,
+                                          bool new_win) {}
   virtual ~ContentRendererClient() {}
 
   // Notifies us that the RenderThread has been created.
