diff -r -u --color up/chromium/content/app_shim_remote_cocoa/ns_view_bridge_factory_impl.mm nw/chromium/content/app_shim_remote_cocoa/ns_view_bridge_factory_impl.mm
--- up/chromium/content/app_shim_remote_cocoa/ns_view_bridge_factory_impl.mm	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/app_shim_remote_cocoa/ns_view_bridge_factory_impl.mm	2023-02-25 19:07:15.973817763 +0000
@@ -67,6 +67,8 @@
  private:
   void OnMojoDisconnect() { delete this; }
 
+  content::RenderWidgetHostViewMac* GetRenderWidgetHostViewMac() override { return nullptr; }
+
   std::unique_ptr<blink::WebCoalescedInputEvent> TranslateEvent(
       const blink::WebInputEvent& web_event) {
     return std::make_unique<blink::WebCoalescedInputEvent>(
