diff -r -u --color up/chromium/content/public/common/content_features.cc nw/chromium/content/public/common/content_features.cc
--- up/chromium/content/public/common/content_features.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/public/common/content_features.cc	2023-02-25 19:07:16.561820969 +0000
@@ -16,6 +16,9 @@
 
 namespace features {
 
+const base::Feature kNWNewWin{
+    "nw2", base::FEATURE_ENABLED_BY_DEFAULT};
+
 // All features in alphabetical order.
 
 // Enables content-initiated, main frame navigations to data URLs.
@@ -790,7 +793,7 @@
 // at a time.
 BASE_FEATURE(kProcessSharingWithDefaultSiteInstances,
              "ProcessSharingWithDefaultSiteInstances",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT); //NWJS#7197
 
 // Whether cross-site frames should get their own SiteInstance even when
 // strict site isolation is disabled. These SiteInstances will still be
@@ -1123,7 +1126,7 @@
 // This feature turns on site isolation support in <webview> guests.
 BASE_FEATURE(kSiteIsolationForGuests,
              "SiteIsolationForGuests",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 // When enabled, OOPIFs will not try to reuse compatible processes from
 // unrelated tabs.
@@ -1136,7 +1139,7 @@
 // This feature is only consulted in site-per-process mode.
 BASE_FEATURE(kSpareRendererForSitePerProcess,
              "SpareRendererForSitePerProcess",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kStopVideoCaptureOnScreenLock,
              "StopVideoCaptureOnScreenLock",
