diff -r -u --color up/chromium/content/public/browser/desktop_streams_registry.h nw/chromium/content/public/browser/desktop_streams_registry.h
--- up/chromium/content/public/browser/desktop_streams_registry.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/public/browser/desktop_streams_registry.h	2023-02-25 19:07:16.533820816 +0000
@@ -52,7 +52,7 @@
       int render_frame_id,
       const url::Origin& origin,
       std::string* extension_name,
-      const DesktopStreamRegistryType type) = 0;
+      const DesktopStreamRegistryType type, bool nodejs = false) = 0;
 };
 
 }  // namespace content
