diff -r -u --color up/chromium/content/public/common/content_switches.h nw/chromium/content/public/common/content_switches.h
--- up/chromium/content/public/common/content_switches.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/public/common/content_switches.h	2023-02-24 20:12:09.484866334 +0000
@@ -52,7 +52,12 @@
 CONTENT_EXPORT extern const char kDisableFileSystem[];
 CONTENT_EXPORT extern const char kDisableGestureRequirementForPresentation[];
 CONTENT_EXPORT extern const char kDisableGpu[];
+CONTENT_EXPORT extern const char kDisableTransparency[];
+CONTENT_EXPORT extern const char kForceCpuDraw[];
 CONTENT_EXPORT extern const char kDisableGpuCompositing[];
+CONTENT_EXPORT extern const char kDisableRAFThrottling[];
+CONTENT_EXPORT extern const char kEnableNodeWorker[];
+CONTENT_EXPORT extern const char kNWAppPath[];
 CONTENT_EXPORT extern const char kDisableGpuEarlyInit[];
 CONTENT_EXPORT extern const char kDisableGpuMemoryBufferCompositorResources[];
 CONTENT_EXPORT extern const char kDisableGpuMemoryBufferVideoFrames[];
