diff -r -u --color up/chromium/content/browser/devtools/devtools_http_handler.cc nw/chromium/content/browser/devtools/devtools_http_handler.cc
--- up/chromium/content/browser/devtools/devtools_http_handler.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/browser/devtools/devtools_http_handler.cc	2023-02-25 19:07:16.073818309 +0000
@@ -39,6 +39,8 @@
 #include "content/public/browser/devtools_frontend_host.h"
 #include "content/public/browser/devtools_manager_delegate.h"
 #include "content/public/browser/devtools_socket_factory.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/web_contents.h"
 #include "content/public/common/content_client.h"
 #include "content/public/common/url_constants.h"
 #include "content/public/common/user_agent.h"
@@ -70,10 +72,14 @@
 const base::FilePath::CharType kDevToolsActivePortFileName[] =
     FILE_PATH_LITERAL("DevToolsActivePort");
 
+#if defined(NWJS_SDK)
 const char kDevToolsHandlerThreadName[] = "Chrome_DevToolsHandlerThread";
+#endif
 
 const char kPageUrlPrefix[] = "/devtools/page/";
+#if defined(NWJS_SDK)
 const char kBrowserUrlPrefix[] = "/devtools/browser";
+#endif
 
 const char kTargetIdField[] = "id";
 const char kTargetParentIdField[] = "parentId";
@@ -724,6 +730,10 @@
   std::sort(agent_hosts.begin(), agent_hosts.end(), TimeComparator);
   base::Value::List list_value;
   for (auto& agent_host : agent_hosts) {
+    WebContents* web_contents = agent_host->GetWebContents();
+    if (web_contents && web_contents->GetPrimaryMainFrame())
+      if (!web_contents->GetPrimaryMainFrame()->context_created())
+        continue;
     // TODO(caseq): figure out if it makes sense exposing tab target to
     // HTTP clients and potentially compatibility risks involved.
     if (agent_host->GetType() != DevToolsAgentHost::kTypeTab)
@@ -815,6 +825,7 @@
     const base::FilePath& output_directory,
     const base::FilePath& debug_frontend_dir)
     : delegate_(delegate) {
+#if defined(NWJS_SDK)
   browser_guid_ = delegate_->IsBrowserTargetDiscoverable()
                       ? kBrowserUrlPrefix
                       : base::StringPrintf("%s/%s", kBrowserUrlPrefix,
@@ -832,6 +843,7 @@
                        output_directory, debug_frontend_dir, browser_guid_,
                        delegate_->HasBundledFrontendResources()));
   }
+#endif
 }
 
 void DevToolsHttpHandler::ServerStarted(
