diff -r -u --color up/chromium/content/app_shim_remote_cocoa/render_widget_host_ns_view_host_helper.h nw/chromium/content/app_shim_remote_cocoa/render_widget_host_ns_view_host_helper.h
--- up/chromium/content/app_shim_remote_cocoa/render_widget_host_ns_view_host_helper.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/app_shim_remote_cocoa/render_widget_host_ns_view_host_helper.h	2023-02-19 17:36:35.219463418 +0000
@@ -22,6 +22,7 @@
 
 namespace content {
 struct NativeWebKeyboardEvent;
+class RenderWidgetHostViewMac;
 }  // namespace content
 
 namespace remote_cocoa {
@@ -30,6 +31,7 @@
 class RenderWidgetHostNSViewHost;
 }  // namespace mojom
 
+
 // An interface through which the NSView for a RenderWidgetHostViewMac is to
 // communicate with the RenderWidgetHostViewMac (potentially in another
 // process). Unlike mojom::RenderWidgetHostNSViewHost, this object is always
@@ -56,6 +58,8 @@
 
   // Return the currently focused accessibility element.
   virtual id GetFocusedBrowserAccessibilityElement() = 0;
+  virtual content::RenderWidgetHostViewMac* GetRenderWidgetHostViewMac() = 0;
+
 
   // Set the NSWindow that will be the accessibility parent of the NSView.
   virtual void SetAccessibilityWindow(NSWindow* window) = 0;
