diff -r -u --color up/chromium/content/public/common/common_param_traits_macros.h nw/chromium/content/public/common/common_param_traits_macros.h
--- up/chromium/content/public/common/common_param_traits_macros.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/public/common/common_param_traits_macros.h	2023-02-19 17:36:35.795464873 +0000
@@ -135,6 +135,8 @@
 #if BUILDFLAG(IS_OZONE)
   IPC_STRUCT_TRAITS_MEMBER(selection_clipboard_buffer_available)
 #endif
+  IPC_STRUCT_TRAITS_MEMBER(nw_inject_js_doc_start)
+  IPC_STRUCT_TRAITS_MEMBER(nw_inject_js_doc_end)
 IPC_STRUCT_TRAITS_END()
 
 IPC_ENUM_TRAITS_MAX_VALUE(content::DropData::Kind,
