diff -r -u --color up/chromium/content/browser/zygote_host/zygote_host_impl_linux.cc nw/chromium/content/browser/zygote_host/zygote_host_impl_linux.cc
--- up/chromium/content/browser/zygote_host/zygote_host_impl_linux.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/browser/zygote_host/zygote_host_impl_linux.cc	2023-02-25 19:07:16.457820402 +0000
@@ -1,6 +1,7 @@
 // Copyright 2012 The Chromium Authors
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
+#pragma clang diagnostic ignored "-Wunreachable-code"
 
 #include "content/browser/zygote_host/zygote_host_impl_linux.h"
 
@@ -34,6 +35,8 @@
 #include "content/public/common/zygote/zygote_handle.h"
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
+#include "content/nw/src/common/shell_switches.h"
+
 namespace content {
 
 namespace {
@@ -83,7 +86,7 @@
 }
 
 void ZygoteHostImpl::Init(const base::CommandLine& command_line) {
-  if (command_line.HasSwitch(sandbox::policy::switches::kNoSandbox)) {
+  if (true || command_line.HasSwitch(sandbox::policy::switches::kNoSandbox)) {
     return;
   }
 
@@ -116,7 +119,7 @@
   } else if (!command_line.HasSwitch(
                  sandbox::policy::switches::kDisableSetuidSandbox) &&
              !sandbox_binary_.empty()) {
-    use_suid_sandbox_ = true;
+    use_suid_sandbox_ = false;
 
     // Use the SUID sandbox for adjusting OOM scores when we are using
     // the setuid sandbox. This is needed beacuse the processes are
