diff -r -u --color up/chromium/content/browser/webui/url_data_manager_backend.cc nw/chromium/content/browser/webui/url_data_manager_backend.cc
--- up/chromium/content/browser/webui/url_data_manager_backend.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/browser/webui/url_data_manager_backend.cc	2023-02-24 20:54:12.912493303 +0000
@@ -7,6 +7,8 @@
 #include <set>
 #include <utility>
 
+#include "content/nw/src/nw_content.h"
+
 #include "base/bind.h"
 #include "base/containers/contains.h"
 #include "base/location.h"
@@ -42,6 +44,11 @@
 #include "ui/base/webui/i18n_source_stream.h"
 #include "url/url_util.h"
 
+namespace nw {
+typedef bool (*CheckStoragePartitionMatchesFn)(int render_process_id, const GURL& url);
+CONTENT_EXPORT CheckStoragePartitionMatchesFn gCheckStoragePartitionMatches = nullptr;
+}
+
 namespace content {
 
 namespace {
