diff -r -u --color up/chromium/content/browser/renderer_host/input/touch_emulator.cc nw/chromium/content/browser/renderer_host/input/touch_emulator.cc
--- up/chromium/content/browser/renderer_host/input/touch_emulator.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/browser/renderer_host/input/touch_emulator.cc	2023-02-25 19:07:16.273819399 +0000
@@ -5,6 +5,8 @@
 #include "content/browser/renderer_host/input/touch_emulator.h"
 
 #include <memory>
+#include "content/public/browser/render_frame_host.h"
+#include "content/browser/renderer_host/render_frame_host_impl.h"
 
 #include "base/containers/queue.h"
 #include "base/time/time.h"
@@ -69,6 +71,7 @@
 TouchEmulator::TouchEmulator(TouchEmulatorClient* client,
                              float device_scale_factor)
     : client_(client),
+      rfh_limit_(nullptr),
       gesture_provider_config_type_(
           ui::GestureProviderConfigType::CURRENT_PLATFORM),
       double_tap_enabled_(true),
@@ -183,6 +186,11 @@
   if (!enabled() || mode_ != Mode::kEmulatingTouchFromMouse)
     return false;
 
+  if (rfh_limit_) {
+    if (rfh_limit_->GetView() != target_view)
+      return false;
+  }
+
   UpdateCursor();
 
   if (mouse_event.button == WebMouseEvent::Button::kRight &&
