diff -r -u --color up/chromium/content/browser/renderer_host/render_view_host_delegate.h nw/chromium/content/browser/renderer_host/render_view_host_delegate.h
--- up/chromium/content/browser/renderer_host/render_view_host_delegate.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/browser/renderer_host/render_view_host_delegate.h	2023-02-24 20:54:12.828493690 +0000
@@ -46,6 +46,7 @@
 // WebContents type (see //renderer_host/DEPS).
 class RenderViewHostDelegate {
  public:
+  RenderViewHostDelegate();
   // Returns the current delegate associated with a feature. May return NULL if
   // there is no corresponding delegate.
   virtual RenderViewHostDelegateView* GetDelegateView();
@@ -80,6 +81,9 @@
   // JavaScript window.focus() method).
   virtual void Activate() {}
 
+  virtual bool GetSkipBlockingParser();
+  virtual void SetSkipBlockingParser(bool);
+
   // Returns true if RenderWidgets under this RenderViewHost will never be
   // user-visible and thus never need to generate pixels for display.
   virtual bool IsNeverComposited();
@@ -127,6 +131,8 @@
 
  protected:
   virtual ~RenderViewHostDelegate() {}
+ private:
+  bool skip_blocking_parser_;
 };
 
 }  // namespace content
