diff -r -u --color up/chromium/content/browser/renderer_host/render_frame_host_delegate.h nw/chromium/content/browser/renderer_host/render_frame_host_delegate.h
--- up/chromium/content/browser/renderer_host/render_frame_host_delegate.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/browser/renderer_host/render_frame_host_delegate.h	2023-02-19 17:36:35.543464237 +0000
@@ -418,7 +418,7 @@
       int main_frame_widget_route_id,
       WindowOpenDisposition disposition,
       const blink::mojom::WindowFeatures& window_features,
-      bool user_gesture) {}
+      bool user_gesture, std::string manifest) {}
 
   // The main frame document element is ready. This happens when the document
   // has finished parsing.
