diff -r -u --color up/chromium/content/renderer/BUILD.gn nw/chromium/content/renderer/BUILD.gn
--- up/chromium/content/renderer/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/renderer/BUILD.gn	2023-02-19 17:36:35.827464954 +0000
@@ -46,6 +46,8 @@
     "accessibility/ax_tree_snapshotter_impl.h",
     "accessibility/blink_ax_action_target.cc",
     "accessibility/blink_ax_action_target.h",
+    "//base/message_loop/message_pumpuv_mac.mm",
+    "//base/message_loop/message_pumpuv_mac.h",
     "accessibility/render_accessibility_impl.cc",
     "accessibility/render_accessibility_impl.h",
     "accessibility/render_accessibility_manager.cc",
