diff -r -u --color up/chromium/content/renderer/frame_owner_properties_converter.cc nw/chromium/content/renderer/frame_owner_properties_converter.cc
--- up/chromium/content/renderer/frame_owner_properties_converter.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/renderer/frame_owner_properties_converter.cc	2023-02-25 19:07:16.597821165 +0000
@@ -13,6 +13,8 @@
   blink::WebFrameOwnerProperties result;
   result.name = blink::WebString::FromUTF8(mojo_properties.name);
   result.scrollbar_mode = mojo_properties.scrollbar_mode;
+  result.nwFakeTop = mojo_properties.nwfaketop;
+  result.nwuseragent = blink::WebString::FromUTF8(mojo_properties.nwuseragent);
   result.margin_width = mojo_properties.margin_width;
   result.margin_height = mojo_properties.margin_height;
   result.allow_fullscreen = mojo_properties.allow_fullscreen;
@@ -32,6 +34,8 @@
 
   mojo_properties->name = web_frame_owner_properties.name.Utf8();
   mojo_properties->scrollbar_mode = web_frame_owner_properties.scrollbar_mode;
+  mojo_properties->nwfaketop = web_frame_owner_properties.nwFakeTop;
+  mojo_properties->nwuseragent = web_frame_owner_properties.nwuseragent.Utf8();
   mojo_properties->margin_width = web_frame_owner_properties.margin_width;
   mojo_properties->margin_height = web_frame_owner_properties.margin_height;
   mojo_properties->allow_fullscreen =
