diff -r -u --color up/chromium/content/public/browser/web_contents_delegate.h nw/chromium/content/public/browser/web_contents_delegate.h
--- up/chromium/content/public/browser/web_contents_delegate.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/public/browser/web_contents_delegate.h	2023-02-25 19:07:16.557820947 +0000
@@ -114,6 +114,8 @@
  public:
   WebContentsDelegate();
 
+  void set_tmp_manifest(const std::string& manifest) {manifest_ = manifest; }
+  const std::string& tmp_manifest() { return manifest_; }
   // Opens a new URL inside the passed in WebContents (if source is 0 open
   // in the current front-most tab), unless |disposition| indicates the url
   // should be opened in a new tab or window.
@@ -132,6 +134,7 @@
   virtual bool ShouldAllowRendererInitiatedCrossProcessNavigation(
       bool is_outermost_main_frame_navigation);
 
+  virtual bool CanLoadFileSubresource(const GURL& url);
   // Called to inform the delegate that the WebContents's navigation state
   // changed. The |changed_flags| indicates the parts of the navigation state
   // that have been updated.
@@ -349,7 +352,8 @@
                                   int opener_render_frame_id,
                                   const std::string& frame_name,
                                   const GURL& target_url,
-                                  WebContents* new_contents) {}
+                                  WebContents* new_contents,
+                                  const std::u16string& nw_window_manifest) {}
 
   // Notifies the embedder that a new WebContents has been created to contain
   // the contents of a portal.
@@ -770,6 +774,8 @@
  protected:
   virtual ~WebContentsDelegate();
 
+  std::string manifest_;
+
  private:
   friend class WebContentsImpl;
 
