diff -r -u --color up/chromium/device/bluetooth/BUILD.gn nw/chromium/device/bluetooth/BUILD.gn
--- up/chromium/device/bluetooth/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/device/bluetooth/BUILD.gn	2023-02-24 20:54:13.424490942 +0000
@@ -351,8 +351,8 @@
         "bluez/bluetooth_socket_bluez.h",
         "bluez/bluez_features.cc",
         "bluez/bluez_features.h",
-        "bluez/metrics_recorder.cc",
-        "bluez/metrics_recorder.h",
+        #"bluez/metrics_recorder.cc",
+        #"bluez/metrics_recorder.h",
         "dbus/bluetooth_adapter_client.cc",
         "dbus/bluetooth_adapter_client.h",
         "dbus/bluetooth_admin_policy_client.cc",
