diff -r -u --color up/chromium/extensions/renderer/ipc_message_sender.h nw/chromium/extensions/renderer/ipc_message_sender.h
--- up/chromium/extensions/renderer/ipc_message_sender.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/ipc_message_sender.h	2023-02-24 20:54:13.772489337 +0000
@@ -15,6 +15,10 @@
 
 struct ExtensionHostMsg_APIActionOrEvent_Params;
 
+namespace base {
+class ListValue;
+}
+
 namespace extensions {
 class ScriptContext;
 class WorkerThreadDispatcher;
@@ -35,8 +39,12 @@
   enum class ActivityLogCallType { APICALL, EVENT };
 
   // Sends a request message to the browser.
-  virtual void SendRequestIPC(ScriptContext* context,
-                              mojom::RequestParamsPtr params) = 0;
+  virtual void SendRequestIPC(
+      ScriptContext* context,
+      mojom::RequestParamsPtr params,
+      bool sync = false,
+      bool* success = nullptr, ::base::Value::List* response = nullptr,
+      std::string* error = nullptr) = 0;
 
   // Handles sending any additional messages required after receiving a response
   // to a request.
