diff -r -u --color up/chromium/extensions/browser/app_window/native_app_window.h nw/chromium/extensions/browser/app_window/native_app_window.h
--- up/chromium/extensions/browser/app_window/native_app_window.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/app_window/native_app_window.h	2023-02-25 19:07:17.589826572 +0000
@@ -35,6 +35,8 @@
   // |fullscreen_types| is a bit field of AppWindow::FullscreenType.
   virtual void SetFullscreen(int fullscreen_types) = 0;
 
+  virtual void SetResizable(bool flag) = 0;
+  virtual bool IsResizable() const = 0;
   // Returns whether the window is fullscreen or about to enter fullscreen.
   virtual bool IsFullscreenOrPending() const = 0;
 
@@ -74,6 +76,8 @@
   // borders) and the content bounds, if any.
   virtual gfx::Insets GetFrameInsets() const = 0;
 
+  virtual void SetShowInTaskbar(bool) = 0;
+
   // Returns the minimum size constraints of the content.
   virtual gfx::Size GetContentMinimumSize() const = 0;
 
