diff -r -u --color up/chromium/extensions/common/mojom/frame.mojom nw/chromium/extensions/common/mojom/frame.mojom
--- up/chromium/extensions/common/mojom/frame.mojom	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/common/mojom/frame.mojom	2023-02-25 19:07:17.665826987 +0000
@@ -30,6 +30,7 @@
   bool match_about_blank;
   // When to inject the code.
   RunLocation run_at;
+  bool in_main_world;
   // Whether the request is coming from a <webview>.
   bool is_web_view;
 };
@@ -136,6 +137,11 @@
           mojo_base.mojom.ListValue response_wrapper,
           string error,
           ExtraResponseData? extra_data);
+  [Sync]
+  RequestSync(RequestParams params)
+      => (bool success,
+          mojo_base.mojom.ListValue response_wrapper,
+          string error);
 
   // Notifies the browser process that a tab has started or stopped matching
   // certain conditions.  This method is called by response to several events:
