diff -r -u --color up/chromium/extensions/renderer/user_script_injector.h nw/chromium/extensions/renderer/user_script_injector.h
--- up/chromium/extensions/renderer/user_script_injector.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/user_script_injector.h	2023-02-25 19:07:17.709827226 +0000
@@ -46,6 +46,7 @@
   // ScriptInjector implementation.
   mojom::InjectionType script_type() const override;
   blink::mojom::UserActivationOption IsUserGesture() const override;
+  bool ShouldExecuteInMainWorld() const override;
   mojom::ExecutionWorld GetExecutionWorld() const override;
   mojom::CSSOrigin GetCssOrigin() const override;
   mojom::CSSInjection::Operation GetCSSInjectionOperation() const override;
