diff -r -u --color up/chromium/extensions/browser/guest_view/web_view/web_view_constants.cc nw/chromium/extensions/browser/guest_view/web_view/web_view_constants.cc
--- up/chromium/extensions/browser/guest_view/web_view/web_view_constants.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/guest_view/web_view/web_view_constants.cc	2023-02-25 19:07:17.609826681 +0000
@@ -11,6 +11,7 @@
 const char kAttributeAllowScaling[] = "allowscaling";
 const char kAttributeName[] = "name";
 const char kAttributeSrc[] = "src";
+const char kAttributeAllowNW[] = "allownw";
 
 // API namespace.
 const char kAPINamespace[] = "webViewInternal";
