diff -r -u --color up/chromium/extensions/renderer/native_extension_bindings_system.h nw/chromium/extensions/renderer/native_extension_bindings_system.h
--- up/chromium/extensions/renderer/native_extension_bindings_system.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/renderer/native_extension_bindings_system.h	2023-02-19 17:36:36.911467693 +0000
@@ -97,7 +97,7 @@
   // Returns the API with the given |name| for the given |context|. Used for
   // testing purposes.
   v8::Local<v8::Object> GetAPIObjectForTesting(ScriptContext* context,
-                                               const std::string& api_name);
+                                               const std::string& api_name, bool hidden = false);
 
  private:
   // Handles sending a given |request|, forwarding it on to the send_ipc_ after
@@ -116,6 +116,8 @@
   // Getter callback for an extension API, since APIs are constructed lazily.
   static void BindingAccessor(v8::Local<v8::Name> name,
                               const v8::PropertyCallbackInfo<v8::Value>& info);
+  static void BindingAccessorHidden(v8::Local<v8::Name> name,
+                              const v8::PropertyCallbackInfo<v8::Value>& info);
 
   // Callback for accessing a restricted extension API. Access to the API is
   // restricted to the developer mode only.
@@ -125,7 +127,7 @@
 
   // Creates and returns the API binding for the given |name|.
   static v8::Local<v8::Object> GetAPIHelper(v8::Local<v8::Context> context,
-                                            v8::Local<v8::String> name);
+                                            v8::Local<v8::String> name, bool hidden = false);
 
   // Gets the chrome.runtime API binding.
   static v8::Local<v8::Object> GetLastErrorParents(
