diff -r -u --color up/chromium/extensions/common/manifest_constants.cc nw/chromium/extensions/common/manifest_constants.cc
--- up/chromium/extensions/common/manifest_constants.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/common/manifest_constants.cc	2023-02-24 20:12:10.284873027 +0000
@@ -10,6 +10,14 @@
 namespace extensions {
 
 namespace manifest_keys {
+const char kNWJSInternalFlag[] = "__nwjs_app";
+const char kNWJSInternalManifest[] = "__nwjs_manifest";
+const char kNWJSInternalMainFilename[] = "__nwjs_filename";
+const char kNWJSContentVerifyFlag[] = "__nwjs_cv";
+const char kNWJSMain[] = "main";
+const char kNWJSMixedContext[] = "mixed_context";
+const char kNWJSEnableNode[] = "nodejs";
+const char kNWJSDomain[] = "domain";
 
 const char kAboutPage[] = "about_page";
 const char kAction[] = "action";
@@ -88,6 +96,7 @@
 const char kLinkedAppIconURL[] = "url";
 const char kLinkedAppIconSize[] = "size";
 const char kManifestVersion[] = "manifest_version";
+const char kInMainWorld[] = "in_main_world";
 const char kMatches[] = "matches";
 const char kMinimumChromeVersion[] = "minimum_chrome_version";
 const char kMIMETypes[] = "mime_types";
@@ -165,7 +174,6 @@
 const char kActionHandlerActionKey[] = "action";
 const char kActionHandlerEnabledOnLockScreenKey[] = "enabled_on_lock_screen";
 #endif
-
 }  // namespace manifest_keys
 
 namespace manifest_values {
@@ -476,6 +484,8 @@
     "See developer.chrome.com/*/manifestVersion for details.";
 const char kInvalidMatch[] =
     "Invalid value for 'content_scripts[*].matches[*]': *";
+const char kInvalidInMainWorld[] =
+    "Invalid value for 'content_scripts[*].in_main_world'.";
 const char kInvalidMatchCount[] =
     "Invalid value for 'content_scripts[*].matches'. There must be at least "
     "one match specified.";
