diff -r -u --color up/chromium/extensions/common/manifest_constants.h nw/chromium/extensions/common/manifest_constants.h
--- up/chromium/extensions/common/manifest_constants.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/common/manifest_constants.h	2023-02-19 17:36:36.879467612 +0000
@@ -12,6 +12,14 @@
 
 // Keys used in JSON representation of extensions.
 namespace manifest_keys {
+extern const char kNWJSInternalFlag[];
+extern const char kNWJSInternalMainFilename[];
+extern const char kNWJSInternalManifest[];
+extern const char kNWJSContentVerifyFlag[];
+extern const char kNWJSMain[];
+extern const char kNWJSMixedContext[];
+extern const char kNWJSEnableNode[];
+extern const char kNWJSDomain[];
 
 extern const char kAboutPage[];
 extern const char kAction[];
@@ -92,6 +100,7 @@
 extern const char kLinkedAppIconURL[];
 extern const char kLinkedAppIconSize[];
 extern const char kManifestVersion[];
+extern const char kInMainWorld[];
 extern const char kMatches[];
 extern const char kMIMETypes[];
 extern const char kMimeTypesHandler[];
@@ -351,6 +360,7 @@
 extern const char kInvalidManifestVersionMissingKey[];
 extern const char kInvalidManifestVersionUnsupported[];
 extern const char kInvalidMatch[];
+extern const char kInvalidInMainWorld[];
 extern const char kInvalidMatchCount[];
 extern const char kInvalidMatches[];
 extern const char16_t kInvalidMIMETypes[];
