diff -r -u --color up/chromium/extensions/browser/api/web_request/web_request_api.cc nw/chromium/extensions/browser/api/web_request/web_request_api.cc
--- up/chromium/extensions/browser/api/web_request/web_request_api.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/browser/api/web_request/web_request_api.cc	2023-02-19 17:36:36.811467440 +0000
@@ -2200,7 +2200,7 @@
   if (request.is_web_view) {
     // If this is a navigation request, then we can skip this check. IDs will
     // be -1 and the request is trusted.
-    if (!request.is_navigation_request &&
+    if (//!request.is_navigation_request &&
         listener.id.render_process_id != request.web_view_embedder_process_id) {
       return false;
     }
