diff -r -u --color up/chromium/extensions/renderer/script_context_set.h nw/chromium/extensions/renderer/script_context_set.h
--- up/chromium/extensions/renderer/script_context_set.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/script_context_set.h	2023-02-25 19:07:17.709827226 +0000
@@ -56,6 +56,7 @@
   // This may also include invalid contexts. TODO(kalman): Useful?
   size_t size() const { return contexts_.size(); }
 
+  const std::set<ScriptContext*>& contexts() const { return contexts_; }
   // Creates and starts managing a new ScriptContext. Ownership is held.
   // Returns a weak reference to the new ScriptContext.
   ScriptContext* Register(blink::WebLocalFrame* frame,
@@ -124,7 +125,9 @@
       int32_t world_id,
       const GURL& url,
       const blink::WebSecurityOrigin& origin,
-      mojom::ViewType view_type);
+      mojom::ViewType view_type,
+      const blink::WebLocalFrame* frame = nullptr
+                                             );
 
   // Weak reference to all installed Extensions that are also active in this
   // process.
