diff -r -u --color up/chromium/extensions/browser/verified_contents.h nw/chromium/extensions/browser/verified_contents.h
--- up/chromium/extensions/browser/verified_contents.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/verified_contents.h	2023-02-24 20:12:10.268872894 +0000
@@ -71,7 +71,7 @@
 
   // Returns the base64url-decoded "payload" field from the |contents|, if
   // the signature was valid.
-  bool GetPayload(base::StringPiece contents, std::string* payload);
+  bool GetPayload(base::StringPiece contents, std::string* payload, const char* manifest = nullptr);
 
   // The |protected_value| and |payload| arguments should be base64url encoded
   // strings, and |signature_bytes| should be a byte array. See comments in the
