diff -r -u --color up/chromium/extensions/common/file_util.cc nw/chromium/extensions/common/file_util.cc
--- up/chromium/extensions/common/file_util.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/common/file_util.cc	2023-02-25 19:07:17.653826921 +0000
@@ -44,6 +44,9 @@
 #include "ui/base/l10n/l10n_util.h"
 #include "url/gurl.h"
 
+#include "base/command_line.h"
+#include "content/public/common/content_client.h"
+
 using extensions::mojom::ManifestLocation;
 
 namespace extensions {
@@ -250,7 +253,20 @@
 absl::optional<base::Value::Dict> LoadManifest(
     const base::FilePath& extension_path,
     std::string* error) {
-  return LoadManifest(extension_path, kManifestFilename, error);
+  base::FilePath manifest_path = extension_path.Append(kNWJSManifestFilename);
+
+  if (!base::PathExists(manifest_path))
+    return LoadManifest(extension_path, kManifestFilename, error);
+
+  absl::optional<base::Value::Dict> manifest =
+    LoadManifest(extension_path, kNWJSManifestFilename, error);
+  content::GetContentClient()->LoadNWAppAsExtension(&manifest.value(), extension_path, error);
+
+  base::CommandLine* cmdline = base::CommandLine::ForCurrentProcess();
+  if (cmdline->HasSwitch("mixed-context"))
+    manifest->Set(manifest_keys::kNWJSMixedContext, true);
+
+  return manifest;
 }
 
 absl::optional<base::Value::Dict> LoadManifest(
@@ -297,6 +313,7 @@
   // Check children of extension root to see if any of them start with _ and is
   // not on the reserved list. We only warn, and do not block the loading of the
   // extension.
+#if 0
   std::string warning;
   if (!CheckForIllegalFilenames(extension->path(), &warning))
     warnings->push_back(InstallWarning(warning));
@@ -328,6 +345,7 @@
     }
     // Only warn; don't block loading the extension.
   }
+#endif
   return true;
 }
 
@@ -360,6 +378,7 @@
 
 bool CheckForIllegalFilenames(const base::FilePath& extension_path,
                               std::string* error) {
+#if 0
   // Enumerate all files and directories in the extension root.
   // There is a problem when using pattern "_*" with FileEnumerator, so we have
   // to cheat with find_first_of and match all.
@@ -387,7 +406,7 @@
         file.BaseName().AsUTF8Unsafe().c_str());
     return false;
   }
-
+#endif
   return true;
 }
 
@@ -537,11 +556,11 @@
 }
 
 base::FilePath GetVerifiedContentsPath(const base::FilePath& extension_path) {
-  return extension_path.Append(kMetadataFolder)
+  return extension_path
       .Append(kVerifiedContentsFilename);
 }
 base::FilePath GetComputedHashesPath(const base::FilePath& extension_path) {
-  return extension_path.Append(kMetadataFolder).Append(kComputedHashesFilename);
+  return extension_path.Append(kComputedHashesFilename);
 }
 base::FilePath GetIndexedRulesetDirectoryRelativePath() {
   return base::FilePath(kMetadataFolder).Append(kIndexedRulesetDirectory);
