diff -r -u --color up/chromium/extensions/browser/script_executor.cc nw/chromium/extensions/browser/script_executor.cc
--- up/chromium/extensions/browser/script_executor.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/script_executor.cc	2023-02-25 19:07:17.621826747 +0000
@@ -361,6 +361,7 @@
                                    const std::set<int>& frame_ids,
                                    ScriptExecutor::MatchAboutBlank about_blank,
                                    mojom::RunLocation run_at,
+                                   ScriptExecutor::WorldType world_type,
                                    ScriptExecutor::ProcessType process_type,
                                    const GURL& webview_src,
                                    ScriptFinishedCallback callback) {
@@ -396,6 +397,7 @@
   auto params = mojom::ExecuteCodeParams::New();
   params->host_id = host_id.Clone();
   params->injection = std::move(injection);
+  params->in_main_world = (world_type == MAIN_WORLD);
   params->match_about_blank = (about_blank == MATCH_ABOUT_BLANK);
   params->run_at = run_at;
   params->is_web_view = (process_type == WEB_VIEW_PROCESS);
