diff -r -u --color up/chromium/extensions/renderer/bindings/api_binding_js_util.h nw/chromium/extensions/renderer/bindings/api_binding_js_util.h
--- up/chromium/extensions/renderer/bindings/api_binding_js_util.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/bindings/api_binding_js_util.h	2023-02-25 19:07:17.685827096 +0000
@@ -7,6 +7,7 @@
 
 #include <string>
 
+#include "base/values.h"
 #include "gin/wrappable.h"
 #include "v8/include/v8.h"
 
@@ -44,10 +45,22 @@
  private:
   // A handler to initiate an API request through the APIRequestHandler. A
   // replacement for custom bindings that utilize require('sendRequest').
+  void SendRequestSync(gin::Arguments* arguments,
+                   const std::string& name,
+                   const std::vector<v8::Local<v8::Value>>& request_args,
+                   v8::Local<v8::Value> options);
   void SendRequest(gin::Arguments* arguments,
                    const std::string& name,
                    const std::vector<v8::Local<v8::Value>>& request_args,
                    v8::Local<v8::Value> options);
+  void SendRequestHelper(gin::Arguments* arguments,
+                   const std::string& name,
+                   const std::vector<v8::Local<v8::Value>>& request_args,
+                   v8::Local<v8::Value> options,
+                   bool sync = false,
+                   bool* success = nullptr,
+                   base::Value::List* response = nullptr,
+                   std::string* error = nullptr);
 
   // A handler to register an argument massager for a specific event.
   // Replacement for event_bindings.registerArgumentMassager.
