diff -r -u --color up/chromium/extensions/renderer/resources/guest_view/web_view/web_view_attributes.js nw/chromium/extensions/renderer/resources/guest_view/web_view/web_view_attributes.js
--- up/chromium/extensions/renderer/resources/guest_view/web_view/web_view_attributes.js	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/resources/guest_view/web_view/web_view_attributes.js	2023-02-24 20:12:10.316873295 +0000
@@ -52,6 +52,19 @@
                                        this.getValue());
 };
 
+function AllowNWAttribute(view) {
+  $Function.call(
+    GuestViewAttributes.BooleanAttribute,
+    this, WebViewConstants.ATTRIBUTE_ALLOWNW, view);
+}
+
+AllowNWAttribute.prototype.__proto__ =
+    GuestViewAttributes.BooleanAttribute.prototype;
+
+AllowNWAttribute.prototype.handleMutation = function(oldValue,
+                                                               newValue) {
+};
+
 // -----------------------------------------------------------------------------
 // AutosizeDimensionAttribute object.
 
@@ -257,6 +270,7 @@
 };
 
 var WebViewAttributes = {
+  AllowNWAttribute: AllowNWAttribute,
   AllowScalingAttribute: AllowScalingAttribute,
   AllowTransparencyAttribute: AllowTransparencyAttribute,
   AutosizeDimensionAttribute: AutosizeDimensionAttribute,
