diff -r -u --color up/chromium/extensions/renderer/ipc_message_sender.cc nw/chromium/extensions/renderer/ipc_message_sender.cc
--- up/chromium/extensions/renderer/ipc_message_sender.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/renderer/ipc_message_sender.cc	2023-02-19 17:36:36.911467693 +0000
@@ -46,17 +46,25 @@
 
   ~MainThreadIPCMessageSender() override {}
 
-  void SendRequestIPC(ScriptContext* context,
-                      mojom::RequestParamsPtr params) override {
+  void SendRequestIPC(
+      ScriptContext* context,
+      mojom::RequestParamsPtr params,
+      bool sync, bool* success, base::Value::List* response,
+      std::string* error) override {
     content::RenderFrame* frame = context->GetRenderFrame();
     if (!frame)
       return;
 
     int request_id = params->request_id;
+    if (!sync)
     ExtensionFrameHelper::Get(frame)->GetLocalFrameHost()->Request(
         std::move(params),
         base::BindOnce(&MainThreadIPCMessageSender::OnResponse,
                        weak_ptr_factory_.GetWeakPtr(), request_id));
+    else
+      ExtensionFrameHelper::Get(frame)->GetLocalFrameHost()->RequestSync(
+                                                                         std::move(params),
+                                                                         success, response, error);
   }
 
   void SendOnRequestResponseReceivedIPC(int request_id) override {}
@@ -262,7 +270,9 @@
   ~WorkerThreadIPCMessageSender() override {}
 
   void SendRequestIPC(ScriptContext* context,
-                      mojom::RequestParamsPtr params) override {
+                      mojom::RequestParamsPtr params,
+                      bool sync, bool* success, base::Value::List* response,
+                      std::string* error) override {
     DCHECK(!context->GetRenderFrame());
     DCHECK(context->IsForServiceWorker());
     DCHECK_NE(kMainThreadId, content::WorkerThread::GetCurrentId());
