diff -r -u --color up/chromium/extensions/BUILD.gn nw/chromium/extensions/BUILD.gn
--- up/chromium/extensions/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/BUILD.gn	2023-02-19 17:36:36.755467299 +0000
@@ -178,7 +178,7 @@
     "//services/network/public/mojom",
     "//testing/gmock",
     "//testing/gtest",
-    "//third_party/cld_3/src/src:cld_3",
+    #"//third_party/cld_3/src/src:cld_3",
     "//third_party/zlib/google:zip",
   ]
 
