diff -r -u --color up/chromium/extensions/browser/api/guest_view/web_view/web_view_internal_api.cc nw/chromium/extensions/browser/api/guest_view/web_view/web_view_internal_api.cc
--- up/chromium/extensions/browser/api/guest_view/web_view/web_view_internal_api.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/browser/api/guest_view/web_view/web_view_internal_api.cc	2023-02-19 17:36:36.783467369 +0000
@@ -155,7 +155,7 @@
   // if needed.
   bool allowed_everywhere =
       extension && extensions::PermissionsData::CanExecuteScriptEverywhere(
-                       extension->id(), extension->location());
+                                                                           extension->id(), extension->location(), extension->GetType());
   for (const std::string& match : script_value.matches) {
     URLPattern pattern(UserScript::ValidUserScriptSchemes(allowed_everywhere));
     if (pattern.Parse(match) != URLPattern::ParseResult::kSuccess) {
@@ -869,6 +869,40 @@
   return RespondNow(Error(std::move(error)));
 }
 
+WebViewInternalShowDevToolsFunction::WebViewInternalShowDevToolsFunction() {
+}
+
+WebViewInternalShowDevToolsFunction::~WebViewInternalShowDevToolsFunction() {
+}
+
+ExtensionFunction::ResponseAction
+WebViewInternalShowDevToolsFunction::Run() {
+  std::unique_ptr<web_view_internal::ShowDevTools::Params> params(
+             web_view_internal::ShowDevTools::Params::Create(args()));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  int proc_id = params->proc_id ? *params->proc_id : -1;
+  int guest_id = params->guest_id ? *params->guest_id : -1;
+  guest_->ShowDevTools(params->show, proc_id, guest_id);
+  return RespondNow(NoArguments());
+}
+
+WebViewInternalInspectElementAtFunction::WebViewInternalInspectElementAtFunction() {
+}
+
+WebViewInternalInspectElementAtFunction::~WebViewInternalInspectElementAtFunction() {
+}
+
+ExtensionFunction::ResponseAction
+WebViewInternalInspectElementAtFunction::Run() {
+  std::unique_ptr<web_view_internal::InspectElementAt::Params> params(
+           web_view_internal::InspectElementAt::Params::Create(args()));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  guest_->InspectElement(params->x, params->y);
+  return RespondNow(NoArguments());
+}
+
 WebViewInternalGoFunction::WebViewInternalGoFunction() {
 }
 
