diff -r -u --color up/chromium/extensions/browser/api/execute_code_function.cc nw/chromium/extensions/browser/api/execute_code_function.cc
--- up/chromium/extensions/browser/api/execute_code_function.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/api/execute_code_function.cc	2023-02-25 19:07:17.557826398 +0000
@@ -6,6 +6,7 @@
 #define EXTENSIONS_BROWSER_API_EXECUTE_CODE_FUNCTION_IMPL_H_
 
 #include "extensions/browser/api/execute_code_function.h"
+#include "content/nw/src/browser/nw_chrome_browser_hooks.h"
 
 #include <utility>
 
@@ -109,6 +110,9 @@
       break;
   }
 
+  ScriptExecutor::WorldType world_type = details_->main_world && *details_->main_world
+    ? ScriptExecutor::MAIN_WORLD : ScriptExecutor::ISOLATED_WORLD;
+
   mojom::CodeInjectionPtr injection;
   bool is_css_injection = ShouldInsertCSS() || ShouldRemoveCSS();
   if (is_css_injection) {
@@ -143,6 +147,7 @@
   executor->ExecuteScript(
       host_id_, std::move(injection), frame_scope, {root_frame_id_},
       match_about_blank, run_at,
+      world_type,
       IsWebView() ? ScriptExecutor::WEB_VIEW_PROCESS
                   : ScriptExecutor::DEFAULT_PROCESS,
       GetWebViewSrc(),
