diff -r -u --color up/chromium/extensions/extensions_resources.grd nw/chromium/extensions/extensions_resources.grd
--- up/chromium/extensions/extensions_resources.grd	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/extensions_resources.grd	2023-02-19 17:36:36.891467642 +0000
@@ -9,6 +9,7 @@
   <release seq="1">
     <includes>
       <include name="IDR_EXTENSION_API_FEATURES" file="common\api\_api_features.json" type="BINDATA" />
+      <include name="IDR_NW_EXTENSION_API_FEATURES" file="..\content\nw\src\api\_api_features.json" type="BINDATA" />
     </includes>
   </release>
 </grit>
