diff -r -u --color up/chromium/extensions/renderer/module_system.h nw/chromium/extensions/renderer/module_system.h
--- up/chromium/extensions/renderer/module_system.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/module_system.h	2023-02-24 20:54:13.772489337 +0000
@@ -126,7 +126,7 @@
   // code to set up various hooks.
   // TODO(devlin): We can get rid of this once we convert all our custom
   // bindings.
-  void OnNativeBindingCreated(const std::string& api_name,
+  v8::Local<v8::Value> OnNativeBindingCreated(const std::string& api_name,
                               v8::Local<v8::Value> api_bridge_value);
 
   void SetGetInternalAPIHook(v8::Local<v8::FunctionTemplate> get_internal_api);
