diff -r -u --color up/chromium/extensions/browser/extension_protocols.cc nw/chromium/extensions/browser/extension_protocols.cc
--- up/chromium/extensions/browser/extension_protocols.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/browser/extension_protocols.cc	2023-02-19 17:36:36.831467491 +0000
@@ -256,7 +256,7 @@
                                 const ProcessMap& process_map) {
   const bool is_main_frame =
       destination == network::mojom::RequestDestination::kDocument;
-  if (is_incognito &&
+  if (is_incognito && !extension->is_nwjs_app() &&
       !ExtensionCanLoadInIncognito(is_main_frame, extension,
                                    extension_enabled_in_incognito)) {
     return false;
