diff -r -u --color up/chromium/extensions/browser/content_verifier/test_utils.h nw/chromium/extensions/browser/content_verifier/test_utils.h
--- up/chromium/extensions/browser/content_verifier/test_utils.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/content_verifier/test_utils.h	2023-02-25 19:07:17.593826594 +0000
@@ -186,7 +186,9 @@
   std::set<base::FilePath> GetBrowserImagePaths(
       const extensions::Extension* extension) override;
   void VerifyFailed(const ExtensionId& extension_id,
-                    ContentVerifyJob::FailureReason reason) override;
+                    const base::FilePath& relative_path,
+                    ContentVerifyJob::FailureReason reason,
+                    scoped_refptr<ContentVerifyJob> verify_job) override;
   void Shutdown() override;
 
   // Modifier.
