diff -r -u --color up/chromium/extensions/browser/app_window/test_app_window_contents.h nw/chromium/extensions/browser/app_window/test_app_window_contents.h
--- up/chromium/extensions/browser/app_window/test_app_window_contents.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/app_window/test_app_window_contents.h	2023-02-25 19:07:17.589826572 +0000
@@ -31,7 +31,8 @@
   // apps:AppWindowContents:
   void Initialize(content::BrowserContext* context,
                   content::RenderFrameHost* creator_frame,
-                  const GURL& url) override;
+                  const GURL& url,
+                  const Extension* extension, bool skip_blocking_parser) override;
   void LoadContents(int32_t creator_process_id) override;
   void NativeWindowChanged(NativeAppWindow* native_app_window) override;
   void NativeWindowClosed(bool send_onclosed) override;
