diff -r -u --color up/chromium/extensions/common/permissions/permissions_data.h nw/chromium/extensions/common/permissions/permissions_data.h
--- up/chromium/extensions/common/permissions/permissions_data.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/common/permissions/permissions_data.h	2023-02-25 19:07:17.669827008 +0000
@@ -89,7 +89,8 @@
   // NOTE: This is static because it is used during extension initialization,
   // before the extension has an associated PermissionsData object.
   static bool CanExecuteScriptEverywhere(const ExtensionId& extension_id,
-                                         mojom::ManifestLocation location);
+                                         mojom::ManifestLocation location,
+                                         Manifest::Type type);
 
   // Returns true if the given |url| is restricted for the given |extension|,
   // as is commonly the case for chrome:// urls.
@@ -159,8 +160,10 @@
   // Note this does not include APIs with no corresponding permission, like
   // "runtime" or "browserAction".
   // TODO(mpcomplete): drop the "API" from these names, it's confusing.
-  bool HasAPIPermission(mojom::APIPermissionID permission) const;
-  bool HasAPIPermission(const std::string& permission_name) const;
+  bool HasAPIPermission(mojom::APIPermissionID permission,
+                        bool ignore_override = false) const;
+  bool HasAPIPermission(const std::string& permission_name,
+                        bool ignore_override = false) const;
   bool HasAPIPermissionForTab(int tab_id,
                               mojom::APIPermissionID permission) const;
   bool CheckAPIPermissionWithParam(
@@ -303,6 +306,7 @@
 #endif
 
  private:
+  bool allow_all_override_;
   // Gets the tab-specific host permissions of |tab_id|, or NULL if there
   // aren't any.
   // Must be called with |runtime_lock_| acquired.
