diff -r -u --color up/chromium/extensions/browser/guest_view/web_view/web_view_constants.h nw/chromium/extensions/browser/guest_view/web_view/web_view_constants.h
--- up/chromium/extensions/browser/guest_view/web_view/web_view_constants.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/browser/guest_view/web_view/web_view_constants.h	2023-02-19 17:36:36.839467511 +0000
@@ -16,6 +16,7 @@
 extern const char kAttributeAllowScaling[];
 extern const char kAttributeName[];
 extern const char kAttributeSrc[];
+extern const char kAttributeAllowNW[];
 
 // API namespace.
 // TODO(kalman): Consolidate this with the other API constants.
