diff -r -u --color up/chromium/extensions/renderer/bindings/api_request_handler.h nw/chromium/extensions/renderer/bindings/api_request_handler.h
--- up/chromium/extensions/renderer/bindings/api_request_handler.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/bindings/api_request_handler.h	2023-02-24 20:54:13.764489374 +0000
@@ -37,6 +37,11 @@
 
     ~Request();
 
+    bool sync = false;
+    bool* success = nullptr;
+    std::string* error = nullptr;
+    base::Value::List* response = nullptr;
+
     int request_id = -1;
     std::string method_name;
     bool has_async_response_handler = false;
@@ -79,7 +84,11 @@
       binding::AsyncResponseType async_type,
       v8::Local<v8::Function> callback,
       v8::Local<v8::Function> custom_callback,
-      binding::ResultModifierFunction result_modifier);
+      binding::ResultModifierFunction result_modifier,
+      bool sync = false,
+      bool* success = nullptr,
+      base::Value::List* response = nullptr,
+      std::string* error = nullptr);
 
   // Adds a pending request for the request handler to manage (and complete via
   // CompleteRequest). This is used by renderer-side implementations that
