diff -r -u --color up/chromium/extensions/browser/api/guest_view/web_view/web_view_internal_api.h nw/chromium/extensions/browser/api/guest_view/web_view/web_view_internal_api.h
--- up/chromium/extensions/browser/api/guest_view/web_view/web_view_internal_api.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/api/guest_view/web_view/web_view_internal_api.h	2023-02-24 20:54:13.676489780 +0000
@@ -407,6 +407,35 @@
   ResponseAction Run() override;
 };
 
+class WebViewInternalShowDevToolsFunction
+    : public WebViewInternalExtensionFunction {
+ public:
+  DECLARE_EXTENSION_FUNCTION("webViewInternal.showDevTools", UNKNOWN)
+
+  WebViewInternalShowDevToolsFunction();
+
+ protected:
+  ~WebViewInternalShowDevToolsFunction() override;
+
+  ResponseAction Run() override;
+ private:
+
+};
+
+class WebViewInternalInspectElementAtFunction
+    : public WebViewInternalExtensionFunction {
+ public:
+  DECLARE_EXTENSION_FUNCTION("webViewInternal.inspectElementAt", UNKNOWN)
+
+  WebViewInternalInspectElementAtFunction();
+
+ protected:
+  ~WebViewInternalInspectElementAtFunction() override;
+  ResponseAction Run() override;
+ private:
+
+};
+
 class WebViewInternalGoFunction : public WebViewInternalExtensionFunction {
  public:
   DECLARE_EXTENSION_FUNCTION("webViewInternal.go", WEBVIEWINTERNAL_GO)
