diff -r -u --color up/chromium/extensions/renderer/object_backed_native_handler.h nw/chromium/extensions/renderer/object_backed_native_handler.h
--- up/chromium/extensions/renderer/object_backed_native_handler.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/object_backed_native_handler.h	2023-02-24 20:54:13.772489337 +0000
@@ -79,6 +79,7 @@
                                      const v8::Local<v8::Object>& object,
                                      bool allow_null_context);
 
+ public:
   // The following methods are convenience wrappers for methods on v8::Object
   // with the corresponding names.
   // Returns whether or not setting privates was successful.
