diff -r -u --color up/chromium/extensions/renderer/api/app_window_custom_bindings.cc nw/chromium/extensions/renderer/api/app_window_custom_bindings.cc
--- up/chromium/extensions/renderer/api/app_window_custom_bindings.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/renderer/api/app_window_custom_bindings.cc	2023-02-19 17:36:36.895467652 +0000
@@ -5,6 +5,8 @@
 #include "extensions/renderer/api/app_window_custom_bindings.h"
 
 #include "base/bind.h"
+#include "third_party/blink/public/web/blink.h"
+
 #include "base/command_line.h"
 #include "content/public/renderer/render_frame.h"
 #include "content/public/renderer/render_thread.h"
@@ -35,6 +37,18 @@
       "ResumeParser",
       base::BindRepeating(&AppWindowCustomBindings::ResumeParser,
                           base::Unretained(this)));
+  RouteHandlerFunction("FixGamePadAPI",
+                base::BindRepeating(&AppWindowCustomBindings::FixGamePadAPI,
+                           base::Unretained(this)));
+}
+
+void AppWindowCustomBindings::FixGamePadAPI(
+    const v8::FunctionCallbackInfo<v8::Value>& args) {
+  content::RenderFrame* render_frame = context()->GetRenderFrame();
+  if (!render_frame)
+    return;
+  blink::WebLocalFrame* main_frame = render_frame->GetWebFrame();
+  blink::fix_gamepad_nw(main_frame);
 }
 
 void AppWindowCustomBindings::GetFrame(
