diff -r -u --color up/chromium/headless/BUILD.gn nw/chromium/headless/BUILD.gn
--- up/chromium/headless/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/headless/BUILD.gn	2023-02-24 20:12:10.492874768 +0000
@@ -647,8 +647,8 @@
   static_library("headless_renderer") {
     defines = []
     sources = [
-      "lib/browser/headless_web_contents_impl.cc",
-      "lib/browser/headless_web_contents_impl.h",
+      #"lib/browser/headless_web_contents_impl.cc",
+      #"lib/browser/headless_web_contents_impl.h",
       "lib/renderer/headless_content_renderer_client.cc",
       "lib/renderer/headless_content_renderer_client.h",
       "lib/utility/headless_content_utility_client.cc",
@@ -952,8 +952,8 @@
     sources += [
       "lib/browser/headless_content_browser_client.cc",
       "lib/browser/headless_content_browser_client.h",
-      "lib/utility/headless_content_utility_client.cc",
-      "lib/utility/headless_content_utility_client.h",
+      #"lib/utility/headless_content_utility_client.cc",
+      #"lib/utility/headless_content_utility_client.h",
     ]
   }
 
@@ -986,6 +986,13 @@
     deps += [ "//components/crash/content/browser" ]
   }
 
+  if ((is_linux || is_mac) && is_component_build) {
+    sources += [
+      "app/headless_shell_switches.cc",
+      "app/headless_shell_switches.h",
+    ]
+  }
+
   if (enable_printing) {
     deps += [
       "//components/printing/browser",
