diff -r -u --color up/chromium/ipc/ipc_logging.cc nw/chromium/ipc/ipc_logging.cc
--- up/chromium/ipc/ipc_logging.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ipc/ipc_logging.cc	2023-02-24 20:54:14.964483417 +0000
@@ -242,7 +242,7 @@
       }
     }
   }
-  if (enabled_on_stderr_) {
+  if (enabled_on_stderr_ && !sender_) {
     std::string message_name;
     if (data.message_name.empty()) {
       message_name = base::StringPrintf("[unknown type %d]", data.type);
