diff -r -u --color up/chromium/media/base/feedback_signal_accumulator.h nw/chromium/media/base/feedback_signal_accumulator.h
--- up/chromium/media/base/feedback_signal_accumulator.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/media/base/feedback_signal_accumulator.h	2023-02-25 19:07:19.233835534 +0000
@@ -10,6 +10,7 @@
 #include <ostream>
 
 #include "base/time/time.h"
+#include <algorithm>
 
 namespace media {
 
