diff -r -u --color up/chromium/mojo/public/mojom/base/message_pump_type.mojom nw/chromium/mojo/public/mojom/base/message_pump_type.mojom
--- up/chromium/mojo/public/mojom/base/message_pump_type.mojom	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/mojo/public/mojom/base/message_pump_type.mojom	2023-02-25 19:07:20.049839982 +0000
@@ -16,4 +16,5 @@
   kNsRunloop,
   [EnableIf=is_ios]
   kNsRunloop,
+  kNode,
 };
