diff -r -u --color up/chromium/net/proxy_resolution/configured_proxy_resolution_service.cc nw/chromium/net/proxy_resolution/configured_proxy_resolution_service.cc
--- up/chromium/net/proxy_resolution/configured_proxy_resolution_service.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/net/proxy_resolution/configured_proxy_resolution_service.cc	2023-02-25 19:07:20.881844517 +0000
@@ -1298,6 +1298,23 @@
   return previous_state;
 }
 
+void ConfiguredProxyResolutionService::ResetConfigService(
+       std::unique_ptr<ProxyConfigService> new_proxy_config_service) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  State previous_state = ResetProxyConfig(true);
+
+  // Release the old configuration service.
+  if (config_service_.get())
+    config_service_->RemoveObserver(this);
+
+  // Set the new configuration service.
+  config_service_ = std::move(new_proxy_config_service);
+  config_service_->AddObserver(this);
+
+  if (previous_state != STATE_NONE)
+    ApplyProxyConfigIfAvailable();
+}
+
 void ConfiguredProxyResolutionService::ForceReloadProxyConfig() {
   DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
   ResetProxyConfig(false);
