diff -r -u --color up/chromium/services/network/public/mojom/network_context.mojom nw/chromium/services/network/public/mojom/network_context.mojom
--- up/chromium/services/network/public/mojom/network_context.mojom	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/services/network/public/mojom/network_context.mojom	2023-02-25 19:07:21.525848028 +0000
@@ -1107,6 +1107,8 @@
   [EnableIf=is_chromeos]
   UpdateAdditionalCertificates(AdditionalCertificates? additional_certificates);
 
+  SetTrustAnchors(array<X509Certificate> trust_anchors);
+
   // Updates the CT policy to be used for requests. Only applies if the
   // NetworkContextParams set enforce_chrome_ct_policy to true.
   // TODO(rsleevi): Remove this once Chrome-specific policies are moved out
