diff -r -u --color up/chromium/storage/browser/file_system/isolated_context.cc nw/chromium/storage/browser/file_system/isolated_context.cc
--- up/chromium/storage/browser/file_system/isolated_context.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/storage/browser/file_system/isolated_context.cc	2023-02-24 20:54:16.792473092 +0000
@@ -206,6 +206,7 @@
       filesystem_id_(filesystem_id),
       file_info_(file_info),
       path_type_(path_type),
+      files_(),
       ref_counts_(0) {
   DCHECK(IsSinglePathIsolatedFileSystem(type_));
 }
