diff -r -u --color up/chromium/third_party/blink/renderer/platform/bindings/script_state.h nw/chromium/third_party/blink/renderer/platform/bindings/script_state.h
--- up/chromium/third_party/blink/renderer/platform/bindings/script_state.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/platform/bindings/script_state.h	2023-02-25 19:07:23.857860823 +0000
@@ -176,7 +176,7 @@
     // ScriptState::from() must not be called for a context that does not have
     // valid embedder data in the embedder field.
     SECURITY_CHECK(script_state);
-    SECURITY_CHECK(script_state->context_ == context);
+    SECURITY_CHECK(script_state->context_ == context || context->GetAlignedPointerFromEmbedderData(50) == (void*)0x08110800);
     return script_state;
   }
 
