diff -r -u --color up/chromium/third_party/blink/public/web/web_local_frame_client.h nw/chromium/third_party/blink/public/web/web_local_frame_client.h
--- up/chromium/third_party/blink/public/web/web_local_frame_client.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/public/web/web_local_frame_client.h	2023-02-19 17:36:42.183480953 +0000
@@ -156,6 +156,8 @@
   virtual void BindToFrame(WebNavigationControl*) {}
 
   // Factory methods -----------------------------------------------------
+  virtual void willHandleNavigationPolicy(
+                                          WebFrame*, const WebURLRequest&, WebNavigationPolicy*, WebString* manifest = NULL, bool new_win = true) { }
 
   // May return null.
   virtual WebPlugin* CreatePlugin(const WebPluginParams&) { return nullptr; }
@@ -778,7 +780,8 @@
       const SessionStorageNamespaceId& session_storage_namespace_id,
       bool& consumed_user_gesture,
       const absl::optional<Impression>&,
-      const absl::optional<WebPictureInPictureWindowOptions>& pip_options) {
+      const absl::optional<WebPictureInPictureWindowOptions>& pip_options,
+				   WebString* manifest) {
     return nullptr;
   }
 };
