diff -r -u --color up/chromium/third_party/blink/public/mojom/renderer_preferences.mojom nw/chromium/third_party/blink/public/mojom/renderer_preferences.mojom
--- up/chromium/third_party/blink/public/mojom/renderer_preferences.mojom	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/public/mojom/renderer_preferences.mojom	2023-02-25 19:07:22.581853820 +0000
@@ -107,6 +107,9 @@
   // Whether or not caret browsing is enabled.
   bool caret_browsing_enabled = false;
 
+  string nw_inject_js_doc_start;
+  string nw_inject_js_doc_end;
+
   [EnableIf=renderer_pref_system_font_family_name]
   string system_font_family_name;
 
@@ -161,4 +164,5 @@
   bool selection_clipboard_buffer_available = false;
 
   array<uint16> explicitly_allowed_network_ports;
+
 };
