diff -r -u --color up/chromium/third_party/blink/renderer/core/html/html_iframe_element.cc nw/chromium/third_party/blink/renderer/core/html/html_iframe_element.cc
--- up/chromium/third_party/blink/renderer/core/html/html_iframe_element.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/html_iframe_element.cc	2023-02-25 19:07:23.057856433 +0000
@@ -23,6 +23,7 @@
  */
 
 #include "third_party/blink/renderer/core/html/html_iframe_element.h"
+#include "third_party/blink/renderer/core/frame/local_frame.h"
 
 #include "base/metrics/histogram_macros.h"
 #include "services/network/public/cpp/features.h"
@@ -216,6 +217,17 @@
     }
     SetSandboxFlags(current_flags);
     UseCounter::Count(GetDocument(), WebFeature::kSandboxViaIFrame);
+  } else if (name == html_names::kNwuseragentAttr) {
+    if (nwuseragent_ != value) {
+      nwuseragent_ = value;
+      FrameOwnerPropertiesChanged();
+    }
+  } else if (name == html_names::kNwfaketopAttr) {
+    bool old_faketop = nwfaketop_;
+    nwfaketop_ = !value.IsNull();
+    if (nwfaketop_ != old_faketop) {
+      FrameOwnerPropertiesChanged();
+    }
   } else if (name == html_names::kReferrerpolicyAttr) {
     referrer_policy_ = network::mojom::ReferrerPolicy::kDefault;
     if (!value.IsNull()) {
