diff -r -u --color up/chromium/third_party/blink/renderer/core/frame/remote_frame_owner.h nw/chromium/third_party/blink/renderer/core/frame/remote_frame_owner.h
--- up/chromium/third_party/blink/renderer/core/frame/remote_frame_owner.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/frame/remote_frame_owner.h	2023-02-24 20:54:17.752467670 +0000
@@ -79,6 +79,11 @@
 
   void Trace(Visitor*) const override;
 
+  bool nwfaketop() const override { return nwfaketop_; }
+  AtomicString nwuseragent() const override { return nwuseragent_; }
+  void set_nwfaketop(bool flag) { nwfaketop_ = flag; }
+  void set_nwuseragent(const WebString& agent) { nwuseragent_ = agent; }
+
  private:
   // Intentionally private to prevent redundant checks when the type is
   // already HTMLFrameOwnerElement.
@@ -89,6 +94,8 @@
   FramePolicy frame_policy_;
   AtomicString browsing_context_container_name_;
   mojom::blink::ScrollbarMode scrollbar_;
+  bool nwfaketop_;
+  WebString nwuseragent_;
   int margin_width_;
   int margin_height_;
   bool allow_fullscreen_;
