diff -r -u --color up/chromium/third_party/blink/renderer/core/html/html_iframe_element.idl nw/chromium/third_party/blink/renderer/core/html/html_iframe_element.idl
--- up/chromium/third_party/blink/renderer/core/html/html_iframe_element.idl	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/html_iframe_element.idl	2023-02-19 17:36:42.575481935 +0000
@@ -27,6 +27,9 @@
     Exposed=Window,
     HTMLConstructor
 ] interface HTMLIFrameElement : HTMLElement {
+    [CEReactions, Reflect] attribute boolean nwdisable;
+    [CEReactions, Reflect] attribute boolean nwfaketop;
+    [CEReactions, Reflect, Custom=Setter] attribute [TreatNullAs=EmptyString] DOMString nwUserAgent;
     [CEReactions, Reflect, URL] attribute USVString src;
     [CEReactions, Reflect, RaisesException=Setter] attribute HTMLString srcdoc;
     [CEReactions, Reflect] attribute DOMString name;
