diff -r -u --color up/chromium/third_party/blink/renderer/core/frame/frame_owner.h nw/chromium/third_party/blink/renderer/core/frame/frame_owner.h
--- up/chromium/third_party/blink/renderer/core/frame/frame_owner.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/frame/frame_owner.h	2023-02-19 17:36:42.503481755 +0000
@@ -51,6 +51,9 @@
   // child frame is occluded or has visual effects applied.
   virtual void SetNeedsOcclusionTracking(bool) = 0;
 
+  virtual AtomicString nwuseragent() const = 0;
+  virtual bool nwfaketop() const = 0;
+
   // Returns the 'name' content attribute value of the browsing context
   // container.
   // https://html.spec.whatwg.org/C/#browsing-context-container
@@ -137,6 +140,8 @@
   mojom::blink::ScrollbarMode ScrollbarMode() const override {
     return mojom::blink::ScrollbarMode::kAuto;
   }
+  AtomicString nwuseragent() const override { return AtomicString(); }
+  bool nwfaketop() const override { return false; }
   int MarginWidth() const override { return -1; }
   int MarginHeight() const override { return -1; }
   bool AllowFullscreen() const override { return false; }
