diff -r -u --color up/chromium/third_party/blink/renderer/core/loader/empty_clients.h nw/chromium/third_party/blink/renderer/core/loader/empty_clients.h
--- up/chromium/third_party/blink/renderer/core/loader/empty_clients.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/loader/empty_clients.h	2023-02-19 17:36:42.751482376 +0000
@@ -106,7 +106,7 @@
             LocalFrame& opener_frame,
             NavigationPolicy navigation_policy,
             const mojom::blink::WindowFeatures& window_features,
-            bool consumed_user_gesture) override {}
+            bool consumed_user_gesture, WebString* manifest = nullptr) override {}
   void DidOverscroll(const gfx::Vector2dF&,
                      const gfx::Vector2dF&,
                      const gfx::PointF&,
@@ -155,7 +155,7 @@
                              const WebWindowFeatures&,
                              network::mojom::blink::WebSandboxFlags,
                              const SessionStorageNamespaceId&,
-                             bool& consumed_user_gesture) override {
+                             bool& consumed_user_gesture, WebString*) override {
     return nullptr;
   }
   bool OpenJavaScriptAlertDelegate(LocalFrame*, const String&) override {
