diff -r -u --color up/chromium/third_party/blink/renderer/core/xmlhttprequest/xml_http_request.cc nw/chromium/third_party/blink/renderer/core/xmlhttprequest/xml_http_request.cc
--- up/chromium/third_party/blink/renderer/core/xmlhttprequest/xml_http_request.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/xmlhttprequest/xml_http_request.cc	2023-02-24 20:54:18.092465750 +0000
@@ -685,11 +685,13 @@
     // exception thrown.
     // Refer : https://xhr.spec.whatwg.org/#sync-warning
     // Use count for XHR synchronous requests on main thread only.
+#if 0
     if (!window->document()->ProcessingBeforeUnload()) {
       Deprecation::CountDeprecation(
           GetExecutionContext(),
           WebFeature::kXMLHttpRequestSynchronousInNonWorkerOutsideBeforeUnload);
     }
+#endif
   }
 
   method_ = FetchUtils::NormalizeMethod(method);
