diff -r -u --color up/chromium/third_party/blink/renderer/platform/wtf/BUILD.gn nw/chromium/third_party/blink/renderer/platform/wtf/BUILD.gn
--- up/chromium/third_party/blink/renderer/platform/wtf/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/platform/wtf/BUILD.gn	2023-02-19 17:36:43.651484632 +0000
@@ -15,6 +15,7 @@
   "//:gn_all",
   "//mojo/public/cpp/bindings/*",
   "//third_party/blink/*",
+  "//content/nw/*",
 ]
 
 config("wtf_config") {
