diff -r -u --color up/chromium/third_party/blink/renderer/core/frame/platform_event_controller.h nw/chromium/third_party/blink/renderer/core/frame/platform_event_controller.h
--- up/chromium/third_party/blink/renderer/core/frame/platform_event_controller.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/frame/platform_event_controller.h	2023-02-25 19:07:23.001856125 +0000
@@ -30,6 +30,7 @@
 
   void Trace(Visitor*) const override;
   LocalDOMWindow& GetWindow() const { return *window_; }
+  void SetWindow(LocalDOMWindow* win) { window_ = win; }
 
  protected:
   explicit PlatformEventController(LocalDOMWindow&);
