diff -r -u --color up/chromium/third_party/blink/renderer/core/script/modulator_impl_base.h nw/chromium/third_party/blink/renderer/core/script/modulator_impl_base.h
--- up/chromium/third_party/blink/renderer/core/script/modulator_impl_base.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/script/modulator_impl_base.h	2023-02-24 20:54:17.996466292 +0000
@@ -45,6 +45,8 @@
 
   bool IsScriptingDisabled() const override;
 
+  void AddToMap(const KURL& url, ModuleScript* script) override;
+
   mojom::blink::V8CacheOptions GetV8CacheOptions() const final;
 
   ModuleRecordResolver* GetModuleRecordResolver() override {
