diff -r -u --color up/chromium/third_party/blink/renderer/modules/websockets/websocket_channel_impl.cc nw/chromium/third_party/blink/renderer/modules/websockets/websocket_channel_impl.cc
--- up/chromium/third_party/blink/renderer/modules/websockets/websocket_channel_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/modules/websockets/websocket_channel_impl.cc	2023-02-25 19:07:23.809860560 +0000
@@ -1011,7 +1011,7 @@
     uint32_t readable_size;
     const MojoResult begin_result = readable_->BeginReadData(
         &buffer, &readable_size, MOJO_READ_DATA_FLAG_NONE);
-    if (begin_result == MOJO_RESULT_SHOULD_WAIT) {
+    if (begin_result == MOJO_RESULT_SHOULD_WAIT || begin_result == MOJO_RESULT_BUSY) {
       readable_watcher_.ArmOrNotify();
       return;
     }
