diff -r -u --color up/chromium/third_party/blink/common/privacy_budget/identifiability_study_settings.cc nw/chromium/third_party/blink/common/privacy_budget/identifiability_study_settings.cc
--- up/chromium/third_party/blink/common/privacy_budget/identifiability_study_settings.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/common/privacy_budget/identifiability_study_settings.cc	2023-02-19 17:36:41.611479519 +0000
@@ -71,7 +71,8 @@
   // Same restrictions as IdentifiabilityStudySettings::SetGlobalProvider().
   void SetProvider(
       std::unique_ptr<IdentifiabilityStudySettingsProvider> provider) {
-    DCHECK(!initialized_.IsSet());
+    if (initialized_.IsSet())
+      return;
     initialized_settings_.emplace(std::move(provider));
     initialized_.Set();
   }
