diff -r -u --color up/chromium/third_party/blink/renderer/core/html/html_iframe_element.h nw/chromium/third_party/blink/renderer/core/html/html_iframe_element.h
--- up/chromium/third_party/blink/renderer/core/html/html_iframe_element.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/html_iframe_element.h	2023-02-25 19:07:23.057856433 +0000
@@ -93,7 +93,10 @@
   // browser changed, such as 'id' and 'src'. This will send an IPC to the
   // browser about the updates.
   void DidChangeAttributes() override;
+  AtomicString nwuseragent() const override { return nwuseragent_; }
+  bool nwfaketop() const override { return nwfaketop_; }
 
+  AtomicString nwuseragent_;
   AtomicString name_;
   AtomicString required_csp_;
   AtomicString allow_;
@@ -105,6 +108,7 @@
   // API). If present, this is parsed in ConstructTrustTokenParams.
   AtomicString trust_token_;
   bool allow_fullscreen_;
+  bool nwfaketop_;
   bool allow_payment_request_;
   bool collapsed_by_client_;
   bool credentialless_ = false;
