diff -r -u --color up/chromium/third_party/zlib/google/zip_reader.cc nw/chromium/third_party/zlib/google/zip_reader.cc
--- up/chromium/third_party/zlib/google/zip_reader.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/zlib/google/zip_reader.cc	2023-02-19 17:37:07.127539695 +0000
@@ -150,7 +150,7 @@
   zip_file_ = internal::OpenHandleForUnzipping(zip_fd);
 #endif
   if (!zip_file_) {
-    LOG(ERROR) << "Cannot open ZIP from file handle " << zip_fd;
+    VLOG(1) << "Cannot open ZIP from file handle " << zip_fd;
     return false;
   }
 
