diff -r -u --color up/chromium/third_party/blink/renderer/core/html/html_attribute_names.json5 nw/chromium/third_party/blink/renderer/core/html/html_attribute_names.json5
--- up/chromium/third_party/blink/renderer/core/html/html_attribute_names.json5	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/html_attribute_names.json5	2023-02-19 17:36:42.567481915 +0000
@@ -350,5 +350,12 @@
     "webkitdirectory",
     "width",
     "wrap",
+    "nwworkingdir",
+    "nwdisable",
+    "nwfaketop",
+    "nwdirectory",
+    "nwdirectorydesc",
+    "nwsaveas",
+    "nwuseragent",
   ],
 }
