diff -r -u --color up/chromium/third_party/blink/renderer/core/paint/timing/text_paint_timing_detector.h nw/chromium/third_party/blink/renderer/core/paint/timing/text_paint_timing_detector.h
--- up/chromium/third_party/blink/renderer/core/paint/timing/text_paint_timing_detector.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/paint/timing/text_paint_timing_detector.h	2023-02-25 19:07:23.325857904 +0000
@@ -16,6 +16,7 @@
 #include "ui/gfx/geometry/rect_conversions.h"
 
 namespace blink {
+class WebString;
 class LayoutBoxModelObject;
 class LocalFrameView;
 class PropertyTreeStateOrAlias;
