diff -r -u --color up/chromium/third_party/blink/renderer/core/html/media/autoplay_policy.cc nw/chromium/third_party/blink/renderer/core/html/media/autoplay_policy.cc
--- up/chromium/third_party/blink/renderer/core/html/media/autoplay_policy.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/media/autoplay_policy.cc	2023-02-25 19:07:23.069856499 +0000
@@ -70,6 +70,9 @@
   if (document.IsInWebAppScope())
     return Type::kNoUserGestureRequired;
 
+  if (document.GetFrame()->isNodeJS())
+    return Type::kNoUserGestureRequired;
+
   if (DocumentHasUserExceptionFlag(document))
     return Type::kNoUserGestureRequired;
 
@@ -265,7 +268,7 @@
 
 absl::optional<DOMExceptionCode> AutoplayPolicy::RequestPlay() {
   if (!LocalFrame::HasTransientUserActivation(
-          element_->GetDocument().GetFrame())) {
+                                              element_->GetDocument().GetFrame()) && !element_->GetDocument().GetFrame()->isNodeJS()) {
     autoplay_uma_helper_->OnAutoplayInitiated(AutoplaySource::kMethod);
     if (IsGestureNeededForPlayback())
       return DOMExceptionCode::kNotAllowedError;
