diff -r -u --color up/chromium/third_party/blink/public/common/renderer_preferences/renderer_preferences.h nw/chromium/third_party/blink/public/common/renderer_preferences/renderer_preferences.h
--- up/chromium/third_party/blink/public/common/renderer_preferences/renderer_preferences.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/public/common/renderer_preferences/renderer_preferences.h	2023-02-19 17:36:42.087480712 +0000
@@ -75,6 +75,8 @@
   bool plugin_fullscreen_allowed{true};
   bool caret_browsing_enabled{false};
   std::vector<uint16_t> explicitly_allowed_network_ports;
+  std::string nw_inject_js_doc_start;
+  std::string nw_inject_js_doc_end;
 
   RendererPreferences();
   RendererPreferences(const RendererPreferences& other);
