diff -r -u --color up/chromium/third_party/blink/renderer/core/frame/web_local_frame_impl.cc nw/chromium/third_party/blink/renderer/core/frame/web_local_frame_impl.cc
--- up/chromium/third_party/blink/renderer/core/frame/web_local_frame_impl.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/frame/web_local_frame_impl.cc	2023-02-19 17:36:42.527481815 +0000
@@ -2342,6 +2342,8 @@
             std::move(policy_container),
             GetFrame()->DomWindow()->GetStorageKey(), document_ukm_source_id);
   };
+  owner_properties.nwFakeTop = owner_element->FastHasAttribute(html_names::kNwfaketopAttr);
+  owner_properties.nwuseragent = owner_element->nwuseragent();
 
   // FIXME: Using subResourceAttributeName as fallback is not a perfect
   // solution. subResourceAttributeName returns just one attribute name. The
