diff -r -u --color up/chromium/third_party/blink/renderer/core/probe/core_probes.h nw/chromium/third_party/blink/renderer/core/probe/core_probes.h
--- up/chromium/third_party/blink/renderer/core/probe/core_probes.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/probe/core_probes.h	2023-02-24 20:54:17.992466315 +0000
@@ -124,6 +124,26 @@
 }
 
 inline CoreProbeSink* ToCoreProbeSink(Document& document) {
+  LocalFrame* frame = document.GetFrame();
+  if (!frame && document.TemplateDocumentHost())
+    frame = document.TemplateDocumentHost()->GetFrame();
+  // filter out non-jail frame instrumentations
+  if (frame) {
+    Frame* jail_frame = NULL;
+    if ((jail_frame = frame->getDevtoolsJail()) != NULL) {
+      Frame* f = document.GetFrame();
+      bool in_jail_frame = false;
+      while (f) {
+        if (f == jail_frame) {
+          in_jail_frame = true;
+          break;
+        }
+        f = f->Tree().Parent();
+      }
+      if (!in_jail_frame)
+        return NULL;
+    }
+  }
   return ToCoreProbeSink(document.GetExecutionContext());
 }
 
