diff -r -u --color up/chromium/third_party/blink/renderer/platform/weborigin/security_origin.h nw/chromium/third_party/blink/renderer/platform/weborigin/security_origin.h
--- up/chromium/third_party/blink/renderer/platform/weborigin/security_origin.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/platform/weborigin/security_origin.h	2023-02-25 19:07:24.081862053 +0000
@@ -135,6 +135,7 @@
   String Protocol() const { return protocol_; }
   String Host() const { return host_; }
   String Domain() const { return domain_; }
+  bool hasUniversalAccess() const { return universal_access_; }
 
   // Returns the registrable domain if available.
   // For non-tuple origin, IP address URL, and public suffixes, this returns a
