diff -r -u --color up/chromium/third_party/blink/public/web/web_shared_worker.h nw/chromium/third_party/blink/public/web/web_shared_worker.h
--- up/chromium/third_party/blink/public/web/web_shared_worker.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/public/web/web_shared_worker.h	2023-02-24 20:12:13.956899939 +0000
@@ -51,6 +51,11 @@
 #include "third_party/blink/public/platform/web_policy_container.h"
 #include "third_party/blink/public/platform/web_security_origin.h"
 
+namespace base {
+class SingleThreadTaskRunner;
+class FilePath;
+}
+
 namespace blink {
 
 class MessagePortDescriptor;
@@ -70,6 +75,7 @@
   // starts a worker context.
   // WebSharedWorkerClient given here should own this instance.
   static std::unique_ptr<WebSharedWorker> CreateAndStart(
+      bool nodejs, const base::FilePath& root_path,
       const blink::SharedWorkerToken& token,
       const WebURL& script_url,
       mojom::ScriptType script_type,
