diff -r -u --color up/chromium/third_party/blink/renderer/core/html/forms/file_chooser.cc nw/chromium/third_party/blink/renderer/core/html/forms/file_chooser.cc
--- up/chromium/third_party/blink/renderer/core/html/forms/file_chooser.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/forms/file_chooser.cc	2023-02-19 17:36:42.547481865 +0000
@@ -110,6 +110,7 @@
 }
 
 void FileChooser::DidChooseFiles(mojom::blink::FileChooserResultPtr result) {
+  bool canceled = false;
   // TODO(tkent): If |result| is nullptr, we should not clear the
   // already-selected files in <input type=file> like other browsers.
   FileChooserFileInfoList files;
@@ -135,14 +136,16 @@
     }
     if (!was_changed) {
       DidCloseChooser();
-      return;
+      canceled = true;
     }
   }
 
   if (client_) {
     client_->FilesChosen(std::move(files),
-                         result ? result->base_directory : base::FilePath());
+                         result ? result->base_directory : base::FilePath(), canceled);
   }
+  if (canceled)
+    return;
   DidCloseChooser();
 }
 
