diff -r -u --color up/chromium/third_party/blink/renderer/core/page/chrome_client.cc nw/chromium/third_party/blink/renderer/core/page/chrome_client.cc
--- up/chromium/third_party/blink/renderer/core/page/chrome_client.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/page/chrome_client.cc	2023-02-24 20:54:17.956466518 +0000
@@ -79,15 +79,17 @@
     const WebWindowFeatures& features,
     network::mojom::blink::WebSandboxFlags sandbox_flags,
     const SessionStorageNamespaceId& session_storage_namespace_id,
-    bool& consumed_user_gesture) {
+    bool& consumed_user_gesture, WebString* manifest) {
+#if 0
   if (!CanOpenUIElementIfDuringPageDismissal(
           frame->Tree().Top(), UIElementType::kPopup, g_empty_string)) {
     return nullptr;
   }
+#endif
 
   return CreateWindowDelegate(frame, r, frame_name, features, sandbox_flags,
                               session_storage_namespace_id,
-                              consumed_user_gesture);
+                              consumed_user_gesture, manifest);
 }
 
 template <typename Delegate>
