diff -r -u --color up/chromium/third_party/blink/public/mojom/frame/frame.mojom nw/chromium/third_party/blink/public/mojom/frame/frame.mojom
--- up/chromium/third_party/blink/public/mojom/frame/frame.mojom	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/public/mojom/frame/frame.mojom	2023-02-19 17:36:42.107480762 +0000
@@ -1068,7 +1068,8 @@
   ShowCreatedWindow(blink.mojom.LocalFrameToken opener_frame_token,
                     ui.mojom.WindowOpenDisposition disposition,
                     blink.mojom.WindowFeatures window_features,
-                    bool opened_by_user_gesture) => ();
+                    bool opened_by_user_gesture,
+                    mojo_base.mojom.String16 manifest) => ();
 
   // Request that the browser change the bounds of the window.
   // This corresponds to the window.resizeTo() and window.moveTo() APIs, and the browser
