diff -r -u --color up/chromium/third_party/blink/public/web/web_frame.h nw/chromium/third_party/blink/public/web/web_frame.h
--- up/chromium/third_party/blink/public/web/web_frame.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/public/web/web_frame.h	2023-02-24 20:54:17.504469071 +0000
@@ -90,6 +90,12 @@
       CrossVariantMojoAssociatedReceiver<mojom::RemoteFrameInterfaceBase>
           receiver,
       mojom::FrameReplicationStatePtr replicated_state);
+  virtual bool isNwFakeTop() const {return false;}
+  virtual void setNodeJS(bool) {}
+  virtual bool isNodeJS() const {return false;}
+  virtual bool isNwDisabledChildFrame() const {return false;}
+  virtual void setDevtoolsJail(WebFrame*) {}
+  virtual WebFrame* getDevtoolsJail() {return nullptr;}
 
   // This method closes and deletes the WebFrame. This is typically called by
   // the embedder in response to a frame detached callback to the WebFrame
@@ -187,8 +193,8 @@
   // RemoteFrame <--> RenderFrameProxyHost in the browser process.
   const FrameToken& GetFrameToken() const { return frame_token_; }
 
-#if INSIDE_BLINK
   static WebFrame* FromCoreFrame(Frame*);
+#if INSIDE_BLINK
   static Frame* ToCoreFrame(const WebFrame&);
 #endif
 
