diff -r -u --color up/chromium/third_party/blink/renderer/core/script/module_map.cc nw/chromium/third_party/blink/renderer/core/script/module_map.cc
--- up/chromium/third_party/blink/renderer/core/script/module_map.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/script/module_map.cc	2023-02-25 19:07:23.337857969 +0000
@@ -34,6 +34,8 @@
   ModuleScript* GetModuleScript() const;
 
  private:
+  friend class ModuleMap;
+
   void DispatchFinishedNotificationAsync(SingleModuleClient*);
 
   // Implements ModuleScriptLoaderClient
@@ -52,6 +54,14 @@
   DCHECK(map_);
 }
 
+void ModuleMap::AddToMap(const KURL& url, ModuleScript* script) {
+  MapImpl::AddResult result = map_.insert(std::make_pair(url, ModuleType::kJavaScript), nullptr);
+  Member<Entry>& entry = result.stored_value->value;
+  entry = MakeGarbageCollected<Entry>(this);
+  entry->module_script_ = script;
+  entry->is_fetching_ = false;
+}
+
 void ModuleMap::Entry::Trace(Visitor* visitor) const {
   visitor->Trace(module_script_);
   visitor->Trace(map_);
