diff -r -u --color up/chromium/third_party/blink/renderer/core/fileapi/file_list.idl nw/chromium/third_party/blink/renderer/core/fileapi/file_list.idl
--- up/chromium/third_party/blink/renderer/core/fileapi/file_list.idl	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/fileapi/file_list.idl	2023-02-25 19:07:22.965855928 +0000
@@ -29,6 +29,9 @@
     Exposed=(Window,Worker),
     Serializable
 ] interface FileList {
+    [Custom] constructor();
     getter File? item(unsigned long index);
     readonly attribute unsigned long length;
+    void clear();
+    void append(File item);
 };
