diff -r -u --color up/chromium/third_party/blink/public/web/web_node.h nw/chromium/third_party/blink/public/web/web_node.h
--- up/chromium/third_party/blink/public/web/web_node.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/public/web/web_node.h	2023-02-24 20:54:17.504469071 +0000
@@ -120,8 +120,8 @@
   template <typename T>
   T DynamicTo() const;
 
-#if INSIDE_BLINK
   WebNode(Node*);
+#if INSIDE_BLINK
   WebNode& operator=(Node*);
   operator Node*() const;
 
