diff -r -u --color up/chromium/third_party/blink/renderer/modules/csspaint/paint_worklet_global_scope_proxy.cc nw/chromium/third_party/blink/renderer/modules/csspaint/paint_worklet_global_scope_proxy.cc
--- up/chromium/third_party/blink/renderer/modules/csspaint/paint_worklet_global_scope_proxy.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/modules/csspaint/paint_worklet_global_scope_proxy.cc	2023-02-25 19:07:23.541859089 +0000
@@ -44,7 +44,7 @@
           ? frame_client->ReducedUserAgent()
           : frame_client->UserAgent();
 
-  auto creation_params = std::make_unique<GlobalScopeCreationParams>(
+  auto creation_params = std::make_unique<GlobalScopeCreationParams>(false, std::string(),
       window->Url(), mojom::blink::ScriptType::kModule, global_scope_name,
       user_agent, frame_client->UserAgentMetadata(),
       frame_client->CreateWorkerFetchContext(),
