diff -r -u --color up/chromium/third_party/blink/renderer/platform/BUILD.gn nw/chromium/third_party/blink/renderer/platform/BUILD.gn
--- up/chromium/third_party/blink/renderer/platform/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/platform/BUILD.gn	2023-02-19 17:36:43.367483920 +0000
@@ -240,7 +240,7 @@
   visibility = []
   visibility = [ "./*" ]
   if (enable_precompiled_headers) {
-    if (is_win) {
+    if (is_win && false) {
       # This is a string rather than a file GN knows about. It has to match
       # exactly what's in the /FI flag below, and what might appear in the
       # source code in quotes for an #include directive.
