diff -r -u --color up/chromium/third_party/blink/renderer/core/inspector/inspector_page_agent.cc nw/chromium/third_party/blink/renderer/core/inspector/inspector_page_agent.cc
--- up/chromium/third_party/blink/renderer/core/inspector/inspector_page_agent.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/inspector/inspector_page_agent.cc	2023-02-25 19:07:23.113856740 +0000
@@ -146,6 +146,8 @@
       return DispositionEnum::NewWindow;
     case kNavigationPolicyPictureInPicture:
       return DispositionEnum::NewWindow;
+    case kNavigationPolicyIgnore:
+      return DispositionEnum::Nwignore;
   }
   return DispositionEnum::CurrentTab;
 }
