diff -r -u --color up/chromium/third_party/blink/renderer/core/BUILD.gn nw/chromium/third_party/blink/renderer/core/BUILD.gn
--- up/chromium/third_party/blink/renderer/core/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/BUILD.gn	2023-02-25 19:07:22.717854567 +0000
@@ -190,7 +190,7 @@
   output_name = "blink_core"
 
   visibility = []  # Allow re-assignment of list.
-  visibility = [ "//third_party/blink/*" ]
+  visibility = [ "//third_party/blink/*", "//content/nw/*" ]
 
   sources = [
     "core_initializer.cc",
