diff -r -u --color up/chromium/third_party/blink/renderer/core/loader/frame_loader.h nw/chromium/third_party/blink/renderer/core/loader/frame_loader.h
--- up/chromium/third_party/blink/renderer/core/loader/frame_loader.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/loader/frame_loader.h	2023-02-24 20:12:14.368901512 +0000
@@ -210,6 +210,8 @@
   void SaveScrollAnchor();
   void SaveScrollState();
   void RestoreScrollPositionAndViewState();
+  void setUserAgentOverride(const String& agent);
+  String userAgentOverride() const;
 
   bool HasProvisionalNavigation() const {
     return committing_navigation_ || client_navigation_.get();
@@ -316,6 +318,8 @@
   enum class State { kUninitialized, kInitialized, kDetached };
   State state_ = State::kUninitialized;
 
+  String user_agent_override_;
+
   bool dispatching_did_clear_window_object_in_main_world_;
   bool committing_navigation_ = false;
   bool has_accessed_initial_document_ = false;
