diff -r -u --color up/chromium/third_party/blink/public/mojom/worker/shared_worker_info.mojom nw/chromium/third_party/blink/public/mojom/worker/shared_worker_info.mojom
--- up/chromium/third_party/blink/public/mojom/worker/shared_worker_info.mojom	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/public/mojom/worker/shared_worker_info.mojom	2023-02-25 19:07:22.593853886 +0000
@@ -8,12 +8,15 @@
 import "third_party/blink/public/mojom/loader/fetch_client_settings_object.mojom";
 import "third_party/blink/public/mojom/worker/worker_options.mojom";
 import "url/mojom/url.mojom";
+import "mojo/public/mojom/base/file_path.mojom";
 
 // Meta data that is necessary to create a new shared worker context. This
 // structure gets populated when a new SharedWorker object is created in the
 // parent context (e.g. Document), and passed onto the destination child
 // process where the shared worker runs.
 struct SharedWorkerInfo {
+  bool is_node_js;
+  mojo_base.mojom.FilePath root_path;
   url.mojom.Url url;
   WorkerOptions options;
   array<network.mojom.ContentSecurityPolicy> content_security_policies;
