diff -r -u --color up/chromium/third_party/blink/renderer/core/page/drag_controller.cc nw/chromium/third_party/blink/renderer/core/page/drag_controller.cc
--- up/chromium/third_party/blink/renderer/core/page/drag_controller.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/page/drag_controller.cc	2023-02-19 17:36:42.779482446 +0000
@@ -231,7 +231,9 @@
 
   LocalFrameView* frame_view(local_root.View());
   if (frame_view) {
-    DataTransferAccessPolicy policy = DataTransferAccessPolicy::kTypesReadable;
+    DataTransferAccessPolicy policy =
+      document_under_mouse_->domWindow()->GetSecurityOrigin()->hasUniversalAccess()
+      ? DataTransferAccessPolicy::kReadable : DataTransferAccessPolicy::kTypesReadable;
     DataTransfer* data_transfer = CreateDraggingDataTransfer(policy, drag_data);
     data_transfer->SetSourceOperation(drag_data->DraggingSourceOperationMask());
     local_root.GetEventHandler().CancelDragAndDrop(CreateMouseEvent(drag_data),
@@ -801,7 +803,10 @@
   if (!local_root.View())
     return false;
 
-  DataTransferAccessPolicy policy = DataTransferAccessPolicy::kTypesReadable;
+  DataTransferAccessPolicy policy =
+    document_under_mouse_->domWindow()->GetSecurityOrigin()->hasUniversalAccess()
+    ? DataTransferAccessPolicy::kReadable
+    : DataTransferAccessPolicy::kTypesReadable;
   DataTransfer* data_transfer = CreateDraggingDataTransfer(policy, drag_data);
   DragOperationsMask src_op_mask = drag_data->DraggingSourceOperationMask();
   data_transfer->SetSourceOperation(src_op_mask);
