diff -r -u --color up/chromium/tools/licenses/licenses.py nw/chromium/tools/licenses/licenses.py
--- up/chromium/tools/licenses/licenses.py	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/tools/licenses/licenses.py	2023-02-25 19:07:48.037990515 +0000
@@ -129,6 +129,7 @@
 #   - Directories that contain example code, or build tooling.
 #   - Nested third_party code inside other third_party libraries.
 ADDITIONAL_PATHS = (
+    os.path.join('content', 'nw'),
     os.path.join('chrome', 'common', 'extensions', 'docs', 'examples'),
     os.path.join('chrome', 'test', 'chromeos', 'autotest'),
     os.path.join('chrome', 'test', 'data'),
@@ -154,6 +155,16 @@
 # can't provide a README.chromium.  Please prefer a README.chromium
 # wherever possible.
 SPECIAL_CASES = {
+    os.path.join('content', 'nw'): {
+        "Name": "NW.js",
+        "URL": "http://nwjs.io",
+        "License": "MIT",
+    },
+    os.path.join('third_party', 'node'): {
+        "Name": "IO.js",
+        "URL": "https://iojs.org",
+        "License": "MIT",
+    },
     os.path.join('native_client'): {
         "Name": "native client",
         "URL": "http://code.google.com/p/nativeclient",
