diff -r -u --color up/chromium/tools/grit/grit_args.gni nw/chromium/tools/grit/grit_args.gni
--- up/chromium/tools/grit/grit_args.gni	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/tools/grit/grit_args.gni	2023-02-25 19:07:48.017990395 +0000
@@ -37,6 +37,7 @@
 
   # Mac and iOS want Title Case strings.
   "use_titlecase=${is_apple}",
+  "nwjs_sdk=${nwjs_sdk}",
 ]
 
 # Must match `enable_hidpi` in ui/base/ui_features.gni.
@@ -44,6 +45,11 @@
   _grit_defines += [ "scale_factors=2x" ]
 }
 
+if (nwjs_sdk) {
+  _grit_defines += [ "nwjs_sdk" ]
+}
+
+
 # Environment variables to be used by grit and its subproceses.
 
 _grit_environment_vars = [
