diff -r -u --color up/chromium/ui/accessibility/platform/inspect/ax_tree_formatter_auralinux.cc nw/chromium/ui/accessibility/platform/inspect/ax_tree_formatter_auralinux.cc
--- up/chromium/ui/accessibility/platform/inspect/ax_tree_formatter_auralinux.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/accessibility/platform/inspect/ax_tree_formatter_auralinux.cc	2023-02-25 19:07:48.661994262 +0000
@@ -239,6 +239,7 @@
       DCHECK_GE(index_in_parent, 0);
 #endif
 
+      link = atk_hypertext_get_link(hypertext, link_index);
       int utf8_offset = atk_hyperlink_get_start_index(link);
       gchar* link_start = g_utf8_offset_to_pointer(character_text, utf8_offset);
       int offset = link_start - character_text;
