diff -r -u --color up/chromium/ui/base/models/simple_menu_model.h nw/chromium/ui/base/models/simple_menu_model.h
--- up/chromium/ui/base/models/simple_menu_model.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/base/models/simple_menu_model.h	2023-02-25 19:07:48.761994862 +0000
@@ -72,6 +72,8 @@
     bool GetAcceleratorForCommandId(
         int command_id,
         ui::Accelerator* accelerator) const override;
+
+    virtual bool HasIcon(int command_id);
   };
 
   // The Delegate can be NULL, though if it is items can't be checked or
