diff -r -u --color up/chromium/ui/base/ui_base_types.h nw/chromium/ui/base/ui_base_types.h
--- up/chromium/ui/base/ui_base_types.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/base/ui_base_types.h	2023-02-25 19:07:48.773994934 +0000
@@ -29,7 +29,8 @@
   SHOW_STATE_MAXIMIZED = 3,
   SHOW_STATE_INACTIVE = 4,  // Views only, not persisted.
   SHOW_STATE_FULLSCREEN = 5,
-  SHOW_STATE_END = 6  // The end of show state enum.
+  SHOW_STATE_HIDDEN = 6,
+  SHOW_STATE_END = 7  // The end of show state enum.
 };
 
 #if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS_LACROS)
