diff -r -u --color up/chromium/ui/base/resource/resource_bundle_mac.mm nw/chromium/ui/base/resource/resource_bundle_mac.mm
--- up/chromium/ui/base/resource/resource_bundle_mac.mm	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/base/resource/resource_bundle_mac.mm	2023-02-25 19:07:48.769994910 +0000
@@ -51,13 +51,13 @@
 }  // namespace
 
 void ResourceBundle::LoadCommonResources() {
-  AddDataPackFromPath(GetResourcesPakFilePath(@"chrome_100_percent", nil),
+  AddDataPackFromPath(GetResourcesPakFilePath(@"nw_100_percent", nil),
                       k100Percent);
 
   // On Mac we load 1x and 2x resources and we let the UI framework decide
   // which one to use.
   if (IsScaleFactorSupported(k200Percent)) {
-    AddDataPackFromPath(GetResourcesPakFilePath(@"chrome_200_percent", nil),
+    AddDataPackFromPath(GetResourcesPakFilePath(@"nw_200_percent", nil),
                         k200Percent);
   }
 }
