diff -r -u --color up/chromium/ui/base/x/visual_picker_glx.cc nw/chromium/ui/base/x/visual_picker_glx.cc
--- up/chromium/ui/base/x/visual_picker_glx.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/base/x/visual_picker_glx.cc	2023-02-24 20:54:37.264356838 +0000
@@ -296,6 +296,9 @@
   if (configs) {
     system_visual_ = PickBestSystemVisual(*configs.reply);
     rgba_visual_ = PickBestRgbaVisual(*configs.reply);
+  } else {
+    system_visual_ = x11::VisualId{};
+    rgba_visual_   = x11::VisualId{};
   }
 }
 
