diff -r -u --color up/chromium/ui/display/display.cc nw/chromium/ui/display/display.cc
--- up/chromium/ui/display/display.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/display/display.cc	2023-02-25 19:07:49.021996423 +0000
@@ -21,6 +21,16 @@
 #include "ui/gfx/geometry/transform.h"
 #include "ui/gfx/icc_profile.h"
 
+namespace content {
+
+// Switch to enable / disable code for window's transparency
+DISPLAY_EXPORT bool g_support_transparency = true;
+
+// Switch to force cpu drawing, is used to enable click through on alpha pixels
+extern bool g_force_cpu_draw;
+bool g_force_cpu_draw = false;
+}
+
 namespace display {
 namespace {
 
