diff -r -u --color up/chromium/ui/gtk/select_file_dialog_linux_gtk.cc nw/chromium/ui/gtk/select_file_dialog_linux_gtk.cc
--- up/chromium/ui/gtk/select_file_dialog_linux_gtk.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/ui/gtk/select_file_dialog_linux_gtk.cc	2023-02-19 17:37:08.667543397 +0000
@@ -421,12 +421,12 @@
   std::string title_string = title;
   if (title_string.empty()) {
     title_string =
-        (type == SELECT_UPLOAD_FOLDER)
+      (false)
             ? l10n_util::GetStringUTF8(IDS_SELECT_UPLOAD_FOLDER_DIALOG_TITLE)
             : l10n_util::GetStringUTF8(IDS_SELECT_FOLDER_DIALOG_TITLE);
   }
   std::string accept_button_label =
-      (type == SELECT_UPLOAD_FOLDER)
+    (false)
           ? l10n_util::GetStringUTF8(
                 IDS_SELECT_UPLOAD_FOLDER_DIALOG_UPLOAD_BUTTON)
           : GetOpenLabel();
