diff -r -u --color up/chromium/ui/gl/gl_features.cc nw/chromium/ui/gl/gl_features.cc
--- up/chromium/ui/gl/gl_features.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/gl/gl_features.cc	2023-02-24 20:54:39.004346829 +0000
@@ -75,7 +75,7 @@
              base::FEATURE_DISABLED_BY_DEFAULT);
 #endif
 
-#if !defined(PASSTHROUGH_COMMAND_DECODER_LAUNCHED)
+#if !defined(PASSTHROUGH_COMMAND_DECODER_LAUNCHED) || BUILDFLAG(IS_LINUX)
 // Use the passthrough command decoder by default.  This can be overridden with
 // the --use-cmd-decoder=passthrough or --use-cmd-decoder=validating flags.
 // Feature lives in ui/gl because it affects the GL binding initialization on
