diff -r -u --color up/chromium/ui/message_center/public/cpp/notification.h nw/chromium/ui/message_center/public/cpp/notification.h
--- up/chromium/ui/message_center/public/cpp/notification.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/message_center/public/cpp/notification.h	2023-02-24 20:54:39.020346735 +0000
@@ -472,8 +472,7 @@
   }
 
   bool should_show_settings_button() const {
-    return optional_fields_.settings_button_handler !=
-           SettingsButtonHandler::NONE;
+    return false;
   }
 
   void set_settings_button_handler(SettingsButtonHandler handler) {
