diff -r -u --color up/chromium/ui/ozone/common/egl_util.cc nw/chromium/ui/ozone/common/egl_util.cc
--- up/chromium/ui/ozone/common/egl_util.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/ozone/common/egl_util.cc	2023-02-25 19:07:49.293998056 +0000
@@ -130,7 +130,7 @@
     if (!base::PathService::Get(base::DIR_MODULE, &module_path))
       return false;
 #endif
-
+    module_path = module_path.Append("lib/");
     glesv2_path = module_path.Append(kAngleGlesSoname);
     egl_path = module_path.Append(kAngleEglSoname);
   } else {
