diff -r -u --color up/chromium/ui/shell_dialogs/execute_select_file_win.cc nw/chromium/ui/shell_dialogs/execute_select_file_win.cc
--- up/chromium/ui/shell_dialogs/execute_select_file_win.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/shell_dialogs/execute_select_file_win.cc	2023-02-25 19:07:49.373998536 +0000
@@ -277,11 +277,12 @@
   }
 
   std::u16string ok_button_label;
+#if 0
   if (type == SelectFileDialog::SELECT_UPLOAD_FOLDER) {
     ok_button_label = l10n_util::GetStringUTF16(
         IDS_SELECT_UPLOAD_FOLDER_DIALOG_UPLOAD_BUTTON);
   }
-
+#endif
   DWORD dialog_options = FOS_PICKFOLDERS;
 
   std::vector<FileFilterSpec> no_filter;
