diff -r -u --color up/chromium/ui/views/widget/widget_delegate.cc nw/chromium/ui/views/widget/widget_delegate.cc
--- up/chromium/ui/views/widget/widget_delegate.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/ui/views/widget/widget_delegate.cc	2023-02-19 17:37:08.963544109 +0000
@@ -68,6 +68,8 @@
 
 void WidgetDelegate::OnWidgetMove() {}
 
+void WidgetDelegate::OnWidgetResize() {}
+
 void WidgetDelegate::OnDisplayChanged() {}
 
 void WidgetDelegate::OnWorkAreaChanged() {}
@@ -108,6 +110,10 @@
   return can_activate_;
 }
 
+bool WidgetDelegate::NWCanClose(bool user_force) const {
+  return true;
+}
+
 ui::ModalType WidgetDelegate::GetModalType() const {
   return params_.modal_type;
 }
@@ -141,6 +147,10 @@
   return params_.show_close_button;
 }
 
+bool WidgetDelegate::ShouldHandleOnSize() const {
+  return false;
+}
+
 ui::ImageModel WidgetDelegate::GetWindowAppIcon() {
   // Prefer app icon if available.
   if (!params_.app_icon.IsEmpty())
@@ -162,6 +172,14 @@
   return false;
 }
 
+bool WidgetDelegate::ExecuteAppCommand(int command_id) {
+  return ExecuteWindowsCommand(command_id);
+}
+
+bool WidgetDelegate::HandleSize(unsigned int param, const gfx::Size& size) {
+  return false;
+}
+
 std::string WidgetDelegate::GetWindowName() const {
   return std::string();
 }
