diff -r -u --color up/chromium/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc nw/chromium/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
--- up/chromium/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc	2023-02-19 17:37:08.955544090 +0000
@@ -869,6 +869,14 @@
   desktop_window_tree_host_->SetBoundsInDIP(bounds);
 }
 
+#if defined(OS_WIN)
+void DesktopNativeWidgetAura::SetPosition(const gfx::Point& pos) {
+  if (!content_window_)
+    return;
+  desktop_window_tree_host_->SetPositionInDIP(pos);
+}
+#endif
+
 void DesktopNativeWidgetAura::SetBoundsConstrained(const gfx::Rect& bounds) {
   if (!content_window_)
     return;
