diff -r -u --color up/chromium/ui/views/widget/native_widget_delegate.h nw/chromium/ui/views/widget/native_widget_delegate.h
--- up/chromium/ui/views/widget/native_widget_delegate.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/views/widget/native_widget_delegate.h	2023-02-25 19:07:49.553999617 +0000
@@ -40,6 +40,7 @@
  public:
   virtual ~NativeWidgetDelegate() = default;
 
+  virtual bool NWCanClose(bool user_force = false) const = 0;
   // Returns true if the window is modal.
   virtual bool IsModal() const = 0;
 
