diff -r -u --color up/chromium/ui/views/win/hwnd_message_handler.h nw/chromium/ui/views/win/hwnd_message_handler.h
--- up/chromium/ui/views/win/hwnd_message_handler.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/ui/views/win/hwnd_message_handler.h	2023-02-19 17:37:08.967544118 +0000
@@ -458,6 +458,7 @@
     CR_MSG_WM_SETTINGCHANGE(OnSettingChange)
     CR_MSG_WM_SIZE(OnSize)
     CR_MSG_WM_SIZING(OnSizing)
+    CR_MSG_WM_STYLECHANGING(OnStyleChanging)
     CR_MSG_WM_SYSCOMMAND(OnSysCommand)
     CR_MSG_WM_THEMECHANGED(OnThemeChanged)
     CR_MSG_WM_TIMECHANGE(OnTimeChange)
@@ -519,6 +520,7 @@
   void OnSettingChange(UINT flags, const wchar_t* section);
   void OnSize(UINT param, const gfx::Size& size);
   void OnSizing(UINT param, RECT* rect);
+  void OnStyleChanging(UINT nStyleType, LPSTYLESTRUCT lpStyleStruct);
   void OnSysCommand(UINT notification_code, const gfx::Point& point);
   void OnThemeChanged();
   void OnTimeChange();
@@ -596,6 +598,7 @@
   // please refer to the SetBounds() function.
   void SetBoundsInternal(const gfx::Rect& bounds_in_pixels,
                          bool force_size_changed);
+  void SetPosition(const gfx::Point& pos_in_pixels);
 
   // Checks if there is a full screen window on the same monitor as the
   // |window| which is becoming active. If yes then we reduce the size of the
