diff -r -u --color up/chromium/ui/views/controls/menu/native_menu_win.h nw/chromium/ui/views/controls/menu/native_menu_win.h
--- up/chromium/ui/views/controls/menu/native_menu_win.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/views/controls/menu/native_menu_win.h	2023-02-25 19:07:49.493999257 +0000
@@ -37,6 +37,9 @@
   void Rebuild(MenuInsertionDelegateWin* delegate);
   void UpdateStates();
 
+  void set_is_popup_menu(bool flag) { is_popup_menu_ = flag; }
+  HMENU menu() const { return menu_; }
+
  private:
   // IMPORTANT: Note about indices.
   //            Functions in this class deal in two index spaces:
@@ -103,6 +106,9 @@
 
   // If we're a submenu, this is our parent.
   raw_ptr<NativeMenuWin> parent_;
+
+  // A flag to indicate whether to create a menubar or popupmenu.
+  bool is_popup_menu_;
 };
 
 }  // namespace views
