diff -r -u --color up/chromium/ui/views/widget/desktop_aura/desktop_native_widget_aura.h nw/chromium/ui/views/widget/desktop_aura/desktop_native_widget_aura.h
--- up/chromium/ui/views/widget/desktop_aura/desktop_native_widget_aura.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/views/widget/desktop_aura/desktop_native_widget_aura.h	2023-02-25 19:07:49.549999593 +0000
@@ -154,6 +154,9 @@
   gfx::Rect GetRestoredBounds() const override;
   std::string GetWorkspace() const override;
   void SetBounds(const gfx::Rect& bounds) override;
+#if defined(OS_WIN)
+  void SetPosition(const gfx::Point& pos) override;
+#endif
   void SetBoundsConstrained(const gfx::Rect& bounds) override;
   void SetSize(const gfx::Size& size) override;
   void StackAbove(gfx::NativeView native_view) override;
