diff -r -u --color up/chromium/ui/views/widget/native_widget_private.cc nw/chromium/ui/views/widget/native_widget_private.cc
--- up/chromium/ui/views/widget/native_widget_private.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/views/widget/native_widget_private.cc	2023-02-24 20:54:39.220345562 +0000
@@ -10,6 +10,11 @@
 
 namespace views::internal {
 
+#if defined(OS_WIN)
+void NativeWidgetPrivate::SetPosition(const gfx::Point& pos) {
+
+}
+#endif
 // static
 gfx::Rect NativeWidgetPrivate::ConstrainBoundsToDisplayWorkArea(
     const gfx::Rect& bounds) {
