diff -r -u --color up/chromium/ui/views/widget/widget_hwnd_utils.cc nw/chromium/ui/views/widget/widget_hwnd_utils.cc
--- up/chromium/ui/views/widget/widget_hwnd_utils.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/views/widget/widget_hwnd_utils.cc	2023-02-24 20:12:34.109074229 +0000
@@ -11,6 +11,7 @@
 #include "ui/base/l10n/l10n_util_win.h"
 #include "ui/base/ui_base_features.h"
 #include "ui/base/ui_base_switches.h"
+#include "ui/display/display.h"
 #include "ui/views/widget/widget_delegate.h"
 #include "ui/views/win/hwnd_message_handler.h"
 
@@ -86,8 +87,19 @@
           native_widget_delegate->IsDialogBox() ? WS_EX_DLGMODALFRAME : 0;
 
       // See layered window comment below.
-      if (is_translucent)
-        *style &= static_cast<DWORD>(~(WS_THICKFRAME | WS_CAPTION));
+      if (content::g_support_transparency && is_translucent) {
+        if (params.remove_standard_frame) {
+          *style &= static_cast<DWORD>(~(WS_CAPTION));
+        }
+        if (!native_widget_delegate->IsDialogBox() && !native_widget_delegate->IsModal()) {
+          if (content::g_force_cpu_draw)
+            *ex_style |= WS_EX_LAYERED;
+        }
+      } else {
+        if (is_translucent)
+          *style &= static_cast<DWORD>(~(WS_THICKFRAME | WS_CAPTION));
+      }
+
       break;
     }
     case Widget::InitParams::TYPE_CONTROL:
@@ -165,9 +177,10 @@
   //
   // This doesn't work when Aero is disabled, so disable it in that case.
   // Software composited windows can continue to use WS_EX_LAYERED.
-  bool is_translucent =
+  bool is_translucent = !content::g_support_transparency ?
       (params.opacity == Widget::InitParams::WindowOpacity::kTranslucent &&
-       (ui::win::IsAeroGlassEnabled() || params.force_software_compositing));
+       (ui::win::IsAeroGlassEnabled() || params.force_software_compositing)) :
+    (params.opacity == Widget::InitParams::WindowOpacity::kTranslucent || params.force_software_compositing);
 
   CalculateWindowStylesFromInitParams(params, widget_delegate,
                                       native_widget_delegate, is_translucent,
