diff -r -u --color up/chromium/ui/views/widget/widget.h nw/chromium/ui/views/widget/widget.h
--- up/chromium/ui/views/widget/widget.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/ui/views/widget/widget.h	2023-02-19 17:37:08.963544109 +0000
@@ -632,6 +632,9 @@
 
   // Sizes and/or places the widget to the specified bounds, size or position.
   void SetBounds(const gfx::Rect& bounds);
+#if defined(OS_WIN)
+  void SetPosition(const gfx::Point& position);
+#endif
   void SetSize(const gfx::Size& size);
 
   // Sizes the window to the specified size and centers it.
@@ -681,13 +684,13 @@
 
   // Equivalent to CloseWithReason(ClosedReason::kUnspecified).
   // DEPRECATED: Please use CloseWithReason() instead.
-  void Close();
+  void Close(bool force = false);
 
   // Hides the widget, then closes it after a return to the message loop,
   // specifying the reason for it having been closed.
   // Note that while you can pass ClosedReason::kUnspecified, it is highly
   // discouraged and only supported for backwards-compatibility with Close().
-  void CloseWithReason(ClosedReason closed_reason);
+  void CloseWithReason(ClosedReason closed_reason, bool force = false);
 
   // A UI test which tries to asynchronously examine a widget (e.g. the pixel
   // tests) will fail if the widget is closed before that.  This can happen
@@ -1072,6 +1075,7 @@
 
   base::WeakPtr<Widget> GetWeakPtr();
 
+  bool NWCanClose(bool user_force = false) const override;
   // Overridden from NativeWidgetDelegate:
   bool IsModal() const override;
   bool IsDialogBox() const override;
@@ -1337,6 +1341,7 @@
   // Note that this may be ClosedReason::kUnspecified if the deprecated Close()
   // method was called rather than CloseWithReason().
   ClosedReason closed_reason_ = ClosedReason::kUnspecified;
+  bool force_closing_ = false;
 
   // The saved "show" state for this window. See note in SetInitialBounds
   // that explains why we save this.
