diff -r -u --color up/chromium/ui/views/widget/desktop_aura/desktop_window_tree_host.h nw/chromium/ui/views/widget/desktop_aura/desktop_window_tree_host.h
--- up/chromium/ui/views/widget/desktop_aura/desktop_window_tree_host.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/views/widget/desktop_aura/desktop_window_tree_host.h	2023-02-25 19:07:49.549999593 +0000
@@ -213,6 +213,11 @@
   virtual void UpdateWindowShapeIfNeeded(const ui::PaintContext& context);
 
   virtual DesktopNativeCursorManager* GetSingletonDesktopNativeCursorManager();
+
+#if defined(OS_WIN)
+  virtual void SetPositionInDIP(const gfx::Point& pos);
+#endif
+
 };
 
 }  // namespace views
