diff -r -u --color up/chromium/BUILD.gn nw/chromium/BUILD.gn
--- up/chromium/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/BUILD.gn	2023-02-19 17:36:23.891434763 +0000
@@ -128,6 +128,7 @@
 
   if (!is_ios && !is_android && !is_castos) {
     deps += [
+      "//content/nw:nwjs",
       "//chrome",
       "//chrome/browser/ui/color:dump_colors",
       "//chrome/test:browser_tests",
@@ -226,7 +227,7 @@
     ]
   }
 
-  if (!is_ios && !is_android) {
+  if (false && !is_ios && !is_android) {
     deps += [
       "//components/cronet:cronet_tests",
       "//components/cronet:cronet_unittests",
@@ -1635,16 +1636,6 @@
   }
 }
 
-# TODO(cassew): Add more OS's that don't support x86.
-is_valid_x86_target =
-    target_os != "ios" && target_os != "mac" &&
-    (target_os != "linux" || use_libfuzzer || !build_with_chromium)
-
-# Note: v8_target_cpu == arm allows using the V8 arm simulator on x86 for fuzzing.
-assert(
-    is_valid_x86_target || target_cpu != "x86" || v8_target_cpu == "arm",
-    "'target_cpu=x86' is not supported for 'target_os=$target_os'. Consider omitting 'target_cpu' (default) or using 'target_cpu=x64' instead.")
-
 group("chromium_builder_perf") {
   testonly = true
 
