diff -r -u --color up/chromium/build/config/dcheck_always_on.gni nw/chromium/build/config/dcheck_always_on.gni
--- up/chromium/build/config/dcheck_always_on.gni	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/build/config/dcheck_always_on.gni	2023-02-19 17:36:25.191438054 +0000
@@ -21,8 +21,8 @@
   # Other clients typically set this to false. If another client wants to use
   # the same default value as Chromium, we'd need to add a separate gclient
   # variable to replace build_with_chromium here.
-  dcheck_always_on =
-      (build_with_chromium && !is_official_build) || dcheck_is_configurable
+  dcheck_always_on = false
+      #(build_with_chromium && !is_official_build) || dcheck_is_configurable
 }
 
 declare_args() {
