diff -r -u --color up/chromium/chrome/browser/ui/views/frame/opaque_browser_frame_view.h nw/chromium/chrome/browser/ui/views/frame/opaque_browser_frame_view.h
--- up/chromium/chrome/browser/ui/views/frame/opaque_browser_frame_view.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/frame/opaque_browser_frame_view.h	2023-02-19 17:36:29.563449120 +0000
@@ -68,6 +68,7 @@
   void PaintAsActiveChanged() override;
   void UpdateFrameColor() override;
   void OnThemeChanged() override;
+  gfx::Size GetMaximumSize() const override;
 
   // views::NonClientFrameView:
   gfx::Rect GetBoundsForClientView() const override;
@@ -93,6 +94,7 @@
   std::u16string GetWindowTitle() const override;
   int GetIconSize() const override;
   gfx::Size GetBrowserViewMinimumSize() const override;
+  gfx::Size GetBrowserViewMaximumSize() const override;
   bool ShouldShowCaptionButtons() const override;
   bool IsRegularOrGuestSession() const override;
   bool CanMaximize() const override;
@@ -213,6 +215,7 @@
   void UpdateCaptionButtonToolTipsForWindowControlsOverlay();
 #endif
 
+  bool frameless_;
   // Our layout manager also calculates various bounds.
   raw_ptr<OpaqueBrowserFrameViewLayout> layout_;
 
