diff -r -u --color up/chromium/extensions/browser/extension_util.cc nw/chromium/extensions/browser/extension_util.cc
--- up/chromium/extensions/browser/extension_util.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/browser/extension_util.cc	2023-02-19 17:36:36.831467491 +0000
@@ -197,7 +197,7 @@
   return ui_util::ShouldDisplayInExtensionSettings(type, location) &&
          !Manifest::IsPolicyLocation(location) &&
          !Manifest::IsComponentLocation(location) &&
-         !PermissionsData::CanExecuteScriptEverywhere(extension_id, location);
+    !PermissionsData::CanExecuteScriptEverywhere(extension_id, location, type);
 }
 
 // The below functionality maps a context to a unique id by increasing a static
