diff -r -u --color up/chromium/extensions/common/api/content_scripts.idl nw/chromium/extensions/common/api/content_scripts.idl
--- up/chromium/extensions/common/api/content_scripts.idl	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/common/api/content_scripts.idl	2023-02-19 17:36:36.859467561 +0000
@@ -62,6 +62,8 @@
     // parent or opener frame matches one of the patterns declared in matches.
     // Defaults to false.
     boolean? match_about_blank;
+
+    boolean? in_main_world;
     // Applied after matches to include only those URLs that also match this
     // glob. Intended to emulate the
     // <a href="http://wiki.greasespot.net/Metadata_Block#.40include">@include
