diff -r -u --color up/chromium/chrome/test/base/test_browser_window.cc nw/chromium/chrome/test/base/test_browser_window.cc
--- up/chromium/chrome/test/base/test_browser_window.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/test/base/test_browser_window.cc	2023-02-19 17:36:30.203450740 +0000
@@ -75,6 +75,13 @@
 
 // TestBrowserWindow ----------------------------------------------------------
 
+void TestBrowserWindow::UpdateDraggableRegions(
+                                               const std::vector<extensions::DraggableRegion>& regions) {
+}
+SkRegion* TestBrowserWindow::GetDraggableRegion() {
+  return nullptr;
+}
+
 TestBrowserWindow::TestBrowserWindow() {}
 
 TestBrowserWindow::~TestBrowserWindow() {}
