diff -r -u --color up/chromium/chrome/browser/net/profile_network_context_service.h nw/chromium/chrome/browser/net/profile_network_context_service.h
--- up/chromium/chrome/browser/net/profile_network_context_service.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/net/profile_network_context_service.h	2023-02-19 17:36:28.047445283 +0000
@@ -100,6 +100,7 @@
 
   // Flushes all pending proxy configuration changes.
   void FlushProxyConfigMonitorForTesting();
+  void UpdateProxyConfig(const net::ProxyConfigWithAnnotation& proxy_config);
 
   static void SetDiscardDomainReliabilityUploadsForTesting(bool value);
 
