diff -r -u --color up/chromium/chrome/browser/ui/webui/print_preview/print_preview_handler.h nw/chromium/chrome/browser/ui/webui/print_preview/print_preview_handler.h
--- up/chromium/chrome/browser/ui/webui/print_preview/print_preview_handler.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/webui/print_preview/print_preview_handler.h	2023-02-19 17:36:29.767449637 +0000
@@ -41,6 +41,16 @@
 class WebContents;
 }
 
+namespace chrome {
+  base::Value::List PrintersToValues(const printing::PrinterList& printer_list);
+  void NWPrintSetCustomPrinting(bool value);
+  bool NWPrintGetCustomPrinting();
+  void NWPrintSetDefaultPrinter(const std::string& printer_name);
+  void NWPrintSetPDFPath(const base::FilePath& path);
+  const base::FilePath& NWPrintGetPDFPath();
+void NWPrintSetOptions(const base::DictionaryValue* dict, content::WebContents*);
+}
+
 namespace printing {
 
 class PdfPrinterHandler;
