diff -r -u --color up/chromium/third_party/blink/renderer/core/page/chrome_client.h nw/chromium/third_party/blink/renderer/core/page/chrome_client.h
--- up/chromium/third_party/blink/renderer/core/page/chrome_client.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/page/chrome_client.h	2023-02-19 17:36:42.779482446 +0000
@@ -235,7 +235,7 @@
                      const WebWindowFeatures&,
                      network::mojom::blink::WebSandboxFlags,
                      const SessionStorageNamespaceId&,
-                     bool& consumed_user_gesture);
+                     bool& consumed_user_gesture, WebString* manifest = nullptr);
 
   // Show a previously created Page that was created via CreateWindow. This
   // should only be called once the newly created window when it is ready to be
@@ -246,7 +246,7 @@
                     LocalFrame& opener_frame,
                     NavigationPolicy navigation_policy,
                     const mojom::blink::WindowFeatures& window_features,
-                    bool consumed_user_gesture) = 0;
+                    bool consumed_user_gesture, WebString* manifest = nullptr) = 0;
 
   // All the parameters should be in viewport space. That is, if an event
   // scrolls by 10 px, but due to a 2X page scale we apply a 5px scroll to the
@@ -589,7 +589,7 @@
                                      const WebWindowFeatures&,
                                      network::mojom::blink::WebSandboxFlags,
                                      const SessionStorageNamespaceId&,
-                                     bool& consumed_user_gesture) = 0;
+                                     bool& consumed_user_gesture, WebString*) = 0;
 
  private:
   bool CanOpenUIElementIfDuringPageDismissal(Frame& main_frame,
