diff -r -u --color up/chromium/chrome/browser/browser_resources.grd nw/chromium/chrome/browser/browser_resources.grd
--- up/chromium/chrome/browser/browser_resources.grd	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/browser_resources.grd	2023-02-19 17:36:27.475443835 +0000
@@ -22,6 +22,8 @@
       </if>
     </structures>
     <includes>
+      <include name="IDR_NWJS_DEFAPP_MANIFEST" file="resources\nwjs_default_app\manifest.json" type="BINDATA" />
+      <include name="IDR_NWJS_DEFAPP_MANIFEST_NEWWIN" file="resources\nwjs_newwin_app\manifest.json" type="BINDATA" />
       <if expr="is_win">
         <include name="IDR_ABOUT_CONFLICTS_HTML" file="resources\conflicts\about_conflicts.html" type="BINDATA" />
         <include name="IDR_ABOUT_CONFLICTS_JS" file="${root_gen_dir}\chrome\browser\resources\conflicts\tsc\about_conflicts.js" use_base_dir="false" type="BINDATA" />
