diff -r -u --color up/chromium/content/public/browser/content_browser_client.cc nw/chromium/content/public/browser/content_browser_client.cc
--- up/chromium/content/public/browser/content_browser_client.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/public/browser/content_browser_client.cc	2023-02-19 17:36:35.763464792 +0000
@@ -4,6 +4,8 @@
 
 #include "content/public/browser/content_browser_client.h"
 
+#include "content/nw/src/browser/nw_content_browser_hooks.h"
+
 #include <utility>
 
 #include "base/callback_helpers.h"
@@ -168,6 +170,8 @@
     BrowserContext* browser_context,
     const GURL& effective_url) {
   DCHECK(browser_context);
+  if (nw::PinningRenderer())
+    return false;
   return true;
 }
 
@@ -182,6 +186,7 @@
   return false;
 }
 
+
 bool ContentBrowserClient::ShouldIgnoreSameSiteCookieRestrictionsWhenTopLevel(
     base::StringPiece scheme,
     bool is_embedded_origin_secure) {
@@ -194,6 +199,10 @@
   return site_url.spec();
 }
 
+bool ContentBrowserClient::IsNWOrigin(const url::Origin& origin, BrowserContext* context) {
+  return false;
+}
+
 void ContentBrowserClient::OverrideURLLoaderFactoryParams(
     BrowserContext* browser_context,
     const url::Origin& origin,
@@ -459,6 +468,10 @@
   std::move(callback).Run(true);
 }
 
+base::FilePath ContentBrowserClient::GetRootPath() {
+  return base::FilePath();
+}
+
 bool ContentBrowserClient::AllowWorkerIndexedDB(
     const GURL& url,
     BrowserContext* browser_context,
