diff -r -u --color up/chromium/third_party/blink/renderer/bindings/bindings.gni nw/chromium/third_party/blink/renderer/bindings/bindings.gni
--- up/chromium/third_party/blink/renderer/bindings/bindings.gni	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/bindings/bindings.gni	2023-02-19 17:36:42.187480963 +0000
@@ -23,6 +23,8 @@
                     "core/v8/callback_promise_adapter.h",
                     "core/v8/capture_source_location.cc",
                     "core/v8/capture_source_location.h",
+                    "core/v8/custom/v8_html_iframe_element_custom.cc",
+                    "core/v8/custom/v8_filelist_custom.cc",
                     "core/v8/custom/v8_dev_tools_host_custom.cc",
                     "core/v8/custom/v8_html_all_collection_custom.cc",
                     "core/v8/custom/v8_html_plugin_element_custom.cc",
