diff -r -u --color up/chromium/extensions/browser/user_script_loader.cc nw/chromium/extensions/browser/user_script_loader.cc
--- up/chromium/extensions/browser/user_script_loader.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/browser/user_script_loader.cc	2023-02-19 17:36:36.855467551 +0000
@@ -87,7 +87,7 @@
           ->GetExtensionById(host_id.id, ExtensionRegistry::ENABLED);
 
   return extension && PermissionsData::CanExecuteScriptEverywhere(
-                          extension->id(), extension->location());
+                                                                  extension->id(), extension->location(), extension->GetType());
 }
 
 }  // namespace
