diff -r -u --color up/chromium/chrome/chrome_paks.gni nw/chromium/chrome/chrome_paks.gni
--- up/chromium/chrome/chrome_paks.gni	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/chrome_paks.gni	2023-02-19 17:36:29.995450214 +0000
@@ -77,7 +77,7 @@
       deps += [ "//extensions:extensions_browser_resources" ]
     }
 
-    output = "${invoker.output_dir}/chrome_${percent}_percent.pak"
+    output = "${invoker.output_dir}/nw_${percent}_percent.pak"
   }
 }
 
@@ -136,6 +136,12 @@
     if (defined(invoker.additional_paks)) {
       sources += invoker.additional_paks
     }
+    if (nwjs_sdk) {
+      sources += [
+        "$root_gen_dir/content/browser/tracing/tracing_resources.pak",
+        "$root_gen_dir/content/browser/devtools/devtools_resources.pak",
+      ]
+    }
 
     if (!is_android) {
       # New paks should be added here by default.
@@ -171,8 +177,6 @@
         "$root_gen_dir/chrome/tab_search_resources.pak",
         "$root_gen_dir/chrome/webui_gallery_resources.pak",
         "$root_gen_dir/chrome/whats_new_resources.pak",
-        "$root_gen_dir/content/browser/devtools/devtools_resources.pak",
-        "$root_gen_dir/content/browser/tracing/tracing_resources.pak",
       ]
       deps += [
         "//chrome/browser/resources:component_extension_resources",
