diff -r -u --color up/chromium/chrome/browser/ui/views/download/download_item_view.cc nw/chromium/chrome/browser/ui/views/download/download_item_view.cc
--- up/chromium/chrome/browser/ui/views/download/download_item_view.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/download/download_item_view.cc	2023-02-19 17:36:29.535449050 +0000
@@ -811,6 +811,7 @@
     StyleFilename(*warning_label_, filename_offset, filename.length());
     warning_label_->SizeToFit(GetLabelWidth(*warning_label_));
   }
+#if 0
 
   deep_scanning_label_->SetVisible(mode_ ==
                                    download::DownloadItemMode::kDeepScanning);
@@ -827,6 +828,7 @@
     StyleFilename(*deep_scanning_label_, filename_offset, filename.length());
     deep_scanning_label_->SizeToFit(GetLabelWidth(*deep_scanning_label_));
   }
+#endif
 }
 
 void DownloadItemView::UpdateButtons() {
@@ -1226,6 +1228,9 @@
 }
 
 void DownloadItemView::ShowOpenDialog(content::WebContents* web_contents) {
+#if 1
+  ExecuteCommand(DownloadCommands::BYPASS_DEEP_SCANNING);
+#else
   if (mode_ == download::DownloadItemMode::kDeepScanning) {
     TabModalConfirmDialog::Create(
         std::make_unique<safe_browsing::DeepScanningModalDialog>(
@@ -1243,6 +1248,7 @@
                        weak_ptr_factory_.GetWeakPtr(),
                        DownloadCommands::BYPASS_DEEP_SCANNING));
   }
+#endif
 }
 
 void DownloadItemView::ShowContextMenuImpl(const gfx::Rect& rect,
@@ -1279,7 +1285,7 @@
 }
 
 void DownloadItemView::OpenDownloadDuringAsyncScanning() {
-  model_->CompleteSafeBrowsingScan();
+  //model_->CompleteSafeBrowsingScan();
   model_->SetOpenWhenComplete(true);
 }
 
