diff -r -u --color up/chromium/chrome/browser/ui/apps/chrome_app_delegate.h nw/chromium/chrome/browser/ui/apps/chrome_app_delegate.h
--- up/chromium/chrome/browser/ui/apps/chrome_app_delegate.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/apps/chrome_app_delegate.h	2023-02-19 17:36:29.331448533 +0000
@@ -93,6 +93,7 @@
   std::unique_ptr<NewWindowContentsDelegate> new_window_contents_delegate_;
   base::OnceClosure terminating_callback_;
   base::CallbackListSubscription subscription_;
+  content::WebContents* web_contents_;
   base::WeakPtrFactory<ChromeAppDelegate> weak_factory_{this};
 };
 
