diff -r -u --color up/chromium/chrome/installer/linux/BUILD.gn nw/chromium/chrome/installer/linux/BUILD.gn
--- up/chromium/chrome/installer/linux/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/installer/linux/BUILD.gn	2023-02-19 17:36:30.075450416 +0000
@@ -35,7 +35,7 @@
 assert(is_linux || is_chromeos)
 
 packaging_files_executables = [
-  "$root_out_dir/chrome",
+  "$root_out_dir/nw",
   "$root_out_dir/chrome_crashpad_handler",
   "$root_out_dir/chrome_management_service",
   "$root_out_dir/chrome_sandbox",
@@ -71,16 +71,16 @@
 
 if (use_egl) {
   packaging_files_shlibs += [
-    "$root_out_dir/libEGL.so",
-    "$root_out_dir/libGLESv2.so",
+    "$root_out_dir/lib/libEGL.so",
+    "$root_out_dir/lib/libGLESv2.so",
   ]
   if (angle_shared_libvulkan && !is_chromeos) {
-    packaging_files_shlibs += [ "$root_out_dir/libvulkan.so.1" ]
+    packaging_files_shlibs += [ "$root_out_dir/lib/libvulkan.so.1" ]
   }
 }
 
 if (enable_swiftshader) {
-  packaging_files_shlibs += [ "$root_out_dir/libvk_swiftshader.so" ]
+  packaging_files_shlibs += [ "$root_out_dir/lib/libvk_swiftshader.so" ]
 }
 
 if (build_with_internal_optimization_guide) {
@@ -183,7 +183,7 @@
 }
 
 strip_binary("strip_chrome_binary") {
-  binary_input = "$root_out_dir/chrome"
+  binary_input = "$root_out_dir/nw"
   deps = [ "//chrome" ]
 }
 
@@ -203,22 +203,22 @@
 }
 
 strip_binary("strip_libEGL_shlib") {
-  binary_input = "$root_out_dir/libEGL.so"
+  binary_input = "$root_out_dir/lib/libEGL.so"
   deps = [ "//third_party/angle:libEGL" ]
 }
 
 strip_binary("strip_libGLESv2_shlib") {
-  binary_input = "$root_out_dir/libGLESv2.so"
+  binary_input = "$root_out_dir/lib/libGLESv2.so"
   deps = [ "//third_party/angle:libGLESv2" ]
 }
 
 strip_binary("strip_libvulkan_shlib") {
-  binary_input = "$root_out_dir/libvulkan.so.1"
+  binary_input = "$root_out_dir/lib/libvulkan.so.1"
   deps = [ "//third_party/vulkan-deps/vulkan-loader/src:libvulkan" ]
 }
 
 strip_binary("strip_libvk_swiftshader.shlib") {
-  binary_input = "$root_out_dir/libvk_swiftshader.so"
+  binary_input = "$root_out_dir/lib/libvk_swiftshader.so"
   deps = [ "//third_party/swiftshader/src/Vulkan:swiftshader_libvulkan" ]
 }
 
@@ -396,6 +396,9 @@
     "//components/crash/core/app:chrome_crashpad_handler",
     "//sandbox/linux:chrome_sandbox",
   ]
+  if (is_linux) {
+    public_deps += [ "//chrome:xdg_mime" ]
+  }
   if (package_nacl) {
     public_deps += [
       ":strip_nacl_helper",
