diff -r -u --color up/chromium/third_party/blink/renderer/core/page/create_window.h nw/chromium/third_party/blink/renderer/core/page/create_window.h
--- up/chromium/third_party/blink/renderer/core/page/create_window.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/page/create_window.h	2023-02-19 17:36:42.779482446 +0000
@@ -36,11 +36,12 @@
 class KURL;
 class LocalDOMWindow;
 class LocalFrame;
+class WebString;
 struct FrameLoadRequest;
 
 Frame* CreateNewWindow(LocalFrame& opener_frame,
                        FrameLoadRequest&,
-                       const AtomicString& name);
+                       const AtomicString& name, WebString* manifes);
 
 CORE_EXPORT WebWindowFeatures
 GetWindowFeaturesFromString(const String& feature_string,
