diff -r -u --color up/chromium/chrome/browser/extensions/chrome_content_verifier_delegate.cc nw/chromium/chrome/browser/extensions/chrome_content_verifier_delegate.cc
--- up/chromium/chrome/browser/extensions/chrome_content_verifier_delegate.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/chrome_content_verifier_delegate.cc	2023-02-19 17:36:27.767444574 +0000
@@ -193,7 +193,9 @@
 
 void ChromeContentVerifierDelegate::VerifyFailed(
     const std::string& extension_id,
-    ContentVerifyJob::FailureReason reason) {
+    const base::FilePath& relative_path,
+    ContentVerifyJob::FailureReason reason,
+    scoped_refptr<ContentVerifyJob> verify_job) {
   ExtensionRegistry* registry = ExtensionRegistry::Get(context_);
   const Extension* extension =
       registry->GetExtensionById(extension_id, ExtensionRegistry::ENABLED);
@@ -322,6 +324,8 @@
   if (should_repair)
     return VerifyInfo(default_mode_, is_from_webstore, should_repair);
 
+  if (extension.is_nwjs_app() && !Manifest::IsComponentLocation(extension.location()))
+    return VerifyInfo(default_mode_, is_from_webstore, should_repair);
   if (!extension.is_extension() && !extension.is_legacy_packaged_app())
     return VerifyInfo(VerifyInfo::Mode::NONE, is_from_webstore, should_repair);
   if (!Manifest::IsAutoUpdateableLocation(extension.location()))
