diff -r -u --color up/chromium/third_party/blink/renderer/core/workers/global_scope_creation_params.cc nw/chromium/third_party/blink/renderer/core/workers/global_scope_creation_params.cc
--- up/chromium/third_party/blink/renderer/core/workers/global_scope_creation_params.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/workers/global_scope_creation_params.cc	2023-02-19 17:36:43.003483008 +0000
@@ -14,6 +14,7 @@
 namespace blink {
 
 GlobalScopeCreationParams::GlobalScopeCreationParams(
+    bool isNodeJS, const std::string& main_script,
     const KURL& script_url,
     mojom::blink::ScriptType script_type,
     const String& global_scope_name,
@@ -52,6 +53,7 @@
     : script_url(script_url),
       script_type(script_type),
       global_scope_name(global_scope_name),
+      nodejs_(isNodeJS), main_script_(main_script),
       user_agent(user_agent),
       ua_metadata(ua_metadata.value_or(blink::UserAgentMetadata())),
       web_worker_fetch_context(std::move(web_worker_fetch_context)),
