diff -r -u --color up/chromium/components/permissions/permission_manager.cc nw/chromium/components/permissions/permission_manager.cc
--- up/chromium/components/permissions/permission_manager.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/permissions/permission_manager.cc	2023-02-19 17:36:33.459458971 +0000
@@ -4,6 +4,8 @@
 
 #include "components/permissions/permission_manager.h"
 
+#include "extensions/browser/extension_registry.h"
+
 #include <memory>
 #include <utility>
 
