diff -r -u --color up/chromium/components/renderer_context_menu/context_menu_content_type.cc nw/chromium/components/renderer_context_menu/context_menu_content_type.cc
--- up/chromium/components/renderer_context_menu/context_menu_content_type.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/renderer_context_menu/context_menu_content_type.cc	2023-02-19 17:36:33.923460144 +0000
@@ -146,7 +146,7 @@
       return true;
 
     case ITEM_GROUP_DEVTOOLS_UNPACKED_EXT:
-      return false;
+      return true;
 
     case ITEM_GROUP_PRINT_PREVIEW:
 #if BUILDFLAG(ENABLE_PRINT_PREVIEW)
