diff -r -u --color up/chromium/components/sessions/core/tab_restore_service_impl.cc nw/chromium/components/sessions/core/tab_restore_service_impl.cc
--- up/chromium/components/sessions/core/tab_restore_service_impl.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/sessions/core/tab_restore_service_impl.cc	2023-02-19 17:36:34.087460558 +0000
@@ -181,6 +181,7 @@
   kSerializedShowStateMaximized = 3,
   kSerializedShowStateInactive = 4,
   kSerializedShowStateFullscreen = 5,
+  kSerializedShowStateHidden = 6,
 };
 
 // Converts a window show state to an integer. This function needs to be kept
@@ -199,6 +200,8 @@
       return kSerializedShowStateInactive;
     case ui::SHOW_STATE_FULLSCREEN:
       return kSerializedShowStateFullscreen;
+    case ui::SHOW_STATE_HIDDEN:
+      return kSerializedShowStateHidden;
     case ui::SHOW_STATE_END:
       // This should never happen.
       NOTREACHED();
@@ -230,6 +233,9 @@
     case kSerializedShowStateFullscreen:
       *show_state = ui::SHOW_STATE_FULLSCREEN;
       return true;
+    case kSerializedShowStateHidden:
+      *show_state = ui::SHOW_STATE_HIDDEN;
+      return true;
     case kSerializedShowStateInvalid:
     default:
       // Ignore unknown values. This could happen if the data is corrupt.
