diff -r -u --color up/chromium/components/safe_browsing/content/common/proto/BUILD.gn nw/chromium/components/safe_browsing/content/common/proto/BUILD.gn
--- up/chromium/components/safe_browsing/content/common/proto/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/safe_browsing/content/common/proto/BUILD.gn	2023-02-19 17:36:33.963460245 +0000
@@ -5,7 +5,7 @@
 import("//components/safe_browsing/buildflags.gni")
 import("//third_party/protobuf/proto_library.gni")
 
-if (safe_browsing_mode != 0) {
+if (true) {
   proto_library("download_file_types_proto") {
     sources = [ "download_file_types.proto" ]
   }
