diff -r -u --color up/chromium/components/remote_cocoa/common/native_widget_ns_window.mojom nw/chromium/components/remote_cocoa/common/native_widget_ns_window.mojom
--- up/chromium/components/remote_cocoa/common/native_widget_ns_window.mojom	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/remote_cocoa/common/native_widget_ns_window.mojom	2023-02-19 17:36:33.919460134 +0000
@@ -206,6 +206,10 @@
 
   // Zoom or un-zoom the window.
   SetZoomed(bool zoomed);
+  SetMaximized(bool maximized);
+  [Sync] IsMaximized() => (bool maximized);
+  [Sync] GetRestoredBounds() => (gfx.mojom.Rect bounds);
+  SetRestoredBounds(gfx.mojom.Rect bounds);
 
   // Called by NativeWidgetMac when the window size constraints change.
   SetSizeConstraints(gfx.mojom.Size min_size,
