diff -r -u --color up/chromium/content/browser/renderer_host/render_widget_host_view_mac.h nw/chromium/content/browser/renderer_host/render_widget_host_view_mac.h
--- up/chromium/content/browser/renderer_host/render_widget_host_view_mac.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/browser/renderer_host/render_widget_host_view_mac.h	2023-02-19 17:36:35.571464307 +0000
@@ -100,6 +100,9 @@
 
   RenderWidgetHostViewCocoa* GetInProcessNSView() const;
 
+  // Used by g_force_cpu_draw
+  CALayer* background_layer() const;
+
   // |delegate| is used to separate out the logic from the NSResponder delegate.
   // |delegate| is retained by this class.
   // |delegate| should be set at most once.
@@ -317,6 +320,7 @@
   id GetRootBrowserAccessibilityElement() override;
   id GetFocusedBrowserAccessibilityElement() override;
   void SetAccessibilityWindow(NSWindow* window) override;
+  RenderWidgetHostViewMac* GetRenderWidgetHostViewMac() override;
   void ForwardKeyboardEvent(const NativeWebKeyboardEvent& key_event,
                             const ui::LatencyInfo& latency_info) override;
   void ForwardKeyboardEventWithCommands(
