diff -r -u --color up/chromium/content/browser/worker_host/shared_worker_host.cc nw/chromium/content/browser/worker_host/shared_worker_host.cc
--- up/chromium/content/browser/worker_host/shared_worker_host.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/browser/worker_host/shared_worker_host.cc	2023-02-19 17:36:35.687464600 +0000
@@ -290,6 +290,7 @@
   auto options = blink::mojom::WorkerOptions::New(
       instance_.script_type(), instance_.credentials_mode(), instance_.name());
   blink::mojom::SharedWorkerInfoPtr info(blink::mojom::SharedWorkerInfo::New(
+                                                                             instance_.nodejs(), instance_.root_path(),
       instance_.url(), std::move(options),
       mojo::Clone(content_security_policies_),
       std::move(outside_fetch_client_settings_object)));
