diff -r -u --color up/chromium/content/BUILD.gn nw/chromium/content/BUILD.gn
--- up/chromium/content/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/BUILD.gn	2023-02-19 17:36:35.219463418 +0000
@@ -60,6 +60,7 @@
 # linking problems) for when check is enabled.
 
 content_shared_components = [
+  "//content/nw:nw_base",
   "//content/gpu:gpu_sources",
   "//content/public/browser:browser_sources",
   "//content/public/child:child_sources",
@@ -77,6 +78,9 @@
   component("content") {
     public_deps =
         content_shared_components + [ "//content/public/app:app_sources" ]
+    deps = [
+      "//content/nw:nw_content",
+    ]
   }
 } else {
   group("content") {
