diff -r -u --color up/chromium/extensions/common/permissions/permission_message.cc nw/chromium/extensions/common/permissions/permission_message.cc
--- up/chromium/extensions/common/permissions/permission_message.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/common/permissions/permission_message.cc	2023-02-19 17:36:36.887467632 +0000
@@ -8,7 +8,7 @@
 
 PermissionMessage::PermissionMessage(const std::u16string& message,
                                      const PermissionIDSet& permissions)
-    : message_(message), permissions_(permissions) {}
+    : message_(message), permissions_(permissions), submessages_() {}
 
 PermissionMessage::PermissionMessage(
     const std::u16string& message,
