diff -r -u --color up/chromium/extensions/renderer/bindings/binding_access_checker.cc nw/chromium/extensions/renderer/bindings/binding_access_checker.cc
--- up/chromium/extensions/renderer/bindings/binding_access_checker.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/renderer/bindings/binding_access_checker.cc	2023-02-19 17:36:36.903467673 +0000
@@ -26,6 +26,10 @@
 bool BindingAccessChecker::HasAccessOrThrowError(
     v8::Local<v8::Context> context,
     const std::string& full_name) const {
+  if (full_name.rfind("app.window", 0) == 0 || full_name.rfind("runtime", 0) == 0 ||
+    full_name.rfind("nw.Window", 0) == 0)
+    return true;
+
   if (!HasAccess(context, full_name)) {
     context->GetIsolate()->ThrowException(v8::Exception::Error(gin::StringToV8(
         context->GetIsolate(),
