diff -r -u --color up/chromium/extensions/renderer/api/app_window_custom_bindings.h nw/chromium/extensions/renderer/api/app_window_custom_bindings.h
--- up/chromium/extensions/renderer/api/app_window_custom_bindings.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/renderer/api/app_window_custom_bindings.h	2023-02-19 17:36:36.895467652 +0000
@@ -24,6 +24,8 @@
  private:
   void GetFrame(const v8::FunctionCallbackInfo<v8::Value>& args);
   void ResumeParser(const v8::FunctionCallbackInfo<v8::Value>& args);
+
+  void FixGamePadAPI(const v8::FunctionCallbackInfo<v8::Value>& args);
 };
 
 }  // namespace extensions
