diff -r -u --color up/chromium/extensions/browser/api/content_settings/content_settings_store.cc nw/chromium/extensions/browser/api/content_settings/content_settings_store.cc
--- up/chromium/extensions/browser/api/content_settings/content_settings_store.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/browser/api/content_settings/content_settings_store.cc	2023-02-19 17:36:36.767467329 +0000
@@ -114,6 +114,22 @@
   {
     base::AutoLock lock(lock_);
     OriginIdentifierValueMap* map = GetValueMap(ext_id, scope);
+    if (!map) {
+      ExtensionEntry* entry = new ExtensionEntry;
+      entry->install_time = base::Time::Now();
+ 
+      auto unique_entry = base::WrapUnique(entry);
+      auto location =
+        std::upper_bound(entries_.begin(), entries_.end(), unique_entry,
+                         [](const std::unique_ptr<ExtensionEntry>& a,
+                            const std::unique_ptr<ExtensionEntry>& b) {
+                           return a->install_time > b->install_time;
+                         });
+      entries_.insert(location, std::move(unique_entry));
+      entry->id = ext_id;
+      entry->enabled = true;
+      map = GetValueMap(ext_id, scope);
+    }
     if (setting == CONTENT_SETTING_DEFAULT) {
       map->DeleteValue(primary_pattern, secondary_pattern, type);
     } else {
