diff -r -u --color up/chromium/net/cert/test_root_certs.h nw/chromium/net/cert/test_root_certs.h
--- up/chromium/net/cert/test_root_certs.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/net/cert/test_root_certs.h	2023-02-19 17:36:40.111475760 +0000
@@ -74,6 +74,7 @@
   TestRootCerts();
   ~TestRootCerts();
 
+public:
   // Marks |certificate| as trusted in the effective trust store
   // used by CertVerifier::Verify(). Returns false if the
   // certificate could not be marked trusted.
