diff -r -u --color up/chromium/net/proxy_resolution/configured_proxy_resolution_service.h nw/chromium/net/proxy_resolution/configured_proxy_resolution_service.h
--- up/chromium/net/proxy_resolution/configured_proxy_resolution_service.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/net/proxy_resolution/configured_proxy_resolution_service.h	2023-02-19 17:36:40.491476712 +0000
@@ -153,6 +153,15 @@
   // ProxyResolutionService
   void OnShutdown() override;
 
+  // Tells this ProxyResolutionService to start using a new
+  // ProxyConfigService
+  // to retrieve its ProxyConfig from. The new ProxyConfigService
+  // will
+  // immediately be queried for new config info which will be used
+  // for all
+  // subsequent ResolveProxy calls.
+  void ResetConfigService(
+                          std::unique_ptr<ProxyConfigService> new_proxy_config_service);
   // Returns the last configuration fetched from ProxyConfigService.
   const absl::optional<ProxyConfigWithAnnotation>& fetched_config() const {
     return fetched_config_;
