diff -r -u --color up/chromium/ppapi/native_client/src/untrusted/pnacl_support_extension/BUILD.gn nw/chromium/ppapi/native_client/src/untrusted/pnacl_support_extension/BUILD.gn
--- up/chromium/ppapi/native_client/src/untrusted/pnacl_support_extension/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/ppapi/native_client/src/untrusted/pnacl_support_extension/BUILD.gn	2023-02-19 17:36:40.703477244 +0000
@@ -35,7 +35,10 @@
   shim_cpu = nacl_target_cpu
 
   output_prefix = "$root_out_dir/pnacl/pnacl_public_"
-  outputs = [ "${output_prefix}pnacl_json" ]
+  outputs = [
+    "$root_out_dir/pnacl",
+    "${output_prefix}pnacl_json",
+  ]
 
   # Files that will be copied from the toolchain to output.
   outputs_from_toolchain = [
