diff -r -u --color up/chromium/third_party/blink/renderer/core/frame/csp/content_security_policy.cc nw/chromium/third_party/blink/renderer/core/frame/csp/content_security_policy.cc
--- up/chromium/third_party/blink/renderer/core/frame/csp/content_security_policy.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/frame/csp/content_security_policy.cc	2023-02-19 17:36:42.495481735 +0000
@@ -1240,6 +1240,10 @@
   if (!context)
     return false;
 
+  if (const_cast<ExecutionContext*>(context)->GetSecurityOrigin() &&
+      const_cast<ExecutionContext*>(context)->GetSecurityOrigin()->hasUniversalAccess())
+    return true;
+
   return ShouldBypassMainWorldDeprecated(context->GetCurrentWorld().get());
 }
 
