diff -r -u --color up/chromium/third_party/blink/renderer/bindings/scripts/bind_gen/interface.py nw/chromium/third_party/blink/renderer/bindings/scripts/bind_gen/interface.py
--- up/chromium/third_party/blink/renderer/bindings/scripts/bind_gen/interface.py	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/bindings/scripts/bind_gen/interface.py	2023-02-19 17:36:42.223481053 +0000
@@ -2138,7 +2138,12 @@
         EmptyNode(),
     ])
 
-    if "HTMLConstructor" in cg_context.constructor.extended_attributes:
+    if "Custom" in cg_context.constructor.extended_attributes:
+        body.append(T("// [Custom Constructor]"))
+        text = _format(
+            "${class_name}::ConstructorCustom(${info});")
+        body.append(T(text))
+    elif "HTMLConstructor" in cg_context.constructor.extended_attributes:
         body.append(T("// [HTMLConstructor]"))
         text = _format(
             "V8HTMLConstructor::HtmlConstructor("
@@ -7434,6 +7439,12 @@
             add_custom_callback_impl_decl(
                 operation_group=operation_group,
                 arg_decls=["const v8::FunctionCallbackInfo<v8::Value>&"])
+    for constructor_group in class_like.constructor_groups:
+        if "Custom" in constructor_group.extended_attributes:
+            add_custom_callback_impl_decl(
+                name="ConstructorCustom",
+                constructor_group=constructor_group,
+                arg_decls=["const v8::FunctionCallbackInfo<v8::Value>&"])
     if interface and interface.indexed_and_named_properties:
         props = interface.indexed_and_named_properties
         operation = props.own_named_getter
