diff -r -u --color up/chromium/third_party/blink/renderer/core/html/html_frame_owner_element.cc nw/chromium/third_party/blink/renderer/core/html/html_frame_owner_element.cc
--- up/chromium/third_party/blink/renderer/core/html/html_frame_owner_element.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/html_frame_owner_element.cc	2023-02-19 17:36:42.571481925 +0000
@@ -468,6 +468,8 @@
   properties->allow_payment_request = AllowPaymentRequest();
   properties->is_display_none = IsDisplayNone();
   properties->color_scheme = GetColorScheme();
+  properties->nwfaketop = FastHasAttribute(html_names::kNwfaketopAttr);
+  properties->nwuseragent = nwuseragent().IsNull() ? WTF::g_empty_string : nwuseragent();
 
   GetDocument()
       .GetFrame()
