diff -r -u --color up/chromium/third_party/blink/public/web/blink.h nw/chromium/third_party/blink/public/web/blink.h
--- up/chromium/third_party/blink/public/web/blink.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/public/web/blink.h	2023-02-19 17:36:42.175480933 +0000
@@ -47,6 +47,7 @@
 
 class Platform;
 
+class WebLocalFrame;
 // Initialize the entire Blink (wtf, platform, core, modules and web).
 // If you just need wtf and platform, use Platform::Initialize instead.
 //
@@ -79,6 +80,10 @@
 // Alters whether the browser can handle focus events while running web tests.
 BLINK_EXPORT void SetBrowserCanHandleFocusForWebTest(bool);
 
+BLINK_EXPORT void set_web_worker_hooks(void*);
+
+BLINK_EXPORT void fix_gamepad_nw(WebLocalFrame*);
+
 // Alters the rendering of fonts for web tests.
 BLINK_EXPORT void SetFontAntialiasingEnabledForTest(bool);
 BLINK_EXPORT bool FontAntialiasingEnabledForTest();
