diff -r -u --color up/chromium/third_party/blink/renderer/modules/gamepad/navigator_gamepad.cc nw/chromium/third_party/blink/renderer/modules/gamepad/navigator_gamepad.cc
--- up/chromium/third_party/blink/renderer/modules/gamepad/navigator_gamepad.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/modules/gamepad/navigator_gamepad.cc	2023-02-19 17:36:43.119483299 +0000
@@ -45,7 +45,18 @@
 #include "third_party/blink/renderer/platform/privacy_budget/identifiability_digest_helpers.h"
 #include "third_party/blink/renderer/platform/wtf/text/atomic_string.h"
 
+#include "third_party/blink/public/web/web_document.h"
+#include "third_party/blink/public/web/web_frame.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "third_party/blink/renderer/core/dom/document.h"
+
 namespace blink {
+void fix_gamepad_nw(WebLocalFrame* frame) {
+  Document* doc = frame->GetDocument();
+  NavigatorGamepad* gamepad = &NavigatorGamepad::From(*(doc->domWindow()->navigator()));
+  gamepad->SetWindow(doc->domWindow());
+  gamepad->SampleAndCompareGamepadState();
+}
 
 namespace {
 
