diff -r -u --color up/chromium/third_party/blink/renderer/platform/bindings/script_state.cc nw/chromium/third_party/blink/renderer/platform/bindings/script_state.cc
--- up/chromium/third_party/blink/renderer/platform/bindings/script_state.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/platform/bindings/script_state.cc	2023-02-19 17:36:43.395483990 +0000
@@ -23,6 +23,10 @@
   context->SetAlignedPointerInEmbedderData(kV8ContextPerContextDataIndex, this);
   RendererResourceCoordinator::Get()->OnScriptStateCreated(this,
                                                            execution_context);
+  for (int i = 32; i <= 36; i++) { //node_context_data.h
+    context->SetAlignedPointerInEmbedderData(i, nullptr);
+  }
+  context->SetAlignedPointerInEmbedderData(50, nullptr);
 }
 
 ScriptState::~ScriptState() {
