diff -r -u --color up/chromium/third_party/blink/renderer/core/html/forms/html_input_element.h nw/chromium/third_party/blink/renderer/core/html/forms/html_input_element.h
--- up/chromium/third_party/blink/renderer/core/html/forms/html_input_element.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/forms/html_input_element.h	2023-02-19 17:36:42.551481875 +0000
@@ -310,6 +310,11 @@
   static Vector<String> FilesFromFileInputFormControlState(
       const FormControlState&);
 
+  const AtomicString& nwworkingdir() const;
+  void setNwworkingdir(const AtomicString& value);
+  String nwsaveas() const;
+  void setNwsaveas(const String& value);
+
   bool MatchesReadOnlyPseudoClass() const final;
   bool MatchesReadWritePseudoClass() const final;
   ControlPart AutoAppearance() const;
