diff -r -u --color up/chromium/third_party/blink/renderer/core/html/html_html_element.cc nw/chromium/third_party/blink/renderer/core/html/html_html_element.cc
--- up/chromium/third_party/blink/renderer/core/html/html_html_element.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/html/html_html_element.cc	2023-02-19 17:36:42.571481925 +0000
@@ -58,6 +58,7 @@
   GetDocument().Parser()->DocumentElementAvailable();
   if (GetDocument().GetFrame()) {
     GetDocument().GetFrame()->Loader().DispatchDocumentElementAvailable();
+    if (GetDocument().GetFrame())
     GetDocument().GetFrame()->Loader().RunScriptsAtDocumentElementAvailable();
     // RunScriptsAtDocumentElementAvailable might have invalidated
     // GetDocument().
