diff -r -u --color up/chromium/ui/aura/window_tree_host.h nw/chromium/ui/aura/window_tree_host.h
--- up/chromium/ui/aura/window_tree_host.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/ui/aura/window_tree_host.h	2023-02-19 17:37:08.043541897 +0000
@@ -223,6 +223,9 @@
   // depending on the platform. The |local_surface_id| takes effect when (and
   // if) the new size is confirmed (potentially asynchronously) by the platform.
   virtual void SetBoundsInPixels(const gfx::Rect& bounds_in_pixels) = 0;
+#if defined(OS_WIN)
+  virtual void SetPositionInPixels(const gfx::Point& pos_in_pixels);
+#endif
   virtual gfx::Rect GetBoundsInPixels() const = 0;
 
   // Gets the bounds in DIP.
