diff -r -u --color up/chromium/apps/launcher.cc nw/chromium/apps/launcher.cc
--- up/chromium/apps/launcher.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/apps/launcher.cc	2023-02-24 06:16:17.751875263 +0000
@@ -112,7 +112,10 @@
         extension_id(app->id()),
         entry_paths_(entry_paths),
         mime_type_collector_(context),
-        is_directory_collector_(context) {}
+        is_directory_collector_(context) {
+    if (app->is_nwjs_app()) //NWJS#5097
+      entry_paths_.clear();
+  }
 
   PlatformAppPathLauncher(content::BrowserContext* context,
                           const Extension* app,
@@ -121,7 +124,7 @@
         extension_id(app->id()),
         mime_type_collector_(context),
         is_directory_collector_(context) {
-    if (!file_path.empty())
+    if (!file_path.empty() && !app->is_nwjs_app()) //NWJS#5097
       entry_paths_.push_back(file_path);
   }
   PlatformAppPathLauncher(const PlatformAppPathLauncher&) = delete;
@@ -497,7 +500,7 @@
 
   extensions::ExtensionPrefs* extension_prefs =
       extensions::ExtensionPrefs::Get(context);
-  bool had_windows = extension_prefs->IsActive(app->id());
+  bool had_windows = extension_prefs->IsActive(app->id()) || true;
   extension_prefs->SetIsActive(app->id(), false);
   bool listening_to_launch = event_router->ExtensionHasEventListener(
       app->id(), app_runtime::OnLaunched::kEventName);
