diff -r -u --color up/chromium/base/test/test_message_loop.cc nw/chromium/base/test/test_message_loop.cc
--- up/chromium/base/test/test_message_loop.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/base/test/test_message_loop.cc	2023-02-24 06:16:18.667929510 +0000
@@ -19,6 +19,7 @@
     MessagePumpType type) {
   switch (type) {
     case MessagePumpType::DEFAULT:
+    case MessagePumpType::NODE:
       return test::SingleThreadTaskEnvironment::MainThreadType::DEFAULT;
     case MessagePumpType::IO:
       return test::SingleThreadTaskEnvironment::MainThreadType::IO;
