diff -r -u --color up/chromium/base/allocator/partition_allocator/partition_root.cc nw/chromium/base/allocator/partition_allocator/partition_root.cc
--- up/chromium/base/allocator/partition_allocator/partition_root.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/base/allocator/partition_allocator/partition_root.cc	2023-02-24 20:54:05.124528976 +0000
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#pragma clang diagnostic ignored "-Wmisleading-indentation"
 #include "base/allocator/partition_allocator/partition_root.h"
 
 #include <cstdint>
