diff -r -u --color up/chromium/build/config/mac/BUILD.gn nw/chromium/build/config/mac/BUILD.gn
--- up/chromium/build/config/mac/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/build/config/mac/BUILD.gn	2023-02-24 20:12:01.472792989 +0000
@@ -84,6 +84,7 @@
 
       # Path for loading shared libraries for bundled binaries. Get back from
       # Binary.app/Contents/MacOS.
+      "-lc++abi",
       "-Wl,-rpath,@loader_path/../../..",
     ]
 
