diff -r -u --color up/chromium/buildtools/third_party/libc++/BUILD.gn nw/chromium/buildtools/third_party/libc++/BUILD.gn
--- up/chromium/buildtools/third_party/libc++/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/buildtools/third_party/libc++/BUILD.gn	2023-02-24 20:12:01.500793274 +0000
@@ -44,7 +44,11 @@
 if (libcxx_is_shared) {
   _libcxx_target_type = "shared_library"
 } else {
-  _libcxx_target_type = "source_set"
+  if (is_linux) {
+    _libcxx_target_type = "static_library"
+  } else {
+    _libcxx_target_type = "source_set"
+  }
 }
 target(_libcxx_target_type, "libc++") {
   # Most things that need to depend on libc++ should do so via the implicit
@@ -53,6 +57,7 @@
   visibility = [
     "//build/config:common_deps",
     "//third_party/catapult/devil:devil",
+    ":*",
   ]
   if (is_linux) {
     # This target packages libc++.so, so must have an explicit dependency on
@@ -211,3 +216,8 @@
     }
   }
 }
+
+static_library("libcpp") {
+   complete_static_lib = true
+   deps = [":libc++"]
+}
