diff -r -u --color up/chromium/chrome/test/base/test_browser_window_aura.cc nw/chromium/chrome/test/base/test_browser_window_aura.cc
--- up/chromium/chrome/test/base/test_browser_window_aura.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/test/base/test_browser_window_aura.cc	2023-02-24 20:54:08.904511784 +0000
@@ -51,6 +51,14 @@
   return native_window_.get();
 }
 
+void TestBrowserWindowAura::UpdateDraggableRegions(
+    const std::vector<extensions::DraggableRegion>& regions) {
+}
+
+SkRegion* TestBrowserWindowAura::GetDraggableRegion() {
+  return nullptr;
+}
+
 void TestBrowserWindowAura::Show() {
   native_window_->Show();
 }
