diff -r -u --color up/chromium/chrome/browser/BUILD.gn nw/chromium/chrome/browser/BUILD.gn
--- up/chromium/chrome/browser/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/BUILD.gn	2023-02-24 06:16:20.156017632 +0000
@@ -1029,8 +1029,8 @@
     "page_load_metrics/observers/tab_restore_page_load_metrics_observer.h",
     "page_load_metrics/observers/third_party_metrics_observer.cc",
     "page_load_metrics/observers/third_party_metrics_observer.h",
-    "page_load_metrics/observers/translate_page_load_metrics_observer.cc",
-    "page_load_metrics/observers/translate_page_load_metrics_observer.h",
+    #    "page_load_metrics/observers/translate_page_load_metrics_observer.cc",
+    #  "page_load_metrics/observers/translate_page_load_metrics_observer.h",
     "page_load_metrics/page_load_metrics_initialize.cc",
     "page_load_metrics/page_load_metrics_initialize.h",
     "page_load_metrics/page_load_metrics_memory_tracker_factory.cc",
@@ -1810,18 +1810,18 @@
     "tracing/trace_event_system_stats_monitor.h",
     "transition_manager/full_browser_transition_manager.cc",
     "transition_manager/full_browser_transition_manager.h",
-    "translate/chrome_translate_client.cc",
-    "translate/chrome_translate_client.h",
-    "translate/translate_frame_binder.cc",
-    "translate/translate_frame_binder.h",
-    "translate/translate_model_service_factory.cc",
-    "translate/translate_model_service_factory.h",
-    "translate/translate_ranker_factory.cc",
-    "translate/translate_ranker_factory.h",
-    "translate/translate_ranker_metrics_provider.cc",
-    "translate/translate_ranker_metrics_provider.h",
-    "translate/translate_service.cc",
-    "translate/translate_service.h",
+    #"translate/chrome_translate_client.cc",
+    #"translate/chrome_translate_client.h",
+    #"translate/translate_frame_binder.cc",
+    #"translate/translate_frame_binder.h",
+    #"translate/translate_model_service_factory.cc",
+    #"translate/translate_model_service_factory.h",
+    #"translate/translate_ranker_factory.cc",
+    #"translate/translate_ranker_factory.h",
+    #"translate/translate_ranker_metrics_provider.cc",
+    #"translate/translate_ranker_metrics_provider.h",
+    #"translate/translate_service.cc",
+    #"translate/translate_service.h",
     "undo/bookmark_undo_service_factory.cc",
     "undo/bookmark_undo_service_factory.h",
     "unexpire_flags.cc",
@@ -1910,10 +1910,10 @@
     "//chrome/browser/share",
     "//chrome/browser/ui",
     "//chrome/browser/storage_access_api:permissions",
-    "//chrome/browser/safe_browsing",
-    "//chrome/browser/safe_browsing:verdict_cache_manager_factory",
-    "//chrome/browser/safe_browsing:advanced_protection",
-    "//chrome/browser/safe_browsing:metrics_collector",
+    #"//chrome/browser/safe_browsing",
+    #"//chrome/browser/safe_browsing:verdict_cache_manager_factory",
+    #"//chrome/browser/safe_browsing:advanced_protection",
+    #"//chrome/browser/safe_browsing:metrics_collector",
 
     # TODO(crbug.com/1030821): Eliminate usages of browser.h from Media Router.
     "//chrome/browser/media/router",
@@ -1973,6 +1973,7 @@
     "//build/config/chromebox_for_meetings:buildflags",
     "//build/config/compiler:compiler_buildflags",
     "//cc",
+    "//content/nw:nw_browser",
     "//chrome:extra_resources",
     "//chrome:resources",
     "//chrome:strings",
@@ -2014,10 +2015,10 @@
     "//chrome/browser/push_messaging:budget_proto",
     "//chrome/browser/resource_coordinator:mojo_bindings",
     "//chrome/browser/resource_coordinator:tab_manager_features",
-    "//chrome/browser/safe_browsing",
-    "//chrome/browser/safe_browsing:advanced_protection",
-    "//chrome/browser/safe_browsing:metrics_collector",
-    "//chrome/browser/safe_browsing:verdict_cache_manager_factory",
+    #"//chrome/browser/safe_browsing",
+    #"//chrome/browser/safe_browsing:advanced_protection",
+    #"//chrome/browser/safe_browsing:metrics_collector",
+    #"//chrome/browser/safe_browsing:verdict_cache_manager_factory",
     "//chrome/browser/share",
     "//chrome/browser/sharing:buildflags",
     "//chrome/browser/sharing/proto",
@@ -2080,9 +2081,9 @@
     "//components/commerce/core:persisted_state_db_content_proto",
     "//components/commerce/core:pref_names",
     "//components/commerce/core:shopping_service",
-    "//components/component_updater",
-    "//components/component_updater:crl_set_remover",
-    "//components/component_updater/installer_policies",
+    #"//components/component_updater",
+    #"//components/component_updater:crl_set_remover",
+    #"//components/component_updater/installer_policies",
     "//components/consent_auditor",
     "//components/content_capture/browser",
     "//components/content_settings/browser",
@@ -2251,22 +2252,22 @@
     "//components/reputation/core:proto",
     "//components/resources",
     "//components/safe_browsing/content/browser",
-    "//components/safe_browsing/content/browser:client_side_detection",
-    "//components/safe_browsing/content/browser:safe_browsing_service",
+    #"//components/safe_browsing/content/browser:client_side_detection",
+    #"//components/safe_browsing/content/browser:safe_browsing_service",
     "//components/safe_browsing/content/browser/password_protection",
-    "//components/safe_browsing/content/browser/web_ui",
-    "//components/safe_browsing/core/browser",
-    "//components/safe_browsing/core/browser:safe_browsing_metrics_collector",
-    "//components/safe_browsing/core/browser:verdict_cache_manager",
-    "//components/safe_browsing/core/browser/db:database_manager",
-    "//components/safe_browsing/core/browser/password_protection:password_protection_metrics_util",
-    "//components/safe_browsing/core/browser/realtime:policy_engine",
-    "//components/safe_browsing/core/browser/realtime:url_lookup_service",
-    "//components/safe_browsing/core/browser/sync",
-    "//components/safe_browsing/core/common",
-    "//components/safe_browsing/core/common:safe_browsing_policy_handler",
-    "//components/safe_browsing/core/common/proto:csd_proto",
-    "//components/safe_browsing/core/common/proto:realtimeapi_proto",
+    #"//components/safe_browsing/content/browser/web_ui",
+    #"//components/safe_browsing/core/browser",
+    #"//components/safe_browsing/core/browser:safe_browsing_metrics_collector",
+    #"//components/safe_browsing/core/browser:verdict_cache_manager",
+    #"//components/safe_browsing/core/browser/db:database_manager",
+    #"//components/safe_browsing/core/browser/password_protection:password_protection_metrics_util",
+    #"//components/safe_browsing/core/browser/realtime:policy_engine",
+    #"//components/safe_browsing/core/browser/realtime:url_lookup_service",
+    #"//components/safe_browsing/core/browser/sync",
+    #"//components/safe_browsing/core/common",
+    #"//components/safe_browsing/core/common:safe_browsing_policy_handler",
+    #"//components/safe_browsing/core/common/proto:csd_proto",
+    #"//components/safe_browsing/core/common/proto:realtimeapi_proto",
     "//components/safe_search_api",
     "//components/safe_search_api:safe_search_client",
     "//components/saved_tab_groups:core",
@@ -3691,8 +3692,8 @@
       "enterprise/connectors/analysis/analysis_service_settings.h",
       "enterprise/connectors/analysis/analysis_settings.cc",
       "enterprise/connectors/analysis/analysis_settings.h",
-      "enterprise/connectors/analysis/content_analysis_delegate.cc",
-      "enterprise/connectors/analysis/content_analysis_delegate.h",
+      #"enterprise/connectors/analysis/content_analysis_delegate.cc",
+      #"enterprise/connectors/analysis/content_analysis_delegate.h",
       "enterprise/connectors/analysis/content_analysis_delegate_base.cc",
       "enterprise/connectors/analysis/content_analysis_delegate_base.h",
       "enterprise/connectors/analysis/content_analysis_dialog.cc",
@@ -6064,8 +6065,8 @@
       "//base/win:base_win_buildflags",
       "//chrome/app:chrome_exe_main_exports",
       "//chrome/app/theme:chrome_unscaled_resources_grit",
-      "//chrome/browser/safe_browsing/chrome_cleaner",
-      "//chrome/browser/safe_browsing/chrome_cleaner:public",
+      #"//chrome/browser/safe_browsing/chrome_cleaner",
+      #"//chrome/browser/safe_browsing/chrome_cleaner:public",
       "//chrome/browser/web_applications/chrome_pwa_launcher:util",
       "//chrome/browser/win/conflicts:module_info",
       "//chrome/chrome_elf:constants",
@@ -6095,8 +6096,8 @@
       "//ui/color:accent_color_observer",
       "//ui/display/util:util",
     ]
-    allow_circular_includes_from +=
-        [ "//chrome/browser/safe_browsing/chrome_cleaner" ]
+    #allow_circular_includes_from +=
+        #[ "//chrome/browser/safe_browsing/chrome_cleaner" ]
 
     all_dependent_configs = [ ":browser_win_linker_flags" ]
 
@@ -6304,7 +6305,7 @@
   }
 
   if (is_win || is_linux || is_android || is_chromeos) {
-    deps += [ "//chrome/browser/privacy:traffic_annotation_proto" ]
+    #deps += [ "//chrome/browser/privacy:traffic_annotation_proto" ]
   }
 
   if (is_android || is_mac || is_win) {
@@ -6376,8 +6377,8 @@
     sources += [
       "enterprise/chrome_browser_main_extra_parts_enterprise.cc",
       "enterprise/chrome_browser_main_extra_parts_enterprise.h",
-      "enterprise/connectors/analysis/content_analysis_sdk_manager.cc",
-      "enterprise/connectors/analysis/content_analysis_sdk_manager.h",
+      #"enterprise/connectors/analysis/content_analysis_sdk_manager.cc",
+      #"enterprise/connectors/analysis/content_analysis_sdk_manager.h",
       "enterprise/connectors/analysis/local_binary_upload_service.cc",
       "enterprise/connectors/analysis/local_binary_upload_service.h",
       "enterprise/connectors/analysis/local_binary_upload_service_factory.cc",
@@ -7892,7 +7893,6 @@
       "//components/safe_browsing/content/browser:safe_browsing_blocking_page",
       "//components/safe_browsing/content/browser/download:download_stats",
       "//components/safe_browsing/content/common:file_type_policies",
-      "//components/safe_browsing/content/common/proto:download_file_types_proto",
     ]
   }
 
@@ -7903,6 +7903,9 @@
     ]
     deps += [ "//components/segmentation_platform/internal:optimization_guide_segmentation_handler" ]
   }
+  deps += [
+      "//components/safe_browsing/content/common/proto:download_file_types_proto",
+  ]
 }
 
 if (is_android) {
@@ -7997,7 +8000,11 @@
 }
 
 grit("resources") {
-  source = "browser_resources.grd"
+  if (nwjs_sdk) {
+    source = "browser_resources.grd"
+  } else {
+    source = "nwjs_resources.grd"
+  }
 
   use_brotli = true
 
@@ -8005,6 +8012,9 @@
   if (enable_hangout_services_extension) {
     defines += [ "enable_hangout_services_extension" ]
   }
+  if (nwjs_sdk) {
+    defines += [ "nwjs_sdk" ]
+  }
 
   output_dir = "$root_gen_dir/chrome"
   outputs = [
