diff -r -u --color up/chromium/chrome/browser/background/background_application_list_model.cc nw/chromium/chrome/browser/background/background_application_list_model.cc
--- up/chromium/chrome/browser/background/background_application_list_model.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/background/background_application_list_model.cc	2023-02-24 20:54:06.992520601 +0000
@@ -244,8 +244,10 @@
   //    manifest.
 
   // Not a background app if we don't have the background permission.
+  // NWJS: nwjs_default_app is listed as background app and prevents
+  // quit so we need to disable it here
   if (!extension.permissions_data()->HasAPIPermission(
-          APIPermissionID::kBackground)) {
+                     APIPermissionID::kBackground, true)) {
     return false;
   }
 
@@ -382,7 +384,8 @@
         // Policy changes are only used for host permissions, so the
         // "background"
         // permission would never be present in  permissions .
-        NOTREACHED();
+        break;
+        //NOTREACHED();
     }
   }
 }
