diff -r -u --color up/chromium/chrome/install_static/chromium_install_modes.cc nw/chromium/chrome/install_static/chromium_install_modes.cc
--- up/chromium/chrome/install_static/chromium_install_modes.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/install_static/chromium_install_modes.cc	2023-02-24 06:16:22.932182031 +0000
@@ -16,7 +16,7 @@
 
 const wchar_t kCompanyPathName[] = L"";
 
-const wchar_t kProductPathName[] = L"Chromium";
+const wchar_t kProductPathName[] = L"nwjs";
 
 const size_t kProductPathNameLength = _countof(kProductPathName) - 1;
 
@@ -33,9 +33,9 @@
             L"",  // Empty install_suffix for the primary install mode.
         .logo_suffix = L"",  // No logo suffix for the primary install mode.
         .app_guid =
-            L"",  // Empty app_guid since no integration with Google Update.
-        .base_app_name = L"Chromium",      // A distinct base_app_name.
-        .base_app_id = L"Chromium",        // A distinct base_app_id.
+            L"",  // Empty app_guid since no integraion with Google Update.
+        .base_app_name = L"nwjs",      // A distinct base_app_name.
+        .base_app_id = L"nwjs",        // A distinct base_app_id.
         .prog_id_prefix = L"ChromiumHTM",  // ProgID prefix.
         .prog_id_description =
             L"Chromium HTML Document",  // ProgID description.
