diff -r -u --color up/chromium/chrome/browser/resources/BUILD.gn nw/chromium/chrome/browser/resources/BUILD.gn
--- up/chromium/chrome/browser/resources/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/resources/BUILD.gn	2023-02-24 06:16:21.480096041 +0000
@@ -161,6 +161,9 @@
     if (enable_hangout_services_extension) {
       defines += [ "enable_hangout_services_extension" ]
     }
+    if (nwjs_sdk) {
+      defines += [ "nwjs_sdk" ]
+    }
 
     # Some resources within component_extension_resources.grd are generated at
     # build time so we include them as explicit dependencies here
