diff -r -u --color up/chromium/chrome/browser/extensions/safe_browsing_verdict_handler.cc nw/chromium/chrome/browser/extensions/safe_browsing_verdict_handler.cc
--- up/chromium/chrome/browser/extensions/safe_browsing_verdict_handler.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/safe_browsing_verdict_handler.cc	2023-02-24 20:12:03.324811847 +0000
@@ -47,6 +47,7 @@
 void SafeBrowsingVerdictHandler::Init() {
   TRACE_EVENT0("browser,startup", "SafeBrowsingVerdictHandler::Init");
 
+#if 0
   std::unique_ptr<ExtensionSet> all_extensions =
       registry_->GenerateInstalledExtensionsSet();
 
@@ -67,6 +68,7 @@
       blocklist_.Insert(extension);
     }
   }
+#endif
 }
 
 void SafeBrowsingVerdictHandler::ManageBlocklist(
