diff -r -u --color up/chromium/chrome/browser/ui/views/constrained_web_dialog_delegate_views.cc nw/chromium/chrome/browser/ui/views/constrained_web_dialog_delegate_views.cc
--- up/chromium/chrome/browser/ui/views/constrained_web_dialog_delegate_views.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/constrained_web_dialog_delegate_views.cc	2023-02-24 20:54:08.388514164 +0000
@@ -209,6 +209,7 @@
 
   // ui::WebDialogWebContentsDelegate:
   void CloseContents(content::WebContents* source) override {
+    if (view_->GetWidget())
     view_->GetWidget()->Close();
   }
 
