diff -r -u --color up/chromium/chrome/common/logging_chrome.cc nw/chromium/chrome/common/logging_chrome.cc
--- up/chromium/chrome/common/logging_chrome.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/common/logging_chrome.cc	2023-02-24 06:16:22.896179899 +0000
@@ -20,6 +20,7 @@
 #define IPC_LOG_TABLE_ADD_ENTRY(msg_id, logger) \
   content::RegisterIPCLogger(msg_id, logger)
 #include "chrome/common/all_messages.h"
+#include "extensions/common/extension_messages.h"
 #endif
 
 #if BUILDFLAG(IS_WIN)
