diff -r -u --color up/chromium/chrome/browser/extensions/extension_allowlist.cc nw/chromium/chrome/browser/extensions/extension_allowlist.cc
--- up/chromium/chrome/browser/extensions/extension_allowlist.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/extension_allowlist.cc	2023-02-24 06:16:21.004067851 +0000
@@ -399,6 +399,7 @@
 }
 
 void ExtensionAllowlist::ReportExtensionReEnabledEvent() {
+#if 0
   auto* metrics_collector =
       safe_browsing::SafeBrowsingMetricsCollectorFactory::GetForProfile(
           profile_);
@@ -408,6 +409,7 @@
         safe_browsing::SafeBrowsingMetricsCollector::EventType::
             NON_ALLOWLISTED_EXTENSION_RE_ENABLED);
   }
+#endif
 }
 
 }  // namespace extensions
