diff -r -u --color up/chromium/chrome/browser/download/download_stats.cc nw/chromium/chrome/browser/download/download_stats.cc
--- up/chromium/chrome/browser/download/download_stats.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/download/download_stats.cc	2023-02-24 20:12:03.184810421 +0000
@@ -27,10 +27,12 @@
     const base::FilePath& file_path,
     bool is_https,
     bool has_user_gesture) {
+#if 0
   base::UmaHistogramEnumeration("Download.ShowedDownloadWarning", danger_type,
                                 download::DOWNLOAD_DANGER_TYPE_MAX);
   safe_browsing::RecordDangerousDownloadWarningShown(
       danger_type, file_path, is_https, has_user_gesture);
+#endif
 }
 
 void RecordOpenedDangerousConfirmDialog(
