diff -r -u --color up/chromium/chrome/browser/extensions/api/tab_capture/tab_capture_registry.cc nw/chromium/chrome/browser/extensions/api/tab_capture/tab_capture_registry.cc
--- up/chromium/chrome/browser/extensions/api/tab_capture/tab_capture_registry.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/api/tab_capture/tab_capture_registry.cc	2023-02-24 20:12:03.280811399 +0000
@@ -22,6 +22,10 @@
 #include "extensions/common/extension.h"
 #include "url/origin.h"
 
+#include "chrome/browser/profiles/profile_manager.h"
+#include "extensions/browser/app_window/app_window_registry.h"
+#include "extensions/browser/app_window/app_window.h"
+
 using content::BrowserThread;
 using extensions::tab_capture::TabCaptureState;
 
