diff -r -u --color up/chromium/chrome/browser/ui/views/omnibox/omnibox_popup_contents_view.cc nw/chromium/chrome/browser/ui/views/omnibox/omnibox_popup_contents_view.cc
--- up/chromium/chrome/browser/ui/views/omnibox/omnibox_popup_contents_view.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/omnibox/omnibox_popup_contents_view.cc	2023-02-24 06:16:22.500156447 +0000
@@ -97,7 +97,7 @@
 
     // Destroy the popup when done. The observer deletes itself on completion.
     scoped_settings->AddObserver(new ui::ClosureAnimationObserver(
-        base::BindOnce(&AutocompletePopupWidget::Close, AsWeakPtr())));
+                                                                  base::BindOnce(&AutocompletePopupWidget::Close, AsWeakPtr(), false)));
   }
 
   void OnNativeWidgetDestroying() override {
