diff -r -u --color up/chromium/chrome/chrome_elf/chrome_elf_x86.def nw/chromium/chrome/chrome_elf/chrome_elf_x86.def
--- up/chromium/chrome/chrome_elf/chrome_elf_x86.def	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/chrome_elf/chrome_elf_x86.def	2023-02-24 06:16:22.860177767 +0000
@@ -1,7 +1,8 @@
 ; Copyright 2013 The Chromium Authors
 ; Use of this source code is governed by a BSD-style license that can be
 ; found in the LICENSE file.
-LIBRARY  "chrome_elf.dll"
+
+LIBRARY  "nw_elf.dll"
 
 EXPORTS
   ; When functions are added to this file, they must also be added to
@@ -43,3 +44,4 @@
 
   ; From chrome/chrome_elf/third_party_dlls/main.cc
   IsThirdPartyInitialized
+  ElfGetReporterClient
