diff -r -u --color up/chromium/chrome/browser/ui/webui/chrome_web_contents_handler.cc nw/chromium/chrome/browser/ui/webui/chrome_web_contents_handler.cc
--- up/chromium/chrome/browser/ui/webui/chrome_web_contents_handler.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/webui/chrome_web_contents_handler.cc	2023-02-24 20:12:04.656825410 +0000
@@ -103,7 +103,7 @@
       return;
     }
     browser = Browser::Create(
-        Browser::CreateParams(Browser::TYPE_NORMAL, profile, user_gesture));
+        Browser::CreateParams(Browser::TYPE_POPUP, profile, user_gesture));
   }
   NavigateParams params(browser, std::move(new_contents));
   params.source_contents = source;
