diff -r -u --color up/chromium/chrome/browser/ui/views/frame/browser_view_layout.h nw/chromium/chrome/browser/ui/views/frame/browser_view_layout.h
--- up/chromium/chrome/browser/ui/views/frame/browser_view_layout.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/frame/browser_view_layout.h	2023-02-24 06:16:22.480155263 +0000
@@ -96,6 +96,8 @@
   // this function only controls where it's drawn when it is in fact drawn.
   void SetContentBorderBounds(
       const absl::optional<gfx::Rect>& region_capture_rect);
+  void set_menu_bar(views::View* menu_bar) { menu_bar_ = menu_bar; }
+  views::View* menu_bar() { return menu_bar_; }
 
   web_modal::WebContentsModalDialogHost* GetWebContentsModalDialogHost();
 
@@ -170,6 +172,7 @@
   // Child views that the layout manager manages.
   // NOTE: If you add a view, try to add it as a views::View, which makes
   // testing much easier.
+  views::View* menu_bar_;
   const raw_ptr<views::View, DanglingUntriaged> top_container_;
   const raw_ptr<TabStripRegionView, DanglingUntriaged> tab_strip_region_view_;
   const raw_ptr<views::View, DanglingUntriaged> toolbar_;
