diff -r -u --color up/chromium/chrome/common/chrome_features.cc nw/chromium/chrome/common/chrome_features.cc
--- up/chromium/chrome/common/chrome_features.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/common/chrome_features.cc	2023-02-24 06:16:22.868178241 +0000
@@ -1048,7 +1048,7 @@
 #if BUILDFLAG(IS_ANDROID)
              base::FEATURE_DISABLED_BY_DEFAULT
 #else
-             base::FEATURE_ENABLED_BY_DEFAULT
+      base::FEATURE_DISABLED_BY_DEFAULT
 #endif
 );
 
