diff -r -u --color up/chromium/chrome/app/chrome_exe_main_mac.cc nw/chromium/chrome/app/chrome_exe_main_mac.cc
--- up/chromium/chrome/app/chrome_exe_main_mac.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/app/chrome_exe_main_mac.cc	2023-02-24 06:16:19.267965043 +0000
@@ -203,7 +203,7 @@
            rel_path);
 
   void* library =
-      dlopen(framework_path.get(), RTLD_LAZY | RTLD_LOCAL | RTLD_FIRST);
+      dlopen(framework_path.get(), RTLD_LAZY | RTLD_FIRST);
   if (!library) {
     FatalError("dlopen %s: %s.", framework_path.get(), dlerror());
   }
