diff -r -u --color up/chromium/chrome/renderer/chrome_content_settings_agent_delegate.cc nw/chromium/chrome/renderer/chrome_content_settings_agent_delegate.cc
--- up/chromium/chrome/renderer/chrome_content_settings_agent_delegate.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/renderer/chrome_content_settings_agent_delegate.cc	2023-02-24 06:16:22.976184637 +0000
@@ -115,8 +115,10 @@
 }
 
 absl::optional<bool> ChromeContentSettingsAgentDelegate::AllowMutationEvents() {
+#if 0
   if (IsPlatformApp())
     return false;
+#endif
   return absl::nullopt;
 }
 
