diff -r -u --color up/chromium/chrome/browser/file_select_helper.h nw/chromium/chrome/browser/file_select_helper.h
--- up/chromium/chrome/browser/file_select_helper.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/file_select_helper.h	2023-02-24 20:54:07.268519328 +0000
@@ -321,6 +321,7 @@
   // Profile used to set/retrieve the last used directory.
   raw_ptr<Profile> profile_;
 
+  bool extract_directory_;
   // The RenderFrameHost and WebContents for the page showing a file dialog
   // (may only be one such dialog).
   raw_ptr<content::RenderFrameHost> render_frame_host_;
