diff -r -u --color up/chromium/chrome/app/chrome_main.cc nw/chromium/chrome/app/chrome_main.cc
--- up/chromium/chrome/app/chrome_main.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/app/chrome_main.cc	2023-02-24 06:16:19.267965043 +0000
@@ -106,7 +106,7 @@
   auto crash_on_detach_resetter = base::ScopedClosureRunner(
       base::BindOnce(&base::win::SetShouldCrashOnProcessDetach,
                      base::win::ShouldCrashOnProcessDetach()));
-  base::win::SetShouldCrashOnProcessDetach(true);
+  base::win::SetShouldCrashOnProcessDetach(false);
   base::win::SetAbortBehaviorForCrashReporting();
   params.instance = instance;
   params.sandbox_info = sandbox_info;
