diff -r -u --color up/chromium/chrome/browser/ui/views/media_router/presentation_receiver_window_view.cc nw/chromium/chrome/browser/ui/views/media_router/presentation_receiver_window_view.cc
--- up/chromium/chrome/browser/ui/views/media_router/presentation_receiver_window_view.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/media_router/presentation_receiver_window_view.cc	2023-02-24 06:16:22.500156447 +0000
@@ -166,7 +166,7 @@
   infobars::ContentInfoBarManager::CreateForWebContents(web_contents);
 
   SecurityStateTabHelper::CreateForWebContents(web_contents);
-  ChromeTranslateClient::CreateForWebContents(web_contents);
+  //ChromeTranslateClient::CreateForWebContents(web_contents);
   autofill::ChromeAutofillClient::CreateForWebContents(web_contents);
   autofill::ContentAutofillDriverFactory::CreateForWebContentsAndDelegate(
       web_contents,
