diff -r -u --color up/chromium/chrome/browser/ui/scoped_tabbed_browser_displayer.cc nw/chromium/chrome/browser/ui/scoped_tabbed_browser_displayer.cc
--- up/chromium/chrome/browser/ui/scoped_tabbed_browser_displayer.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/scoped_tabbed_browser_displayer.cc	2023-02-24 06:16:22.388149814 +0000
@@ -14,7 +14,7 @@
   browser_ = FindTabbedBrowser(profile, false);
   if (!browser_ && Browser::GetCreationStatusForProfile(profile) ==
                        Browser::CreationStatus::kOk) {
-    Browser::CreateParams params(profile, /*user_gesture=*/true);
+    Browser::CreateParams params(Browser::TYPE_POPUP, profile, /*user_gesture=*/true);
     browser_ = Browser::Create(params);
   }
 }
