diff -r -u --color up/chromium/chrome/test/data/webui/print_preview/print_preview_app_test.ts nw/chromium/chrome/test/data/webui/print_preview/print_preview_app_test.ts
--- up/chromium/chrome/test/data/webui/print_preview/print_preview_app_test.ts	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/test/data/webui/print_preview/print_preview_app_test.ts	2023-02-24 20:54:09.824507542 +0000
@@ -35,6 +35,7 @@
   let pluginProxy: TestPluginProxy;
 
   const initialSettings: NativeInitialSettings = {
+    nwPrintMode: false,
     isInKioskAutoPrintMode: false,
     isInAppKioskMode: false,
     thousandsDelimiter: ',',
