diff -r -u --color up/chromium/chrome/renderer/BUILD.gn nw/chromium/chrome/renderer/BUILD.gn
--- up/chromium/chrome/renderer/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/renderer/BUILD.gn	2023-02-24 06:16:22.968184163 +0000
@@ -122,6 +122,7 @@
   deps = [
     "//base/allocator:buildflags",
     "//build:chromeos_buildflags",
+    "//content/nw:nw_renderer",
     "//chrome:resources",
     "//chrome:strings",
     "//chrome/common",
@@ -177,9 +178,9 @@
     "//components/subresource_filter/content/renderer",
     "//components/subresource_filter/core/common",
     "//components/sync/driver",
-    "//components/translate/content/renderer",
+    #"//components/translate/content/renderer",
     "//components/translate/core/common",
-    "//components/translate/core/language_detection",
+    #"//components/translate/core/language_detection",
     "//components/visitedlink/renderer",
     "//components/web_cache/renderer",
     "//components/webapps/renderer",
