diff -r -u --color up/chromium/chrome/browser/new_tab_page/one_google_bar/one_google_bar_loader_impl.cc nw/chromium/chrome/browser/new_tab_page/one_google_bar/one_google_bar_loader_impl.cc
--- up/chromium/chrome/browser/new_tab_page/one_google_bar/one_google_bar_loader_impl.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/new_tab_page/one_google_bar/one_google_bar_loader_impl.cc	2023-02-24 06:16:21.256082775 +0000
@@ -280,14 +280,15 @@
     const std::string& application_locale,
     bool account_consistency_mirror_required)
     : url_loader_factory_(url_loader_factory),
-      application_locale_(application_locale),
-      account_consistency_mirror_required_(
-          account_consistency_mirror_required) {}
+      application_locale_(application_locale)
+      /* account_consistency_mirror_required_(
+         account_consistency_mirror_required)*/ {}
 
 OneGoogleBarLoaderImpl::~OneGoogleBarLoaderImpl() = default;
 
 void OneGoogleBarLoaderImpl::Load(OneGoogleCallback callback) {
   callbacks_.push_back(std::move(callback));
+#if 0
 
   // Note: If there is an ongoing request, abandon it. It's possible that
   // something has changed in the meantime (e.g. signin state) that would make
@@ -297,6 +298,7 @@
       base::BindOnce(&OneGoogleBarLoaderImpl::LoadDone,
                      base::Unretained(this)));
   pending_request_->Start();
+#endif
 }
 
 GURL OneGoogleBarLoaderImpl::GetLoadURLForTesting() const {
