diff -r -u --color up/chromium/chrome/browser/infobars/simple_alert_infobar_creator.cc nw/chromium/chrome/browser/infobars/simple_alert_infobar_creator.cc
--- up/chromium/chrome/browser/infobars/simple_alert_infobar_creator.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/infobars/simple_alert_infobar_creator.cc	2023-02-24 20:54:07.300519181 +0000
@@ -19,6 +19,7 @@
     const std::u16string& message,
     bool auto_expire,
     bool should_animate) {
+  if (infobar_manager) //NWJS#6340
   infobar_manager->AddInfoBar(
       CreateConfirmInfoBar(std::make_unique<SimpleAlertInfoBarDelegate>(
           infobar_identifier, vector_icon, message, auto_expire,
