diff -r -u --color up/chromium/chrome/browser/ui/cocoa/status_icons/status_icon_mac.mm nw/chromium/chrome/browser/ui/cocoa/status_icons/status_icon_mac.mm
--- up/chromium/chrome/browser/ui/cocoa/status_icons/status_icon_mac.mm	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/cocoa/status_icons/status_icon_mac.mm	2023-02-24 20:54:08.300514569 +0000
@@ -100,7 +100,7 @@
   return menu_.get() != nil;
 }
 
-void StatusIconMac::UpdatePlatformContextMenu(StatusIconMenuModel* model) {
+void StatusIconMac::UpdatePlatformContextMenu(ui::MenuModel* model) {
   if (!model) {
     menu_.reset();
   } else {
