diff -r -u --color up/chromium/chrome/common/BUILD.gn nw/chromium/chrome/common/BUILD.gn
--- up/chromium/chrome/common/BUILD.gn	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/common/BUILD.gn	2023-02-24 20:12:04.860827487 +0000
@@ -222,8 +222,8 @@
     "//components/services/app_service/public/cpp:app_types",
     "//components/services/heap_profiling/public/cpp",
     "//components/strings",
-    "//components/translate/content/common",
-    "//components/translate/core/common",
+    #"//components/translate/content/common",
+    #"//components/translate/core/common",
     "//components/url_formatter",
     "//components/variations",
     "//components/variations/net",
@@ -334,6 +334,7 @@
       "//components/services/app_service/public/mojom",
     ]
     public_deps += [
+      "//content/nw/src/api:nw_api",
       "//chrome/common/extensions/api",
       "//chrome/common/extensions/api:extensions_features",
       "//extensions:extensions_resources",
