diff -r -u --color up/chromium/chrome/browser/resources_integrity.cc nw/chromium/chrome/browser/resources_integrity.cc
--- up/chromium/chrome/browser/resources_integrity.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/resources_integrity.cc	2023-02-24 06:16:21.856118308 +0000
@@ -109,10 +109,10 @@
   base::span<const uint8_t, crypto::kSHA256Length> resources_hash =
       kSha256_resources_pak;
   base::span<const uint8_t, crypto::kSHA256Length> chrome_100_hash =
-      kSha256_chrome_100_percent_pak;
+      kSha256_nw_100_percent_pak;
 #if BUILDFLAG(ENABLE_HIDPI)
   base::span<const uint8_t, crypto::kSHA256Length> chrome_200_hash =
-      kSha256_chrome_200_percent_pak;
+      kSha256_nw_200_percent_pak;
 #endif
 #endif  // BUILDFLAG(IS_WIN)
 
