diff -r -u --color up/chromium/chrome/renderer/url_loader_throttle_provider_impl.cc nw/chromium/chrome/renderer/url_loader_throttle_provider_impl.cc
--- up/chromium/chrome/renderer/url_loader_throttle_provider_impl.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/renderer/url_loader_throttle_provider_impl.cc	2023-02-24 06:16:22.988185347 +0000
@@ -140,6 +140,7 @@
   DCHECK(!is_frame_resource ||
          type_ == blink::URLLoaderThrottleProviderType::kFrame);
 
+#if 0
   if (!is_frame_resource) {
     if (safe_browsing_remote_)
       safe_browsing_.Bind(std::move(safe_browsing_remote_));
@@ -147,6 +148,7 @@
         std::make_unique<safe_browsing::RendererURLLoaderThrottle>(
             safe_browsing_.get(), render_frame_id));
   }
+#endif
 
   if (type_ == blink::URLLoaderThrottleProviderType::kFrame &&
       !is_frame_resource) {
