diff -r -u --color up/chromium/chrome/app/version_assembly/version_assembly_manifest.template nw/chromium/chrome/app/version_assembly/version_assembly_manifest.template
--- up/chromium/chrome/app/version_assembly/version_assembly_manifest.template	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/app/version_assembly/version_assembly_manifest.template	2023-02-24 20:12:02.592804393 +0000
@@ -4,5 +4,5 @@
       name='@MAJOR@.@MINOR@.@BUILD@.@PATCH@'
       version='@MAJOR@.@MINOR@.@BUILD@.@PATCH@'
       type='win32'/>
-  <file name='chrome_elf.dll'/>
+  <file name='nw_elf.dll'/>
 </assembly>
