diff -r -u --color up/chromium/chrome/browser/ui/views/status_icons/status_tray_win.cc nw/chromium/chrome/browser/ui/views/status_icons/status_tray_win.cc
--- up/chromium/chrome/browser/ui/views/status_icons/status_tray_win.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/status_icons/status_tray_win.cc	2023-02-24 06:16:22.548159290 +0000
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/views/status_icons/status_tray_win.h"
 
+#include "ui/views/controls/menu/menu_controller.h"
 #include <commctrl.h>
 #include <wrl/client.h>
 
@@ -100,7 +101,7 @@
 };
 
 StatusTrayWin::StatusTrayWin()
-    : next_icon_id_(1),
+    : key_handler_(nullptr), next_icon_id_(1),
       atom_(0),
       instance_(NULL),
       window_(NULL) {
@@ -210,6 +211,13 @@
     // it gets that Windows is exiting. Make sure we shutdown in an orderly
     // fashion.
     chrome::SessionEnding();
+  } else if (message == WM_KEYDOWN || message == WM_KEYUP) {
+    CHROME_MSG msg = {hwnd, message, wparam, lparam,
+               static_cast<DWORD>(GetMessageTime())};
+    ui::KeyEvent key(msg);
+    views::MenuController* controller = views::MenuController::GetActiveInstance();
+    if (controller)
+      controller->OnWillDispatchKeyEvent(&key);
   }
   return ::DefWindowProc(hwnd, message, wparam, lparam);
 }
