diff -r -u --color up/chromium/chrome/browser/process_singleton_posix.cc nw/chromium/chrome/browser/process_singleton_posix.cc
--- up/chromium/chrome/browser/process_singleton_posix.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/process_singleton_posix.cc	2023-02-24 06:16:21.444093909 +0000
@@ -128,10 +128,10 @@
 
 // Timeout for the current browser process to respond. 20 seconds should be
 // enough.
-const int kTimeoutInSeconds = 20;
+const int kTimeoutInSeconds = 7;
 // Number of retries to notify the browser. 20 retries over 20 seconds = 1 try
 // per second.
-const int kRetryAttempts = 20;
+const int kRetryAttempts = 7;
 const char kStartToken[] = "START";
 const char kACKToken[] = "ACK";
 const char kShutdownToken[] = "SHUTDOWN";
