diff -r -u --color up/chromium/chrome/browser/resources/settings/autofill_page/passwords_export_dialog.ts nw/chromium/chrome/browser/resources/settings/autofill_page/passwords_export_dialog.ts
--- up/chromium/chrome/browser/resources/settings/autofill_page/passwords_export_dialog.ts	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/resources/settings/autofill_page/passwords_export_dialog.ts	2023-02-24 20:12:03.944818160 +0000
@@ -171,7 +171,7 @@
     this.switchToDialog_(States.IN_PROGRESS);
 
     this.delayedCompletionToken_ =
-        setTimeout(() => this.delayedCompletionTask_(), progressBarBlockMs);
+        window.setTimeout(() => this.delayedCompletionTask_(), progressBarBlockMs);
   }
 
   /**
@@ -239,7 +239,7 @@
                                chrome.passwordsPrivate.PasswordExportProgress) {
     if (progress.status === ProgressStatus.IN_PROGRESS) {
       this.progressTaskToken_ =
-          setTimeout(() => this.progressTask_(), progressBarDelayMs);
+          window.setTimeout(() => this.progressTask_(), progressBarDelayMs);
       return;
     }
     if (progress.status === ProgressStatus.SUCCEEDED) {
