diff -r -u --color up/chromium/chrome/browser/safe_browsing/BUILD.gn nw/chromium/chrome/browser/safe_browsing/BUILD.gn
--- up/chromium/chrome/browser/safe_browsing/BUILD.gn	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/safe_browsing/BUILD.gn	2023-02-24 06:16:21.856118308 +0000
@@ -41,7 +41,7 @@
     "//components/resources:components_resources_grit",
     "//components/safe_browsing:buildflags",
     "//components/safe_browsing/content/browser",
-    "//components/safe_browsing/content/browser:client_side_detection",
+    #"//components/safe_browsing/content/browser:client_side_detection",
     "//components/safe_browsing/content/browser/triggers:suspicious_site_trigger",
     "//components/safe_browsing/content/browser/web_ui",
     "//components/safe_browsing/core/browser",
