diff -r -u --color up/chromium/chrome/browser/ui/views/frame/browser_non_client_frame_view_mac.h nw/chromium/chrome/browser/ui/views/frame/browser_non_client_frame_view_mac.h
--- up/chromium/chrome/browser/ui/views/frame/browser_non_client_frame_view_mac.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/frame/browser_non_client_frame_view_mac.h	2023-02-24 06:16:22.476155026 +0000
@@ -67,6 +67,7 @@
   gfx::Size GetMinimumSize() const override;
   void AddedToWidget() override;
   void PaintChildren(const views::PaintInfo& info) override;
+  gfx::Size GetMaximumSize() const override;
 
   // web_app::AppRegistrarObserver
   void OnAlwaysShowToolbarInFullscreenChanged(const web_app::AppId& app_id,
