diff -r -u --color up/chromium/chrome/browser/file_system_access/chrome_file_system_access_permission_context.h nw/chromium/chrome/browser/file_system_access/chrome_file_system_access_permission_context.h
--- up/chromium/chrome/browser/file_system_access/chrome_file_system_access_permission_context.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/browser/file_system_access/chrome_file_system_access_permission_context.h	2023-02-24 06:16:21.068071642 +0000
@@ -54,6 +54,8 @@
     : public content::FileSystemAccessPermissionContext,
       public permissions::ObjectPermissionContextBase {
  public:
+  content::BrowserContext* browser_context_; //optimal place to patch
+                                             //this in NW
   explicit ChromeFileSystemAccessPermissionContext(
       content::BrowserContext* context,
       const base::Clock* clock = base::DefaultClock::GetInstance());
@@ -216,7 +218,7 @@
   // Checks whether the file or directory at `path` corresponds to a directory
   // Chrome considers sensitive (i.e. system files). Calls `callback` with
   // whether the path is on the blocklist.
-  void CheckPathAgainstBlocklist(PathType path_type,
+  void CheckPathAgainstBlocklist(const url::Origin& origin, PathType path_type,
                                  const base::FilePath& path,
                                  HandleType handle_type,
                                  base::OnceCallback<void(bool)> callback);
