diff -r -u --color up/chromium/chrome/browser/component_updater/component_updater_prefs.cc nw/chromium/chrome/browser/component_updater/component_updater_prefs.cc
--- up/chromium/chrome/browser/component_updater/component_updater_prefs.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/component_updater/component_updater_prefs.cc	2023-02-24 20:12:03.132809892 +0000
@@ -20,7 +20,7 @@
 void RegisterPrefs(PrefRegistrySimple* registry) {
   RegisterComponentUpdateServicePrefs(registry);
   RegisterPrefsForRecoveryComponent(registry);
-  AutofillStatesComponentInstallerPolicy::RegisterPrefs(registry);
+  //AutofillStatesComponentInstallerPolicy::RegisterPrefs(registry);
 }
 
 }  // namespace component_updater
