diff -r -u --color up/chromium/chrome/renderer/chrome_render_frame_observer.h nw/chromium/chrome/renderer/chrome_render_frame_observer.h
--- up/chromium/chrome/renderer/chrome_render_frame_observer.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/chrome/renderer/chrome_render_frame_observer.h	2023-02-24 06:16:22.976184637 +0000
@@ -5,6 +5,7 @@
 #ifndef CHROME_RENDERER_CHROME_RENDER_FRAME_OBSERVER_H_
 #define CHROME_RENDERER_CHROME_RENDER_FRAME_OBSERVER_H_
 
+#pragma clang diagnostic ignored "-Wunused-private-field"
 #include <string>
 #include <vector>
 
