diff -r -u --color up/chromium/chrome/browser/extensions/api/sessions/sessions_api.cc nw/chromium/chrome/browser/extensions/api/sessions/sessions_api.cc
--- up/chromium/chrome/browser/extensions/api/sessions/sessions_api.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/extensions/api/sessions/sessions_api.cc	2023-02-24 20:12:03.272811317 +0000
@@ -327,6 +327,9 @@
 
   api::windows::WindowState state = api::windows::WINDOW_STATE_NONE;
   switch (window.show_state) {
+    case ui::SHOW_STATE_HIDDEN:
+      state = api::windows::WINDOW_STATE_HIDDEN;
+      break;
     case ui::SHOW_STATE_NORMAL:
       state = api::windows::WINDOW_STATE_NORMAL;
       break;
