diff -r -u --color up/chromium/chrome/browser/ui/views/frame/contents_web_view.h nw/chromium/chrome/browser/ui/views/frame/contents_web_view.h
--- up/chromium/chrome/browser/ui/views/frame/contents_web_view.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/chrome/browser/ui/views/frame/contents_web_view.h	2023-02-24 20:54:08.412514053 +0000
@@ -25,7 +25,7 @@
       public WebContentsCloseHandlerDelegate {
  public:
   METADATA_HEADER(ContentsWebView);
-  explicit ContentsWebView(content::BrowserContext* browser_context);
+  explicit ContentsWebView(content::BrowserContext* browser_context, bool transparent);
   ContentsWebView(const ContentsWebView&) = delete;
   ContentsWebView& operator=(const ContentsWebView&) = delete;
   ~ContentsWebView() override;
@@ -55,6 +55,7 @@
  private:
   void UpdateBackgroundColor();
   StatusBubbleViews* status_bubble_;
+  bool transparent_;
 
   bool background_visible_ = true;
 
