diff -r -u --color up/chromium/components/component_updater/component_updater_service.h nw/chromium/components/component_updater/component_updater_service.h
--- up/chromium/components/component_updater/component_updater_service.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/component_updater/component_updater_service.h	2023-02-24 20:54:10.576504074 +0000
@@ -24,6 +24,10 @@
 class ComponentsHandler;
 class PluginObserver;
 
+namespace extensions {
+class NwAppUpdateComponentFunction;
+}
+
 namespace policy {
 class ComponentUpdaterPolicyTest;
 }
@@ -199,6 +203,7 @@
   virtual ~OnDemandUpdater() = default;
 
  private:
+  friend class extensions::NwAppUpdateComponentFunction;
   friend class OnDemandTester;
   friend class policy::ComponentUpdaterPolicyTest;
   friend class ::ComponentsHandler;
