diff -r -u --color up/chromium/components/os_crypt/keychain_password_mac.mm nw/chromium/components/os_crypt/keychain_password_mac.mm
--- up/chromium/components/os_crypt/keychain_password_mac.mm	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/os_crypt/keychain_password_mac.mm	2023-02-24 20:54:11.148501437 +0000
@@ -31,8 +31,8 @@
 const char kDefaultServiceName[] = "Chrome Safe Storage";
 const char kDefaultAccountName[] = "Chrome";
 #else
-const char kDefaultServiceName[] = "Chromium Safe Storage";
-const char kDefaultAccountName[] = "Chromium";
+const char kDefaultServiceName[] = "NWJS Safe Storage";
+const char kDefaultAccountName[] = "nwjs";
 #endif
 
 // Generates a random password and adds it to the Keychain.  The added password
