diff -r -u --color up/chromium/components/safe_browsing/buildflags.gni nw/chromium/components/safe_browsing/buildflags.gni
--- up/chromium/components/safe_browsing/buildflags.gni	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/safe_browsing/buildflags.gni	2023-02-24 06:16:26.568397199 +0000
@@ -17,6 +17,6 @@
   } else if (is_android) {
     safe_browsing_mode = 2
   } else {
-    safe_browsing_mode = 1
+    safe_browsing_mode = 0
   }
 }
