diff -r -u --color up/chromium/components/keep_alive_registry/keep_alive_registry.cc nw/chromium/components/keep_alive_registry/keep_alive_registry.cc
--- up/chromium/components/keep_alive_registry/keep_alive_registry.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/keep_alive_registry/keep_alive_registry.cc	2023-02-24 20:54:10.972502248 +0000
@@ -118,7 +118,8 @@
 
 void KeepAliveRegistry::Register(KeepAliveOrigin origin,
                                  KeepAliveRestartOption restart) {
-  CHECK(!is_shutting_down_);
+  //nw1 crash when quit from dock
+//  CHECK(!is_shutting_down_);
 
   bool old_keeping_alive = IsKeepingAlive();
   bool old_restart_allowed = IsRestartAllowed();
