diff -r -u --color up/chromium/components/network_time/network_time_tracker.cc nw/chromium/components/network_time/network_time_tracker.cc
--- up/chromium/components/network_time/network_time_tracker.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/network_time/network_time_tracker.cc	2023-02-24 20:54:11.056501861 +0000
@@ -66,7 +66,7 @@
 #else
 BASE_FEATURE(kNetworkTimeServiceQuerying,
              "NetworkTimeServiceQuerying",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 #endif
 
 namespace {
