diff -r -u --color up/chromium/components/remote_cocoa/app_shim/views_nswindow_delegate.h nw/chromium/components/remote_cocoa/app_shim/views_nswindow_delegate.h
--- up/chromium/components/remote_cocoa/app_shim/views_nswindow_delegate.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/remote_cocoa/app_shim/views_nswindow_delegate.h	2023-02-24 20:12:08.080854587 +0000
@@ -54,10 +54,10 @@
 // a backing store). |notification| is optional and can be set when redirecting
 // a notification such as NSApplicationDidHideNotification.
 - (void)onWindowOrderChanged:(NSNotification*)notification;
-
 // Notify that the system control tint changed.
 - (void)onSystemControlTintChanged:(NSNotification*)notification;
 
+- (void)onWindowWillStartLiveResize:(NSNotification*)notification;
 // Called on the delegate of a modal sheet when its modal session ends.
 - (void)sheetDidEnd:(NSWindow*)sheet
          returnCode:(NSInteger)returnCode
