diff -r -u --color up/chromium/components/remote_cocoa/app_shim/native_widget_ns_window_bridge.h nw/chromium/components/remote_cocoa/app_shim/native_widget_ns_window_bridge.h
--- up/chromium/components/remote_cocoa/app_shim/native_widget_ns_window_bridge.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/remote_cocoa/app_shim/native_widget_ns_window_bridge.h	2023-02-24 06:16:26.540395543 +0000
@@ -143,6 +143,8 @@
   // being reordered in (or out of) the screen list.
   void OnVisibilityChanged();
 
+  void OnWindowWillStartLiveResize();
+
   // Called by the NSWindowDelegate when the system control tint changes.
   void OnSystemControlTintChanged();
 
@@ -257,6 +259,12 @@
   void SetCanAppearInExistingFullscreenSpaces(
       bool can_appear_in_existing_fullscreen_spaces) override;
   void SetMiniaturized(bool miniaturized) override;
+  void SetMaximized(bool maximized) override;
+  bool IsMaximized(bool* maximized) override;
+  void IsMaximized(IsMaximizedCallback callback) override;
+  void GetRestoredBounds(GetRestoredBoundsCallback callback) override;
+  bool GetRestoredBounds(gfx::Rect* bounds) override;
+  void SetRestoredBounds(const gfx::Rect& bounds) override;
   void SetSizeConstraints(const gfx::Size& min_size,
                           const gfx::Size& max_size,
                           bool is_resizable,
@@ -387,6 +395,8 @@
   // (and its compositor).
   gfx::Size content_dip_size_;
 
+  NSRect bounds_before_maximize_;
+
   // The size of the frame most recently *received from* the compositor. Note
   // that during resize (and showing new windows), this will lag behind
   // |content_dip_size_|, which is the frame size most recently *sent to* the
