diff -r -u --color up/chromium/components/metrics_services_manager/metrics_services_manager.cc nw/chromium/components/metrics_services_manager/metrics_services_manager.cc
--- up/chromium/components/metrics_services_manager/metrics_services_manager.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/metrics_services_manager/metrics_services_manager.cc	2023-02-24 20:12:07.536850036 +0000
@@ -1,6 +1,7 @@
 // Copyright 2014 The Chromium Authors
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
+#pragma clang diagnostic ignored "-Wunreachable-code"
 
 #include "components/metrics_services_manager/metrics_services_manager.h"
 
@@ -137,7 +138,7 @@
   DCHECK(thread_checker_.CalledOnValidThread());
   metrics::MetricsService* metrics = GetMetricsService();
 
-  if (metrics::IsMetricsRecordingOnlyEnabled()) {
+  if (true) { //metrics::IsMetricsRecordingOnlyEnabled()) {
     metrics->StartRecordingForTests();
     return;
   }
