diff -r -u --color up/chromium/components/guest_view/browser/guest_view_manager.cc nw/chromium/components/guest_view/browser/guest_view_manager.cc
--- up/chromium/components/guest_view/browser/guest_view_manager.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/guest_view/browser/guest_view_manager.cc	2023-02-24 20:12:07.284847927 +0000
@@ -512,10 +512,9 @@
 
   // The embedder is trying to access a guest with a negative or zero
   // instance ID.
-  if (guest_instance_id <= kInstanceIDNone) {
-    bad_access_key.Set("Nonpositive");
-    return false;
-  }
+  if (guest_instance_id <= kInstanceIDNone)
+    return true; // the 'alert' case will fail when keeps clicking
+                 // 'detach' otherwise
 
   // The embedder is trying to access an instance ID that has not yet been
   // allocated by GuestViewManager. This could cause instance ID
@@ -540,6 +539,7 @@
 
   // MimeHandlerViewGuests (PDF) may be embedded in a cross-process frame.
   // Other than MimeHandlerViewGuest, all other guest types are only permitted
+#if 0
   // to run in the main frame or its local subframes.
   const int allowed_embedder_render_process_id =
       guest_view->CanBeEmbeddedInsideCrossProcessFrames()
@@ -553,7 +553,7 @@
     bad_access_key.Set("Bad embedder process");
     return false;
   }
-
+#endif
   return true;
 }
 
