diff -r -u --color up/chromium/components/storage_monitor/portable_device_watcher_win.cc nw/chromium/components/storage_monitor/portable_device_watcher_win.cc
--- up/chromium/components/storage_monitor/portable_device_watcher_win.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/components/storage_monitor/portable_device_watcher_win.cc	2023-02-24 06:16:26.752408086 +0000
@@ -32,7 +32,7 @@
 namespace {
 
 // Name of the client application that communicates with the MTP device.
-const wchar_t kClientName[] = L"Chromium";
+const wchar_t kClientName[] = L"nwjs";
 
 // Returns true if |data| represents a class of portable devices.
 bool IsPortableDeviceStructure(LPARAM data) {
