diff -r -u --color up/chromium/components/media_router/browser/presentation/presentation_service_delegate_impl.cc nw/chromium/components/media_router/browser/presentation/presentation_service_delegate_impl.cc
--- up/chromium/components/media_router/browser/presentation/presentation_service_delegate_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/components/media_router/browser/presentation/presentation_service_delegate_impl.cc	2023-02-24 20:12:07.500849734 +0000
@@ -472,7 +472,9 @@
     const content::PresentationRequest& request,
     content::PresentationConnectionCallback success_cb,
     content::PresentationConnectionErrorCallback error_cb) {
+#if defined(NWJS_SDK)
   const auto& render_frame_host_id = request.render_frame_host_id;
+#endif
   const auto& presentation_urls = request.presentation_urls;
   if (presentation_urls.empty()) {
     std::move(error_cb).Run(PresentationError(
@@ -485,6 +487,8 @@
                           "Invalid presentation URL."));
     return;
   }
+
+#if defined(NWJS_SDK)
   auto presentation_context = std::make_unique<StartPresentationContext>(
       request,
       base::BindOnce(
@@ -500,6 +504,7 @@
       MediaRouterDialogController::GetOrCreateForWebContents(&GetWebContents());
   controller->ShowMediaRouterDialogForPresentation(
       std::move(presentation_context));
+#endif
 }
 
 void PresentationServiceDelegateImpl::ReconnectPresentation(
