diff -r -u --color up/chromium/content/test/test_render_frame.cc nw/chromium/content/test/test_render_frame.cc
--- up/chromium/content/test/test_render_frame.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/test/test_render_frame.cc	2023-02-24 20:12:09.884869681 +0000
@@ -121,6 +121,8 @@
 
  protected:
   // mojom::FrameHost:
+  void SetNodeJS(bool) override {}
+  void SetContextCreated(bool) override {}
   void CreateNewWindow(mojom::CreateNewWindowParamsPtr,
                        CreateNewWindowCallback) override {
     NOTREACHED() << "We should never dispatch to the service side signature.";
