diff -r -u --color up/chromium/content/renderer/render_thread_impl.h nw/chromium/content/renderer/render_thread_impl.h
--- up/chromium/content/renderer/render_thread_impl.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/renderer/render_thread_impl.h	2023-02-24 06:16:28.216494706 +0000
@@ -181,6 +181,7 @@
     return resource_request_sender_delegate_;
   }
   int PostTaskToAllWebWorkers(base::RepeatingClosure closure) override;
+  bool ResolveProxy(const GURL& url, std::string* proxy_list) override;
   base::WaitableEvent* GetShutdownEvent() override;
   int32_t GetClientId() override;
   void SetRendererProcessType(
