diff -r -u --color up/chromium/content/child/child_thread_impl.cc nw/chromium/content/child/child_thread_impl.cc
--- up/chromium/content/child/child_thread_impl.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/child/child_thread_impl.cc	2023-02-24 06:16:28.084486896 +0000
@@ -673,7 +673,7 @@
   // Requires base::PowerMonitor to be initialized first.
   power_scheduler::PowerModeArbiter::GetInstance()->OnThreadPoolAvailable();
 
-#if BUILDFLAG(IS_POSIX)
+#if 0 //BUILDFLAG(IS_POSIX)
   // Check that --process-type is specified so we don't do this in unit tests
   // and single-process mode.
   if (base::CommandLine::ForCurrentProcess()->HasSwitch(
