diff -r -u --color up/chromium/content/browser/web_contents/web_contents_impl.h nw/chromium/content/browser/web_contents/web_contents_impl.h
--- up/chromium/content/browser/web_contents/web_contents_impl.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/browser/web_contents/web_contents_impl.h	2023-02-24 06:16:28.044484529 +0000
@@ -693,7 +693,7 @@
                          int main_frame_widget_route_id,
                          WindowOpenDisposition disposition,
                          const blink::mojom::WindowFeatures& window_features,
-                         bool user_gesture) override;
+                         bool user_gesture, std::string manifest) override;
   void PrimaryMainDocumentElementAvailable() override;
   void PassiveInsecureContentFound(const GURL& resource_url) override;
   bool ShouldAllowRunningInsecureContent(bool allowed_per_prefs,
