diff -r -u --color up/chromium/content/browser/media/capture/desktop_streams_registry_impl.h nw/chromium/content/browser/media/capture/desktop_streams_registry_impl.h
--- up/chromium/content/browser/media/capture/desktop_streams_registry_impl.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/browser/media/capture/desktop_streams_registry_impl.h	2023-02-24 06:16:27.844472696 +0000
@@ -40,7 +40,7 @@
       int render_frame_id,
       const url::Origin& origin,
       std::string* extension_name,
-      const DesktopStreamRegistryType type) override;
+      const DesktopStreamRegistryType type, bool nodejs = false) override;
 
  private:
   // Type used to store list of accepted desktop media streams.
