diff -r -u --color up/chromium/content/common/user_agent.cc nw/chromium/content/common/user_agent.cc
--- up/chromium/content/common/user_agent.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/common/user_agent.cc	2023-02-24 20:12:09.428865866 +0000
@@ -333,7 +333,17 @@
   return model;
 }
 
-#if BUILDFLAG(IS_ANDROID)
+std::string BuildOSInfo() {
+  std::string os_info;
+  base::StringAppendF(
+      &os_info,
+      "%s%s",
+      GetUserAgentPlatform().c_str(),
+      BuildOSCpuInfo(IncludeAndroidBuildNumber::Exclude,
+                     IncludeAndroidModel::Exclude).c_str());
+  return os_info;
+}
+
 std::string BuildUserAgentFromProductAndExtraOSInfo(
     const std::string& product,
     const std::string& extra_os_info,
@@ -346,6 +356,7 @@
   return BuildUserAgentFromOSAndProduct(os_info, product);
 }
 
+#if 0
 std::string GetAndroidOSInfo(
     IncludeAndroidBuildNumber include_android_build_number,
     IncludeAndroidModel include_android_model) {
