diff -r -u --color up/chromium/content/public/renderer/render_thread.h nw/chromium/content/public/renderer/render_thread.h
--- up/chromium/content/public/renderer/render_thread.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/public/renderer/render_thread.h	2023-02-24 20:12:09.488866368 +0000
@@ -19,6 +19,8 @@
 #include "third_party/blink/public/platform/web_string.h"
 #include "third_party/perfetto/include/perfetto/tracing/traced_proto.h"
 
+class GURL;
+
 namespace base {
 class UnguessableToken;
 class WaitableEvent;
@@ -98,6 +100,11 @@
   // Post task to all worker threads. Returns number of workers.
   virtual int PostTaskToAllWebWorkers(base::RepeatingClosure closure) = 0;
 
+  // Resolve the proxy servers to use for a given url. On success true is
+  // returned and |proxy_list| is set to a PAC string containing a list of
+  // proxy servers.
+  virtual bool ResolveProxy(const GURL& url, std::string* proxy_list) = 0;
+
   // Gets the shutdown event for the process.
   virtual base::WaitableEvent* GetShutdownEvent() = 0;
 
