diff -r -u --color up/chromium/content/ppapi_plugin/ppapi_plugin_main.cc nw/chromium/content/ppapi_plugin/ppapi_plugin_main.cc
--- up/chromium/content/ppapi_plugin/ppapi_plugin_main.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/ppapi_plugin/ppapi_plugin_main.cc	2023-02-24 06:16:28.104488079 +0000
@@ -148,7 +148,7 @@
   gin::V8Initializer::LoadV8Snapshot();
 #endif
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#if 0 //BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
   sandbox::policy::SandboxLinux::GetInstance()->InitializeSandbox(
       sandbox::policy::SandboxTypeFromCommandLine(command_line),
       sandbox::policy::SandboxLinux::PreSandboxHook(),
