diff -r -u --color up/chromium/content/renderer/agent_scheduling_group.cc nw/chromium/content/renderer/agent_scheduling_group.cc
--- up/chromium/content/renderer/agent_scheduling_group.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/renderer/agent_scheduling_group.cc	2023-02-24 06:16:28.192493286 +0000
@@ -336,7 +336,7 @@
 
     if (!local_params->previous_frame_token) {
       // Create a local non-provisional main frame.
-      RenderFrameImpl::CreateMainFrame(
+      RenderFrameImpl* main_frame = RenderFrameImpl::CreateMainFrame(
           *this, web_view, opener_frame,
           /*is_for_nested_main_frame=*/params->type !=
               mojom::ViewWidgetType::kTopLevel,
@@ -344,6 +344,7 @@
               mojom::ViewWidgetType::kFencedFrame,
           std::move(params->replication_state),
           params->devtools_main_frame_token, std::move(local_params));
+      main_frame->SetSkipBlockingParser(params->skip_blocking_parser);
     } else {
       // Create a local provisional main frame and a placeholder RemoteFrame as
       // a placeholder main frame for the new WebView. This can only happen for
