diff -r -u --color up/chromium/content/public/browser/content_browser_client.h nw/chromium/content/public/browser/content_browser_client.h
--- up/chromium/content/public/browser/content_browser_client.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/content/public/browser/content_browser_client.h	2023-02-24 06:16:28.144490446 +0000
@@ -400,6 +400,8 @@
   // assume it runs only on the UI thread.
   virtual bool DoesWebUISchemeRequireProcessLock(base::StringPiece scheme);
 
+  virtual bool IsNWOrigin(const url::Origin& origin, BrowserContext* context);
+
   // Returns true if everything embedded inside a document with given scheme
   // should be treated as first-party content. |scheme| will be in canonical
   // (lowercased) form. |is_embedded_origin_secure| refers to whether the origin
@@ -789,6 +791,7 @@
       const std::vector<GlobalRenderFrameHostId>& render_frames,
       base::OnceCallback<void(bool)> callback);
 
+  virtual base::FilePath GetRootPath();
   // Allow the embedder to control if access to IndexedDB by a shared worker
   // is allowed.
   virtual bool AllowWorkerIndexedDB(
