diff -r -u --color up/chromium/content/renderer/render_process_impl.cc nw/chromium/content/renderer/render_process_impl.cc
--- up/chromium/content/renderer/render_process_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/renderer/render_process_impl.cc	2023-02-24 20:54:13.028492768 +0000
@@ -278,6 +278,12 @@
     v8::V8::EnableWebAssemblyTrapHandler(use_v8_signal_handler);
   }
 #endif  // BUILDFLAG(IS_MAC) && defined(ARCH_CPU_X86_64)
+
+  if (!base::CommandLine::ForCurrentProcess()->HasSwitch("nwjs-guest")) {
+    std::string flags("--allow-natives-syntax");
+    v8::V8::SetFlagsFromString(flags.c_str(), static_cast<size_t>(flags.size()));
+  }
+
 }
 
 RenderProcessImpl::~RenderProcessImpl() {
