diff -r -u --color up/chromium/content/browser/site_info.h nw/chromium/content/browser/site_info.h
--- up/chromium/content/browser/site_info.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/browser/site_info.h	2023-02-24 20:12:09.364865330 +0000
@@ -105,7 +105,7 @@
   // only translates an origin into a site (i.e., scheme and eTLD+1) and is
   // used internally by GetSiteForURLInternal().  For making process model
   // decisions, Create() should be used instead.
-  static GURL GetSiteForOrigin(const url::Origin& origin);
+  static GURL GetSiteForOrigin(const url::Origin& origin, const GURL& real_url = GURL());
 
   // Returns a StoragePartitionConfig for the specified URL.
   // If |is_site_url| is set to true, then |url| MUST be a site URL that
