diff -r -u --color up/chromium/content/renderer/renderer_blink_platform_impl.h nw/chromium/content/renderer/renderer_blink_platform_impl.h
--- up/chromium/content/renderer/renderer_blink_platform_impl.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/renderer/renderer_blink_platform_impl.h	2023-02-24 20:54:13.032492749 +0000
@@ -183,7 +183,7 @@
       const blink::BrowserInterfaceBrokerProxy&) override;
   void DidStartWorkerThread() override;
   void WillStopWorkerThread() override;
-  void WorkerContextCreated(const v8::Local<v8::Context>& worker) override;
+  void WorkerContextCreated(const v8::Local<v8::Context>& worker, bool, const std::string&) override;
   bool AllowScriptExtensionForServiceWorker(
       const blink::WebSecurityOrigin& script_origin) override;
   blink::ProtocolHandlerSecurityLevel GetProtocolHandlerSecurityLevel()
