diff -r -u --color up/chromium/content/public/test/browser_test_base.cc nw/chromium/content/public/test/browser_test_base.cc
--- up/chromium/content/public/test/browser_test_base.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/public/test/browser_test_base.cc	2023-02-24 20:12:09.508866535 +0000
@@ -316,6 +316,8 @@
       switches::kIPCConnectionTimeout,
       base::NumberToString(TestTimeouts::action_max_timeout().InSeconds()));
 
+  command_line->AppendSwitch("nwjs-test-mode");
+
   command_line->AppendSwitch(switches::kDomAutomationController);
 
   // It is sometimes useful when looking at browser test failures to know which
