diff -r -u --color up/chromium/content/browser/devtools/protocol/emulation_handler.cc nw/chromium/content/browser/devtools/protocol/emulation_handler.cc
--- up/chromium/content/browser/devtools/protocol/emulation_handler.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/browser/devtools/protocol/emulation_handler.cc	2023-02-24 20:12:09.160863623 +0000
@@ -602,13 +602,18 @@
   if (touch_emulation_enabled_) {
     if (auto* touch_emulator =
             host_->GetRenderWidgetHost()->GetTouchEmulator()) {
+      touch_emulator->set_rfh_limit(host_->GetWeakPtr());
       touch_emulator->Enable(
           TouchEmulator::Mode::kEmulatingTouchFromMouse,
           TouchEmulationConfigurationToType(touch_emulation_configuration_));
     }
   } else {
-    if (auto* touch_emulator = host_->GetRenderWidgetHost()->GetTouchEmulator())
+    if (auto* touch_emulator =
+            host_->GetRenderWidgetHost()->GetTouchEmulator()) {
       touch_emulator->Disable();
+      touch_emulator->set_rfh_limit(nullptr);
+    }
+
   }
   GetWebContents()->SetForceDisableOverscrollContent(touch_emulation_enabled_);
 }
