diff -r -u --color up/chromium/content/test/test_web_contents.cc nw/chromium/content/test/test_web_contents.cc
--- up/chromium/content/test/test_web_contents.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/content/test/test_web_contents.cc	2023-02-24 20:54:13.376491163 +0000
@@ -367,7 +367,7 @@
     int route_id,
     WindowOpenDisposition disposition,
     const blink::mojom::WindowFeatures& window_features,
-    bool user_gesture) {}
+    bool user_gesture, std::string) {}
 
 void TestWebContents::ShowCreatedWidget(int process_id,
                                         int route_id,
