diff -r -u --color up/node/deps/nghttp2/lib/includes/nghttp2/nghttp2.h nw/node/deps/nghttp2/lib/includes/nghttp2/nghttp2.h
--- up/node/deps/nghttp2/lib/includes/nghttp2/nghttp2.h	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/deps/nghttp2/lib/includes/nghttp2/nghttp2.h	2023-02-19 17:37:30.471593853 +0000
@@ -36,6 +36,10 @@
 #  define __has_declspec_attribute(x) 0
 #endif
 
+#if defined(WIN32)
+#define ssize_t ptrdiff_t
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
