diff -r -u --color up/node/deps/cares/src/lib/ares_process.c nw/node/deps/cares/src/lib/ares_process.c
--- up/node/deps/cares/src/lib/ares_process.c	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/deps/cares/src/lib/ares_process.c	2023-02-19 17:37:30.243593415 +0000
@@ -509,7 +509,7 @@
           else
             fromlen = sizeof(from.sa6);
           count = socket_recvfrom(channel, server->udp_socket, (void *)buf,
-                                  sizeof(buf), 0, &from.sa, &fromlen);
+                                  sizeof(buf), 0, &from.sa, (socklen_t *)&fromlen);
         }
 
         if (count == -1 && try_again(SOCKERRNO))
