diff -r -u --color up/node/deps/uv/src/uv-common.h nw/node/deps/uv/src/uv-common.h
--- up/node/deps/uv/src/uv-common.h	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/deps/uv/src/uv-common.h	2023-02-25 19:08:12.225999565 +0000
@@ -323,6 +323,7 @@
 # define UV_REQ_INIT(req, typ)                                                \
   do {                                                                        \
     (req)->type = (typ);                                                      \
+    (req)->next_req = NULL;                                                   \
     (req)->u.io.overlapped.Internal = 0;  /* SET_REQ_SUCCESS() */             \
   }                                                                           \
   while (0)
