diff -r -u --color up/chromium/extensions/browser/guest_view/extensions_guest_view_manager_delegate.cc nw/chromium/extensions/browser/guest_view/extensions_guest_view_manager_delegate.cc
--- up/chromium/extensions/browser/guest_view/extensions_guest_view_manager_delegate.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/guest_view/extensions_guest_view_manager_delegate.cc	2023-02-24 20:54:13.712489614 +0000
@@ -76,9 +76,11 @@
   if (!owner)
     return;  // Could happen at tab shutdown.
 
+  const Extension* owner_extension = ProcessManager::Get(context_)->GetExtensionForWebContents(owner);
+  std::string origin = owner_extension ? owner_extension->id() : guest->owner_host();
   EventRouter::DispatchEventToSender(
       owner->GetPrimaryMainFrame()->GetProcess(), guest->browser_context(),
-      guest->owner_host(), histogram_value, event_name,
+      origin, histogram_value, event_name,
       content::ChildProcessHost::kInvalidUniqueID, extensions::kMainThreadId,
       blink::mojom::kInvalidServiceWorkerVersionId, std::move(event_args),
       std::move(info));
