diff -r -u --color up/chromium/extensions/renderer/programmatic_script_injector.cc nw/chromium/extensions/renderer/programmatic_script_injector.cc
--- up/chromium/extensions/renderer/programmatic_script_injector.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/renderer/programmatic_script_injector.cc	2023-02-24 20:54:13.772489337 +0000
@@ -39,6 +39,10 @@
   return mojom::InjectionType::kProgrammaticScript;
 }
 
+bool ProgrammaticScriptInjector::ShouldExecuteInMainWorld() const {
+  return params_->in_main_world;
+}
+
 blink::mojom::UserActivationOption ProgrammaticScriptInjector::IsUserGesture()
     const {
   DCHECK(params_->injection->is_js());
