diff -r -u --color up/chromium/extensions/renderer/bindings/event_emitter.h nw/chromium/extensions/renderer/bindings/event_emitter.h
--- up/chromium/extensions/renderer/bindings/event_emitter.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/renderer/bindings/event_emitter.h	2023-02-24 06:16:29.112547720 +0000
@@ -73,7 +73,9 @@
   void RemoveListener(gin::Arguments* arguments);
   bool HasListener(v8::Local<v8::Function> function);
   bool HasListeners();
+  void GetListeners(gin::Arguments* arguments);
   void Dispatch(gin::Arguments* arguments);
+  void DispatchNW(gin::Arguments* arguments);
 
   // Dispatches an event synchronously to listeners, returning the result.
   v8::Local<v8::Value> DispatchSync(v8::Local<v8::Context> context,
