diff -r -u --color up/chromium/extensions/common/api/app_current_window_internal.idl nw/chromium/extensions/common/api/app_current_window_internal.idl
--- up/chromium/extensions/common/api/app_current_window_internal.idl	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/common/api/app_current_window_internal.idl	2023-02-24 06:16:29.068545116 +0000
@@ -47,6 +47,7 @@
     static void setBounds(DOMString boundsType, Bounds bounds);
     static void setSizeConstraints(DOMString boundsType,
                                    SizeConstraints constraints);
+    static void setResizable(boolean flag);
     static void setIcon(DOMString icon_url);
     static void setShape(Region region);
     static void setAlwaysOnTop(boolean always_on_top);
@@ -57,6 +58,8 @@
   interface Events {
     static void onClosed();
     static void onBoundsChanged();
+    static void onResized();
+    static void onMoved();
     static void onFullscreened();
     static void onMinimized();
     static void onMaximized();
