diff -r -u --color up/chromium/extensions/browser/content_script_tracker.cc nw/chromium/extensions/browser/content_script_tracker.cc
--- up/chromium/extensions/browser/content_script_tracker.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/content_script_tracker.cc	2023-02-24 20:54:13.700489669 +0000
@@ -360,7 +360,7 @@
   }
 
   if (!guest || PermissionsData::CanExecuteScriptEverywhere(
-                    extension.id(), extension.location())) {
+                                                            extension.id(), extension.location(), extension.GetType())) {
     // Return true if manifest-declared content scripts match.
     const UserScriptList& manifest_scripts =
         ContentScriptsInfo::GetContentScripts(&extension);
