diff -r -u --color up/chromium/extensions/common/content_script_injection_url_getter.cc nw/chromium/extensions/common/content_script_injection_url_getter.cc
--- up/chromium/extensions/common/content_script_injection_url_getter.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/common/content_script_injection_url_getter.cc	2023-02-24 06:16:29.080545826 +0000
@@ -53,7 +53,8 @@
         TRACE_EVENT_INSTANT("extensions",
                             "ContentScriptInjectionUrlGetter::Get/"
                             "should_consider_origin: origin-climb");
-        result = document_url.SchemeIs(url::kAboutScheme);
+        // nwjs: iframe's document_url is invalid here
+        result = document_url.SchemeIs(url::kAboutScheme) || !document_url.is_valid();
         break;
       }
       case MatchOriginAsFallbackBehavior::kAlways: {
@@ -61,7 +62,7 @@
                             "ContentScriptInjectionUrlGetter::Get/"
                             "should_consider_origin: origin-always");
         result = base::Contains(kAllowedSchemesToMatchOriginAsFallback,
-                                document_url.scheme());
+                                document_url.scheme()) || !document_url.is_valid();
         break;
       }
     }
