diff -r -u --color up/chromium/extensions/common/constants.h nw/chromium/extensions/common/constants.h
--- up/chromium/extensions/common/constants.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/common/constants.h	2023-02-24 06:16:29.080545826 +0000
@@ -17,6 +17,7 @@
 
 // The name of the manifest inside an extension.
 EXTENSIONS_EXPORT extern const base::FilePath::CharType kManifestFilename[];
+EXTENSIONS_EXPORT extern const base::FilePath::CharType kNWJSManifestFilename[];
 
 // The name of the differential fingerprint file inside an extension.
 EXTENSIONS_EXPORT extern const base::FilePath::CharType
@@ -67,6 +68,7 @@
 // The filename to use for a background page generated from
 // background.scripts.
 EXTENSIONS_EXPORT extern const char kGeneratedBackgroundPageFilename[];
+EXTENSIONS_EXPORT extern const char kNWJSDefaultAppJS[];
 
 // The URL piece between the extension ID and favicon URL.
 EXTENSIONS_EXPORT extern const char kFaviconSourcePath[];
