diff -r -u --color up/chromium/extensions/renderer/user_script_set.cc nw/chromium/extensions/renderer/user_script_set.cc
--- up/chromium/extensions/renderer/user_script_set.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/renderer/user_script_set.cc	2023-02-24 06:16:29.128548666 +0000
@@ -203,12 +203,28 @@
     injection_host = std::make_unique<WebUIInjectionHost>(host_id_);
   }
 
-  GURL effective_document_url =
+  // nwjs#6324: move the upstream logic of GetEffectiveDocumentURL() here
+  // in upstream is not matched first with empty invalid url for iframe
+  GURL effective_document_url;
+  bool match_about_blank = false; //copied from 6d96d3d1cb83d
+  switch (script->match_origin_as_fallback()) {
+  case MatchOriginAsFallbackBehavior::kAlways:
+  case MatchOriginAsFallbackBehavior::kMatchForAboutSchemeAndClimbTree:
+    match_about_blank = true;
+    break;
+  case MatchOriginAsFallbackBehavior::kNever:
+    break;  // `false` is correct for |match_about_blank|.
+  }
+
+  if (!match_about_blank || !document_url.SchemeIs(url::kAboutScheme))
+    effective_document_url = document_url;
+  else
+    effective_document_url =
       ScriptContext::GetEffectiveDocumentURLForInjection(
           web_frame, document_url, script->match_origin_as_fallback());
 
   bool is_subframe = !web_frame->IsOutermostMainFrame();
-  if (!script->MatchesDocument(effective_document_url, is_subframe))
+  if (!script->MatchesDocument(effective_document_url, is_subframe) || !effective_document_url.is_valid())
     return injection;
 
   // Extension dynamic scripts are treated as declarative scripts and should use
