diff -r -u --color up/chromium/extensions/renderer/user_script_injector.h nw/chromium/extensions/renderer/user_script_injector.h
--- up/chromium/extensions/renderer/user_script_injector.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/renderer/user_script_injector.h	2023-02-24 06:16:29.128548666 +0000
@@ -46,6 +46,7 @@
   // ScriptInjector implementation.
   mojom::InjectionType script_type() const override;
   blink::mojom::UserActivationOption IsUserGesture() const override;
+  bool ShouldExecuteInMainWorld() const override;
   mojom::ExecutionWorld GetExecutionWorld() const override;
   mojom::CSSOrigin GetCssOrigin() const override;
   mojom::CSSInjection::Operation GetCSSInjectionOperation() const override;
