diff -r -u --color up/chromium/extensions/renderer/runtime_custom_bindings.cc nw/chromium/extensions/renderer/runtime_custom_bindings.cc
--- up/chromium/extensions/renderer/runtime_custom_bindings.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/renderer/runtime_custom_bindings.cc	2023-02-24 06:16:29.124548430 +0000
@@ -5,6 +5,7 @@
 #include "extensions/renderer/runtime_custom_bindings.h"
 
 #include <stdint.h>
+#include "extensions/renderer/script_context_set.h"
 
 #include <memory>
 
@@ -54,8 +55,13 @@
     CHECK_EQ("ALL", view_type_string);
 
   const std::string& extension_id = context()->GetExtensionID();
-  if (extension_id.empty())
-    return;
+  // id is empty while calling from external page. we want to do
+  // this for window controlling. note the case that there are
+  // multiple extensions in the process, e.g. the automation extension
+  // for chromedriver
+
+  // if (extension_id.empty())
+  //   return;
 
   // We ignore iframes here. (Returning subframes can cause broken behavior by
   // treating an app window's iframe as its main frame, and maybe other
