diff -r -u --color up/chromium/extensions/browser/extension_function_dispatcher.h nw/chromium/extensions/browser/extension_function_dispatcher.h
--- up/chromium/extensions/browser/extension_function_dispatcher.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/browser/extension_function_dispatcher.h	2023-02-24 06:16:29.040543460 +0000
@@ -72,6 +72,12 @@
       content::BrowserContext* browser_context);
   ~ExtensionFunctionDispatcher();
 
+  void DispatchSync(mojom::RequestParamsPtr params,
+                    bool* success,
+                    base::Value::List* response,
+                    std::string* error,
+                    content::RenderFrameHost* render_frame_host,
+                    int render_process_id);
   // Dispatches a request and the response is sent in |callback| that is a reply
   // of mojom::LocalFrameHost::Request.
   void Dispatch(mojom::RequestParamsPtr params,
@@ -147,7 +153,12 @@
       const mojom::RequestParams& params,
       content::RenderFrameHost* render_frame_host,
       int render_process_id,
-      ExtensionFunction::ResponseCallback callback);
+      ExtensionFunction::ResponseCallback callback,
+      bool sync = false,
+      bool* success = nullptr,
+      base::Value::List* response = nullptr,
+      std::string* error = nullptr
+                                    );
 
   void RemoveWorkerCallbacksForProcess(int render_process_id);
 
