diff -r -u --color up/chromium/extensions/browser/process_manager.cc nw/chromium/extensions/browser/process_manager.cc
--- up/chromium/extensions/browser/process_manager.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/extensions/browser/process_manager.cc	2023-02-24 06:16:29.056544406 +0000
@@ -324,7 +324,7 @@
 
 scoped_refptr<content::SiteInstance> ProcessManager::GetSiteInstanceForURL(
     const GURL& url) {
-  return site_instance_->GetRelatedSiteInstance(url);
+  return site_instance_->GetRelatedSiteInstance(url, false);
 }
 
 const ProcessManager::FrameSet ProcessManager::GetAllFrames() const {
@@ -842,6 +842,8 @@
 
 void ProcessManager::OnLazyBackgroundPageIdle(const std::string& extension_id,
                                               uint64_t sequence_id) {
+  // bg page will be destroyed if main points to remote page NWJS#5282
+#if 0
   ExtensionHost* host = GetBackgroundHostForExtension(extension_id);
   if (host && !background_page_data_[extension_id].is_closing &&
       sequence_id == background_page_data_[extension_id].close_sequence_id) {
@@ -860,6 +862,7 @@
           extension_id, sequence_id));
     }
   }
+#endif
 }
 
 void ProcessManager::OnLazyBackgroundPageActive(
