diff -r -u --color up/chromium/extensions/browser/content_verifier/content_hash.cc nw/chromium/extensions/browser/content_verifier/content_hash.cc
--- up/chromium/extensions/browser/content_verifier/content_hash.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/browser/content_verifier/content_hash.cc	2023-02-24 20:12:10.236872626 +0000
@@ -1,6 +1,7 @@
 // Copyright 2018 The Chromium Authors
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
+#pragma clang diagnostic ignored "-Wunreachable-code"
 
 #include "extensions/browser/content_verifier/content_hash.h"
 
@@ -53,7 +54,7 @@
       VerifiedContents::CreateFromFile(key.verifier_key,
                                        verified_contents_path);
   if (!verified_contents) {
-    if (delete_invalid_file && !base::DeleteFile(verified_contents_path)) {
+    if (delete_invalid_file && false && !base::DeleteFile(verified_contents_path)) {
       LOG(WARNING) << "Failed to delete " << verified_contents_path.value();
     }
     return nullptr;
