diff -r -u --color up/chromium/extensions/common/url_pattern.cc nw/chromium/extensions/common/url_pattern.cc
--- up/chromium/extensions/common/url_pattern.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/extensions/common/url_pattern.cc	2023-02-24 20:54:13.756489411 +0000
@@ -377,7 +377,7 @@
   spec_.clear();
   scheme_.assign(scheme.data(), scheme.size());
   if (scheme_ == "*") {
-    valid_schemes_ &= (SCHEME_HTTP | SCHEME_HTTPS);
+    valid_schemes_ &= (SCHEME_HTTP | SCHEME_HTTPS | SCHEME_EXTENSION | SCHEME_FILE);
   } else if (!IsValidScheme(scheme_)) {
     return false;
   }
