diff -r -u --color up/node/lib/internal/bootstrap/switches/is_main_thread.js nw/node/lib/internal/bootstrap/switches/is_main_thread.js
--- up/node/lib/internal/bootstrap/switches/is_main_thread.js	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/lib/internal/bootstrap/switches/is_main_thread.js	2023-02-19 17:37:33.247599187 +0000
@@ -124,6 +124,18 @@
   }
 }
 
+function createWritableDummyStream(fd) {
+  var DummyStream = require('dummystream');
+  var stream = new DummyStream();
+  stream.fd = fd;
+
+  stream._isStdio = true;
+  stream.isTTY = false;
+
+  return stream;
+
+}
+
 let stdin;
 let stdout;
 let stderr;
@@ -147,7 +159,10 @@
 
 function getStdout() {
   if (stdout) return stdout;
-  stdout = createWritableStdioStream(1);
+  if (process.platform === 'win32')
+    stdout = createWritableDummyStream(1);
+  else
+    stdout = createWritableStdioStream(1);
   stdout.destroySoon = stdout.destroy;
   // Override _destroy so that the fd is never actually closed.
   stdoutDestroy = stdout._destroy;
@@ -169,6 +184,9 @@
 
 function getStderr() {
   if (stderr) return stderr;
+  if (process.platform === 'win32')
+    stderr = createWritableDummyStream(2);
+  else
   stderr = createWritableStdioStream(2);
   stderr.destroySoon = stderr.destroy;
   stderrDestroy = stderr._destroy;
