diff -r -u --color up/node/lib/internal/process/execution.js nw/node/lib/internal/process/execution.js
--- up/node/lib/internal/process/execution.js	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/lib/internal/process/execution.js	2023-02-25 19:08:13.798006601 +0000
@@ -157,18 +157,8 @@
     if (exceptionHandlerState.captureFn !== null) {
       exceptionHandlerState.captureFn(er);
     } else if (!process.emit('uncaughtException', er, type)) {
-      // If someone handled it, then great. Otherwise, die in C++ land
-      // since that means that we'll exit the process, emit the 'exit' event.
-      try {
-        if (!process._exiting) {
-          process._exiting = true;
-          process.exitCode = kGenericUserError;
-          process.emit('exit', kGenericUserError);
-        }
-      } catch {
-        // Nothing to be done about it at this point.
-      }
-      return false;
+      // NWJS#6153: prevent the crash
+      // return true;
     }
 
     // If we handled an error, then make sure any ticks get processed
