diff -r -u --color up/node/lib/internal/modules/helpers.js nw/node/lib/internal/modules/helpers.js
--- up/node/lib/internal/modules/helpers.js	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/lib/internal/modules/helpers.js	2023-02-25 19:08:13.794006579 +0000
@@ -45,8 +45,6 @@
   cjsConditions = new SafeSet([
     'require',
     'node',
-    ...addonConditions,
-    ...userConditions,
   ]);
 }
 
