diff -r -u --color up/node/lib/internal/source_map/source_map_cache.js nw/node/lib/internal/source_map/source_map_cache.js
--- up/node/lib/internal/source_map/source_map_cache.js	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/lib/internal/source_map/source_map_cache.js	2023-02-25 19:08:13.802006623 +0000
@@ -50,6 +50,8 @@
 }
 
 function setSourceMapsEnabled(val) {
+  if (val === undefined)
+    val = false;
   validateBoolean(val, 'val');
 
   const {
