diff -r -u --color up/node/lib/internal/modules/run_main.js nw/node/lib/internal/modules/run_main.js
--- up/node/lib/internal/modules/run_main.js	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/lib/internal/modules/run_main.js	2023-02-19 17:37:33.255599202 +0000
@@ -34,7 +34,7 @@
    * (or an empty list when none have been registered).
    */
   const userImports = getOptionValue('--import');
-  if (userLoaders.length > 0 || userImports.length > 0)
+  if ((userLoaders && userLoaders.length > 0) || (userImports && userImports.length > 0))
     return true;
   const { readPackageScope } = require('internal/modules/cjs/loader');
   // Determine the module format of the main
