diff -r -u --color up/node/lib/internal/modules/esm/utils.js nw/node/lib/internal/modules/esm/utils.js
--- up/node/lib/internal/modules/esm/utils.js	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/lib/internal/modules/esm/utils.js	2023-02-25 19:08:13.794006579 +0000
@@ -48,8 +48,6 @@
   defaultConditions = ObjectFreeze([
     'node',
     'import',
-    ...addonConditions,
-    ...userConditions,
   ]);
   defaultConditionsSet = new SafeSet(defaultConditions);
 }
@@ -93,8 +91,8 @@
   initializeDefaultConditions();
   // Setup per-isolate callbacks that locate data or callbacks that we keep
   // track of for different ESM modules.
-  setInitializeImportMetaObjectCallback(initializeImportMetaObject);
-  setImportModuleDynamicallyCallback(importModuleDynamicallyCallback);
+  //setInitializeImportMetaObjectCallback(initializeImportMetaObject);
+  //setImportModuleDynamicallyCallback(importModuleDynamicallyCallback);
 }
 
 module.exports = {
