diff -r -u --color up/node/lib/internal/bootstrap/browser.js nw/node/lib/internal/bootstrap/browser.js
--- up/node/lib/internal/bootstrap/browser.js	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/lib/internal/bootstrap/browser.js	2023-02-25 19:08:13.786006535 +0000
@@ -1,5 +1,7 @@
 'use strict';
 
+if (process.__nwjs) {
+} else {
 const {
   ObjectDefineProperty,
   globalThis,
@@ -143,3 +145,5 @@
   [
     'CompressionStream', 'DecompressionStream',
   ]);
+
+} //not nwjs
