diff -r -u --color up/chromium/ppapi/shared_impl/media_stream_buffer_manager.h nw/chromium/ppapi/shared_impl/media_stream_buffer_manager.h
--- up/chromium/ppapi/shared_impl/media_stream_buffer_manager.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ppapi/shared_impl/media_stream_buffer_manager.h	2023-02-24 20:12:12.824894745 +0000
@@ -9,6 +9,7 @@
 
 #include <memory>
 #include <vector>
+#include <ostream>
 
 #include "base/compiler_specific.h"
 #include "base/containers/circular_deque.h"
