diff -r -u --color up/node/src/api/hooks.cc nw/node/src/api/hooks.cc
--- up/node/src/api/hooks.cc	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/src/api/hooks.cc	2023-02-25 19:08:13.814006688 +0000
@@ -22,6 +22,14 @@
   env->RunAtExitCallbacks();
 }
 
+void AtExit(void (*cb)(void* arg), void* arg) {
+  //auto env = Environment::GetThreadLocalEnv();
+  thread_ctx_st* tls_ctx = (struct thread_ctx_st*)uv_key_get(&node::thread_ctx_key);
+  if (tls_ctx && tls_ctx->env) {
+    AtExit(tls_ctx->env, cb, arg);
+  }
+}
+
 void AtExit(Environment* env, void (*cb)(void* arg), void* arg) {
   CHECK_NOT_NULL(env);
   env->AtExit(cb, arg);
