diff -r -u --color up/node/src/node_messaging.cc nw/node/src/node_messaging.cc
--- up/node/src/node_messaging.cc	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/src/node_messaging.cc	2023-02-25 19:08:13.834006798 +0000
@@ -296,6 +296,14 @@
   SerializerDelegate(Environment* env, Local<Context> context, Message* m)
       : env_(env), context_(context), msg_(m) {}
 
+  void* ReallocateBufferMemory(void* old_buffer, size_t size, size_t* actual_size) override {
+    *actual_size = size;
+    return realloc(old_buffer, size);
+  }
+
+  void FreeBufferMemory(void* buffer) override {
+    free(buffer);
+  }
   void ThrowDataCloneError(Local<String> message) override {
     ThrowDataCloneException(context_, message);
   }
