diff -r -u --color up/node/src/env.h nw/node/src/env.h
--- up/node/src/env.h	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/src/env.h	2023-02-25 19:08:13.822006732 +0000
@@ -990,6 +990,7 @@
   inline void ThrowError(v8::Local<v8::Value> (*fun)(v8::Local<v8::String>),
                          const char* errmsg);
 
+  bool uv_initialized_ = false;
   std::list<binding::DLib> loaded_addons_;
   v8::Isolate* const isolate_;
   IsolateData* const isolate_data_;
