diff -r -u --color up/node/src/heap_utils.cc nw/node/src/heap_utils.cc
--- up/node/src/heap_utils.cc	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/src/heap_utils.cc	2023-02-25 19:08:13.822006732 +0000
@@ -345,9 +345,11 @@
 inline void TakeSnapshot(Environment* env,
                          v8::OutputStream* out,
                          HeapProfiler::HeapSnapshotOptions options) {
+#if 0
   HeapSnapshotPointer snapshot{
       env->isolate()->GetHeapProfiler()->TakeHeapSnapshot(options)};
   snapshot->Serialize(out, HeapSnapshot::kJSON);
+#endif
 }
 
 }  // namespace
@@ -433,6 +435,7 @@
 }
 
 void CreateHeapSnapshotStream(const FunctionCallbackInfo<Value>& args) {
+#if 0
   Environment* env = Environment::GetCurrent(args);
   CHECK_EQ(args.Length(), 1);
   auto options = GetHeapSnapshotOptions(args[0]);
@@ -443,6 +446,7 @@
       CreateHeapSnapshotStream(env, std::move(snapshot));
   if (stream)
     args.GetReturnValue().Set(stream->object());
+#endif
 }
 
 void TriggerHeapSnapshot(const FunctionCallbackInfo<Value>& args) {
