diff -r -u --color up/node/src/tracing/agent.h nw/node/src/tracing/agent.h
--- up/node/src/tracing/agent.h	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/src/tracing/agent.h	2023-02-25 19:08:13.846006863 +0000
@@ -84,7 +84,7 @@
 class Agent {
  public:
   Agent();
-  ~Agent();
+  ~Agent() {}
 
   TracingController* GetTracingController() {
     TracingController* controller = tracing_controller_.get();
@@ -156,8 +156,8 @@
 };
 
 void AgentWriterHandle::reset() {
-  if (agent_ != nullptr)
-    agent_->Disconnect(id_);
+  //if (agent_ != nullptr)
+  //  agent_->Disconnect(id_);
   agent_ = nullptr;
 }
 
