diff -r -u --color up/node/src/node_contextify.cc nw/node/src/node_contextify.cc
--- up/node/src/node_contextify.cc	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/src/node_contextify.cc	2023-02-25 19:08:13.830006776 +0000
@@ -33,6 +33,7 @@
 #include "util-inl.h"
 
 namespace node {
+extern bool node_is_nwjs;
 namespace contextify {
 
 using errors::TryCatchScope;
@@ -266,6 +267,11 @@
   // the context from its constructor.
   v8_context->SetEmbedderData(ContextEmbedderIndex::kSandboxObject,
                               sandbox_obj);
+  if (node_is_nwjs) {
+    void* data = env->context()->GetAlignedPointerFromEmbedderData(2); //v8ContextPerContextDataIndex
+    v8_context->SetAlignedPointerInEmbedderData(2, data);
+    v8_context->SetAlignedPointerInEmbedderData(50, (void*)0x08110800);
+  }
 
   // Delegate the code generation validation to
   // node::ModifyCodeGenerationFromStrings.
@@ -799,6 +805,7 @@
   ContextifyScript* contextify_script =
       new ContextifyScript(env, args.This());
 
+#if 0
   if (*TRACE_EVENT_API_GET_CATEGORY_GROUP_ENABLED(
           TRACING_CATEGORY_NODE2(vm, script)) != 0) {
     Utf8Value fn(isolate, filename);
@@ -807,6 +814,7 @@
                        "filename",
                        TRACE_STR_COPY(*fn));
   }
+#endif
 
   ScriptCompiler::CachedData* cached_data = nullptr;
   if (!cached_data_buf.IsEmpty()) {
