diff -r -u --color up/node/src/inspector/tracing_agent.cc nw/node/src/inspector/tracing_agent.cc
--- up/node/src/inspector/tracing_agent.cc	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/src/inspector/tracing_agent.cc	2023-02-25 19:08:13.826006754 +0000
@@ -88,7 +88,7 @@
   void AppendTraceEvent(
       v8::platform::tracing::TraceObject* trace_event) override {
     if (!json_writer_)
-      json_writer_.reset(TraceWriter::CreateJSONTraceWriter(stream_, "value"));
+      json_writer_.reset(TraceWriter::CreateJSONTraceWriter(stream_));
     json_writer_->AppendTraceEvent(trace_event);
   }
 
@@ -153,6 +153,7 @@
   if (categories_set.empty())
     return DispatchResponse::Error("At least one category should be enabled");
 
+#if 0
   tracing::AgentWriterHandle* writer = GetTracingAgentWriter();
   if (writer != nullptr) {
     trace_writer_ =
@@ -161,6 +162,7 @@
                                        frontend_object_id_, main_thread_),
                                    tracing::Agent::kIgnoreDefaultCategories);
   }
+#endif
   return DispatchResponse::OK();
 }
 
