diff -r -u --color up/node/src/node_metadata.cc nw/node/src/node_metadata.cc
--- up/node/src/node_metadata.cc	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/src/node_metadata.cc	2023-02-25 19:08:13.834006798 +0000
@@ -21,7 +21,8 @@
 #endif
 #endif  // HAVE_OPENSSL
 
-#ifdef OPENSSL_INFO_QUIC
+#if 0
+//#ifdef OPENSSL_INFO_QUIC
 #include <ngtcp2/version.h>
 #include <nghttp3/version.h>
 #endif
@@ -110,7 +111,7 @@
   unicode = U_UNICODE_VERSION;
 #endif  // NODE_HAVE_I18N_SUPPORT
 
-#ifdef OPENSSL_INFO_QUIC
+#if 0 //def OPENSSL_INFO_QUIC
   ngtcp2 = NGTCP2_VERSION;
   nghttp3 = NGHTTP3_VERSION;
 #endif
