diff -r -u --color up/node/src/node_http2.cc nw/node/src/node_http2.cc
--- up/node/src/node_http2.cc	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/src/node_http2.cc	2023-02-19 17:37:33.295599279 +0000
@@ -635,6 +635,7 @@
 
 void Http2Stream::EmitStatistics() {
   CHECK_NOT_NULL(session());
+#if 0
   if (LIKELY(!HasHttp2Observer(env())))
     return;
 
@@ -652,9 +653,11 @@
     if (HasHttp2Observer(env))
       entry->Notify(env);
   });
+#endif
 }
 
 void Http2Session::EmitStatistics() {
+#if 0
   if (LIKELY(!HasHttp2Observer(env())))
     return;
 
@@ -672,6 +675,7 @@
     if (HasHttp2Observer(env))
       entry->Notify(env);
   });
+#endif
 }
 
 // Closes the session and frees the associated resources
