diff -r -u --color up/node/src/large_pages/node_large_page.cc nw/node/src/large_pages/node_large_page.cc
--- up/node/src/large_pages/node_large_page.cc	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/src/large_pages/node_large_page.cc	2023-02-19 17:37:33.287599264 +0000
@@ -99,8 +99,8 @@
 // This symbol must be declared weak because this file becomes part of all
 // Node.js targets (like node_mksnapshot, node_mkcodecache, and cctest) and
 // those files do not supply the symbol.
-extern char __attribute__((weak)) __node_text_start;
-extern char __start_lpstub;
+char __node_text_start;
+char __start_lpstub;
 }  // extern "C"
 #endif  // defined(__linux__) || defined(__FreeBSD__)
 
