diff -r -u --color up/node/src/util.cc nw/node/src/util.cc
--- up/node/src/util.cc	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/src/util.cc	2023-02-25 19:08:13.850006885 +0000
@@ -582,13 +582,9 @@
     return String::Empty(isolate);
   }
   if (is_one_byte()) {
-    NonOwningExternalOneByteResource* source =
-        new NonOwningExternalOneByteResource(*this);
-    return String::NewExternalOneByte(isolate, source).ToLocalChecked();
+    return v8::String::NewFromOneByte(isolate, one_bytes_data(), v8::NewStringType::kNormal, length_).ToLocalChecked();
   } else {
-    NonOwningExternalTwoByteResource* source =
-        new NonOwningExternalTwoByteResource(*this);
-    return String::NewExternalTwoByte(isolate, source).ToLocalChecked();
+    return v8::String::NewFromTwoByte(isolate, two_bytes_data(), v8::NewStringType::kNormal, length_).ToLocalChecked();
   }
 }
 
