diff -r -u --color up/node/src/node_file.cc nw/node/src/node_file.cc
--- up/node/src/node_file.cc	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/src/node_file.cc	2023-02-19 17:37:33.291599271 +0000
@@ -966,9 +966,9 @@
   } else {  // access(path, mode, undefined, ctx)
     CHECK_EQ(argc, 4);
     FSReqWrapSync req_wrap_sync;
-    FS_SYNC_TRACE_BEGIN(access);
+    //FS_SYNC_TRACE_BEGIN(access);
     SyncCall(env, args[3], &req_wrap_sync, "access", uv_fs_access, *path, mode);
-    FS_SYNC_TRACE_END(access);
+    //FS_SYNC_TRACE_END(access);
   }
 }
 
