diff -r -u --color up/node/src/inspector_js_api.cc nw/node/src/inspector_js_api.cc
--- up/node/src/inspector_js_api.cc	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/src/inspector_js_api.cc	2023-02-19 17:37:33.287599264 +0000
@@ -94,6 +94,7 @@
                        : AsyncWrap(env, wrap, PROVIDER_INSPECTORJSBINDING),
                          callback_(env->isolate(), callback) {
     Agent* inspector = env->inspector_agent();
+    if (inspector)
     session_ = ConnectionType::Connect(
         inspector, std::make_unique<JSBindingsSessionDelegate>(env, this));
   }
