diff -r -u --color up/node/src/inspector_agent.cc nw/node/src/inspector_agent.cc
--- up/node/src/inspector_agent.cc	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/src/inspector_agent.cc	2023-02-19 17:37:33.283599256 +0000
@@ -769,13 +769,14 @@
 std::unique_ptr<InspectorSession> Agent::Connect(
     std::unique_ptr<InspectorSessionDelegate> delegate,
     bool prevent_shutdown) {
+  if (!client_)
+    return nullptr;
+
   if (!parent_env_->should_create_inspector() && !client_) {
     ThrowUninitializedInspectorError(parent_env_);
     return std::unique_ptr<InspectorSession>{};
   }
 
-  CHECK_NOT_NULL(client_);
-
   int session_id = client_->connectFrontend(std::move(delegate),
                                             prevent_shutdown);
   return std::unique_ptr<InspectorSession>(
