diff -r -u --color up/node/src/node_realm.h nw/node/src/node_realm.h
--- up/node/src/node_realm.h	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/src/node_realm.h	2023-02-19 17:37:33.299599287 +0000
@@ -47,7 +47,7 @@
 
   Realm(Environment* env,
         v8::Local<v8::Context> context,
-        const RealmSerializeInfo* realm_info);
+        const RealmSerializeInfo* realm_info, bool nwjs = false);
   ~Realm();
 
   Realm(const Realm&) = delete;
@@ -112,6 +112,7 @@
                          const RealmSerializeInfo* realm_info);
   void DoneBootstrapping();
 
+  bool nwjs_ = false;
   Environment* env_;
   // Shorthand for isolate pointer.
   v8::Isolate* isolate_;
