diff -r -u --color up/node/src/node_options.cc nw/node/src/node_options.cc
--- up/node/src/node_options.cc	2023-02-20 19:32:04.000000000 +0000
+++ nw/node/src/node_options.cc	2023-02-25 19:08:13.834006798 +0000
@@ -959,14 +959,15 @@
             &PerProcessOptions::trace_sigint,
             kAllowedInEnvvar);
 
-  Insert(iop, &PerProcessOptions::get_per_isolate_options);
-
+  //Insert(iop, &PerProcessOptions::get_per_isolate_options);
+#if 0
   AddOption("--node-memory-debug",
             "Run with extra debug checks for memory leaks in Node.js itself",
             NoOp{},
             kAllowedInEnvvar);
   Implies("--node-memory-debug", "--debug-arraybuffer-allocations");
   Implies("--node-memory-debug", "--verify-base-objects");
+#endif
 }
 
 inline std::string RemoveBrackets(const std::string& host) {
