diff -r -u --color up/chromium/third_party/blink/renderer/bindings/core/v8/window_proxy.h nw/chromium/third_party/blink/renderer/bindings/core/v8/window_proxy.h
--- up/chromium/third_party/blink/renderer/bindings/core/v8/window_proxy.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/bindings/core/v8/window_proxy.h	2023-02-24 20:54:17.528468935 +0000
@@ -167,6 +167,7 @@
   virtual bool IsLocal() const { return false; }
 
   enum FrameReuseStatus { kFrameWillNotBeReused, kFrameWillBeReused };
+  bool ContextNotReady() const { return lifecycle_ == Lifecycle::kContextIsUninitialized;}
 
  protected:
   // Lifecycle represents the following four states.
@@ -262,7 +263,6 @@
 
   v8::Isolate* GetIsolate() const { return isolate_; }
   Frame* GetFrame() const { return frame_.Get(); }
-
 #if DCHECK_IS_ON()
   void DidAttachGlobalObject() { is_global_object_attached_ = true; }
   void DidDetachGlobalObject() { is_global_object_attached_ = false; }
