diff -r -u --color up/chromium/third_party/blink/renderer/core/page/chrome_client_impl.h nw/chromium/third_party/blink/renderer/core/page/chrome_client_impl.h
--- up/chromium/third_party/blink/renderer/core/page/chrome_client_impl.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/page/chrome_client_impl.h	2023-02-24 06:16:33.780823712 +0000
@@ -98,12 +98,12 @@
                              const WebWindowFeatures&,
                              network::mojom::blink::WebSandboxFlags,
                              const SessionStorageNamespaceId&,
-                             bool& consumed_user_gesture) override;
+                             bool& consumed_user_gesture, WebString*) override;
   void Show(LocalFrame& frame,
             LocalFrame& opener_frame,
             NavigationPolicy navigation_policy,
             const mojom::blink::WindowFeatures& window_features,
-            bool user_gesture) override;
+            bool user_gesture, WebString* manifest = nullptr) override;
   void DidOverscroll(const gfx::Vector2dF& overscroll_delta,
                      const gfx::Vector2dF& accumulated_overscroll,
                      const gfx::PointF& position_in_viewport,
