diff -r -u --color up/chromium/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.h nw/chromium/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.h
--- up/chromium/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.h	2023-02-24 06:16:33.928832455 +0000
@@ -62,7 +62,7 @@
                       size_t element_byte_size,
                       SharingType is_shared,
                       InitializationPolicy);
-
+  ArrayBufferContents(void* data, size_t length, bool nodejs, DataDeleter deleter);
   ArrayBufferContents(
       const base::subtle::PlatformSharedMemoryRegion& shared_memory_region,
       uint64_t offset,
@@ -117,6 +117,7 @@
   static void FreeMemory(void*);
 
  private:
+  bool nodejs_;
   static void* AllocateMemoryWithFlags(size_t,
                                        InitializationPolicy,
                                        unsigned int);
