diff -r -u --color up/chromium/third_party/blink/renderer/core/workers/worker_global_scope.idl nw/chromium/third_party/blink/renderer/core/workers/worker_global_scope.idl
--- up/chromium/third_party/blink/renderer/core/workers/worker_global_scope.idl	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/workers/worker_global_scope.idl	2023-02-24 20:12:14.544902184 +0000
@@ -44,6 +44,7 @@
     // Although RaisesException is not specified, importScripts() can actually
     // throw exceptions directly to V8.
     void importScripts(ScriptURLString... urls);
+    [CallWith=ScriptState] any importNWBin(ArrayBuffer buffer);
     readonly attribute WorkerNavigator navigator;
 
 
