diff -r -u --color up/chromium/third_party/crashpad/crashpad/handler/handler_main.cc nw/chromium/third_party/crashpad/crashpad/handler/handler_main.cc
--- up/chromium/third_party/crashpad/crashpad/handler/handler_main.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/crashpad/crashpad/handler/handler_main.cc	2023-02-24 20:12:28.285020184 +0000
@@ -1017,7 +1017,7 @@
   }
 
   ScopedStoppable upload_thread;
-  if (!options.url.empty()) {
+  if (true) {
     // TODO(scottmg): options.rate_limit should be removed when we have a
     // configurable database setting to control upload limiting.
     // See https://crashpad.chromium.org/bug/23.
