diff -r -u --color up/chromium/third_party/blink/renderer/core/frame/local_frame.cc nw/chromium/third_party/blink/renderer/core/frame/local_frame.cc
--- up/chromium/third_party/blink/renderer/core/frame/local_frame.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/frame/local_frame.cc	2023-02-24 06:16:33.524808589 +0000
@@ -2347,6 +2347,8 @@
 
 // static
 bool LocalFrame::HasTransientUserActivation(LocalFrame* frame) {
+  if (frame && frame->isNodeJS())
+    return true;
   return frame ? frame->Frame::HasTransientUserActivation() : false;
 }
 
