diff -r -u --color up/chromium/third_party/blink/renderer/platform/heap/thread_local.h nw/chromium/third_party/blink/renderer/platform/heap/thread_local.h
--- up/chromium/third_party/blink/renderer/platform/heap/thread_local.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/platform/heap/thread_local.h	2023-02-24 06:16:34.392859866 +0000
@@ -27,7 +27,7 @@
 
 // The call is still cheaper than multiple calls through WTF/base/pthread*
 // layers.
-#if defined(COMPONENT_BUILD)
+#if 1 //defined(COMPONENT_BUILD)
 #define BLINK_HEAP_HIDE_THREAD_LOCAL_IN_LIBRARY 1
 #else
 #define BLINK_HEAP_HIDE_THREAD_LOCAL_IN_LIBRARY 0
