diff -r -u --color up/chromium/third_party/blink/renderer/core/frame/web_frame_test.cc nw/chromium/third_party/blink/renderer/core/frame/web_frame_test.cc
--- up/chromium/third_party/blink/renderer/core/frame/web_frame_test.cc	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/frame/web_frame_test.cc	2023-02-24 06:16:33.540809534 +0000
@@ -7634,7 +7634,8 @@
       const SessionStorageNamespaceId&,
       bool& consumed_user_gesture,
       const absl::optional<Impression>&,
-      const absl::optional<WebPictureInPictureWindowOptions>&) override {
+      const absl::optional<WebPictureInPictureWindowOptions>&,
+			   WebString*) override {
     EXPECT_TRUE(false);
     return nullptr;
   }
