diff -r -u --color up/chromium/third_party/blink/renderer/core/exported/web_view_test.cc nw/chromium/third_party/blink/renderer/core/exported/web_view_test.cc
--- up/chromium/third_party/blink/renderer/core/exported/web_view_test.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/exported/web_view_test.cc	2023-02-24 20:54:17.724467828 +0000
@@ -3835,7 +3835,7 @@
       const SessionStorageNamespaceId&,
       bool& consumed_user_gesture,
       const absl::optional<Impression>&,
-      const absl::optional<WebPictureInPictureWindowOptions>&) override {
+      const absl::optional<WebPictureInPictureWindowOptions>&, WebString*) override {
     return web_view_helper_.InitializeWithOpener(Frame());
   }
   WebView* CreatedWebView() const { return web_view_helper_.GetWebView(); }
