diff -r -u --color up/chromium/third_party/blink/renderer/core/frame/local_frame_client_impl.h nw/chromium/third_party/blink/renderer/core/frame/local_frame_client_impl.h
--- up/chromium/third_party/blink/renderer/core/frame/local_frame_client_impl.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/frame/local_frame_client_impl.h	2023-02-24 06:16:33.524808589 +0000
@@ -69,6 +69,7 @@
   // LocalFrameClient ----------------------------------------------
   WebContentCaptureClient* GetWebContentCaptureClient() const override;
   void DidCommitDocumentReplacementNavigation(DocumentLoader*) override;
+  void willHandleNavigationPolicy(const blink::ResourceRequest& request, blink::NavigationPolicy* policy, WebString* manifest = nullptr, bool new_win = true) override;
   // Notifies the WebView delegate that the JS window object has been cleared,
   // giving it a chance to bind native objects to the window before script
   // parsing begins.
