diff -r -u --color up/chromium/third_party/blink/renderer/core/workers/shared_worker_client_holder.cc nw/chromium/third_party/blink/renderer/core/workers/shared_worker_client_holder.cc
--- up/chromium/third_party/blink/renderer/core/workers/shared_worker_client_holder.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/workers/shared_worker_client_holder.cc	2023-02-24 20:54:18.084465795 +0000
@@ -88,7 +88,7 @@
     const KURL& url,
     mojo::PendingRemote<mojom::blink::BlobURLToken> blob_url_token,
     mojom::blink::WorkerOptionsPtr options,
-    ukm::SourceId client_ukm_source_id) {
+    ukm::SourceId client_ukm_source_id, bool isNodeJS) {
   DCHECK(IsMainThread());
   DCHECK(options);
 
@@ -110,7 +110,7 @@
           ? mojom::InsecureRequestsPolicy::kUpgrade
           : mojom::InsecureRequestsPolicy::kDoNotUpgrade;
 
-  auto info = mojom::blink::SharedWorkerInfo::New(
+  auto info = mojom::blink::SharedWorkerInfo::New(isNodeJS, base::FilePath(),
       url, std::move(options),
       mojo::Clone(worker->GetExecutionContext()
                       ->GetContentSecurityPolicy()
