diff -r -u --color up/chromium/third_party/blink/renderer/core/exported/web_view_impl.h nw/chromium/third_party/blink/renderer/core/exported/web_view_impl.h
--- up/chromium/third_party/blink/renderer/core/exported/web_view_impl.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/exported/web_view_impl.h	2023-02-24 20:54:17.724467828 +0000
@@ -580,7 +580,7 @@
             NavigationPolicy policy,
             const gfx::Rect& requested_rect,
             const gfx::Rect& adjusted_rect,
-            bool opened_by_user_gesture);
+            bool opened_by_user_gesture, WebString* manifest);
 
   // Send the window rect to the browser and call `ack_callback` when the
   // browser has processed it.
