diff -r -u --color up/chromium/third_party/blink/renderer/core/script/modulator_impl_base.cc nw/chromium/third_party/blink/renderer/core/script/modulator_impl_base.cc
--- up/chromium/third_party/blink/renderer/core/script/modulator_impl_base.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/script/modulator_impl_base.cc	2023-02-24 20:54:17.996466292 +0000
@@ -58,6 +58,10 @@
   return GetExecutionContext()->GetV8CacheOptions();
 }
 
+void ModulatorImplBase::AddToMap(const KURL& url, ModuleScript* script) {
+  map_->AddToMap(url, script);
+}
+
 // <specdef label="fetch-a-module-script-tree"
 // href="https://html.spec.whatwg.org/C/#fetch-a-module-script-tree">
 // <specdef label="fetch-a-module-worker-script-tree"
