diff -r -u --color up/chromium/third_party/blink/renderer/platform/scheduler/worker/non_main_thread_impl.h nw/chromium/third_party/blink/renderer/platform/scheduler/worker/non_main_thread_impl.h
--- up/chromium/third_party/blink/renderer/platform/scheduler/worker/non_main_thread_impl.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/platform/scheduler/worker/non_main_thread_impl.h	2023-02-24 20:54:18.476463581 +0000
@@ -71,6 +71,7 @@
 
     explicit SimpleThreadImpl(const WTF::String& name_prefix,
                               const base::SimpleThread::Options& options,
+                              bool is_node,
                               bool supports_gc,
                               NonMainThreadImpl* worker_thread);
 
@@ -118,6 +119,7 @@
         non_main_thread_scheduler_;
     scoped_refptr<base::SingleThreadTaskRunner> default_task_runner_;
     base::RunLoop* run_loop_;
+    bool nodejs_;
     bool supports_gc_;
     std::unique_ptr<GCSupport> gc_support_;
   };
