diff -r -u --color up/chromium/third_party/blink/renderer/core/page/chrome_client_impl.cc nw/chromium/third_party/blink/renderer/core/page/chrome_client_impl.cc
--- up/chromium/third_party/blink/renderer/core/page/chrome_client_impl.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/third_party/blink/renderer/core/page/chrome_client_impl.cc	2023-02-24 20:54:17.956466518 +0000
@@ -284,7 +284,7 @@
     const WebWindowFeatures& features,
     network::mojom::blink::WebSandboxFlags sandbox_flags,
     const SessionStorageNamespaceId& session_storage_namespace_id,
-    bool& consumed_user_gesture) {
+    bool& consumed_user_gesture, WebString* manifest) {
   if (!frame->GetPage() || frame->GetPage()->Paused())
     return nullptr;
 
@@ -300,7 +300,7 @@
           WrappedResourceRequest(r.GetResourceRequest()), features, frame_name,
           static_cast<WebNavigationPolicy>(r.GetNavigationPolicy()),
           sandbox_flags, session_storage_namespace_id, consumed_user_gesture,
-          r.Impression(), r.GetPictureInPictureWindowOptions()));
+          r.Impression(), r.GetPictureInPictureWindowOptions(), manifest));
   if (!new_view)
     return nullptr;
   return new_view->GetPage();
@@ -354,7 +354,7 @@
                             LocalFrame& opener_frame,
                             NavigationPolicy navigation_policy,
                             const mojom::blink::WindowFeatures& window_features,
-                            bool user_gesture) {
+                            bool user_gesture, WebString* manifest) {
   DCHECK(web_view_);
   const gfx::Rect rect_adjusted_for_minimum =
       AdjustWindowRectForMinimum(window_features.bounds);
@@ -367,7 +367,7 @@
   // TODO(crbug.com/897300): Use permission state for better sync estimates or
   // store unadjusted pending window rects if that will not break many sites.
   web_view_->Show(opener_frame.GetLocalFrameToken(), navigation_policy,
-                  rect_adjusted_for_minimum, adjusted_rect, user_gesture);
+                  rect_adjusted_for_minimum, adjusted_rect, user_gesture, manifest);
 }
 
 bool ChromeClientImpl::ShouldReportDetailedMessageForSourceAndSeverity(
