diff -r -u --color up/chromium/tools/gritsettings/resource_ids.spec nw/chromium/tools/gritsettings/resource_ids.spec
--- up/chromium/tools/gritsettings/resource_ids.spec	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/tools/gritsettings/resource_ids.spec	2023-02-24 06:16:53.785973645 +0000
@@ -35,6 +35,7 @@
   #
   # chrome/ and ios/chrome/ must start at the same id.
   # App only use one file depending on whether it is iOS or other platform.
+
   # Chromium strings and Google Chrome strings must start at the same id.
   # We only use one file depending on whether we're building Chromium or
   # Google Chrome.
@@ -1068,4 +1069,10 @@
 
   # Thinking about appending to the end?
   # Please read the header and find the right section above instead.
+
+  # Resource ids starting at 31000 are reserved for projects built on Chromium.
+  "chrome/browser/nwjs_resources.grd": {
+    "includes": [32000],
+    "structures": [32450],
+  },
 }
