diff -r -u --color up/chromium/ui/aura/window_tree_host.cc nw/chromium/ui/aura/window_tree_host.cc
--- up/chromium/ui/aura/window_tree_host.cc	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/aura/window_tree_host.cc	2023-02-24 20:12:33.493068513 +0000
@@ -269,6 +269,12 @@
                                window_->GetLocalSurfaceId());
 }
 
+#if defined(OS_WIN)
+void WindowTreeHost::SetPositionInPixels(const gfx::Point& pos_in_pixels) {
+
+}
+#endif
+
 void WindowTreeHost::ConvertDIPToScreenInPixels(gfx::Point* point) const {
   ConvertDIPToPixels(point);
   gfx::Point location = GetLocationOnScreenInPixels();
