diff -r -u --color up/chromium/ui/gfx/mac/nswindow_frame_controls.h nw/chromium/ui/gfx/mac/nswindow_frame_controls.h
--- up/chromium/ui/gfx/mac/nswindow_frame_controls.h	2023-02-21 19:03:23.000000000 +0000
+++ nw/chromium/ui/gfx/mac/nswindow_frame_controls.h	2023-02-24 20:54:38.964347063 +0000
@@ -29,6 +29,7 @@
                                              const gfx::Size& max_size,
                                              bool can_resize,
                                              bool can_fullscreen);
+GFX_EXPORT void SetNSWindowShowInTaskbar(NSWindow* window, bool show);
 
 }  // namespace gfx
 
