diff -r -u --color up/chromium/ui/views/widget/native_widget_delegate.h nw/chromium/ui/views/widget/native_widget_delegate.h
--- up/chromium/ui/views/widget/native_widget_delegate.h	2023-02-15 18:46:36.000000000 +0000
+++ nw/chromium/ui/views/widget/native_widget_delegate.h	2023-02-24 06:16:56.318112640 +0000
@@ -40,6 +40,7 @@
  public:
   virtual ~NativeWidgetDelegate() = default;
 
+  virtual bool NWCanClose(bool user_force = false) const = 0;
   // Returns true if the window is modal.
   virtual bool IsModal() const = 0;
 
