diff -r -u --color up/node/deps/zlib/zlib.gyp nw/node/deps/zlib/zlib.gyp
--- up/node/deps/zlib/zlib.gyp	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/deps/zlib/zlib.gyp	2023-02-19 17:37:33.159599018 +0000
@@ -7,7 +7,7 @@
     'ZLIB_ROOT': '.',
     'use_system_zlib%': 0,
     'arm_fpu%': '',
-    'llvm_version%': '0.0',
+    'llvm_version': '10.0',
   },
   'conditions': [
     ['use_system_zlib==0', {
@@ -32,6 +32,17 @@
                 }],
               ],
             }],
+            ['OS=="win"', {
+                  'msvs_settings': {
+                    'VCCLCompilerTool': {
+                      'AdditionalOptions': [
+                        '-mssse3',
+                        '-msse4.2',
+                        '-mpclmul',
+                      ],
+                    },
+                  },
+            }],
             ['arm_fpu=="neon"', {
               'defines': [ 'ADLER32_SIMD_NEON' ],
             }],
@@ -122,7 +133,18 @@
                   '-mpclmul',
                 ],
               },
-            }]
+            }],
+            ['OS=="win"', {
+                  'msvs_settings': {
+                    'VCCLCompilerTool': {
+                      'AdditionalOptions': [
+                        '-mssse3',
+                        '-msse4.2',
+                        '-mpclmul',
+                      ],
+                    },
+                  },
+            }],
           ],
           'defines': [ 'CRC32_SIMD_SSE42_PCLMUL' ],
           'include_dirs': [ '<(ZLIB_ROOT)' ],
@@ -210,6 +232,17 @@
               ],
               'defines': [ 'DEFLATE_SLIDE_HASH_SSE2' ],
               'conditions': [
+                ['OS=="win"', {
+                  'msvs_settings': {
+                    'VCCLCompilerTool': {
+                      'AdditionalOptions': [
+                        '-mssse3',
+                        '-msse4.2',
+                        '-mpclmul',
+                      ],
+                    },
+                  },
+                }],
                 ['target_arch=="x64"', {
                   'defines': [ 'INFLATE_CHUNK_READ_64LE' ],
                 }],
Only in nw/node/lib: dummystream.js
