diff -r -u --color up/node/deps/openssl/openssl/providers/implementations/rands/seeding/rand_unix.c nw/node/deps/openssl/openssl/providers/implementations/rands/seeding/rand_unix.c
--- up/node/deps/openssl/openssl/providers/implementations/rands/seeding/rand_unix.c	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/deps/openssl/openssl/providers/implementations/rands/seeding/rand_unix.c	2023-02-19 17:37:31.415595667 +0000
@@ -357,7 +357,7 @@
      * internally. So we need to check errno for ENOSYS
      */
 #  if !defined(__DragonFly__) && !defined(__NetBSD__)
-#    if defined(__GNUC__) && __GNUC__>=2 && defined(__ELF__) && !defined(__hpux)
+#    if 0 //defined(__GNUC__) && __GNUC__>=2 && defined(__ELF__) && !defined(__hpux)
     extern int getentropy(void *buffer, size_t length) __attribute__((weak));
 
     if (getentropy != NULL) {
