diff -r -u --color up/node/deps/uv/src/unix/core.c nw/node/deps/uv/src/unix/core.c
--- up/node/deps/uv/src/unix/core.c	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/deps/uv/src/unix/core.c	2023-02-19 17:37:31.683596182 +0000
@@ -1638,9 +1638,11 @@
    * glibc it's... complicated... so for consistency try sched_getaffinity()
    * before falling back to sysconf(_SC_NPROCESSORS_ONLN).
    */
+#if defined(CPU_COUNT)
   if (0 == sched_getaffinity(0, sizeof(set), &set))
     rc = CPU_COUNT(&set);
   else
+#endif
     rc = sysconf(_SC_NPROCESSORS_ONLN);
 
   if (rc < 1)
