diff -r -u --color up/node/deps/cares/cares.gyp nw/node/deps/cares/cares.gyp
--- up/node/deps/cares/cares.gyp	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/deps/cares/cares.gyp	2023-02-19 17:37:30.219593369 +0000
@@ -1,4 +1,7 @@
 {
+  'variables': {
+    'library%': 'static_library',
+  },
   'target_defaults': {
     'conditions': [
       ['OS!="win"', {
@@ -126,6 +129,7 @@
             '_WINSOCK_DEPRECATED_NO_WARNINGS',
           ],
           'include_dirs': [ 'config/win32' ],
+          'msvs_disabled_warnings': [4133],
           'sources': [
             'src/lib/config-win32.h',
             'src/lib/windows_port.c',
