diff -r -u --color up/node/lib/internal/modules/esm/loader.js nw/node/lib/internal/modules/esm/loader.js
--- up/node/lib/internal/modules/esm/loader.js	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/lib/internal/modules/esm/loader.js	2023-02-19 17:37:33.255599202 +0000
@@ -236,9 +236,9 @@
   translators = getTranslators();
 
   constructor() {
-    if (getOptionValue('--experimental-loader').length > 0) {
+/*    if (getOptionValue('--experimental-loader').length > 0) {
       emitExperimentalWarning('Custom ESM Loaders');
-    }
+    } */
     if (getOptionValue('--experimental-network-imports')) {
       emitExperimentalWarning('Network Imports');
     }
