diff -r -u --color up/node/src/node_perf.cc nw/node/src/node_perf.cc
--- up/node/src/node_perf.cc	2023-02-14 22:09:05.000000000 +0000
+++ nw/node/src/node_perf.cc	2023-02-19 17:37:33.299599287 +0000
@@ -88,10 +88,12 @@
 
 void PerformanceState::Mark(PerformanceMilestone milestone, uint64_t ts) {
   this->milestones[milestone] = static_cast<double>(ts);
+#if 0
   TRACE_EVENT_INSTANT_WITH_TIMESTAMP0(
       TRACING_CATEGORY_NODE1(bootstrap),
       GetPerformanceMilestoneName(milestone),
       TRACE_EVENT_SCOPE_THREAD, ts / 1000);
+#endif
 }
 
 // Allows specific Node.js lifecycle milestones to be set from JavaScript
@@ -196,6 +198,7 @@
 
 static void InstallGarbageCollectionTracking(
     const FunctionCallbackInfo<Value>& args) {
+#if 0
   Environment* env = Environment::GetCurrent(args);
   // Reset current_gc_type to 0
   env->performance_state()->current_gc_type = 0;
@@ -204,6 +207,7 @@
   env->isolate()->AddGCEpilogueCallback(MarkGarbageCollectionEnd,
                                         static_cast<void*>(env));
   env->AddCleanupHook(GarbageCollectionCleanupHook, env);
+#endif
 }
 
 static void RemoveGarbageCollectionTracking(
