diff --git a/base/test/test_message_loop.cc b/base/test/test_message_loop.cc
index e4634f1b35..d1bb8a6ee7 100644
--- a/base/test/test_message_loop.cc
+++ b/base/test/test_message_loop.cc
@@ -19,6 +19,7 @@ test::SingleThreadTaskEnvironment::MainThreadType GetMainThreadType(
     MessagePumpType type) {
   switch (type) {
     case MessagePumpType::DEFAULT:
+    case MessagePumpType::NODE:
       return test::SingleThreadTaskEnvironment::MainThreadType::DEFAULT;
     case MessagePumpType::IO:
       return test::SingleThreadTaskEnvironment::MainThreadType::IO;
