diff --git a/base/message_loop/message_pump_mac.h b/base/message_loop/message_pump_mac.h
index a0693a0bc9..6efd6fb23e 100644
--- a/base/message_loop/message_pump_mac.h
+++ b/base/message_loop/message_pump_mac.h
@@ -156,7 +156,7 @@ class BASE_EXPORT MessagePumpCFRunLoopBase : public MessagePump {
   // Get the current mode mask from |enabled_modes_|.
   int GetModeMask() const;

- private:
+ protected:
   class ScopedModeEnabler;

   // The maximum number of run loop modes that can be monitored.
@@ -172,14 +172,16 @@ class BASE_EXPORT MessagePumpCFRunLoopBase : public MessagePump {
   // the instance method; the instance method returns true if it resignalled
   // |work_source_| to be called again from the loop.
   static void RunWorkSource(void* info);
-  bool RunWork();
+ protected:
+  virtual bool RunWork();

   // Perform idle-priority work.  This is normally called by PreWaitObserver,
   // but is also associated with |idle_work_source_|.  When this function
   // actually does perform idle work, it will resignal that source.  The
   // static method calls the instance method.
   static void RunIdleWorkSource(void* info);
-  void RunIdleWork();
+  virtual void RunIdleWork();
+  virtual void PreWaitObserverHook();

   // Perform work that may have been deferred because it was not runnable
   // within a nested run loop.  This is associated with
