diff --git a/base/process/launch_posix.cc b/base/process/launch_posix.cc
index 9196ad285e..f5b04a6887 100644
--- a/base/process/launch_posix.cc
+++ b/base/process/launch_posix.cc
@@ -458,6 +458,7 @@ Process LaunchProcess(const std::vector<std::string>& argv,
 #ifndef PR_SET_NO_NEW_PRIVS
 #define PR_SET_NO_NEW_PRIVS 38
 #endif
+#if 0
     if (!options.allow_new_privs) {
       if (prctl(PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0) && errno != EINVAL) {
         // Only log if the error is not EINVAL (i.e. not supported).
@@ -471,6 +472,7 @@ Process LaunchProcess(const std::vector<std::string>& argv,
         _exit(127);
       }
     }
+#endif
 #endif

     if (current_directory != nullptr) {
