diff --git a/base/threading/thread_restrictions.h b/base/threading/thread_restrictions.h
index b5f91fe46d..ce83b76549 100644
--- a/base/threading/thread_restrictions.h
+++ b/base/threading/thread_restrictions.h
@@ -130,6 +130,7 @@ bool CameraAppUIShouldEnableLocalOverride(const std::string&);
 namespace audio {
 class OutputDevice;
 }
+
 namespace blink {
 class CategorizedWorkerPool;
 class DiskDataAllocator;
@@ -145,6 +146,16 @@ namespace scheduler {
 class NonMainThreadImpl;
 }
 }  // namespace blink
+
+namespace extensions {
+class NwAppSetProxyConfigFunction;
+class ContentVerifier;
+}
+
+namespace nw {
+class Package;
+}
+
 namespace cc {
 class CompletionEvent;
 class TileTaskManagerImpl;
@@ -693,6 +704,7 @@ class BASE_EXPORT ScopedAllowBaseSyncPrimitivesOutsideBlockingScope {
   friend class service_manager::ServiceProcessLauncher;
   friend class ui::WindowResizeHelperMac;    // http://crbug.com/902829
   friend class content::TextInputClientMac;  // http://crbug.com/121917
+  friend class extensions::NwAppSetProxyConfigFunction;

   ScopedAllowBaseSyncPrimitivesOutsideBlockingScope(
       const Location& from_here = Location::Current());
@@ -834,6 +846,9 @@ class BASE_EXPORT PermanentThreadAllowance {
   friend class content::BrowserTestBase;
   friend class web::WebMainLoop;

+  friend class extensions::ContentVerifier;
+  friend class nw::Package;
+
   static void AllowBlocking() EMPTY_BODY_IF_DCHECK_IS_OFF;
   static void AllowBaseSyncPrimitives() EMPTY_BODY_IF_DCHECK_IS_OFF;
 };
diff --git a/build/common.gypi b/build/common.gypi
new file mode 100644
index 0000000000..2341fd73d2
