diff --git a/base/message_loop/message_pump_mac.mm b/base/message_loop/message_pump_mac.mm
index ac283df047..27fb3e0355 100644
--- a/base/message_loop/message_pump_mac.mm
+++ b/base/message_loop/message_pump_mac.mm
@@ -596,12 +596,16 @@ void MessagePumpCFRunLoopBase::PreWaitObserver(CFRunLoopObserverRef observer,
     // nesting-deferred work may have accumulated.  Schedule it for processing
     // if appropriate.
     self->MaybeScheduleNestingDeferredWork();
+    self->PreWaitObserverHook();

     // Notify the delegate that the loop is about to sleep.
     self->BeforeWait();
   });
 }

+void MessagePumpCFRunLoopBase::PreWaitObserverHook() {
+}
+
 // Called from the run loop.
 // static
 void MessagePumpCFRunLoopBase::AfterWaitObserver(CFRunLoopObserverRef observer,
