diff --git a/build/config/BUILDCONFIG.gn b/build/config/BUILDCONFIG.gn
index c737e53c2c..6788046b10 100644
--- a/build/config/BUILDCONFIG.gn
+++ b/build/config/BUILDCONFIG.gn
@@ -135,6 +135,8 @@ declare_args() {
   # set "is_official_build" to true for any build intended to ship to end-users.
   is_official_build = false

+  nwjs_sdk = true
+
   # Set to true when compiling with the Clang compiler.
   is_clang = current_os != "linux" ||
              (current_cpu != "s390x" && current_cpu != "s390" &&
