diff --git a/build/config/mac/BUILD.gn b/build/config/mac/BUILD.gn
index fa114a5721..3cdb22b6a0 100644
--- a/build/config/mac/BUILD.gn
+++ b/build/config/mac/BUILD.gn
@@ -90,6 +90,7 @@ config("mac_dynamic_flags") {

       # Path for loading shared libraries for bundled binaries. Get back from
       # Binary.app/Contents/MacOS.
+      "-lc++abi",
       "-Wl,-rpath,@loader_path/../../..",
     ]

