diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 0272bf80f3..a657b985af 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -483,7 +483,9 @@ config("compiler") {
     #}
   }

-  if (use_icf && (!is_apple || use_lld)) {
+  if (is_linux && current_cpu == "x86") {
+    ldflags += [ "-Wl,--icf=none" ]
+  } else if (use_icf && (!is_apple || use_lld)) {
     ldflags += [ "-Wl,--icf=all" ]
   }

@@ -1489,6 +1491,9 @@ config("default_warnings") {
       # Disables.
       "-Wno-missing-field-initializers",  # "struct foo f = {0};"
       "-Wno-unused-parameter",  # Unused function parameters.
+      "-Wno-unused-private-field",
+      "-Wno-unused-const-variable",
+      "-Wno-unreachable-code",
     ]
   }

@@ -1500,6 +1505,7 @@ config("default_warnings") {
       # which we no longer use. Check if it makes sense to remove
       # this as well. http://crbug.com/316352
       "-Wno-unneeded-internal-declaration",
+      "-Wno-unused-function",
     ]

     if (!is_nacl || is_nacl_saigo) {
@@ -1680,6 +1686,7 @@ config("no_chromium_code") {
     cflags += [
       "/wd4800",  # Disable warning when forcing value to bool.
       "/wd4267",  # TODO(jschuh): size_t to int.
+      "/wd4275",
     ]
   } else {
     # GCC may emit unsuppressible warnings so don't add -Werror for no chromium
