diff --git a/build/toolchain/gcc_toolchain.gni b/build/toolchain/gcc_toolchain.gni
index 0bfa64ce1e..b816a004ba 100644
--- a/build/toolchain/gcc_toolchain.gni
+++ b/build/toolchain/gcc_toolchain.gni
@@ -498,6 +498,9 @@ template("gcc_toolchain") {
       default_output_extension = default_shlib_extension

       default_output_dir = "{{root_out_dir}}${default_shlib_subdir}"
+      if (shlib_subdir != ".") {
+        default_output_dir += "/$shlib_subdir"
+      }

       output_prefix = "lib"

@@ -579,6 +582,9 @@ template("gcc_toolchain") {
       }

       default_output_dir = "{{root_out_dir}}${default_shlib_subdir}"
+      if (shlib_subdir != ".") {
+        default_output_dir += "/$shlib_subdir"
+      }

       output_prefix = "lib"

