diff --git a/chrome/browser/ui/dialogs/outdated_upgrade_bubble.cc b/chrome/browser/ui/dialogs/outdated_upgrade_bubble.cc
index ed846b5561..54e7071fe8 100644
--- a/chrome/browser/ui/dialogs/outdated_upgrade_bubble.cc
+++ b/chrome/browser/ui/dialogs/outdated_upgrade_bubble.cc
@@ -40,7 +40,9 @@ constexpr int kNumIgnoredBuckets = 5;

 // For ChromeOS Lacros, browser updates are done via system services, thus
 // we redirect to the safetyCheck page that interacts with these. On other
-// platforms it may be possible to download an updated browser via a site.
+// platforms it may be possible to download an updated browser via a
+// site.
+#if 0
 const char* kUpdateBrowserRedirectUrl =
 #if BUILDFLAG(IS_CHROMEOS_LACROS)
     // The URL to be used to update Lacros-Chrome when auto-update failed
@@ -51,7 +53,7 @@ const char* kUpdateBrowserRedirectUrl =
     // too long.
     "https://www.google.com/chrome";
 #endif
-
+#endif
 bool g_upgrade_bubble_is_showing = false;

 // The number of times the user ignored the bubble before finally choosing to
@@ -111,6 +113,7 @@ void OnDialogAccepted(content::PageNavigator* navigator,
 }  // namespace

 void ShowOutdatedUpgradeBubble(Browser* browser, bool auto_update_enabled) {
+#if 0
   if (g_upgrade_bubble_is_showing)
     return;

@@ -139,4 +142,5 @@ void ShowOutdatedUpgradeBubble(Browser* browser, bool auto_update_enabled) {
       auto_update_enabled
           ? base::UserMetricsAction("OutdatedUpgradeBubble.Show")
           : base::UserMetricsAction("OutdatedUpgradeBubble.ShowNoAU"));
+#endif
 }
