diff --git a/chrome/browser/extensions/extension_allowlist.cc b/chrome/browser/extensions/extension_allowlist.cc
index c86f624584..168bd3ced9 100644
--- a/chrome/browser/extensions/extension_allowlist.cc
+++ b/chrome/browser/extensions/extension_allowlist.cc
@@ -399,6 +399,7 @@ void ExtensionAllowlist::NotifyExtensionAllowlistWarningStateChanged(
 }

 void ExtensionAllowlist::ReportExtensionReEnabledEvent() {
+#if 0
   auto* metrics_collector =
       safe_browsing::SafeBrowsingMetricsCollectorFactory::GetForProfile(
           profile_);
@@ -408,6 +409,7 @@ void ExtensionAllowlist::ReportExtensionReEnabledEvent() {
         safe_browsing::SafeBrowsingMetricsCollector::EventType::
             NON_ALLOWLISTED_EXTENSION_RE_ENABLED);
   }
+#endif
 }

 }  // namespace extensions
