diff --git a/chrome/browser/resources/component_extension_resources.grd b/chrome/browser/resources/component_extension_resources.grd
index e9f817da73..ee347d366c 100644
--- a/chrome/browser/resources/component_extension_resources.grd
+++ b/chrome/browser/resources/component_extension_resources.grd
@@ -12,6 +12,9 @@
   </outputs>
   <release seq="1">
     <includes>
+      <include name="IDR_NWJS_DEFAULT_JS" file="nwjs/default.js" type="BINDATA" />
+      <include name="IDR_NWJS_DEFAULT_JS_NEWWIN" file="nwjs/newwin.js" type="BINDATA" />
+
       <include name="IDR_NETWORK_SPEECH_SYNTHESIS_JS" file="network_speech_synthesis/tts_extension.js" type="BINDATA" />

       <if expr="chromeos_ash">
@@ -73,6 +76,14 @@
       <include name="IDS_READ_ANYTHING_LIGHT_PNG" file="side_panel/images/read_anything_light.png" type="BINDATA" />
       <include name="IDS_READ_ANYTHING_DARK_PNG" file="side_panel/images/read_anything_dark.png" type="BINDATA" />
       <include name="IDS_READ_ANYTHING_YELLOW_PNG" file="side_panel/images/read_anything_yellow.png" type="BINDATA" />
+      <include name="IDR_NWJS_DEFAPP_INDEX_HTML" file="nwjs_default_app/nw_blank.html" allowexternalscript="true" type="BINDATA" />
+      <include name="IDR_NWJS_DEFAPP_MAIN_JS"    file="nwjs_default_app/main.js" type="BINDATA" />
+      <include name="IDR_NWJS_NEWWINAPP_INDEX_HTML" file="nwjs_newwin_app/nw_blank.html" allowexternalscript="true" type="BINDATA" />
+      <include name="IDR_NWJS_NEWWINAPP_MAIN_JS"    file="nwjs_newwin_app/main.js" type="BINDATA" />
+      <include name="IDR_NWJS_NEWWINAPP_ROTATE_JS"    file="nwjs_newwin_app/rotate.js" type="BINDATA" />
+      <include name="IDR_NWJS_NEWWINAPP_ROTATE_SVG"    file="nwjs_newwin_app/rotate.svg" type="BINDATA" />
+
+      <if expr="nwjs_sdk">
       <include name="IDR_CRYPTOTOKEN_UTIL_JS" file="cryptotoken/util.js" type="BINDATA" />
       <include name="IDR_CRYPTOTOKEN_B64_JS" file="cryptotoken/b64.js" type="BINDATA" />
       <include name="IDR_CRYPTOTOKEN_COUNTDOWN_JS" file="cryptotoken/countdown.js" type="BINDATA" />
@@ -104,6 +115,7 @@
       <include name="IDR_CRYPTOTOKEN_CRYPTOTOKENAPPROVEDORIGIN_JS" file="cryptotoken/cryptotokenapprovedorigins.js" type="BINDATA" />
       <include name="IDR_CRYPTOTOKEN_CRYPTOTOKENORIGINCHECK_JS" file="cryptotoken/cryptotokenorigincheck.js" type="BINDATA" />
       <include name="IDR_CRYPTOTOKEN_CRYPTOTOKENBACKGROUND_JS" file="cryptotoken/cryptotokenbackground.js" type="BINDATA" />
+      </if> <!-- nwjs_sdk -->
     </includes>
   </release>
 </grit>
diff --git a/chrome/browser/resources/empty.css b/chrome/browser/resources/empty.css
new file mode 100644
index 0000000000..e69de29bb2
diff --git a/chrome/browser/resources/empty.html b/chrome/browser/resources/empty.html
new file mode 100644
index 0000000000..e69de29bb2
diff --git a/chrome/browser/resources/empty.js b/chrome/browser/resources/empty.js
new file mode 100644
index 0000000000..e69de29bb2
diff --git a/chrome/browser/resources/empty.json b/chrome/browser/resources/empty.json
new file mode 100644
index 0000000000..e69de29bb2
diff --git a/chrome/browser/resources/empty.png b/chrome/browser/resources/empty.png
new file mode 100644
index 0000000000..e69de29bb2
diff --git a/chrome/browser/resources/nwjs/default.js b/chrome/browser/resources/nwjs/default.js
new file mode 100644
index 0000000000..a5e1ce27fb
