diff --git a/chrome/test/data/webui/print_preview/print_preview_app_test.ts b/chrome/test/data/webui/print_preview/print_preview_app_test.ts
index 449f20fe3b..4161b9abe7 100644
--- a/chrome/test/data/webui/print_preview/print_preview_app_test.ts
+++ b/chrome/test/data/webui/print_preview/print_preview_app_test.ts
@@ -37,6 +37,7 @@ suite(print_preview_app_test.suiteName, function() {
   let pluginProxy: TestPluginProxy;

   const initialSettings: NativeInitialSettings = {
+    nwPrintMode: false,
     isInKioskAutoPrintMode: false,
     isInAppKioskMode: false,
     thousandsDelimiter: ',',
