diff --git a/chrome/browser/component_updater/file_type_policies_component_installer.cc b/chrome/browser/component_updater/file_type_policies_component_installer.cc
index 5299c424b4..0e556010e1 100644
--- a/chrome/browser/component_updater/file_type_policies_component_installer.cc
+++ b/chrome/browser/component_updater/file_type_policies_component_installer.cc
@@ -39,6 +39,7 @@ const uint8_t kFileTypePoliciesPublicKeySHA256[32] = {
 const char kFileTypePoliciesManifestName[] = "File Type Policies";

 void LoadFileTypesFromDisk(const base::FilePath& pb_path) {
+#if 0
   if (pb_path.empty())
     return;

@@ -53,6 +54,7 @@ void LoadFileTypesFromDisk(const base::FilePath& pb_path) {

   safe_browsing::FileTypePolicies::GetInstance()->PopulateFromDynamicUpdate(
       binary_pb);
+#endif
 }

 }  // namespace
