diff --git a/chrome/browser/ui/views/status_icons/status_icon_linux_wrapper.cc b/chrome/browser/ui/views/status_icons/status_icon_linux_wrapper.cc
index 8e74922bb3..377fbfa123 100644
--- a/chrome/browser/ui/views/status_icons/status_icon_linux_wrapper.cc
+++ b/chrome/browser/ui/views/status_icons/status_icon_linux_wrapper.cc
@@ -43,8 +43,7 @@ StatusIconLinuxWrapper::StatusIconLinuxWrapper(ui::StatusIconLinux* status_icon,
     : status_icon_(status_icon),
       status_icon_type_(status_icon_type),
       image_(GetBestImageRep(image)),
-      tool_tip_(tool_tip),
-      menu_model_(nullptr) {
+      tool_tip_(tool_tip) {
   status_icon_->SetDelegate(this);
 }

@@ -71,8 +70,6 @@ StatusIconLinuxWrapper::StatusIconLinuxWrapper(
 }

 StatusIconLinuxWrapper::~StatusIconLinuxWrapper() {
-  if (menu_model_)
-    menu_model_->RemoveObserver(this);
 }

 void StatusIconLinuxWrapper::SetImage(const gfx::ImageSkia& image) {
@@ -160,17 +157,9 @@ StatusIconLinuxWrapper::CreateWrappedStatusIcon(
 }

 void StatusIconLinuxWrapper::UpdatePlatformContextMenu(
-    StatusIconMenuModel* model) {
+    ui::MenuModel* model) {
   if (!status_icon_)
     return;

-  // If a menu already exists, remove ourself from its observer list.
-  if (menu_model_)
-    menu_model_->RemoveObserver(this);
-
   status_icon_->UpdatePlatformContextMenu(model);
-  menu_model_ = model;
-
-  if (model)
-    model->AddObserver(this);
 }
