diff --git a/chrome/test/base/testing_browser_process.h b/chrome/test/base/testing_browser_process.h
index 0d2454c64c..f797ec3dbe 100644
--- a/chrome/test/base/testing_browser_process.h
+++ b/chrome/test/base/testing_browser_process.h
@@ -129,6 +129,7 @@ class TestingBrowserProcess : public BrowserProcess {
   void StartAutoupdateTimer() override {}
 #endif

+
   component_updater::ComponentUpdateService* component_updater() override;
   MediaFileSystemRegistry* media_file_system_registry() override;

