diff --git a/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout_delegate.h b/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout_delegate.h
index c7f1753349..2108a62a3e 100644
--- a/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout_delegate.h
+++ b/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout_delegate.h
@@ -38,6 +38,7 @@ class OpaqueBrowserFrameViewLayoutDelegate {
   // Returns the browser's minimum view size. Used because we need to calculate
   // the minimum size for the entire non-client area.
   virtual gfx::Size GetBrowserViewMinimumSize() const = 0;
+  virtual gfx::Size GetBrowserViewMaximumSize() const = 0;

   // Whether we should show the (minimize,maximize,close) buttons. This can
   // depend on the current state of the window (e.g., whether it is maximized).
