diff --git a/chrome/browser/resources/settings/autofill_page/passwords_export_dialog.ts b/chrome/browser/resources/settings/autofill_page/passwords_export_dialog.ts
index bfa571bf8b..d1339b9c44 100644
--- a/chrome/browser/resources/settings/autofill_page/passwords_export_dialog.ts
+++ b/chrome/browser/resources/settings/autofill_page/passwords_export_dialog.ts
@@ -166,7 +166,7 @@ export class PasswordsExportDialogElement extends
     this.switchToDialog_(States.IN_PROGRESS);

     this.delayedCompletionToken_ =
-        setTimeout(() => this.delayedCompletionTask_(), progressBarBlockMs);
+        window.setTimeout(() => this.delayedCompletionTask_(), progressBarBlockMs);
   }

   /**
@@ -230,7 +230,7 @@ export class PasswordsExportDialogElement extends
                                chrome.passwordsPrivate.PasswordExportProgress) {
     if (progress.status === ProgressStatus.IN_PROGRESS) {
       this.progressTaskToken_ =
-          setTimeout(() => this.progressTask_(), progressBarDelayMs);
+          window.setTimeout(() => this.progressTask_(), progressBarDelayMs);
       return;
     }
     if (progress.status === ProgressStatus.SUCCEEDED) {
