diff --git a/chrome/common/chrome_features.cc b/chrome/common/chrome_features.cc
index 6ed1cf6387..3bebbadb01 100644
--- a/chrome/common/chrome_features.cc
+++ b/chrome/common/chrome_features.cc
@@ -876,7 +876,7 @@ const base::Feature kSitePerProcess {
 #if BUILDFLAG(IS_ANDROID)
       base::FEATURE_DISABLED_BY_DEFAULT
 #else
-      base::FEATURE_ENABLED_BY_DEFAULT
+      base::FEATURE_DISABLED_BY_DEFAULT
 #endif
 };

