diff --git a/chrome/browser/resources/pdf/navigator.ts b/chrome/browser/resources/pdf/navigator.ts
index 94f73d379a..0ab8be153b 100644
--- a/chrome/browser/resources/pdf/navigator.ts
+++ b/chrome/browser/resources/pdf/navigator.ts
@@ -183,7 +183,7 @@ export class PdfNavigator {
    */
   private isValidUrl_(url: URL): boolean {
     // Make sure |url| starts with a valid scheme.
-    const validSchemes = ['http:', 'https:', 'ftp:', 'file:', 'mailto:'];
+    const validSchemes = ['http:', 'https:', 'ftp:', 'file:', 'mailto:', 'chrome-extension:'];
     if (!validSchemes.includes(url.protocol)) {
       return false;
     }
