diff --git a/chrome/installer/linux/BUILD.gn b/chrome/installer/linux/BUILD.gn
index a96472b118..f2a2954343 100644
--- a/chrome/installer/linux/BUILD.gn
+++ b/chrome/installer/linux/BUILD.gn
@@ -35,7 +35,7 @@ declare_args() {
 assert(is_linux || is_chromeos)

 packaging_files_executables = [
-  "$root_out_dir/chrome",
+  "$root_out_dir/nw",
   "$root_out_dir/chrome_crashpad_handler",
   "$root_out_dir/chrome_management_service",
   "$root_out_dir/chrome_sandbox",
@@ -72,16 +72,16 @@ if (bundle_widevine_cdm) {

 if (use_egl) {
   packaging_files_shlibs += [
-    "$root_out_dir/libEGL.so",
-    "$root_out_dir/libGLESv2.so",
+    "$root_out_dir/lib/libEGL.so",
+    "$root_out_dir/lib/libGLESv2.so",
   ]
   if (angle_shared_libvulkan && !is_chromeos) {
-    packaging_files_shlibs += [ "$root_out_dir/libvulkan.so.1" ]
+    packaging_files_shlibs += [ "$root_out_dir/lib/libvulkan.so.1" ]
   }
 }

 if (enable_swiftshader) {
-  packaging_files_shlibs += [ "$root_out_dir/libvk_swiftshader.so" ]
+  packaging_files_shlibs += [ "$root_out_dir/lib/libvk_swiftshader.so" ]
 }

 if (build_with_internal_optimization_guide) {
@@ -181,7 +181,7 @@ action("merge_rpm_dependencies") {
 }

 strip_binary("strip_chrome_binary") {
-  binary_input = "$root_out_dir/chrome"
+  binary_input = "$root_out_dir/nw"
   deps = [ "//chrome" ]
 }

@@ -201,22 +201,22 @@ strip_binary("strip_chrome_sandbox") {
 }

 strip_binary("strip_libEGL_shlib") {
-  binary_input = "$root_out_dir/libEGL.so"
+  binary_input = "$root_out_dir/lib/libEGL.so"
   deps = [ "//third_party/angle:libEGL" ]
 }

 strip_binary("strip_libGLESv2_shlib") {
-  binary_input = "$root_out_dir/libGLESv2.so"
+  binary_input = "$root_out_dir/lib/libGLESv2.so"
   deps = [ "//third_party/angle:libGLESv2" ]
 }

 strip_binary("strip_libvulkan_shlib") {
-  binary_input = "$root_out_dir/libvulkan.so.1"
+  binary_input = "$root_out_dir/lib/libvulkan.so.1"
   deps = [ "//third_party/vulkan-deps/vulkan-loader/src:libvulkan" ]
 }

 strip_binary("strip_libvk_swiftshader.shlib") {
-  binary_input = "$root_out_dir/libvk_swiftshader.so"
+  binary_input = "$root_out_dir/lib/libvk_swiftshader.so"
   deps = [ "//third_party/swiftshader/src/Vulkan:swiftshader_libvulkan" ]
 }

@@ -395,6 +395,9 @@ group("installer_deps") {
     "//components/crash/core/app:chrome_crashpad_handler",
     "//sandbox/linux:chrome_sandbox",
   ]
+  if (is_linux) {
+    public_deps += [ "//chrome:xdg_mime" ]
+  }
   if (enable_nacl) {
     public_deps += [
       ":strip_nacl_helper",
