diff --git a/chrome/browser/ui/views/frame/browser_view_layout.h b/chrome/browser/ui/views/frame/browser_view_layout.h
index 4b21a2d350..08a9843c3a 100644
--- a/chrome/browser/ui/views/frame/browser_view_layout.h
+++ b/chrome/browser/ui/views/frame/browser_view_layout.h
@@ -97,6 +97,8 @@ class BrowserViewLayout : public views::LayoutManager {
   // this function only controls where it's drawn when it is in fact drawn.
   void SetContentBorderBounds(
       const absl::optional<gfx::Rect>& region_capture_rect);
+  void set_menu_bar(views::View* menu_bar) { menu_bar_ = menu_bar; }
+  views::View* menu_bar() { return menu_bar_; }

   web_modal::WebContentsModalDialogHost* GetWebContentsModalDialogHost();

@@ -176,6 +178,7 @@ class BrowserViewLayout : public views::LayoutManager {
   // testing much easier.
   const raw_ptr<views::View> top_container_;
   const raw_ptr<TabStripRegionView> tab_strip_region_view_;
+  views::View* menu_bar_;
   const raw_ptr<views::View> toolbar_;
   const raw_ptr<InfoBarContainerView> infobar_container_;
   const raw_ptr<views::View> contents_container_;
