diff --git a/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_win.cc b/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_win.cc
index 35018ac6ea..8e3b41ba65 100644
--- a/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_win.cc
+++ b/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_win.cc
@@ -302,6 +302,11 @@ int BrowserDesktopWindowTreeHostWin::GetInitialShowState() const {
 bool BrowserDesktopWindowTreeHostWin::GetClientAreaInsets(
     gfx::Insets* insets,
     HMONITOR monitor) const {
+  //NWJS#7221, this is called during NCCALCSIZE handler
+  //without this, it will lead to a window with white bg shown
+  //initially during first ShowWindow call.
+  if (browser_view_->browser()->is_transparent())
+    return false;
   // Always use default insets for opaque frame.
   if (!ShouldUseNativeFrame())
     return false;
@@ -315,7 +320,7 @@ bool BrowserDesktopWindowTreeHostWin::GetClientAreaInsets(
     // In fullscreen mode there is no frame.
     *insets = gfx::Insets();
   } else {
-    const int frame_thickness = ui::GetFrameThickness(monitor);
+    const int frame_thickness = browser_view_->CanResize() ? ui::GetFrameThickness(monitor) : 1;
     // Reduce the non-client border size; UpdateDWMFrame() will instead extend
     // the border into the window client area. For maximized windows, Windows
     // outdents the window rect from the screen's client rect by
@@ -396,7 +401,8 @@ void BrowserDesktopWindowTreeHostWin::HandleDestroying() {
 void BrowserDesktopWindowTreeHostWin::HandleFrameChanged() {
   // Reinitialize the status bubble, since it needs to be initialized
   // differently depending on whether or not DWM composition is enabled
-  browser_view_->InitStatusBubble();
+  //browser_view_->InitStatusBubble();
+
   DesktopWindowTreeHostWin::HandleFrameChanged();
 }

@@ -498,7 +504,10 @@ bool BrowserDesktopWindowTreeHostWin::ShouldUseNativeFrame() const {
   // context of the BrowserView destructor.
   if (!browser_view_->browser())
     return false;
-
+  if (browser_view_->browser()->is_transparent())
+    return true;
+  if (browser_view_->browser()->is_frameless())
+    return false;
   if (IsOpaqueHostedAppFrame())
     return false;

@@ -514,8 +523,8 @@ bool BrowserDesktopWindowTreeHostWin::ShouldUseNativeFrame() const {

 bool BrowserDesktopWindowTreeHostWin::ShouldWindowContentsBeTransparent()
     const {
-  return !ShouldCustomDrawSystemTitlebar() &&
-         views::DesktopWindowTreeHostWin::ShouldWindowContentsBeTransparent();
+  return browser_view_->browser()->is_transparent() || (!ShouldCustomDrawSystemTitlebar() &&
+         views::DesktopWindowTreeHostWin::ShouldWindowContentsBeTransparent());
 }

 ////////////////////////////////////////////////////////////////////////////////
