diff --git a/chrome/test/data/webui/print_preview/destination_dialog_test.ts b/chrome/test/data/webui/print_preview/destination_dialog_test.ts
index 04ff6c4505..6f4694b21b 100644
--- a/chrome/test/data/webui/print_preview/destination_dialog_test.ts
+++ b/chrome/test/data/webui/print_preview/destination_dialog_test.ts
@@ -114,7 +114,7 @@ suite(destination_dialog_test.suiteName, function() {
         false /* pdfPrinterDisabled */, true /* isDriveMounted */,
         'FooDevice' /* printerName */,
         '' /* serializedDefaultDestinationSelectionRulesStr */,
-        [makeRecentDestination(destinations[4]!)] /* recentDestinations */);
+        [makeRecentDestination(destinations[4]!)] /* recentDestinations */, false);
     await whenPrinterListReady;
     whenPrinterListReady = nativeLayer.waitForGetPrinters(1);
     // This should trigger 1 new getPrinters() call, for extension printers.
@@ -139,7 +139,7 @@ suite(destination_dialog_test.suiteName, function() {
             '' /* serializedDefaultDestinationSelectionRulesStr */, [
               makeRecentDestination(destinations[4]!),
               makeRecentDestination(extensionDestinations[0]!),
-            ] /* recentDestinations */);
+            ] /* recentDestinations */, false);
         await whenAllPreloaded;
         finishSetup();
         flush();
