diff --git a/chrome/browser/devtools/devtools_window.h b/chrome/browser/devtools/devtools_window.h
index 82f0d8040c..7b6349b3a0 100644
--- a/chrome/browser/devtools/devtools_window.h
+++ b/chrome/browser/devtools/devtools_window.h
@@ -248,7 +248,10 @@ class DevToolsWindow : public DevToolsUIBindings::Delegate,

   content::WebContents* GetInspectedWebContents();

- private:
+  void Close();
+
+ public:
+
   friend class DevToolsWindowTesting;
   friend class DevToolsWindowCreationObserver;
   friend class HatsNextWebDialogBrowserTest;
@@ -299,7 +302,8 @@ class DevToolsWindow : public DevToolsUIBindings::Delegate,
                  std::unique_ptr<content::WebContents> main_web_contents,
                  DevToolsUIBindings* bindings,
                  content::WebContents* inspected_web_contents,
-                 bool can_dock);
+                 bool can_dock,
+                 bool headless = false);

   // External frontend is always undocked.
   static void OpenExternalFrontend(
@@ -319,7 +323,9 @@ class DevToolsWindow : public DevToolsUIBindings::Delegate,
                                 const std::string& settings,
                                 const std::string& panel,
                                 bool has_other_clients,
-                                bool browser_connection);
+                                bool browser_connection,
+                                content::WebContents* cdt_web_contents = nullptr);
+
   static GURL GetDevToolsURL(Profile* profile,
                              FrontendType frontend_type,
                              const std::string& frontend_url,
@@ -352,7 +358,7 @@ class DevToolsWindow : public DevToolsUIBindings::Delegate,
                           int opener_render_frame_id,
                           const std::string& frame_name,
                           const GURL& target_url,
-                          content::WebContents* new_contents) override;
+                          content::WebContents* new_contents, const std::u16string& nw_window_manifest) override;
   void CloseContents(content::WebContents* source) override;
   void ContentsZoomChange(bool zoom_in) override;
   void BeforeUnloadFired(content::WebContents* tab,
@@ -386,6 +392,7 @@ class DevToolsWindow : public DevToolsUIBindings::Delegate,
   void SetWhitelistedShortcuts(const std::string& message) override;
   void SetEyeDropperActive(bool active) override;
   void OpenNodeFrontend() override;
+ public:
   void InspectedContentsClosing() override;
   void OnLoadCompleted() override;
   void ReadyForTest() override;
@@ -450,6 +457,7 @@ class DevToolsWindow : public DevToolsUIBindings::Delegate,

   const bool can_dock_;
   bool close_on_detach_;
+  const bool headless_;
   LifeStage life_stage_;
   DevToolsToggleAction action_on_load_;
   DevToolsContentsResizingStrategy contents_resizing_strategy_;
