diff --git a/chrome/browser/ui/startup/startup_tab_provider.cc b/chrome/browser/ui/startup/startup_tab_provider.cc
index 6a74c525dd..b8dfb17f48 100644
--- a/chrome/browser/ui/startup/startup_tab_provider.cc
+++ b/chrome/browser/ui/startup/startup_tab_provider.cc
@@ -100,7 +100,7 @@ bool ValidateUrl(const GURL& url) {
   const GURL reset_settings_url =
       settings_url.Resolve(chrome::kResetProfileSettingsSubPage);
   url_points_to_an_approved_settings_page = url == reset_settings_url;
-#if BUILDFLAG(IS_WIN)
+#if 0 //BUILDFLAG(IS_WIN)
   // On Windows, also allow a hash for the Chrome Cleanup Tool.
   const GURL reset_settings_url_with_cct_hash = reset_settings_url.Resolve(
       std::string("#") + settings::ResetSettingsHandler::kCctResetSettingsHash);
