diff --git a/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.h b/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.h
index 7759883064..3d432f7276 100644
--- a/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.h
+++ b/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.h
@@ -131,6 +131,8 @@ class OpaqueBrowserFrameViewLayout : public views::LayoutManager {
   // Called explicitly from OpaqueBrowserFrameView so we can't group it with
   // the other overrides.
   gfx::Size GetMinimumSize(const views::View* host) const override;
+  gfx::Size GetMaximumSize(const views::View* host) const;
+  gfx::Size GetMinimumSizeHelper(const views::View* host, bool max) const;


  protected:
