diff --git a/chrome/browser/component_updater/registration.cc b/chrome/browser/component_updater/registration.cc
index 06c8a86440..cd4a787060 100644
--- a/chrome/browser/component_updater/registration.cc
+++ b/chrome/browser/component_updater/registration.cc
@@ -102,13 +102,15 @@
 namespace component_updater {

 void RegisterComponentsForUpdate() {
-  auto* const cus = g_browser_process->component_updater();

+#if 0
 #if BUILDFLAG(IS_WIN)
   RegisterRecoveryImprovedComponent(cus, g_browser_process->local_state());
 #endif  // BUILDFLAG(IS_WIN)
+#endif  // 0

 #if BUILDFLAG(IS_MAC)
+  auto* const cus = g_browser_process->component_updater();
   RegisterRecoveryImprovedComponent(cus, g_browser_process->local_state());
   RegisterRecoveryComponent(cus, g_browser_process->local_state());
 #endif  // BUILDFLAG(IS_MAC)
@@ -118,9 +120,10 @@ void RegisterComponentsForUpdate() {
 #endif

 #if BUILDFLAG(ENABLE_WIDEVINE_CDM_COMPONENT)
-  RegisterWidevineCdmComponent(cus);
+  //RegisterWidevineCdmComponent(cus);
 #endif  // BUILDFLAG(ENABLE_WIDEVINE_CDM_COMPONENT)

+#if 0
 #if BUILDFLAG(ENABLE_NACL) && !BUILDFLAG(IS_ANDROID)
 #if BUILDFLAG(IS_CHROMEOS_ASH)
   // PNaCl on Chrome OS is on rootfs and there is no need to download it. But
@@ -223,6 +226,7 @@ void RegisterComponentsForUpdate() {

   RegisterUrlParamClassificationComponent(cus);
   RegisterCommerceHeuristicsComponent(cus);
+#endif // disable component updater
 }

 }  // namespace component_updater
