diff --git a/chrome/browser/profiles/profile_io_data.cc b/chrome/browser/profiles/profile_io_data.cc
index 242c2e3f1f..08b5562c61 100644
--- a/chrome/browser/profiles/profile_io_data.cc
+++ b/chrome/browser/profiles/profile_io_data.cc
@@ -5,6 +5,8 @@
 #include "chrome/browser/profiles/profile_io_data.h"

 #include "base/feature_list.h"
+#include "content/nw/src/policy_cert_verifier.h"
+
 #include "base/logging.h"
 #include "base/strings/string_util.h"
 #include "build/build_config.h"
