diff --git a/chrome/browser/ui/javascript_dialogs/javascript_tab_modal_dialog_manager_delegate_desktop.cc b/chrome/browser/ui/javascript_dialogs/javascript_tab_modal_dialog_manager_delegate_desktop.cc
index 6f8c2dfebd..4e3c448b62 100644
--- a/chrome/browser/ui/javascript_dialogs/javascript_tab_modal_dialog_manager_delegate_desktop.cc
+++ b/chrome/browser/ui/javascript_dialogs/javascript_tab_modal_dialog_manager_delegate_desktop.cc
@@ -39,12 +39,14 @@ void JavaScriptTabModalDialogManagerDelegateDesktop::WillRunDialog() {
   // warnings until user interaction. If the current page has a delayed warning,
   // it'll have a user interaction observer attached. Show the warning
   // immediately in that case.
+#if 0
   safe_browsing::SafeBrowsingUserInteractionObserver* observer =
       safe_browsing::SafeBrowsingUserInteractionObserver::FromWebContents(
           web_contents_);
   if (observer) {
     observer->OnJavaScriptDialog();
   }
+#endif
 }

 void JavaScriptTabModalDialogManagerDelegateDesktop::DidCloseDialog() {
