diff --git a/chrome/browser/extensions/BUILD.gn b/chrome/browser/extensions/BUILD.gn
index 400d719eb1..9002667647 100644
--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -303,14 +303,14 @@ static_library("extensions") {
     "api/resources_private/resources_private_api.h",
     "api/runtime/chrome_runtime_api_delegate.cc",
     "api/runtime/chrome_runtime_api_delegate.h",
-    "api/safe_browsing_private/safe_browsing_private_api.cc",
-    "api/safe_browsing_private/safe_browsing_private_api.h",
-    "api/safe_browsing_private/safe_browsing_private_event_router.cc",
-    "api/safe_browsing_private/safe_browsing_private_event_router.h",
-    "api/safe_browsing_private/safe_browsing_private_event_router_factory.cc",
-    "api/safe_browsing_private/safe_browsing_private_event_router_factory.h",
-    "api/safe_browsing_private/safe_browsing_util.cc",
-    "api/safe_browsing_private/safe_browsing_util.h",
+    #"api/safe_browsing_private/safe_browsing_private_api.cc",
+    #"api/safe_browsing_private/safe_browsing_private_api.h",
+    #"api/safe_browsing_private/safe_browsing_private_event_router.cc",
+    #"api/safe_browsing_private/safe_browsing_private_event_router.h",
+    #"api/safe_browsing_private/safe_browsing_private_event_router_factory.cc",
+    #"api/safe_browsing_private/safe_browsing_private_event_router_factory.h",
+    #"api/safe_browsing_private/safe_browsing_util.cc",
+    #"api/safe_browsing_private/safe_browsing_util.h",
     "api/scripting/scripting_api.cc",
     "api/scripting/scripting_api.h",
     "api/search/search_api.cc",
@@ -742,7 +742,7 @@ static_library("extensions") {
     "//chrome/browser/web_applications/extensions",

     # TODO(crbug/925153): Remove this circular dependency.
-    "//chrome/browser/safe_browsing",
+    #"//chrome/browser/safe_browsing",
   ]

   # Since browser and browser_extensions actually depend on each other,
@@ -791,8 +791,8 @@ static_library("extensions") {
     "//chrome/browser/profiles:profile",
     "//chrome/browser/resource_coordinator:intervention_policy_database_proto",
     "//chrome/browser/resource_coordinator:mojo_bindings",
-    "//chrome/browser/safe_browsing",
-    "//chrome/browser/safe_browsing:metrics_collector",
+    #"//chrome/browser/safe_browsing",
+    #"//chrome/browser/safe_browsing:metrics_collector",
     "//chrome/browser/ui/tabs:tab_enums",
     "//chrome/browser/web_applications",
     "//components/cbor:cbor",
