diff --git a/chrome/browser/ui/views/constrained_web_dialog_delegate_views.cc b/chrome/browser/ui/views/constrained_web_dialog_delegate_views.cc
index 05f48f1996..eedc13e970 100644
--- a/chrome/browser/ui/views/constrained_web_dialog_delegate_views.cc
+++ b/chrome/browser/ui/views/constrained_web_dialog_delegate_views.cc
@@ -209,6 +209,7 @@ class ConstrainedWebDialogDelegateViews

   // ui::WebDialogWebContentsDelegate:
   void CloseContents(content::WebContents* source) override {
+    if (view_->GetWidget())
     view_->GetWidget()->Close();
   }

