diff --git a/chrome/test/data/webui/print_preview/destination_search_test.ts b/chrome/test/data/webui/print_preview/destination_search_test.ts
index 140adc7c06..1a402e3cac 100644
--- a/chrome/test/data/webui/print_preview/destination_search_test.ts
+++ b/chrome/test/data/webui/print_preview/destination_search_test.ts
@@ -44,7 +44,7 @@ suite(destination_search_test.suiteName, function() {
         false /* pdfPrinterDisabled */, false /* isDriveMounted */,
         'FooDevice' /* printerName */,
         '' /* serializedDefaultDestinationSelectionRulesStr */,
-        [] /* recentDestinations */);
+        [] /* recentDestinations */, false);

     dialog = document.createElement('print-preview-destination-dialog');
     dialog.destinationStore = destinationStore;
