diff --git a/chrome/browser/resources/print_preview/ui/destination_settings.ts b/chrome/browser/resources/print_preview/ui/destination_settings.ts
index 9c53c6a41d..32b750eb4f 100644
--- a/chrome/browser/resources/print_preview/ui/destination_settings.ts
+++ b/chrome/browser/resources/print_preview/ui/destination_settings.ts
@@ -255,7 +255,7 @@ export class PrintPreviewDestinationSettingsElement extends
         0, this.getRecentDestinationsDisplayCount_(recentDestinations));
     this.destinationStore_!.init(
         this.pdfPrinterDisabled_, isDriveMounted, defaultPrinter,
-        serializedDefaultDestinationRulesStr, recentDestinations);
+        serializedDefaultDestinationRulesStr, recentDestinations, false);
   }

   /**
