diff --git a/chrome/browser/extensions/api/sessions/sessions_api.cc b/chrome/browser/extensions/api/sessions/sessions_api.cc
index cbf6bc8845..750c5566b5 100644
--- a/chrome/browser/extensions/api/sessions/sessions_api.cc
+++ b/chrome/browser/extensions/api/sessions/sessions_api.cc
@@ -333,6 +333,9 @@ SessionsGetDevicesFunction::CreateWindowModel(

   api::windows::WindowState state = api::windows::WINDOW_STATE_NONE;
   switch (window.show_state) {
+    case ui::SHOW_STATE_HIDDEN:
+      state = api::windows::WINDOW_STATE_HIDDEN;
+      break;
     case ui::SHOW_STATE_NORMAL:
       state = api::windows::WINDOW_STATE_NORMAL;
       break;
