diff --git a/chrome/browser/extensions/api/web_navigation/web_navigation_api_helpers.cc b/chrome/browser/extensions/api/web_navigation/web_navigation_api_helpers.cc
index edba29be7f..d696bcf271 100644
--- a/chrome/browser/extensions/api/web_navigation/web_navigation_api_helpers.cc
+++ b/chrome/browser/extensions/api/web_navigation/web_navigation_api_helpers.cc
@@ -180,6 +180,7 @@ void DispatchOnDOMContentLoaded(content::WebContents* web_contents,
   details.tab_id = ExtensionTabUtil::GetTabId(web_contents);
   details.url = url.spec();
   details.process_id = frame_host->GetProcess()->GetID();
+  details.routing_id = frame_host->GetRoutingID();
   details.frame_id = ExtensionApiFrameIdMap::GetFrameId(frame_host);
   details.parent_frame_id =
       ExtensionApiFrameIdMap::GetParentFrameId(frame_host);
