diff --git a/chrome/services/file_util/BUILD.gn b/chrome/services/file_util/BUILD.gn
index 8a50b0932e..8156bb7556 100644
--- a/chrome/services/file_util/BUILD.gn
+++ b/chrome/services/file_util/BUILD.gn
@@ -38,11 +38,11 @@ source_set("file_util") {
     deps += ["//components/services/filesystem/public/mojom"]
   }

-  if (is_mac) {
+  if (false && is_mac) {
     deps += ["//chrome/utility/safe_browsing/mac"]
   }

-  if (safe_browsing_mode == 1) {
+  if (false && safe_browsing_mode == 1) {
     sources += [
       "safe_archive_analyzer.cc",
       "safe_archive_analyzer.h",
