diff --git a/chrome/browser/win/app_icon.cc b/chrome/browser/win/app_icon.cc
index ab3d600a5f..4b78a5f7c6 100644
--- a/chrome/browser/win/app_icon.cc
+++ b/chrome/browser/win/app_icon.cc
@@ -4,6 +4,8 @@

 #include "chrome/browser/win/app_icon.h"

+#include "content/nw/src/nw_content.h"
+
 #include "chrome/common/chrome_constants.h"
 #include "chrome/install_static/install_details.h"
 #include "third_party/skia/include/core/SkBitmap.h"
@@ -21,6 +23,11 @@ int GetAppIconResourceId() {
 }  // namespace

 HICON GetAppIcon() {
+#if 1
+  HICON ret = nw::GetWindowHIcon();
+  if (ret)
+    return ret;
+#endif
   // TODO(mgiuca): Use GetAppIconImageFamily/CreateExact instead of LoadIcon, to
   // get correct scaling. (See http://crbug.com/551256)
   const int icon_id = GetAppIconResourceId();
@@ -30,6 +37,11 @@ HICON GetAppIcon() {
 }

 HICON GetSmallAppIcon() {
+#if 1
+  HICON ret = nw::GetAppHIcon();
+  if (ret)
+    return ret;
+#endif
   // TODO(mgiuca): Use GetAppIconImageFamily/CreateExact instead of LoadIcon, to
   // get correct scaling. (See http://crbug.com/551256)
   const int icon_id = GetAppIconResourceId();
