diff --git a/chrome/browser/ui/views/fullscreen_control/fullscreen_control_host.cc b/chrome/browser/ui/views/fullscreen_control/fullscreen_control_host.cc
index fb40ed9dfb..975e4c1991 100644
--- a/chrome/browser/ui/views/fullscreen_control/fullscreen_control_host.cc
+++ b/chrome/browser/ui/views/fullscreen_control/fullscreen_control_host.cc
@@ -289,9 +289,13 @@ void FullscreenControlHost::OnPopupTimeout(
 }

 bool FullscreenControlHost::IsExitUiNeeded() {
+#if 1
+  return false;
+#else
   return browser_view_->IsFullscreen() &&
          browser_view_->CanUserExitFullscreen() &&
          browser_view_->ShouldHideUIForFullscreen();
+#endif
 }

 float FullscreenControlHost::CalculateCursorBufferHeight() const {
