diff --git a/chrome/chrome_elf/chrome_elf_x86.def b/chrome/chrome_elf/chrome_elf_x86.def
index 7aa7fa6c4d..63a002d4c8 100644
--- a/chrome/chrome_elf/chrome_elf_x86.def
+++ b/chrome/chrome_elf/chrome_elf_x86.def
@@ -1,7 +1,8 @@
 ; Copyright 2013 The Chromium Authors. All rights reserved.
 ; Use of this source code is governed by a BSD-style license that can be
 ; found in the LICENSE file.
-LIBRARY  "chrome_elf.dll"
+
+LIBRARY  "nw_elf.dll"

 EXPORTS
   ; When functions are added to this file, they must also be added to
@@ -42,3 +43,4 @@ EXPORTS

   ; From chrome/chrome_elf/third_party_dlls/main.cc
   IsThirdPartyInitialized
+  ElfGetReporterClient
