--- /dev/null
+++ b/chrome/browser/resources/nwjs_default_app/main.js
@@ -0,0 +1,7 @@
+var manifest = chrome.runtime.getManifest();
+var url = manifest.cmdlineUrl || 'nw_blank.html';
+chrome.app.runtime.onLaunched.addListener(function() {
+  chrome.app.window.create(
+      url,
+      {'id': 'nwjs_default_app', 'height': 550, 'width': 750});
+});
diff --git a/chrome/browser/resources/nwjs_default_app/manifest.json b/chrome/browser/resources/nwjs_default_app/manifest.json
new file mode 100644
index 0000000000..1e9c9e219e
