diff --git a/chrome/browser/extensions/api/scripting/scripting_api.cc b/chrome/browser/extensions/api/scripting/scripting_api.cc
index 23d0a10405..65d2558573 100644
--- a/chrome/browser/extensions/api/scripting/scripting_api.cc
+++ b/chrome/browser/extensions/api/scripting/scripting_api.cc
@@ -689,6 +689,7 @@ bool ScriptingExecuteScriptFunction::Execute(
                          : blink::mojom::UserActivationOption::kDoNotActivate,
           blink::mojom::PromiseResultOption::kAwait)),
       frame_scope, frame_ids, ScriptExecutor::MATCH_ABOUT_BLANK, run_location,
+      ScriptExecutor::ISOLATED_WORLD,
       ScriptExecutor::DEFAULT_PROCESS,
       /* webview_src */ GURL(),
       base::BindOnce(&ScriptingExecuteScriptFunction::OnScriptExecuted, this));
@@ -814,7 +815,9 @@ bool ScriptingInsertCSSFunction::Execute(
           std::move(sources), ConvertStyleOriginToCSSOrigin(injection_.origin),
           mojom::CSSInjection::Operation::kAdd)),
       frame_scope, frame_ids, ScriptExecutor::MATCH_ABOUT_BLANK,
-      kCSSRunLocation, ScriptExecutor::DEFAULT_PROCESS,
+      kCSSRunLocation,
+      ScriptExecutor::ISOLATED_WORLD,
+      ScriptExecutor::DEFAULT_PROCESS,
       /* webview_src */ GURL(),
       base::BindOnce(&ScriptingInsertCSSFunction::OnCSSInserted, this));

@@ -891,7 +894,9 @@ ExtensionFunction::ResponseAction ScriptingRemoveCSSFunction::Run() {
           std::move(sources), ConvertStyleOriginToCSSOrigin(injection.origin),
           mojom::CSSInjection::Operation::kRemove)),
       frame_scope, frame_ids, ScriptExecutor::MATCH_ABOUT_BLANK,
-      kCSSRunLocation, ScriptExecutor::DEFAULT_PROCESS,
+      kCSSRunLocation,
+      ScriptExecutor::ISOLATED_WORLD,
+      ScriptExecutor::DEFAULT_PROCESS,
       /* webview_src */ GURL(),
       base::BindOnce(&ScriptingRemoveCSSFunction::OnCSSRemoved, this));

