diff --git a/chrome/browser/ui/views/frame/browser_non_client_frame_view_mac.h b/chrome/browser/ui/views/frame/browser_non_client_frame_view_mac.h
index 172fac3c7e..47abf43a01 100644
--- a/chrome/browser/ui/views/frame/browser_non_client_frame_view_mac.h
+++ b/chrome/browser/ui/views/frame/browser_non_client_frame_view_mac.h
@@ -67,6 +67,7 @@ class BrowserNonClientFrameViewMac : public BrowserNonClientFrameView,
   gfx::Size GetMinimumSize() const override;
   void AddedToWidget() override;
   void PaintChildren(const views::PaintInfo& info) override;
+  gfx::Size GetMaximumSize() const override;

   // web_app::AppRegistrarObserver
   void OnAlwaysShowToolbarInFullscreenChanged(const web_app::AppId& app_id,
