diff --git a/chrome/browser/safe_browsing/chrome_safe_browsing_tab_observer_delegate.cc b/chrome/browser/safe_browsing/chrome_safe_browsing_tab_observer_delegate.cc
index 16dd38ac3d..bd7780eea6 100644
--- a/chrome/browser/safe_browsing/chrome_safe_browsing_tab_observer_delegate.cc
+++ b/chrome/browser/safe_browsing/chrome_safe_browsing_tab_observer_delegate.cc
@@ -43,10 +43,12 @@ bool ChromeSafeBrowsingTabObserverDelegate::DoesSafeBrowsingServiceExist() {
   return g_browser_process->safe_browsing_service();
 }

+#if 0
 std::unique_ptr<ClientSideDetectionHost>
 ChromeSafeBrowsingTabObserverDelegate::CreateClientSideDetectionHost(
     content::WebContents* web_contents) {
   return ChromeClientSideDetectionHostDelegate::CreateHost(web_contents);
 }
+#endif

 }  // namespace safe_browsing
