diff --git a/chrome/installer/util/google_update_settings.cc b/chrome/installer/util/google_update_settings.cc
index 973812e86c..83dac65d20 100644
--- a/chrome/installer/util/google_update_settings.cc
+++ b/chrome/installer/util/google_update_settings.cc
@@ -1,6 +1,7 @@
 // Copyright (c) 2012 The Chromium Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
+#pragma clang diagnostic ignored "-Wunused-function"

 #include "chrome/installer/util/google_update_settings.h"

@@ -213,10 +214,14 @@ GoogleUpdateSettings::CollectStatsConsentTaskRunner() {
 }

 bool GoogleUpdateSettings::GetCollectStatsConsent() {
+#if 1
+  return false;
+#else
   return GetCollectStatsConsentImpl(
              &install_static::GetClientStateKeyPath,
              &install_static::GetClientStateMediumKeyPath) ==
          google_update::TRISTATE_TRUE;
+#endif
 }

 bool GoogleUpdateSettings::SetCollectStatsConsent(bool consented) {
