diff --git a/chrome/browser/ui/webui/downloads/downloads_dom_handler.cc b/chrome/browser/ui/webui/downloads/downloads_dom_handler.cc
index 92837d9126..2980c13ce6 100644
--- a/chrome/browser/ui/webui/downloads/downloads_dom_handler.cc
+++ b/chrome/browser/ui/webui/downloads/downloads_dom_handler.cc
@@ -373,7 +373,7 @@ void DownloadsDOMHandler::OpenDuringScanningRequiringGesture(
   if (download) {
     DownloadItemModel model(download);
     model.SetOpenWhenComplete(true);
-#if BUILDFLAG(FULL_SAFE_BROWSING)
+#if 0 //BUILDFLAG(FULL_SAFE_BROWSING)
     model.CompleteSafeBrowsingScan();
 #endif
   }
@@ -387,11 +387,13 @@ void DownloadsDOMHandler::ReviewDangerousRequiringGesture(
   }

   CountDownloadsDOMEvents(DOWNLOADS_DOM_EVENT_REVIEW_DANGEROUS);
+#if 0
   download::DownloadItem* download = GetDownloadByStringId(id);
   if (download) {
     DownloadItemModel model(download);
     model.ReviewScanningVerdict(GetWebUIWebContents());
   }
+#endif
 }

 // DownloadsDOMHandler, private: --------------------------------------------
@@ -420,12 +422,14 @@ void DownloadsDOMHandler::FinalizeRemovals() {

 void DownloadsDOMHandler::ShowDangerPrompt(
     download::DownloadItem* dangerous_item) {
+#if 0
   DownloadDangerPrompt* danger_prompt = DownloadDangerPrompt::Create(
       dangerous_item, GetWebUIWebContents(), false,
       base::BindOnce(&DownloadsDOMHandler::DangerPromptDone,
                      weak_ptr_factory_.GetWeakPtr(), dangerous_item->GetId()));
   // danger_prompt will delete itself.
   DCHECK(danger_prompt);
+#endif
 }

 void DownloadsDOMHandler::DangerPromptDone(
