diff --git a/chrome/browser/ui/views/autofill/autofill_popup_base_view.cc b/chrome/browser/ui/views/autofill/autofill_popup_base_view.cc
index c88e2e355d..982d518f91 100644
--- a/chrome/browser/ui/views/autofill/autofill_popup_base_view.cc
+++ b/chrome/browser/ui/views/autofill/autofill_popup_base_view.cc
@@ -8,6 +8,10 @@
 #include <memory>
 #include <utility>

+#include "chrome/browser/apps/platform_apps/app_window_registry_util.h"
+#include "extensions/browser/app_window/app_window.h"
+#include "extensions/browser/app_window/native_app_window.h"
+
 #include "base/bind.h"
 #include "base/dcheck_is_on.h"
 #include "base/location.h"
