diff --git a/chrome/browser/chrome_browser_main_mac.mm b/chrome/browser/chrome_browser_main_mac.mm
index 7ef669d52b..02afb7949e 100644
--- a/chrome/browser/chrome_browser_main_mac.mm
+++ b/chrome/browser/chrome_browser_main_mac.mm
@@ -4,6 +4,9 @@

 #include "chrome/browser/chrome_browser_main_mac.h"

+#include "content/nw/src/nw_base.h"
+#include "base/strings/utf_string_conversions.h"
+
 #import <Cocoa/Cocoa.h>

 #include "base/bind.h"
@@ -87,6 +90,7 @@ void ChromeBrowserMainPartsMac::PreCreateMainMessageLoop() {
   [[KeystoneGlue defaultKeystoneGlue] registerWithKeystone];
 #endif  // BUILDFLAG(ENABLE_CHROMIUM_UPDATER)

+#if 0
   // Disk image installation is sort of a first-run task, so it shares the
   // no first run switches.
   //
@@ -107,6 +111,7 @@ void ChromeBrowserMainPartsMac::PreCreateMainMessageLoop() {
       exit(0);
     }
   }
+#endif

   // Create the app delegate. This object is intentionally leaked as a global
   // singleton. It is accessed through -[NSApp delegate].
@@ -114,7 +119,7 @@ void ChromeBrowserMainPartsMac::PreCreateMainMessageLoop() {
   [NSApp setDelegate:app_controller];

   chrome::BuildMainMenu(NSApp, app_controller,
-                        l10n_util::GetStringUTF16(IDS_PRODUCT_NAME), false);
+                        base::UTF8ToUTF16(nw::package()->GetName()), true);
   [app_controller mainMenuCreated];

   ui::WarmScreenCapture();
