diff --git a/chrome/test/base/test_browser_window_aura.h b/chrome/test/base/test_browser_window_aura.h
index b6d8e05273..cd06a733b5 100644
--- a/chrome/test/base/test_browser_window_aura.h
+++ b/chrome/test/base/test_browser_window_aura.h
@@ -27,6 +27,9 @@ class TestBrowserWindowAura : public TestBrowserWindow {
   ~TestBrowserWindowAura() override;

   // TestBrowserWindow overrides:
+  void UpdateDraggableRegions(
+         const std::vector<extensions::DraggableRegion>& regions) override;
+  SkRegion* GetDraggableRegion() override;
   gfx::NativeWindow GetNativeWindow() const override;
   void Show() override;
   void Hide() override;
