diff --git a/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc b/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc
index 4bd9021efb..47df0a4e29 100644
--- a/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc
+++ b/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc
@@ -6,6 +6,8 @@

 #include <memory>
 #include <utility>
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"

 #include "base/bind.h"
 #include "base/feature_list.h"
@@ -322,6 +324,9 @@ bool TabLifecycleUnitSource::TabLifecycleUnit::CanDiscard(
     DecisionDetails* decision_details) const {
   DCHECK(decision_details->reasons().empty());

+  Browser* browser = chrome::FindBrowserWithWebContents(web_contents());
+  if (browser && browser->is_type_popup())
+    return false;
   // Leave the |decision_details| empty and return immediately for "trivial"
   // rejection reasons. These aren't worth reporting about, as they have nothing
   // to do with the content itself.
