diff --git a/chrome/browser/translate/translate_frame_binder.cc b/chrome/browser/translate/translate_frame_binder.cc
index c4ee787787..2dfa2699d3 100644
--- a/chrome/browser/translate/translate_frame_binder.cc
+++ b/chrome/browser/translate/translate_frame_binder.cc
@@ -14,6 +14,8 @@ namespace translate {
 void BindContentTranslateDriver(
     content::RenderFrameHost* render_frame_host,
     mojo::PendingReceiver<translate::mojom::ContentTranslateDriver> receiver) {
+  return;
+#if 0
   // Only valid for the primary main frame.
   if (!render_frame_host->IsInPrimaryMainFrame())
     return;
@@ -29,6 +31,7 @@ void BindContentTranslateDriver(
     return;

   translate_client->translate_driver()->AddReceiver(std::move(receiver));
+#endif
 }

 }  // namespace translate
