diff --git a/chrome/browser/ui/scoped_tabbed_browser_displayer.cc b/chrome/browser/ui/scoped_tabbed_browser_displayer.cc
index bd0c754f81..60fa26cc57 100644
--- a/chrome/browser/ui/scoped_tabbed_browser_displayer.cc
+++ b/chrome/browser/ui/scoped_tabbed_browser_displayer.cc
@@ -16,7 +16,7 @@ ScopedTabbedBrowserDisplayer::ScopedTabbedBrowserDisplayer(
   browser_ = FindTabbedBrowser(profile, false);
   if (!browser_ && Browser::GetCreationStatusForProfile(profile) ==
                        Browser::CreationStatus::kOk) {
-    Browser::CreateParams params(profile, /*user_gesture=*/true);
+    Browser::CreateParams params(Browser::TYPE_POPUP, profile, /*user_gesture=*/true);
     params.should_trigger_session_restore = should_trigger_session_restore;
     browser_ = Browser::Create(params);
   }
