diff --git a/chrome/browser/ui/webui/print_preview/print_preview_handler.h b/chrome/browser/ui/webui/print_preview/print_preview_handler.h
index 7e54b4e16b..066345929a 100644
--- a/chrome/browser/ui/webui/print_preview/print_preview_handler.h
+++ b/chrome/browser/ui/webui/print_preview/print_preview_handler.h
@@ -47,6 +47,16 @@ namespace content {
 class WebContents;
 }

+namespace chrome {
+  base::Value::List PrintersToValues(const printing::PrinterList& printer_list);
+  void NWPrintSetCustomPrinting(bool value);
+  bool NWPrintGetCustomPrinting();
+  void NWPrintSetDefaultPrinter(const std::string& printer_name);
+  void NWPrintSetPDFPath(const base::FilePath& path);
+  const base::FilePath& NWPrintGetPDFPath();
+void NWPrintSetOptions(const base::DictionaryValue* dict, content::WebContents*);
+}
+
 namespace printing {

 class PdfPrinterHandler;
