diff --git a/chrome/browser/extensions/api/downloads/downloads_api.cc b/chrome/browser/extensions/api/downloads/downloads_api.cc
index 1d58b5ad00..92c718c0b1 100644
--- a/chrome/browser/extensions/api/downloads/downloads_api.cc
+++ b/chrome/browser/extensions/api/downloads/downloads_api.cc
@@ -1342,6 +1342,7 @@ void DownloadsAcceptDangerFunction::PromptOrWait(int download_id, int retries) {
     return;
   }
   RecordApiFunctions(DOWNLOADS_FUNCTION_ACCEPT_DANGER);
+#if 0
   // DownloadDangerPrompt displays a modal dialog using native widgets that the
   // user must either accept or cancel. It cannot be scripted.
   DownloadDangerPrompt* prompt = DownloadDangerPrompt::Create(
@@ -1352,6 +1353,8 @@ void DownloadsAcceptDangerFunction::PromptOrWait(int download_id, int retries) {
   if (on_prompt_created_ && !on_prompt_created_->is_null())
     std::move(*on_prompt_created_).Run(prompt);
   // Function finishes in DangerPromptCallback().
+#endif
+  DangerPromptCallback(download_id, DownloadDangerPrompt::ACCEPT);
 }

 void DownloadsAcceptDangerFunction::DangerPromptCallback(
