diff --git a/chrome/browser/enterprise/reporting/extension_info.cc b/chrome/browser/enterprise/reporting/extension_info.cc
index 9d1b8809dc..8ed1acf5e2 100644
--- a/chrome/browser/enterprise/reporting/extension_info.cc
+++ b/chrome/browser/enterprise/reporting/extension_info.cc
@@ -109,6 +109,8 @@ em::Extension_ExtensionType ConvertExtensionTypeToProto(
       return em::Extension_ExtensionType_TYPE_LEGACY_PACKAGED_APP;
     case extensions::Manifest::TYPE_PLATFORM_APP:
       return em::Extension_ExtensionType_TYPE_PLATFORM_APP;
+    case extensions::Manifest::TYPE_NWJS_APP:
+      return em::Extension_ExtensionType_TYPE_PLATFORM_APP;
     case extensions::Manifest::TYPE_LOGIN_SCREEN_EXTENSION:
       return em::Extension_ExtensionType_TYPE_LOGIN_SCREEN_EXTENSION;
     case extensions::Manifest::TYPE_CHROMEOS_SYSTEM_EXTENSION:
