diff --git a/chrome/browser/lifetime/application_lifetime.h b/chrome/browser/lifetime/application_lifetime.h
index 07d1f9ffb6..8d412ef567 100644
--- a/chrome/browser/lifetime/application_lifetime.h
+++ b/chrome/browser/lifetime/application_lifetime.h
@@ -66,13 +66,13 @@ bool IsAttemptingShutdown();

 #if !BUILDFLAG(IS_ANDROID)
 // Closes all browsers and if successful, quits.
-void CloseAllBrowsersAndQuit();
+void CloseAllBrowsersAndQuit(bool force = false, bool user_force = false);

 // Closes all browsers. If the session is ending the windows are closed
 // directly. Otherwise the windows are closed by way of posting a WM_CLOSE
 // message. This will quit the application if there is nothing other than
 // browser windows keeping it alive or the application is quitting.
-void CloseAllBrowsers();
+void CloseAllBrowsers(bool force = false, bool user_force = false);

 // If there are no browsers open and we aren't already shutting down,
 // initiate a shutdown.
