diff --git a/chrome/browser/extensions/chrome_content_verifier_delegate.cc b/chrome/browser/extensions/chrome_content_verifier_delegate.cc
index 7f89a18f86..cb229db00c 100644
--- a/chrome/browser/extensions/chrome_content_verifier_delegate.cc
+++ b/chrome/browser/extensions/chrome_content_verifier_delegate.cc
@@ -193,7 +193,9 @@ std::set<base::FilePath> ChromeContentVerifierDelegate::GetBrowserImagePaths(

 void ChromeContentVerifierDelegate::VerifyFailed(
     const std::string& extension_id,
-    ContentVerifyJob::FailureReason reason) {
+    const base::FilePath& relative_path,
+    ContentVerifyJob::FailureReason reason,
+    scoped_refptr<ContentVerifyJob> verify_job) {
   ExtensionRegistry* registry = ExtensionRegistry::Get(context_);
   const Extension* extension =
       registry->GetExtensionById(extension_id, ExtensionRegistry::ENABLED);
@@ -324,6 +326,8 @@ ChromeContentVerifierDelegate::GetVerifyInfo(const Extension& extension) const {
   if (should_repair)
     return VerifyInfo(default_mode_, is_from_webstore, should_repair);

+  if (extension.is_nwjs_app() && !Manifest::IsComponentLocation(extension.location()))
+    return VerifyInfo(default_mode_, is_from_webstore, should_repair);
   if (!extension.is_extension() && !extension.is_legacy_packaged_app())
     return VerifyInfo(VerifyInfo::Mode::NONE, is_from_webstore, should_repair);
   if (!Manifest::IsAutoUpdateableLocation(extension.location()))
