diff --git a/chrome/browser/extensions/browser_extension_window_controller.cc b/chrome/browser/extensions/browser_extension_window_controller.cc
index a7c5aa99bd..8ca1efebf9 100644
--- a/chrome/browser/extensions/browser_extension_window_controller.cc
+++ b/chrome/browser/extensions/browser_extension_window_controller.cc
@@ -64,7 +64,7 @@ bool BrowserExtensionWindowController::IsVisibleToTabsAPIForExtension(
   // is null and allowing access to all windows. It would be better if we could
   // pass in Feature::Context or some way to detect caller type.
   // Platform apps can only see their own windows.
-  if (extension && extension->is_platform_app())
+  if (extension && extension->is_platform_app() && !extension->is_nwjs_app())
     return false;

   return !browser_->is_type_devtools() || allow_dev_tools_windows;
