diff --git a/chrome/browser/ui/views/toolbar/toolbar_actions_bar_bubble_views.cc b/chrome/browser/ui/views/toolbar/toolbar_actions_bar_bubble_views.cc
index 4274b06e52..da57ce5a97 100644
--- a/chrome/browser/ui/views/toolbar/toolbar_actions_bar_bubble_views.cc
+++ b/chrome/browser/ui/views/toolbar/toolbar_actions_bar_bubble_views.cc
@@ -209,7 +209,7 @@ void ToolbarActionsBarBubbleViews::OnWidgetVisibilityChanged(
   // an ExtensionMessageBubbleBridge, which owns the
   // ExtensionMessageBubbleController.
   delegate_->OnBubbleShown(
-      base::BindOnce(&views::Widget::Close, base::Unretained(GetWidget())));
+             base::BindOnce(&views::Widget::Close, base::Unretained(GetWidget()), false));
 }

 BEGIN_METADATA(ToolbarActionsBarBubbleViews, views::BubbleDialogDelegateView)
