diff --git a/chrome/browser/enterprise/connectors/reporting/reporting_service_settings.h b/chrome/browser/enterprise/connectors/reporting/reporting_service_settings.h
index ccd953f091..24c3006f3c 100644
--- a/chrome/browser/enterprise/connectors/reporting/reporting_service_settings.h
+++ b/chrome/browser/enterprise/connectors/reporting/reporting_service_settings.h
@@ -43,6 +43,7 @@ class ReportingServiceSettings {

   // All events that the reporting connector supports.
   static const constexpr char* kAllReportingEvents[] = {
+#if 0
       extensions::SafeBrowsingPrivateEventRouter::kKeyPasswordReuseEvent,
       extensions::SafeBrowsingPrivateEventRouter::kKeyPasswordChangedEvent,
       extensions::SafeBrowsingPrivateEventRouter::kKeyDangerousDownloadEvent,
@@ -53,6 +54,7 @@ class ReportingServiceSettings {
       extensions::SafeBrowsingPrivateEventRouter::kKeyPasswordBreachEvent,
       kExtensionInstallEvent,
       kBrowserCrashEvent,
+#endif
   };

  private:
