diff --git a/chrome/utility/services.cc b/chrome/utility/services.cc
index de86afdfff..ecdf19acd4 100644
--- a/chrome/utility/services.cc
+++ b/chrome/utility/services.cc
@@ -153,12 +153,14 @@ auto RunUnzipper(mojo::PendingReceiver<unzip::mojom::Unzipper> receiver) {
   return std::make_unique<unzip::UnzipperImpl>(std::move(receiver));
 }

+#if 0
 auto RunLanguageDetectionService(
     mojo::PendingReceiver<language_detection::mojom::LanguageDetectionService>
         receiver) {
   return std::make_unique<language_detection::LanguageDetectionServiceImpl>(
       std::move(receiver));
 }
+#endif

 auto RunQRCodeGeneratorService(
     mojo::PendingReceiver<qrcode_generator::mojom::QRCodeGeneratorService>
@@ -439,7 +441,7 @@ void RegisterElevatedMainThreadServices(mojo::ServiceFactory& services) {
 void RegisterMainThreadServices(mojo::ServiceFactory& services) {
   services.Add(RunFilePatcher);
   services.Add(RunUnzipper);
-  services.Add(RunLanguageDetectionService);
+  //services.Add(RunLanguageDetectionService);
   services.Add(RunQRCodeGeneratorService);
   services.Add(RunWebAppOriginAssociationParser);

