diff --git a/chrome/browser/extensions/chrome_extension_web_contents_observer.h b/chrome/browser/extensions/chrome_extension_web_contents_observer.h
index 16bfff4efc..bf154961ff 100644
--- a/chrome/browser/extensions/chrome_extension_web_contents_observer.h
+++ b/chrome/browser/extensions/chrome_extension_web_contents_observer.h
@@ -13,6 +13,8 @@
 #include "extensions/browser/extension_web_contents_observer.h"
 #include "extensions/common/stack_frame.h"

+#include "components/zoom/zoom_observer.h"
+
 namespace content {
 class RenderFrameHost;
 }
@@ -24,6 +26,7 @@ namespace extensions {
 // renderers and updating autoplay policy.
 class ChromeExtensionWebContentsObserver
     : public ExtensionWebContentsObserver,
+      public zoom::ZoomObserver,
       public content::WebContentsUserData<ChromeExtensionWebContentsObserver> {
  public:
   ChromeExtensionWebContentsObserver(
@@ -32,6 +35,9 @@ class ChromeExtensionWebContentsObserver
       const ChromeExtensionWebContentsObserver&) = delete;

   ~ChromeExtensionWebContentsObserver() override;
+  // ZoomObserver implementation.
+  void OnZoomChanged(
+      const zoom::ZoomController::ZoomChangedEventData& data) override;

   // Creates and initializes an instance of this class for the given
   // |web_contents|, if it doesn't already exist.
