diff --git a/chrome/test/base/test_browser_window.cc b/chrome/test/base/test_browser_window.cc
index 84548a780a..788bfafa7c 100644
--- a/chrome/test/base/test_browser_window.cc
+++ b/chrome/test/base/test_browser_window.cc
@@ -75,6 +75,13 @@ bool TestBrowserWindow::TestLocationBar::IsInputTypedUrlWithoutScheme() const {

 // TestBrowserWindow ----------------------------------------------------------

+void TestBrowserWindow::UpdateDraggableRegions(
+                                               const std::vector<extensions::DraggableRegion>& regions) {
+}
+SkRegion* TestBrowserWindow::GetDraggableRegion() {
+  return nullptr;
+}
+
 TestBrowserWindow::TestBrowserWindow() {}

 TestBrowserWindow::~TestBrowserWindow() {}
