diff --git a/chrome/browser/browser_process.h b/chrome/browser/browser_process.h
index d7b9aa164f..37b5c6445c 100644
--- a/chrome/browser/browser_process.h
+++ b/chrome/browser/browser_process.h
@@ -235,6 +235,7 @@ class BrowserProcess {
   virtual void StartAutoupdateTimer() = 0;
 #endif

+
   virtual component_updater::ComponentUpdateService* component_updater() = 0;

   virtual MediaFileSystemRegistry* media_file_system_registry() = 0;
