diff --git a/chrome/utility/chrome_content_utility_client.cc b/chrome/utility/chrome_content_utility_client.cc
index 0d7e6405b1..a0f16a44ff 100644
--- a/chrome/utility/chrome_content_utility_client.cc
+++ b/chrome/utility/chrome_content_utility_client.cc
@@ -25,6 +25,10 @@
 #include "sandbox/policy/mojom/sandbox.mojom.h"
 #include "sandbox/policy/sandbox_type.h"

+#if defined(OS_WIN)
+#pragma warning(disable:4065)
+#endif
+
 namespace {

 base::LazyInstance<ChromeContentUtilityClient::NetworkBinderCreationCallback>::
