diff --git a/chrome/browser/extensions/extension_webkit_preferences.cc b/chrome/browser/extensions/extension_webkit_preferences.cc
index a4ecd58b45..a1c9e07863 100644
--- a/chrome/browser/extensions/extension_webkit_preferences.cc
+++ b/chrome/browser/extensions/extension_webkit_preferences.cc
@@ -33,12 +33,21 @@ void SetPreferences(const extensions::Extension* extension,
   }

   if (extension->is_platform_app()) {
-    webkit_prefs->databases_enabled = false;
-    webkit_prefs->local_storage_enabled = false;
-    webkit_prefs->sync_xhr_in_documents_enabled = false;
-    webkit_prefs->cookie_enabled = false;
-    webkit_prefs->target_blank_implies_no_opener_enabled_will_be_removed =
+    if (extension->is_nwjs_app()) {
+      webkit_prefs->databases_enabled = true;
+      webkit_prefs->local_storage_enabled = true;
+      webkit_prefs->sync_xhr_in_documents_enabled = true;
+      webkit_prefs->cookie_enabled = true;
+      webkit_prefs->target_blank_implies_no_opener_enabled_will_be_removed =
+        true;
+    }else{
+      webkit_prefs->databases_enabled = false;
+      webkit_prefs->local_storage_enabled = false;
+      webkit_prefs->sync_xhr_in_documents_enabled = false;
+      webkit_prefs->cookie_enabled = false;
+      webkit_prefs->target_blank_implies_no_opener_enabled_will_be_removed =
         false;
+    }
   }

   // Prevent font size preferences from affecting the PDF Viewer extension.
