diff --git a/chrome/browser/push_messaging/push_messaging_service_factory.cc b/chrome/browser/push_messaging/push_messaging_service_factory.cc
index 1b02dd1df4..15480c2b25 100644
--- a/chrome/browser/push_messaging/push_messaging_service_factory.cc
+++ b/chrome/browser/push_messaging/push_messaging_service_factory.cc
@@ -31,6 +31,12 @@ PushMessagingServiceImpl* PushMessagingServiceFactory::GetForProfile(
   if (context->IsOffTheRecord())
     return nullptr;

+  if (!instance_id::InstanceIDProfileService::IsInstanceIDEnabled()) {
+    LOG(WARNING) << "PushMessagingService could not be built because "
+                    "InstanceID is unexpectedly disabled";
+    return nullptr;
+  }
+
   return static_cast<PushMessagingServiceImpl*>(
       GetInstance()->GetServiceForBrowserContext(context, true));
 }
