diff --git a/chrome/browser/browser_features.cc b/chrome/browser/browser_features.cc
index 0d12e7d6d6..6ef8c0a7fc 100644
--- a/chrome/browser/browser_features.cc
+++ b/chrome/browser/browser_features.cc
@@ -29,7 +29,7 @@ const base::Feature kColorProviderRedirectionForThemeProvider = {
 // the browser exits.
 const base::Feature kDestroyProfileOnBrowserClose {
   "DestroyProfileOnBrowserClose",
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#if 0 //BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
       base::FEATURE_ENABLED_BY_DEFAULT
 };
 #else
