diff --git a/chrome/browser/devtools/devtools_ui_bindings.cc b/chrome/browser/devtools/devtools_ui_bindings.cc
index 642445c658..b38533ef7c 100644
--- a/chrome/browser/devtools/devtools_ui_bindings.cc
+++ b/chrome/browser/devtools/devtools_ui_bindings.cc
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.

+#include "extensions/browser/extension_protocols.h"
 #include "chrome/browser/devtools/devtools_ui_bindings.h"

 #include <stddef.h>
@@ -584,7 +585,8 @@ bool DevToolsUIBindings::IsValidFrontendURL(const GURL& url) {
     return true;
   }

-  return SanitizeFrontendURL(url).spec() == url.spec();
+  //NW: webview.showDevTools in container; webview-localfile case; changed in c2db881506f
+  return SanitizeFrontendURL(url).spec() == url.spec() || url == GURL(url::kAboutBlankURL);
 }

 bool DevToolsUIBindings::IsValidRemoteFrontendURL(const GURL& url) {
@@ -867,6 +869,15 @@ void DevToolsUIBindings::LoadNetworkResource(DispatchCallback callback,
   resource_request.headers.AddHeadersFromString(headers);

   NetworkResourceLoader::URLLoaderFactoryHolder url_loader_factory;
+  if (gurl.SchemeIs("chrome-extension")) {
+    content::RenderFrameHost* frame_host = web_contents()->GetPrimaryMainFrame();
+    mojo::PendingRemote<network::mojom::URLLoaderFactory> pending_remote
+      = extensions::CreateExtensionURLLoaderFactory(frame_host->GetProcess()->GetID(),
+                                                    frame_host->GetRoutingID());
+    url_loader_factory = network::SharedURLLoaderFactory::Create(
+        std::make_unique<network::WrapperPendingSharedURLLoaderFactory>(
+            std::move(pending_remote)));
+  } else
   if (gurl.SchemeIsFile()) {
     mojo::PendingRemote<network::mojom::URLLoaderFactory> pending_remote =
         content::CreateFileURLLoaderFactory(
@@ -964,8 +975,8 @@ void DevToolsUIBindings::AppendToFile(const std::string& url,
 }

 void DevToolsUIBindings::RequestFileSystems() {
-  CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
-        frontend_host_);
+  //CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
+  //      frontend_host_);
   base::ListValue file_systems_value;
   for (auto const& file_system : file_helper_->GetFileSystems())
     file_systems_value.Append(CreateFileSystemValue(file_system));
@@ -974,23 +985,23 @@ void DevToolsUIBindings::RequestFileSystems() {
 }

 void DevToolsUIBindings::AddFileSystem(const std::string& type) {
-  CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
-        frontend_host_);
+  //CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
+  //      frontend_host_);
   file_helper_->AddFileSystem(
       type, base::BindRepeating(&DevToolsUIBindings::ShowDevToolsInfoBar,
                                 weak_factory_.GetWeakPtr()));
 }

 void DevToolsUIBindings::RemoveFileSystem(const std::string& file_system_path) {
-  CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
-        frontend_host_);
+  //CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
+  //      frontend_host_);
   file_helper_->RemoveFileSystem(file_system_path);
 }

 void DevToolsUIBindings::UpgradeDraggedFileSystemPermissions(
     const std::string& file_system_url) {
-  CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
-        frontend_host_);
+  //CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
+  //      frontend_host_);
   file_helper_->UpgradeDraggedFileSystemPermissions(
       file_system_url,
       base::BindRepeating(&DevToolsUIBindings::ShowDevToolsInfoBar,
@@ -1002,8 +1013,8 @@ void DevToolsUIBindings::IndexPath(
     const std::string& file_system_path,
     const std::string& excluded_folders_message) {
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
-  CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
-        frontend_host_);
+  //CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
+  //      frontend_host_);
   if (!file_helper_->IsFileSystemAdded(file_system_path)) {
     IndexingDone(index_request_id, file_system_path);
     return;
@@ -1049,8 +1060,8 @@ void DevToolsUIBindings::SearchInPath(int search_request_id,
                                       const std::string& file_system_path,
                                       const std::string& query) {
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
-  CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
-        frontend_host_);
+  //CHECK(IsValidFrontendURL(web_contents_->GetLastCommittedURL()) &&
+  //      frontend_host_);
   if (!file_helper_->IsFileSystemAdded(file_system_path)) {
     SearchCompleted(search_request_id,
                     file_system_path,
@@ -1503,11 +1514,15 @@ void DevToolsUIBindings::SearchCompleted(
 void DevToolsUIBindings::ShowDevToolsInfoBar(
     const std::u16string& message,
     DevToolsInfoBarDelegate::Callback callback) {
+#if 1
+  std::move(callback).Run(true); // #4602
+#else
   if (!delegate_->GetInfoBarManager()) {
     std::move(callback).Run(false);
     return;
   }
   DevToolsInfoBarDelegate::Create(message, std::move(callback));
+#endif
 }

 void DevToolsUIBindings::AddDevToolsExtensionsToClient() {
