diff --git a/chrome/browser/google/google_update_settings_posix.cc b/chrome/browser/google/google_update_settings_posix.cc
index 2798348152..692d467ca7 100644
--- a/chrome/browser/google/google_update_settings_posix.cc
+++ b/chrome/browser/google/google_update_settings_posix.cc
@@ -76,6 +76,7 @@ bool GoogleUpdateSettings::GetCollectStatsConsent() {

 // static
 bool GoogleUpdateSettings::SetCollectStatsConsent(bool consented) {
+#if 0
 #if BUILDFLAG(IS_MAC)
   crash_reporter::SetUploadConsent(consented);
 // TODO(crbug.com/1052397): Revisit the macro expression once build flag switch
@@ -84,6 +85,7 @@ bool GoogleUpdateSettings::SetCollectStatsConsent(bool consented) {
   if (crash_reporter::IsCrashpadEnabled()) {
     crash_reporter::SetUploadConsent(consented);
   }
+#endif
 #endif

   base::FilePath consent_dir;
