diff --git a/chrome/browser/ui/apps/chrome_app_window_client.cc b/chrome/browser/ui/apps/chrome_app_window_client.cc
index bb7e1d4925..07e68c84e9 100644
--- a/chrome/browser/ui/apps/chrome_app_window_client.cc
+++ b/chrome/browser/ui/apps/chrome_app_window_client.cc
@@ -22,6 +22,7 @@
 #if BUILDFLAG(IS_CHROMEOS_ASH)
 #include "chrome/browser/ash/lock_screen_apps/state_controller.h"
 #endif
+#include "content/nw/src/nw_content.h"

 // TODO(jamescook): We probably shouldn't compile this class at all on Android.
 // See http://crbug.com/343612
@@ -47,8 +48,10 @@ extensions::AppWindow* ChromeAppWindowClient::CreateAppWindow(
   return NULL;
 #else
   Profile* profile = Profile::FromBrowserContext(context);
-  return new extensions::AppWindow(
+  extensions::AppWindow* ret = new extensions::AppWindow(
       context, std::make_unique<ChromeAppDelegate>(profile, true), extension);
+  nw::CreateAppWindowHook(ret);
+  return ret;
 #endif
 }

