diff --git a/chrome/test/data/webui/print_preview/destination_store_test.ts b/chrome/test/data/webui/print_preview/destination_store_test.ts
index b09759b16c..663f19d73a 100644
--- a/chrome/test/data/webui/print_preview/destination_store_test.ts
+++ b/chrome/test/data/webui/print_preview/destination_store_test.ts
@@ -108,7 +108,7 @@ suite(destination_store_test.suiteName, function() {
         initialSettings.pdfPrinterDisabled, !!initialSettings.isDriveMounted,
         initialSettings.printerName,
         initialSettings.serializedDefaultDestinationSelectionRulesStr,
-        recentDestinations);
+        recentDestinations, false);
     return expectPrinterFailure ? Promise.resolve() : Promise.race([
       nativeLayer.whenCalled('getPrinterCapabilities'),
       whenCapabilitiesReady,
