diff --git a/chrome/browser/extensions/api/preference/preference_helpers.cc b/chrome/browser/extensions/api/preference/preference_helpers.cc
index de1be982bf..4ee2c29b06 100644
--- a/chrome/browser/extensions/api/preference/preference_helpers.cc
+++ b/chrome/browser/extensions/api/preference/preference_helpers.cc
@@ -118,9 +118,12 @@ void DispatchEventToExtensionsImpl(Profile* profile,
       DCHECK(!args_list.empty());
       DCHECK(args_list[0].is_dict());

-      std::string level_of_control =
+      std::string level_of_control;
+      if (extension->is_nwjs_app())
+        level_of_control = kControlledByThisExtension;
+      else
+        level_of_control =
           level_getter.Run(profile, extension->id(), browser_pref, incognito);
-
       args_list[0].SetStringKey(kLevelOfControlKey, level_of_control);

       // If the extension is in incognito split mode,
