diff --git a/chrome/browser/download/download_danger_prompt.cc b/chrome/browser/download/download_danger_prompt.cc
index 44822083a7..c967d1940a 100644
--- a/chrome/browser/download/download_danger_prompt.cc
+++ b/chrome/browser/download/download_danger_prompt.cc
@@ -23,13 +23,15 @@ using safe_browsing::ClientDownloadResponse;
 using safe_browsing::ClientSafeBrowsingReportRequest;

 namespace {
-const char kDownloadDangerPromptPrefix[] = "Download.DownloadDangerPrompt";
+//const char kDownloadDangerPromptPrefix[] = "Download.DownloadDangerPrompt";
+
 }  // namespace

 void DownloadDangerPrompt::SendSafeBrowsingDownloadReport(
     ClientSafeBrowsingReportRequest::ReportType report_type,
     bool did_proceed,
     const download::DownloadItem& download) {
+#if 0
   safe_browsing::SafeBrowsingService* sb_service =
       g_browser_process->safe_browsing_service();
   Profile* profile = Profile::FromBrowserContext(
@@ -52,11 +54,13 @@ void DownloadDangerPrompt::SendSafeBrowsingDownloadReport(
     report->set_token(token);

   sb_service->SendDownloadReport(profile, std::move(report));
+#endif
 }

 void DownloadDangerPrompt::RecordDownloadDangerPrompt(
     bool did_proceed,
     const download::DownloadItem& download) {
+#if 0
   int64_t file_type_uma_value =
       safe_browsing::FileTypePolicies::GetInstance()->UmaValueForFile(
           download.GetTargetFilePath());
@@ -72,4 +76,5 @@ void DownloadDangerPrompt::RecordDownloadDangerPrompt(
                            download::GetDownloadDangerTypeString(danger_type)),
         file_type_uma_value);
   }
+#endif
 }
