diff --git a/chrome/app/chrome_main_mac.mm b/chrome/app/chrome_main_mac.mm
index 82c86521eb..73cd11cbdf 100644
--- a/chrome/app/chrome_main_mac.mm
+++ b/chrome/app/chrome_main_mac.mm
@@ -20,20 +20,14 @@
 #include "content/public/common/content_paths.h"
 #include "content/public/common/content_switches.h"

+#include "content/nw/src/nw_base.h"
+
 void SetUpBundleOverrides() {
   @autoreleasepool {
     base::mac::SetOverrideFrameworkBundlePath(chrome::GetFrameworkBundlePath());

     NSBundle* base_bundle = chrome::OuterAppBundle();
     base::mac::SetBaseBundleID([[base_bundle bundleIdentifier] UTF8String]);
-
-    base::FilePath child_exe_path =
-        chrome::GetFrameworkBundlePath().Append("Helpers").Append(
-            chrome::kHelperProcessExecutablePath);
-
-    // On the Mac, the child executable lives at a predefined location within
-    // the app bundle's versioned directory.
-    base::PathService::Override(content::CHILD_PROCESS_EXE, child_exe_path);
   }
 }

