diff --git a/chrome/browser/extensions/component_loader.h b/chrome/browser/extensions/component_loader.h
index ea0492aa05..7de901e625 100644
--- a/chrome/browser/extensions/component_loader.h
+++ b/chrome/browser/extensions/component_loader.h
@@ -44,6 +44,9 @@ class ComponentLoader {
     return component_extensions_.size();
   }

+  std::string GetExtensionID(
+                             int manifest_resource_id,
+                             const base::FilePath& root_directory);
   // Creates and loads all registered component extensions.
   void LoadAll();

