diff --git a/chrome/common/logging_chrome.cc b/chrome/common/logging_chrome.cc
index 2bf9783eee..7b7c2e5722 100644
--- a/chrome/common/logging_chrome.cc
+++ b/chrome/common/logging_chrome.cc
@@ -20,6 +20,7 @@
 #define IPC_LOG_TABLE_ADD_ENTRY(msg_id, logger) \
     content::RegisterIPCLogger(msg_id, logger)
 #include "chrome/common/all_messages.h"
+#include "extensions/common/extension_messages.h"
 #endif

 #if BUILDFLAG(IS_WIN)
