diff --git a/chrome/browser/metrics/desktop_session_duration/desktop_profile_session_durations_service_factory.cc b/chrome/browser/metrics/desktop_session_duration/desktop_profile_session_durations_service_factory.cc
index 1338c388a9..40b1fd5b62 100644
--- a/chrome/browser/metrics/desktop_session_duration/desktop_profile_session_durations_service_factory.cc
+++ b/chrome/browser/metrics/desktop_session_duration/desktop_profile_session_durations_service_factory.cc
@@ -53,6 +53,8 @@ DesktopProfileSessionDurationsServiceFactory::
 KeyedService*
 DesktopProfileSessionDurationsServiceFactory::BuildServiceInstanceFor(
     content::BrowserContext* context) const {
+  return nullptr;
+#if 0
   Profile* profile = Profile::FromBrowserContext(context);

 // On Ash and Lacros IsGuestSession and IsRegularProfile() are not mutually
@@ -73,6 +75,7 @@ DesktopProfileSessionDurationsServiceFactory::BuildServiceInstanceFor(
       IdentityManagerFactory::GetForProfile(profile);
   return new DesktopProfileSessionDurationsService(
       profile->GetPrefs(), sync_service, identity_manager, tracker);
+#endif
 }

 }  // namespace metrics
