diff --git a/chrome/browser/web_applications/os_integration/web_app_shortcut_mac.mm b/chrome/browser/web_applications/os_integration/web_app_shortcut_mac.mm
index b703ce74a4..3a7f7e70b1 100644
--- a/chrome/browser/web_applications/os_integration/web_app_shortcut_mac.mm
+++ b/chrome/browser/web_applications/os_integration/web_app_shortcut_mac.mm
@@ -208,11 +208,14 @@ std::set<std::string> GetFileHandlerExtensionsWithoutDot(
 }

 bool AppShimCreationDisabledForTest() {
+  return true;
+#if 0
   // Disable app shims in tests if the shortcut folder is not set.
   // Because shims created in ~/Applications will not be cleaned up.
   return base::CommandLine::ForCurrentProcess()->HasSwitch(
              switches::kTestType) &&
          !GetShortcutOverrideForTesting();
+#endif
 }

 bool AppShimRevealDisabledForTest() {
