diff --git a/chrome/browser/ui/views/status_icons/status_tray_win.h b/chrome/browser/ui/views/status_icons/status_tray_win.h
index 7b30415287..bb9e0c2431 100644
--- a/chrome/browser/ui/views/status_icons/status_tray_win.h
+++ b/chrome/browser/ui/views/status_icons/status_tray_win.h
@@ -41,6 +41,9 @@ class StatusTrayWin : public StatusTray {
   LRESULT CALLBACK
       WndProc(HWND hwnd, UINT message, WPARAM wparam, LPARAM lparam);

+  StatusIconWin* key_handler() { return key_handler_; }
+  void set_key_handler(StatusIconWin* win) { key_handler_ = win; }
+
  protected:
   // Overriden from StatusTray:
   std::unique_ptr<StatusIcon> CreatePlatformStatusIcon(
@@ -50,7 +53,7 @@ class StatusTrayWin : public StatusTray {

  private:
   FRIEND_TEST_ALL_PREFIXES(StatusTrayWinTest, EnsureVisibleTest);
-
+  StatusIconWin* key_handler_;
   // Static callback invoked when a message comes in to our messaging window.
   static LRESULT CALLBACK
       WndProcStatic(HWND hwnd, UINT message, WPARAM wparam, LPARAM lparam);
