diff --git a/chrome/browser/extensions/extension_browser_window_helper.cc b/chrome/browser/extensions/extension_browser_window_helper.cc
index f0f0f8ff93..23d5475802 100644
--- a/chrome/browser/extensions/extension_browser_window_helper.cc
+++ b/chrome/browser/extensions/extension_browser_window_helper.cc
@@ -97,6 +97,10 @@ void ExtensionBrowserWindowHelper::OnExtensionUnloaded(
   if (reason != extensions::UnloadedExtensionReason::TERMINATE)
     CleanUpTabsOnUnload(extension);

+  if (extension->is_nwjs_app()) {
+    browser_->window()->Close();
+    return;
+  }
   // If an extension page was active, the toolbar may need to be updated to hide
   // the extension name in the location icon.
   browser_->window()->UpdateToolbar(nullptr);
