diff --git a/chrome/common/extensions/chrome_extensions_client.cc b/chrome/common/extensions/chrome_extensions_client.cc
index 18ffa823ee..6a6ea213d8 100644
--- a/chrome/common/extensions/chrome_extensions_client.cc
+++ b/chrome/common/extensions/chrome_extensions_client.cc
@@ -39,6 +39,8 @@
 #include "ui/base/l10n/l10n_util.h"
 #include "url/gurl.h"

+#include "content/nw/src/api/generated_schemas.h"
+
 namespace extensions {

 namespace {
