diff --git a/chrome/browser/ui/views/frame/glass_browser_frame_view.h b/chrome/browser/ui/views/frame/glass_browser_frame_view.h
index 33eed7e20a..cab9e5d719 100644
--- a/chrome/browser/ui/views/frame/glass_browser_frame_view.h
+++ b/chrome/browser/ui/views/frame/glass_browser_frame_view.h
@@ -43,6 +43,7 @@ class GlassBrowserFrameView : public BrowserNonClientFrameView,
   gfx::Size GetMinimumSize() const override;
   void WindowControlsOverlayEnabledChanged() override;
   TabSearchBubbleHost* GetTabSearchBubbleHost() override;
+  gfx::Size GetMaximumSize() const override;

   // views::NonClientFrameView:
   gfx::Rect GetBoundsForClientView() const override;
