diff --git a/chrome/common/features.gni b/chrome/common/features.gni
index b41360983c..0ee9549fc5 100644
--- a/chrome/common/features.gni
+++ b/chrome/common/features.gni
@@ -88,6 +88,7 @@ use_brlapi = is_chromeos_ash
 # Every grit target in //chrome should apply these defines so that the
 # proper build flags can be set.
 chrome_grit_defines = [
+  "nwjs_sdk=$nwjs_sdk",
   "chrome_root_store_supported=$chrome_root_store_supported",
   "enable_arcore=$enable_arcore",
   "enable_background_mode=$enable_background_mode",
