diff --git a/chrome/browser/ui/views/location_bar/zoom_bubble_view.cc b/chrome/browser/ui/views/location_bar/zoom_bubble_view.cc
index ed7ea9ec70..eb79f3b1fc 100644
--- a/chrome/browser/ui/views/location_bar/zoom_bubble_view.cc
+++ b/chrome/browser/ui/views/location_bar/zoom_bubble_view.cc
@@ -147,6 +147,7 @@ views::View* GetAnchorViewForBrowser(Browser* browser) {
   return nullptr;
 }

+#if 0
 ImmersiveModeController* GetImmersiveModeControllerForBrowser(
     Browser* browser) {
   BrowserView* browser_view = BrowserView::GetBrowserViewForBrowser(browser);
@@ -172,6 +173,7 @@ void ParentToBrowser(Browser* browser,

   views::BubbleDialogDelegateView::CreateBubble(zoom_bubble);
 }
+#endif

 // Find the extension that initiated the zoom change, if any.
 const extensions::ExtensionZoomRequestClient* GetExtensionZoomRequestClient(
@@ -190,6 +192,7 @@ ZoomBubbleView* ZoomBubbleView::zoom_bubble_ = nullptr;
 // static
 void ZoomBubbleView::ShowBubble(content::WebContents* web_contents,
                                 DisplayReason reason) {
+#if 0
   Browser* browser = chrome::FindBrowserWithWebContents(web_contents);
   // |web_contents| could have been unloaded if a tab gets closed and a mouse
   // event arrives before the zoom icon gets hidden.
@@ -227,6 +230,7 @@ void ZoomBubbleView::ShowBubble(content::WebContents* web_contents,
   // disappears after a short timeout.
   zoom_bubble_->ShowForReason(reason, /* allow_refocus_alert */ false);
   zoom_bubble_->UpdateZoomIconVisibility();
+#endif
 }

 // static
