diff --git a/chrome/test/data/webui/print_preview/invalid_settings_browsertest.ts b/chrome/test/data/webui/print_preview/invalid_settings_browsertest.ts
index 210615a8df..1bd53bdc16 100644
--- a/chrome/test/data/webui/print_preview/invalid_settings_browsertest.ts
+++ b/chrome/test/data/webui/print_preview/invalid_settings_browsertest.ts
@@ -30,6 +30,7 @@ suite(invalid_settings_browsertest.suiteName, function() {
   let nativeLayer: NativeLayerStub;

   const initialSettings: NativeInitialSettings = {
+    nwPrintMode: false,
     isInKioskAutoPrintMode: false,
     isInAppKioskMode: false,
     thousandsDelimiter: ',',
