diff --git a/chrome/browser/lifetime/browser_close_manager.cc b/chrome/browser/lifetime/browser_close_manager.cc
index 8601ee1efb..ebd9aedb1a 100644
--- a/chrome/browser/lifetime/browser_close_manager.cc
+++ b/chrome/browser/lifetime/browser_close_manager.cc
@@ -4,6 +4,8 @@

 #include "chrome/browser/lifetime/browser_close_manager.h"

+#include "chrome/browser/ui/views/frame/browser_view.h"
+
 #include <iterator>
 #include <vector>

@@ -45,7 +47,7 @@ void ShowInProgressDownloads(Profile* profile) {

 }  // namespace

-BrowserCloseManager::BrowserCloseManager() : current_browser_(nullptr) {
+BrowserCloseManager::BrowserCloseManager(bool force, bool user_force) : current_browser_(nullptr), force_(force), user_force_(user_force) {
 }

 BrowserCloseManager::~BrowserCloseManager() {
@@ -173,7 +175,10 @@ void BrowserCloseManager::CloseBrowsers() {
   bool ignore_unload_handlers = browser_shutdown::ShouldIgnoreUnloadHandlers();

   for (auto* browser : browser_list_copy) {
-    browser->window()->Close();
+    if (force_)
+      browser->window()->ForceClose();
+    else if (BrowserView::GetBrowserViewForBrowser(browser)->NWCanClose(user_force_))
+      browser->window()->ForceClose();
     if (ignore_unload_handlers) {
       // This path is hit during logoff/power-down. In this case we won't get
       // a final message and so we force the browser to be deleted.
