diff --git a/chrome/browser/lifetime/browser_close_manager.h b/chrome/browser/lifetime/browser_close_manager.h
index 3387a0823c..cdb792c1fb 100644
--- a/chrome/browser/lifetime/browser_close_manager.h
+++ b/chrome/browser/lifetime/browser_close_manager.h
@@ -15,7 +15,7 @@ class Browser;
 // shutdown.
 class BrowserCloseManager : public base::RefCounted<BrowserCloseManager> {
  public:
-  BrowserCloseManager();
+  BrowserCloseManager(bool force = false, bool user_force = false);

   BrowserCloseManager(const BrowserCloseManager&) = delete;
   BrowserCloseManager& operator=(const BrowserCloseManager&) = delete;
@@ -60,6 +60,9 @@ class BrowserCloseManager : public base::RefCounted<BrowserCloseManager> {
   // The browser for which we are waiting for a callback to
   // OnBrowserReportCloseable.
   raw_ptr<Browser> current_browser_;
+  bool force_ = false;
+  bool user_force_ = false;
+
 };

 #endif  // CHROME_BROWSER_LIFETIME_BROWSER_CLOSE_MANAGER_H_
