diff --git a/chrome/browser/web_applications/extensions/web_app_extension_shortcut.cc b/chrome/browser/web_applications/extensions/web_app_extension_shortcut.cc
index 50b0e8d7f8..672ef6b602 100644
--- a/chrome/browser/web_applications/extensions/web_app_extension_shortcut.cc
+++ b/chrome/browser/web_applications/extensions/web_app_extension_shortcut.cc
@@ -221,6 +221,7 @@ bool ShouldCreateShortcutFor(ShortcutCreationReason reason,
   // cannot be shown in the launcher.
   if (extension->location() ==
           extensions::mojom::ManifestLocation::kComponent ||
+      extension->is_nwjs_app() ||
       !extensions::ui_util::CanDisplayInAppLauncher(extension, profile)) {
     return false;
   }
