diff --git a/chrome/browser/download/download_status_updater_mac.mm b/chrome/browser/download/download_status_updater_mac.mm
index 38bdf69f71..781c905d41 100644
--- a/chrome/browser/download/download_status_updater_mac.mm
+++ b/chrome/browser/download/download_status_updater_mac.mm
@@ -115,8 +115,12 @@ void DestroyNSProgress(download::DownloadItem* download) {

 void DownloadStatusUpdater::UpdateAppIconDownloadProgress(
     download::DownloadItem* download) {
-  // Always update overall progress in the Dock icon.

+  NSDockTile *dockTile = [NSApp dockTile];
+  if (dockTile.contentView && [dockTile.contentView.subviews count] > 0)
+    return;
+
+  // Always update overall progress.
   float progress = 0;
   int download_count = 0;
   bool progress_known = GetProgress(&progress, &download_count);
