diff --git a/chrome/chrome_elf/chrome_elf_main.h b/chrome/chrome_elf/chrome_elf_main.h
index 4272be3225..f5d53a02b9 100644
--- a/chrome/chrome_elf/chrome_elf_main.h
+++ b/chrome/chrome_elf/chrome_elf_main.h
@@ -22,7 +22,7 @@ bool GetUserDataDirectoryThunk(wchar_t* user_data_dir,
 // This function is a temporary workaround for https://crbug.com/655788. We
 // need to come up with a better way to initialize crash reporting that can
 // happen inside DllMain().
-void SignalInitializeCrashReporting();
+  void SignalInitializeCrashReporting(void*, void*);
 void SignalChromeElf();

 // Sets the metrics client ID in crash keys.
@@ -33,5 +33,5 @@ void SetMetricsClientId(const char* client_id);
 bool IsBrowserProcess();

 }  // extern "C"
-
+extern "C" void* ElfGetReporterClient();
 #endif  // CHROME_CHROME_ELF_CHROME_ELF_MAIN_H_
