diff --git a/chrome/common/extensions/api/BUILD.gn b/chrome/common/extensions/api/BUILD.gn
index 65b9f44029..5f10185ba9 100644
--- a/chrome/common/extensions/api/BUILD.gn
+++ b/chrome/common/extensions/api/BUILD.gn
@@ -89,7 +89,10 @@ generated_types("generated_api_types") {
 json_features("api_features") {
   feature_type = "APIFeature"
   method_name = "AddChromeAPIFeatures"
-  sources = [ "_api_features.json" ]
+  sources = [
+    "../../../../content/nw/src/api/_api_features.json",
+    "_api_features.json",
+  ]
   visibility = [ ":extensions_features" ]
 }

