diff --git a/components/media_router/browser/presentation/presentation_service_delegate_impl.cc b/components/media_router/browser/presentation/presentation_service_delegate_impl.cc
index a19652f904..3e6dbdfc6e 100644
--- a/components/media_router/browser/presentation/presentation_service_delegate_impl.cc
+++ b/components/media_router/browser/presentation/presentation_service_delegate_impl.cc
@@ -459,7 +459,9 @@ void PresentationServiceDelegateImpl::StartPresentation(
     const content::PresentationRequest& request,
     content::PresentationConnectionCallback success_cb,
     content::PresentationConnectionErrorCallback error_cb) {
+#if defined(NWJS_SDK)
   const auto& render_frame_host_id = request.render_frame_host_id;
+#endif
   const auto& presentation_urls = request.presentation_urls;
   if (presentation_urls.empty()) {
     std::move(error_cb).Run(PresentationError(
@@ -473,6 +475,8 @@ void PresentationServiceDelegateImpl::StartPresentation(
                           "Invalid presentation URL."));
     return;
   }
+
+#if defined(NWJS_SDK)
   auto presentation_context = std::make_unique<StartPresentationContext>(
       request,
       base::BindOnce(
@@ -488,6 +492,7 @@ void PresentationServiceDelegateImpl::StartPresentation(
       MediaRouterDialogController::GetOrCreateForWebContents(&GetWebContents());
   controller->ShowMediaRouterDialogForPresentation(
       std::move(presentation_context));
+#endif
 }

 void PresentationServiceDelegateImpl::ReconnectPresentation(
