diff --git a/components/nacl/zygote/nacl_fork_delegate_linux.cc b/components/nacl/zygote/nacl_fork_delegate_linux.cc
index 65d9da6fac..9329199715 100644
--- a/components/nacl/zygote/nacl_fork_delegate_linux.cc
+++ b/components/nacl/zygote/nacl_fork_delegate_linux.cc
@@ -396,7 +396,7 @@ bool NaClForkDelegate::GetTerminationStatus(pid_t pid, bool known_dead,
   write_pickle.WriteInt(pid);
   write_pickle.WriteBool(known_dead);

-  const std::vector<int> empty_fds;
+  const std::vector<int> empty_fds = std::vector<int>();
   char reply_buf[kNaClMaxIPCMessageLength];
   ssize_t reply_size = 0;
   bool got_reply =
