diff --git a/components/remote_cocoa/common/native_widget_ns_window.mojom b/components/remote_cocoa/common/native_widget_ns_window.mojom
index 6bb631508b..79573180d6 100644
--- a/components/remote_cocoa/common/native_widget_ns_window.mojom
+++ b/components/remote_cocoa/common/native_widget_ns_window.mojom
@@ -191,6 +191,11 @@ interface NativeWidgetNSWindow {
   // Miniaturize or deminiaturize the window.
   SetMiniaturized(bool miniaturized);

+  SetMaximized(bool maximized);
+  [Sync] IsMaximized() => (bool maximized);
+  [Sync] GetRestoredBounds() => (gfx.mojom.Rect bounds);
+  SetRestoredBounds(gfx.mojom.Rect bounds);
+
   // Called by NativeWidgetMac when the window size constraints change.
   SetSizeConstraints(gfx.mojom.Size min_size,
                      gfx.mojom.Size max_size,
