diff --git a/components/nacl/features.gni b/components/nacl/features.gni
index 5b43045821..7196b48c51 100644
--- a/components/nacl/features.gni
+++ b/components/nacl/features.gni
@@ -10,11 +10,8 @@ declare_args() {
   # Temporarily disable nacl on arm64 linux to get rid of compilation errors.
   # Intentionally and permanently disable nacl on arm64 mac.
   # TODO: When mipsel-nacl-clang is available, drop the exclusion.
-  enable_nacl =
-      checkout_nacl && target_os != "ios" && !is_android && !is_fuchsia &&
-      !is_castos && current_cpu != "mipsel" && current_cpu != "mips64el" &&
-      target_cpu != "arm64" && !(is_win && host_os != "win") &&
-      !(is_mac && (host_os != "mac" || target_cpu != "x64"))
+  enable_nacl = false
+
 }

 assert(!(is_win && host_os != "win") || !enable_nacl,
