diff --git a/components/renderer_context_menu/context_menu_content_type.cc b/components/renderer_context_menu/context_menu_content_type.cc
index 54afa8ac03..ea09f30d47 100644
--- a/components/renderer_context_menu/context_menu_content_type.cc
+++ b/components/renderer_context_menu/context_menu_content_type.cc
@@ -149,7 +149,7 @@ bool ContextMenuContentType::SupportsGroupInternal(int group) {
       return true;

     case ITEM_GROUP_DEVTOOLS_UNPACKED_EXT:
-      return false;
+      return true;

     case ITEM_GROUP_PRINT_PREVIEW:
 #if BUILDFLAG(ENABLE_PRINT_PREVIEW)
