diff --git a/components/cookie_config/cookie_store_util.cc b/components/cookie_config/cookie_store_util.cc
index cf65443b37..6925280c31 100644
--- a/components/cookie_config/cookie_store_util.cc
+++ b/components/cookie_config/cookie_store_util.cc
@@ -15,6 +15,8 @@ namespace cookie_config {
     BUILDFLAG(IS_CHROMEOS)
 namespace {

+bool g_should_encrypt = true;
+
 // Use the operating system's mechanisms to encrypt cookies before writing
 // them to persistent store.  Currently this only is done with desktop OS's
 // because ChromeOS and Android already protect the entire profile contents.
@@ -40,7 +42,7 @@ bool CookieOSCryptoDelegate::ShouldEncrypt() {
   // rewritten.
   return false;
 #else
-  return true;
+  return g_should_encrypt;
 #endif
 }

@@ -66,6 +68,11 @@ net::CookieCryptoDelegate* GetCookieCryptoDelegate() {
 }
 #else   // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_APPLE) || BUILDFLAG(IS_LINUX) ||
         // BUILDFLAG(IS_CHROMEOS)
+
+void SetEnableCookieCrypto(bool enable) {
+  g_should_encrypt = enable;
+}
+
 net::CookieCryptoDelegate* GetCookieCryptoDelegate() {
   return NULL;
 }
