diff --git a/components/safe_browsing/content/browser/BUILD.gn b/components/safe_browsing/content/browser/BUILD.gn
index 76f123bef9..237937cbed 100644
--- a/components/safe_browsing/content/browser/BUILD.gn
+++ b/components/safe_browsing/content/browser/BUILD.gn
@@ -141,7 +141,7 @@ source_set("unit_tests") {
   ]

   deps = [
-    ":client_side_detection",
+    #":client_side_detection",
     "//base:base",
     "//base/test:test_support",
     "//components/safe_browsing:buildflags",
