diff --git a/components/translate/content/browser/BUILD.gn b/components/translate/content/browser/BUILD.gn
index 1dfbe926fe..444c940d78 100644
--- a/components/translate/content/browser/BUILD.gn
+++ b/components/translate/content/browser/BUILD.gn
@@ -33,7 +33,7 @@ static_library("browser") {
     "//components/services/language_detection/public/cpp",
     "//components/services/language_detection/public/mojom",
     "//components/translate/core/browser:translate_model_service",
-    "//components/translate/core/language_detection:language_detection",
+    #"//components/translate/core/language_detection:language_detection",
     "//components/ukm/content",
     "//content/public/common",
     "//net",
