diff --git a/components/remote_cocoa/app_shim/views_nswindow_delegate.h b/components/remote_cocoa/app_shim/views_nswindow_delegate.h
index a32d85e291..d6656601de 100644
--- a/components/remote_cocoa/app_shim/views_nswindow_delegate.h
+++ b/components/remote_cocoa/app_shim/views_nswindow_delegate.h
@@ -54,10 +54,10 @@ REMOTE_COCOA_APP_SHIM_EXPORT
 // a backing store). |notification| is optional and can be set when redirecting
 // a notification such as NSApplicationDidHideNotification.
 - (void)onWindowOrderChanged:(NSNotification*)notification;
-
 // Notify that the system control tint changed.
 - (void)onSystemControlTintChanged:(NSNotification*)notification;

+- (void)onWindowWillStartLiveResize:(NSNotification*)notification;
 // Called on the delegate of a modal sheet when its modal session ends.
 - (void)sheetDidEnd:(NSWindow*)sheet
          returnCode:(NSInteger)returnCode
