diff --git a/components/favicon/core/favicon_driver_impl.cc b/components/favicon/core/favicon_driver_impl.cc
index d7a0c99677..f55935f786 100644
--- a/components/favicon/core/favicon_driver_impl.cc
+++ b/components/favicon/core/favicon_driver_impl.cc
@@ -33,7 +33,7 @@ FaviconDriverImpl::FaviconDriverImpl(CoreFaviconService* favicon_service)
         favicon_service_, this, FaviconDriverObserver::TOUCH_LARGEST));
   } else {
     handlers_.push_back(std::make_unique<FaviconHandler>(
-        favicon_service_, this, FaviconDriverObserver::NON_TOUCH_16_DIP));
+        favicon_service_, this, FaviconDriverObserver::NON_TOUCH_LARGEST));
   }
 }

