diff --git a/components/crash/core/app/crash_reporter_client.cc b/components/crash/core/app/crash_reporter_client.cc
index 463f92a6e5..cc930bfdb0 100644
--- a/components/crash/core/app/crash_reporter_client.cc
+++ b/components/crash/core/app/crash_reporter_client.cc
@@ -46,6 +46,10 @@ void CrashReporterClient::SetCrashReporterClientIdFromGUID(
     const std::string& client_guid) {}
 #endif

+void CrashReporterClient::SetUploadDump(bool upload) {
+  enable_upload_ = upload;
+}
+
 #if BUILDFLAG(IS_WIN)
 bool CrashReporterClient::ShouldCreatePipeName(
     const std::wstring& process_type) {
@@ -93,7 +97,7 @@ bool CrashReporterClient::GetShouldDumpLargerDumps() {
 }
 #endif

-#if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_MAC)
+#if BUILDFLAG(IS_POSIX)
 void CrashReporterClient::GetProductNameAndVersion(const char** product_name,
                                                    const char** version) {
 }
