diff --git a/components/sessions/core/tab_restore_service_impl.cc b/components/sessions/core/tab_restore_service_impl.cc
index 35f2ba5ea0..d3131684e5 100644
--- a/components/sessions/core/tab_restore_service_impl.cc
+++ b/components/sessions/core/tab_restore_service_impl.cc
@@ -181,6 +181,7 @@ enum SerializedWindowShowState : int {
   kSerializedShowStateMaximized = 3,
   kSerializedShowStateInactive = 4,
   kSerializedShowStateFullscreen = 5,
+  kSerializedShowStateHidden = 6,
 };

 // Converts a window show state to an integer. This function needs to be kept
@@ -199,6 +200,8 @@ int SerializeWindowShowState(ui::WindowShowState show_state) {
       return kSerializedShowStateInactive;
     case ui::SHOW_STATE_FULLSCREEN:
       return kSerializedShowStateFullscreen;
+    case ui::SHOW_STATE_HIDDEN:
+      return kSerializedShowStateHidden;
     case ui::SHOW_STATE_END:
       // This should never happen.
       NOTREACHED();
@@ -230,6 +233,9 @@ bool DeserializeWindowShowState(int show_state_int,
     case kSerializedShowStateFullscreen:
       *show_state = ui::SHOW_STATE_FULLSCREEN;
       return true;
+    case kSerializedShowStateHidden:
+      *show_state = ui::SHOW_STATE_HIDDEN;
+      return true;
     case kSerializedShowStateInvalid:
     default:
       // Ignore unknown values. This could happen if the data is corrupt.
