diff --git a/components/services/storage/dom_storage/local_storage_impl.cc b/components/services/storage/dom_storage/local_storage_impl.cc
index f4f63f02e6..ec3852ac47 100644
--- a/components/services/storage/dom_storage/local_storage_impl.cc
+++ b/components/services/storage/dom_storage/local_storage_impl.cc
@@ -146,13 +146,13 @@ StorageAreaImpl::Options createOptions() {

   // To avoid excessive IO we apply limits to the amount of data being written
   // and the frequency of writes.
-  static const size_t kMaxBytesPerHour = kPerStorageAreaQuota;
-  static constexpr int kMaxCommitsPerHour = 60;
+  //static const size_t kMaxBytesPerHour = std::numeric_limits<size_t>::max();
+  static constexpr int kMaxCommitsPerHour = std::numeric_limits<int32_t>::max();

   StorageAreaImpl::Options options;
-  options.max_size = kPerStorageAreaQuota + kPerStorageAreaOverQuotaAllowance;
+  options.max_size = std::numeric_limits<size_t>::max();
   options.default_commit_delay = kCommitDefaultDelaySecs;
-  options.max_bytes_per_hour = kMaxBytesPerHour;
+  options.max_bytes_per_hour = std::numeric_limits<int>::max();
   options.max_commits_per_hour = kMaxCommitsPerHour;
 #if BUILDFLAG(IS_ANDROID)
     options.cache_mode = StorageAreaImpl::CacheMode::KEYS_ONLY_WHEN_POSSIBLE;
