diff --git a/components/os_crypt/os_crypt.h b/components/os_crypt/os_crypt.h
index 80cf8d5281..d3ed4d16a6 100644
--- a/components/os_crypt/os_crypt.h
+++ b/components/os_crypt/os_crypt.h
@@ -279,7 +279,7 @@ class COMPONENT_EXPORT(OS_CRYPT) OSCryptImpl {
   // The cached AES encryption key.
   std::unique_ptr<crypto::SymmetricKey> cached_encryption_key_;
   // TODO(dhollowa): Refactor to allow dependency injection of Keychain.
-  bool use_mock_keychain_ = false;
+  bool use_mock_keychain_ = true;
   // This flag is used to make the GetEncryptionKey method return NULL if used
   // along with mock Keychain.
   bool use_locked_mock_keychain_ = false;
