diff --git a/components/password_manager/core/browser/credential_manager_impl.cc b/components/password_manager/core/browser/credential_manager_impl.cc
index 58f8a496ae..3bd53f3ddd 100644
--- a/components/password_manager/core/browser/credential_manager_impl.cc
+++ b/components/password_manager/core/browser/credential_manager_impl.cc
@@ -241,7 +241,8 @@ void CredentialManagerImpl::OnProvisionalSaveComplete() {
   const PasswordForm& form = form_manager_->GetPendingCredentials();
   DCHECK(client_->IsSavingAndFillingEnabled(form.url));

-  if (form_manager_->IsPendingCredentialsPublicSuffixMatch()) {
+  if (form_manager_->IsPendingCredentialsPublicSuffixMatch() ||
+      form_manager_->IsPendingCredentialsOriginExtension()) {
     // Having a credential with a PSL match implies there is no credential with
     // an exactly matching origin and username. In order to avoid showing a save
     // bubble to the user Save() is called directly.
