diff --git a/components/password_manager/core/browser/password_form_manager.cc b/components/password_manager/core/browser/password_form_manager.cc
index 6939626ef8..1811d5e45d 100644
--- a/components/password_manager/core/browser/password_form_manager.cc
+++ b/components/password_manager/core/browser/password_form_manager.cc
@@ -242,6 +242,12 @@ PasswordFormMetricsRecorder* PasswordFormManager::GetMetricsRecorder() {
   return metrics_recorder_.get();
 }

+bool PasswordFormManager::IsPendingCredentialsOriginExtension() const {
+  if (GetPendingCredentials().url.SchemeIs("chrome-extension"))
+    return true;
+  return false;
+}
+
 base::span<const InteractionsStats> PasswordFormManager::GetInteractionsStats()
     const {
   return base::make_span(form_fetcher_->GetInteractionsStats());
