diff --git a/components/javascript_dialogs/app_modal_dialog_manager.cc b/components/javascript_dialogs/app_modal_dialog_manager.cc
index 0f73eb2e20..a8824512c7 100644
--- a/components/javascript_dialogs/app_modal_dialog_manager.cc
+++ b/components/javascript_dialogs/app_modal_dialog_manager.cc
@@ -77,6 +77,8 @@ AppModalDialogManager::~AppModalDialogManager() = default;
 std::u16string AppModalDialogManager::GetTitle(
     content::WebContents* web_contents,
     const url::Origin& alerting_frame_origin) {
+  return std::u16string();
+#if 0
   // For extensions, show the extension name, but only if the origin of
   // the alert matches the top-level WebContents.
   std::string name;
@@ -142,6 +144,7 @@ std::u16string AppModalDialogManager::GetTitleImpl(
       is_same_origin_as_main_frame
           ? IDS_JAVASCRIPT_MESSAGEBOX_TITLE_NONSTANDARD_URL
           : IDS_JAVASCRIPT_MESSAGEBOX_TITLE_NONSTANDARD_URL_IFRAME);
+#endif
 }

 void AppModalDialogManager::RunJavaScriptDialog(
@@ -170,7 +173,7 @@ void AppModalDialogManager::RunJavaScriptDialog(
   AppModalDialogQueue::GetInstance()->AddDialog(new AppModalDialogController(
       web_contents, &javascript_dialog_extra_data_, dialog_title, dialog_type,
       message_text, default_prompt_text,
-      ShouldDisplaySuppressCheckbox(extra_data),
+      false, //ShouldDisplaySuppressCheckbox(extra_data),
       false,  // is_before_unload_dialog
       false,  // is_reload
       base::BindOnce(&AppModalDialogManager::OnDialogClosed,
