diff --git a/components/web_modal/web_contents_modal_dialog_manager.cc b/components/web_modal/web_contents_modal_dialog_manager.cc
index 8ada15ab6b..c9f0df4112 100644
--- a/components/web_modal/web_contents_modal_dialog_manager.cc
+++ b/components/web_modal/web_contents_modal_dialog_manager.cc
@@ -44,7 +44,7 @@ void WebContentsModalDialogManager::ShowDialogWithManager(

   if (child_dialogs_.size() == 1) {
     BlockWebContentsInteraction(true);
-    if (delegate_ && delegate_->IsWebContentsVisible(web_contents()))
+    if (!web_contents()->is_silent_printing() && delegate_ && delegate_->IsWebContentsVisible(web_contents()))
       child_dialogs_.back().manager->Show();
   }
 }
@@ -112,7 +112,8 @@ void WebContentsModalDialogManager::BlockWebContentsInteraction(bool blocked) {
     // The WebContents has already disconnected.
     return;
   }
-
+  if (!blocked && contents->is_silent_printing())
+    contents->set_silent_printing(false);
   contents->SetIgnoreInputEvents(blocked);
   if (delegate_)
     delegate_->SetWebContentsBlocked(contents, blocked);
