diff --git a/components/remote_cocoa/app_shim/views_nswindow_delegate.mm b/components/remote_cocoa/app_shim/views_nswindow_delegate.mm
index db01fcf505..a2b0e2e463 100644
--- a/components/remote_cocoa/app_shim/views_nswindow_delegate.mm
+++ b/components/remote_cocoa/app_shim/views_nswindow_delegate.mm
@@ -75,6 +75,10 @@
   _parent->OnVisibilityChanged();
 }

+- (void)onWindowWillStartLiveResize:(NSNotification*)notification {
+  _parent->OnWindowWillStartLiveResize();
+}
+
 - (void)onSystemControlTintChanged:(NSNotification*)notification {
   _parent->OnSystemControlTintChanged();
 }
