diff --git a/components/metrics/metrics_service.cc b/components/metrics/metrics_service.cc
index d5ce38da14..5d8180bdf4 100644
--- a/components/metrics/metrics_service.cc
+++ b/components/metrics/metrics_service.cc
@@ -245,7 +245,7 @@ MetricsService::MetricsService(MetricsStateManager* state_manager,
       client_(client),
       local_state_(local_state),
       recording_state_(UNSET),
-      test_mode_active_(false),
+      test_mode_active_(true),
       state_(CONSTRUCTED),
       idle_since_last_transmission_(false),
       session_id_(-1) {
@@ -294,9 +294,9 @@ void MetricsService::InitializeMetricsRecordingState() {
 }

 void MetricsService::Start() {
-  HandleIdleSinceLastTransmission(false);
+  test_mode_active_ = true;
   EnableRecording();
-  EnableReporting();
+  DisableReporting();
 }

 void MetricsService::StartRecordingForTests() {
