diff --git a/components/embedder_support/user_agent_utils.cc b/components/embedder_support/user_agent_utils.cc
index 97ddfb060d..1ad89078e2 100644
--- a/components/embedder_support/user_agent_utils.cc
+++ b/components/embedder_support/user_agent_utils.cc
@@ -4,6 +4,8 @@

 #include "components/embedder_support/user_agent_utils.h"

+#include "content/nw/src/common/nw_content_common_hooks.h"
+
 #include "base/command_line.h"
 #include "base/debug/stack_trace.h"
 #include "base/feature_list.h"
@@ -419,6 +421,11 @@ std::string GetUserAgent(
     LOG(WARNING) << "Ignored invalid value for flag --" << kUserAgent;
   }

+  std::string user_agent;
+  if (nw::GetUserAgentFromManifest(&user_agent)) {
+    return user_agent;
+  }
+
   if (base::FeatureList::IsEnabled(blink::features::kFullUserAgent))
     return GetFullUserAgent(force_major_to_minor);

