diff --git a/components/os_crypt/os_crypt_mac.mm b/components/os_crypt/os_crypt_mac.mm
index de92559dd2..3259b5e9cd 100644
--- a/components/os_crypt/os_crypt_mac.mm
+++ b/components/os_crypt/os_crypt_mac.mm
@@ -135,7 +135,7 @@ std::string OSCryptImpl::GetRawEncryptionKey() {

 void OSCryptImpl::SetRawEncryptionKey(const std::string& raw_key) {
   base::AutoLock auto_lock(OSCryptImpl::GetLock());
-  DCHECK(!cached_encryption_key_) << "Encryption key already set.";
+  //DCHECK(!cached_encryption_key_) << "Encryption key already set.";
   cached_encryption_key_ =
       crypto::SymmetricKey::Import(crypto::SymmetricKey::AES, raw_key);
   key_is_cached_ = true;
