diff --git a/components/gcm_driver/gcm_internals_helper.cc b/components/gcm_driver/gcm_internals_helper.cc
index 079092c8e0..9ff8b01e51 100644
--- a/components/gcm_driver/gcm_internals_helper.cc
+++ b/components/gcm_driver/gcm_internals_helper.cc
@@ -4,6 +4,7 @@

 #include "components/gcm_driver/gcm_internals_helper.h"

+#include "content/nw/src/nw_base.h"
 #include <memory>
 #include <utility>

@@ -121,7 +122,7 @@ base::Value::Dict SetGCMInternalsInfo(

     base::Value::Dict device_info;
     device_info.Set(kProfileServiceCreated, profile_service != nullptr);
-    device_info.Set(kGcmEnabled, true);
+    device_info.Set(kGcmEnabled, nw::gcm_enabled());
     device_info.Set(kGcmClientCreated, stats->gcm_client_created);
     device_info.Set(kGcmClientState, stats->gcm_client_state);
     device_info.Set(kConnectionClientCreated, stats->connection_client_created);
