diff --git a/components/network_time/network_time_tracker.cc b/components/network_time/network_time_tracker.cc
index 0a7cdec62c..6154f8b36b 100644
--- a/components/network_time/network_time_tracker.cc
+++ b/components/network_time/network_time_tracker.cc
@@ -64,7 +64,7 @@ const base::Feature kNetworkTimeServiceQuerying{
     "NetworkTimeServiceQuerying", base::FEATURE_DISABLED_BY_DEFAULT};
 #else
 const base::Feature kNetworkTimeServiceQuerying{
-    "NetworkTimeServiceQuerying", base::FEATURE_ENABLED_BY_DEFAULT};
+    "NetworkTimeServiceQuerying", base::FEATURE_DISABLED_BY_DEFAULT};
 #endif

 namespace {
