diff --git a/content/BUILD.gn b/content/BUILD.gn
index a098bc6e8b..52ef63957f 100644
--- a/content/BUILD.gn
+++ b/content/BUILD.gn
@@ -60,6 +60,7 @@ assert(!is_ios, "Chromium/iOS shouldn't use anything in //content")
 # linking problems) for when check is enabled.

 content_shared_components = [
+  "//content/nw:nw_base",
   "//content/gpu:gpu_sources",
   "//content/public/browser:browser_sources",
   "//content/public/child:child_sources",
@@ -77,6 +78,9 @@ if (is_component_build) {
   component("content") {
     public_deps =
         content_shared_components + [ "//content/public/app:app_sources" ]
+    deps = [
+      "//content/nw:nw_content",
+    ]
   }
 } else {
   group("content") {
