diff --git a/content/browser/renderer_host/navigation_request_info.cc b/content/browser/renderer_host/navigation_request_info.cc
index 7ef40dfb24..4869eebf48 100644
--- a/content/browser/renderer_host/navigation_request_info.cc
+++ b/content/browser/renderer_host/navigation_request_info.cc
@@ -30,7 +30,8 @@ NavigationRequestInfo::NavigationRequestInfo(
     const absl::optional<std::vector<net::SourceStream::SourceType>>&
         devtools_accepted_stream_types,
     bool is_pdf,
-    WeakDocumentPtr initiator_document)
+    WeakDocumentPtr initiator_document,
+    bool nw_trusted)
     : common_params(std::move(common_params)),
       begin_params(std::move(begin_params)),
       sandbox_flags(sandbox_flags),
@@ -49,7 +50,8 @@ NavigationRequestInfo::NavigationRequestInfo(
       client_security_state(std::move(client_security_state)),
       devtools_accepted_stream_types(devtools_accepted_stream_types),
       is_pdf(is_pdf),
-      initiator_document(std::move(initiator_document)) {}
+      initiator_document(std::move(initiator_document)),
+      nw_trusted(nw_trusted) {}

 NavigationRequestInfo::~NavigationRequestInfo() {}

