diff --git a/content/common/renderer_host.mojom b/content/common/renderer_host.mojom
index 509b6fb5f8..197255e17c 100644
--- a/content/common/renderer_host.mojom
+++ b/content/common/renderer_host.mojom
@@ -23,4 +23,9 @@ interface RendererHost {

   // Sends a string to be recorded by UserMetrics.
   RecordUserMetricsAction(string action);
+
+  // Gets the list of proxies to use for |url|, as a semicolon delimited list
+  // of "<TYPE> <HOST>:<PORT>" | "DIRECT".
+  [Sync]
+  ResolveProxy(url.mojom.Url url) => (string? proxy_list);
 };
