diff --git a/content/public/common/content_switches.h b/content/public/common/content_switches.h
index 15488266b7..ae091bc697 100644
--- a/content/public/common/content_switches.h
+++ b/content/public/common/content_switches.h
@@ -52,7 +52,12 @@ CONTENT_EXPORT extern const char kDisableWebGL2[];
 CONTENT_EXPORT extern const char kDisableFileSystem[];
 CONTENT_EXPORT extern const char kDisableGestureRequirementForPresentation[];
 CONTENT_EXPORT extern const char kDisableGpu[];
+CONTENT_EXPORT extern const char kDisableTransparency[];
+CONTENT_EXPORT extern const char kForceCpuDraw[];
 CONTENT_EXPORT extern const char kDisableGpuCompositing[];
+CONTENT_EXPORT extern const char kDisableRAFThrottling[];
+CONTENT_EXPORT extern const char kEnableNodeWorker[];
+CONTENT_EXPORT extern const char kNWAppPath[];
 CONTENT_EXPORT extern const char kDisableGpuEarlyInit[];
 CONTENT_EXPORT extern const char kDisableGpuMemoryBufferCompositorResources[];
 CONTENT_EXPORT extern const char kDisableGpuMemoryBufferVideoFrames[];
