diff --git a/content/browser/renderer_host/navigation_request_info.h b/content/browser/renderer_host/navigation_request_info.h
index 1c9a6524ca..9a46e6aa29 100644
--- a/content/browser/renderer_host/navigation_request_info.h
+++ b/content/browser/renderer_host/navigation_request_info.h
@@ -47,7 +47,8 @@ struct CONTENT_EXPORT NavigationRequestInfo {
       const absl::optional<std::vector<net::SourceStream::SourceType>>&
           devtools_accepted_stream_types,
       bool is_pdf,
-      WeakDocumentPtr initiator_document);
+      WeakDocumentPtr initiator_document,
+      bool nw_trust = false);
   NavigationRequestInfo(const NavigationRequestInfo& other) = delete;
   ~NavigationRequestInfo();

@@ -130,6 +131,8 @@ struct CONTENT_EXPORT NavigationRequestInfo {

   // The initiator document, if still available.
   const WeakDocumentPtr initiator_document;
+
+  bool nw_trusted;
 };

 }  // namespace content
