diff --git a/content/renderer/media/render_media_client.cc b/content/renderer/media/render_media_client.cc
index a0b92623bd..0a9993f0e0 100644
--- a/content/renderer/media/render_media_client.cc
+++ b/content/renderer/media/render_media_client.cc
@@ -24,7 +24,7 @@ namespace {
     (BUILDFLAG(IS_WIN) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_LINUX))
 #define NEEDS_PROFILE_UPDATER 1
 #else
-#define NEEDS_PROFILE_UPDATER 0
+#define NEEDS_PROFILE_UPDATER 1
 #endif

 #if NEEDS_PROFILE_UPDATER
