diff --git a/content/renderer/agent_scheduling_group.cc b/content/renderer/agent_scheduling_group.cc
index 909fe4e895..6aa751033b 100644
--- a/content/renderer/agent_scheduling_group.cc
+++ b/content/renderer/agent_scheduling_group.cc
@@ -294,7 +294,7 @@ blink::WebView* AgentSchedulingGroup::CreateWebView(
   web_view->SetWebPreferences(params->web_preferences);

   if (local_main_frame) {
-    RenderFrameImpl::CreateMainFrame(
+    RenderFrameImpl* main_frame = RenderFrameImpl::CreateMainFrame(
         *this, web_view, opener_frame,
         /*is_for_nested_main_frame=*/params->type !=
             mojom::ViewWidgetType::kTopLevel,
@@ -302,6 +302,7 @@ blink::WebView* AgentSchedulingGroup::CreateWebView(
             mojom::ViewWidgetType::kFencedFrame,
         std::move(params->replication_state), params->devtools_main_frame_token,
         std::move(params->main_frame->get_local_params()));
+    main_frame->SetSkipBlockingParser(params->skip_blocking_parser);
   } else {
     blink::WebRemoteFrame::CreateMainFrame(
         web_view, params->main_frame->get_remote_params()->token,
