diff --git a/content/browser/web_contents/file_chooser_impl.cc b/content/browser/web_contents/file_chooser_impl.cc
index 7a3ea45d32..4b023fb4d4 100644
--- a/content/browser/web_contents/file_chooser_impl.cc
+++ b/content/browser/web_contents/file_chooser_impl.cc
@@ -122,12 +122,14 @@ void FileChooserImpl::OpenFileChooser(blink::mojom::FileChooserParamsPtr params,
   // Do not allow messages with absolute paths in them as this can permit a
   // renderer to coerce the browser to perform I/O on a renderer controlled
   // path.
+  #if 0
   if (params->default_file_name != params->default_file_name.BaseName()) {
     mojo::ReportBadMessage(
         "FileChooser: The default file name must not be an absolute path.");
     listener->FileSelectionCanceled();
     return;
   }
+  #endif

   // Don't allow page with open FileChooser to enter BackForwardCache to avoid
   // any unexpected behaviour from BackForwardCache.
