diff --git a/content/public/renderer/content_renderer_client.cc b/content/public/renderer/content_renderer_client.cc
index 24571a0964..8ebe4159cc 100644
--- a/content/public/renderer/content_renderer_client.cc
+++ b/content/public/renderer/content_renderer_client.cc
@@ -18,6 +18,10 @@

 namespace content {

+base::FilePath ContentRendererClient::GetRootPath() {
+  return base::FilePath();
+}
+
 SkBitmap* ContentRendererClient::GetSadPluginBitmap() {
   return nullptr;
 }
