diff --git a/content/renderer/BUILD.gn b/content/renderer/BUILD.gn
index e77edb5346..e2882cd940 100644
--- a/content/renderer/BUILD.gn
+++ b/content/renderer/BUILD.gn
@@ -46,6 +46,8 @@ target(link_target_type, "renderer") {
     "accessibility/ax_tree_snapshotter_impl.h",
     "accessibility/blink_ax_action_target.cc",
     "accessibility/blink_ax_action_target.h",
+    "//base/message_loop/message_pumpuv_mac.mm",
+    "//base/message_loop/message_pumpuv_mac.h",
     "accessibility/blink_ax_tree_source.cc",
     "accessibility/blink_ax_tree_source.h",
     "accessibility/render_accessibility_impl.cc",
