diff --git a/content/test/BUILD.gn b/content/test/BUILD.gn
index cc24c2391c..077f516b9b 100644
--- a/content/test/BUILD.gn
+++ b/content/test/BUILD.gn
@@ -2392,6 +2392,7 @@ test("content_unittests") {
     "../browser/renderer_host/renderer_sandboxed_process_launcher_delegate_unittest.cc",
     "../browser/renderer_host/transient_allow_popup_unittest.cc",
     "../browser/renderer_host/visible_time_request_trigger_unittest.cc",
+    "../browser/resolve_proxy_helper_unittest.cc",
     "../browser/scheduler/browser_io_thread_delegate_unittest.cc",
     "../browser/scheduler/browser_task_executor_unittest.cc",
     "../browser/scheduler/browser_task_queues_unittest.cc",
@@ -2975,7 +2976,7 @@ test("content_unittests") {
       "//device/base",
       "//device/fido",
       "//device/fido:cablev2_authenticator",
-      "//device/fido:cablev2_test_util",
+      #"//device/fido:cablev2_test_util",
       "//device/fido:mocks",
       "//device/fido:test_support",
     ]
