diff --git a/content/test/test_render_frame.cc b/content/test/test_render_frame.cc
index 8cba243998..d6652323a2 100644
--- a/content/test/test_render_frame.cc
+++ b/content/test/test_render_frame.cc
@@ -120,6 +120,8 @@ class MockFrameHost : public mojom::FrameHost {

  protected:
   // mojom::FrameHost:
+  void SetNodeJS(bool) override {}
+  void SetContextCreated(bool) override {}
   void CreateNewWindow(mojom::CreateNewWindowParamsPtr,
                        CreateNewWindowCallback) override {
     NOTREACHED() << "We should never dispatch to the service side signature.";
