diff --git a/content/renderer/render_frame_impl.h b/content/renderer/render_frame_impl.h
index c8e822502d..f1b1265908 100644
--- a/content/renderer/render_frame_impl.h
+++ b/content/renderer/render_frame_impl.h
@@ -173,6 +173,8 @@ class CONTENT_EXPORT RenderFrameImpl
       service_manager::mojom::InterfaceProvider {
  public:
   // Creates a new RenderFrame as the main frame of `web_view`.
+  bool skip_blocking_parser_;
+  void SetSkipBlockingParser(bool) override;
   static RenderFrameImpl* CreateMainFrame(
       AgentSchedulingGroup& agent_scheduling_group,
       blink::WebView* web_view,
@@ -622,7 +624,8 @@ class CONTENT_EXPORT RenderFrameImpl
       bool& consumed_user_gesture,
       const absl::optional<blink::Impression>& impression,
       const absl::optional<blink::WebPictureInPictureWindowOptions>&
-          pip_options) override;
+      pip_options,
+      blink::WebString* manifest) override;

   // Dispatches the current state of selection on the webpage to the browser if
   // it has changed or if the forced flag is passed. The forced flag is used
@@ -670,6 +673,13 @@ class CONTENT_EXPORT RenderFrameImpl
   // started media playback (i.e. played audio and video).
   bool DeferMediaLoad(bool has_played_media_before, base::OnceClosure closure);

+  void willHandleNavigationPolicy(
+                                  blink::WebFrame*,
+                                  const blink::WebURLRequest&,
+                                  blink::WebNavigationPolicy*,
+                                  blink::WebString* manifest = NULL,
+                                  bool new_win = true) override;
+
   // Binds to the MHTML file generation service in the browser.
   void BindMhtmlFileWriter(
       mojo::PendingAssociatedReceiver<mojom::MhtmlFileWriter> receiver);
