diff --git a/content/browser/devtools/protocol/emulation_handler.cc b/content/browser/devtools/protocol/emulation_handler.cc
index 90c7fbebfc..76646d6935 100644
--- a/content/browser/devtools/protocol/emulation_handler.cc
+++ b/content/browser/devtools/protocol/emulation_handler.cc
@@ -596,13 +596,18 @@ void EmulationHandler::UpdateTouchEventEmulationState() {
   if (touch_emulation_enabled_) {
     if (auto* touch_emulator =
             host_->GetRenderWidgetHost()->GetTouchEmulator()) {
+      touch_emulator->set_rfh_limit(host_->GetWeakPtr());
       touch_emulator->Enable(
           TouchEmulator::Mode::kEmulatingTouchFromMouse,
           TouchEmulationConfigurationToType(touch_emulation_configuration_));
     }
   } else {
-    if (auto* touch_emulator = host_->GetRenderWidgetHost()->GetTouchEmulator())
+    if (auto* touch_emulator =
+            host_->GetRenderWidgetHost()->GetTouchEmulator()) {
       touch_emulator->Disable();
+      touch_emulator->set_rfh_limit(nullptr);
+    }
+
   }
   GetWebContents()->SetForceDisableOverscrollContent(touch_emulation_enabled_);
 }
