diff --git a/content/app_shim_remote_cocoa/render_widget_host_ns_view_host_helper.h b/content/app_shim_remote_cocoa/render_widget_host_ns_view_host_helper.h
index 1201128a5b..19a7ea684d 100644
--- a/content/app_shim_remote_cocoa/render_widget_host_ns_view_host_helper.h
+++ b/content/app_shim_remote_cocoa/render_widget_host_ns_view_host_helper.h
@@ -22,6 +22,7 @@ class LatencyInfo;

 namespace content {
 struct NativeWebKeyboardEvent;
+class RenderWidgetHostViewMac;
 }  // namespace content

 namespace remote_cocoa {
@@ -30,6 +31,7 @@ namespace mojom {
 class RenderWidgetHostNSViewHost;
 }  // namespace mojom

+
 // An interface through which the NSView for a RenderWidgetHostViewMac is to
 // communicate with the RenderWidgetHostViewMac (potentially in another
 // process). Unlike mojom::RenderWidgetHostNSViewHost, this object is always
@@ -53,6 +55,8 @@ class RenderWidgetHostNSViewHostHelper {

   // Return the currently focused accessibility element.
   virtual id GetFocusedBrowserAccessibilityElement() = 0;
+  virtual content::RenderWidgetHostViewMac* GetRenderWidgetHostViewMac() = 0;
+

   // Set the NSWindow that will be the accessibility parent of the NSView.
   virtual void SetAccessibilityWindow(NSWindow* window) = 0;
