diff --git a/content/public/test/browser_test_base.cc b/content/public/test/browser_test_base.cc
index 0521a8f721..279804f8ac 100644
--- a/content/public/test/browser_test_base.cc
+++ b/content/public/test/browser_test_base.cc
@@ -317,6 +317,8 @@ void BrowserTestBase::SetUp() {
       switches::kIPCConnectionTimeout,
       base::NumberToString(TestTimeouts::action_max_timeout().InSeconds()));

+  command_line->AppendSwitch("nwjs-test-mode");
+
   command_line->AppendSwitch(switches::kDomAutomationController);

   // It is sometimes useful when looking at browser test failures to know which
