diff --git a/content/browser/gpu/gpu_process_host.h b/content/browser/gpu/gpu_process_host.h
index 9264a01382..5778ec998d 100644
--- a/content/browser/gpu/gpu_process_host.h
+++ b/content/browser/gpu/gpu_process_host.h
@@ -236,6 +236,8 @@ class GpuProcessHost : public BrowserChildProcessHostDelegate,
   // The GPU process reported failure to initialize.
   bool did_fail_initialize_ = false;

+  bool closing_;
+
   // The total number of GPU process crashes.
   static base::subtle::Atomic32 gpu_crash_count_;
   static bool crashed_before_;
