diff --git a/content/public/renderer/content_renderer_client.h b/content/public/renderer/content_renderer_client.h
index a9146ec6c8..9fd26cef84 100644
--- a/content/public/renderer/content_renderer_client.h
+++ b/content/public/renderer/content_renderer_client.h
@@ -12,6 +12,8 @@
 #include <string>
 #include <vector>

+#include "base/files/file_path.h"
+
 #include "base/callback_forward.h"
 #include "base/files/file_path.h"
 #include "base/memory/ref_counted.h"
@@ -46,6 +48,7 @@ class SingleThreadTaskRunner;

 namespace blink {
 class WebElement;
+class WebString;
 class WebFrame;
 class WebLocalFrame;
 class WebPlugin;
@@ -78,10 +81,18 @@ class BinderMap;

 namespace content {
 class RenderFrame;
+class RenderView;

 // Embedder API for participating in renderer logic.
 class CONTENT_EXPORT ContentRendererClient {
  public:
+  virtual base::FilePath GetRootPath();
+  virtual void willHandleNavigationPolicy(RenderFrame* rv,
+                                          blink::WebFrame* frame,
+                                          const blink::WebURLRequest& request,
+                                          blink::WebNavigationPolicy* policy,
+                                          blink::WebString* manifest,
+                                          bool new_win) {}
   virtual ~ContentRendererClient() {}

   // Notifies us that the RenderThread has been created.
