diff --git a/content/public/common/common_param_traits_macros.h b/content/public/common/common_param_traits_macros.h
index 718ea14ca7..f79526bf40 100644
--- a/content/public/common/common_param_traits_macros.h
+++ b/content/public/common/common_param_traits_macros.h
@@ -150,6 +150,8 @@ IPC_STRUCT_TRAITS_BEGIN(blink::RendererPreferences)
 #if defined(USE_OZONE)
   IPC_STRUCT_TRAITS_MEMBER(selection_clipboard_buffer_available)
 #endif
+  IPC_STRUCT_TRAITS_MEMBER(nw_inject_js_doc_start)
+  IPC_STRUCT_TRAITS_MEMBER(nw_inject_js_doc_end)
 IPC_STRUCT_TRAITS_END()

 IPC_ENUM_TRAITS_MAX_VALUE(content::DropData::Kind,
