diff --git a/content/renderer/worker/embedded_shared_worker_stub.h b/content/renderer/worker/embedded_shared_worker_stub.h
index 1470b865cf..6b547fc466 100644
--- a/content/renderer/worker/embedded_shared_worker_stub.h
+++ b/content/renderer/worker/embedded_shared_worker_stub.h
@@ -10,6 +10,7 @@

 #include "base/unguessable_token.h"
 #include "content/renderer/service_worker/service_worker_provider_context.h"
+#include "base/files/file_path.h"
 #include "mojo/public/cpp/bindings/pending_receiver.h"
 #include "mojo/public/cpp/bindings/pending_remote.h"
 #include "mojo/public/cpp/bindings/receiver.h"
@@ -111,6 +112,8 @@ class EmbeddedSharedWorkerStub : public blink::WebSharedWorkerClient,
   std::unique_ptr<blink::WebSharedWorker> impl_;

   scoped_refptr<ServiceWorkerProviderContext> service_worker_provider_context_;
+  bool nodejs_ = false;
+  base::FilePath root_path_;

   // The factory bundle used for loading subresources for this shared worker.
   scoped_refptr<blink::ChildURLLoaderFactoryBundle>
