diff --git a/content/public/common/content_features.cc b/content/public/common/content_features.cc
index 1d8a6de095..18345d07aa 100644
--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -15,6 +15,9 @@

 namespace features {

+const base::Feature kNWNewWin{
+    "nw2", base::FEATURE_ENABLED_BY_DEFAULT};
+
 // All features in alphabetical order.

 // Enables content-initiated, main frame navigations to data URLs.
@@ -690,7 +693,7 @@ const base::Feature kDocumentPictureInPictureAPI{
 // kProcessSharingWithStrictSiteInstances. Only one of these should be enabled
 // at a time.
 const base::Feature kProcessSharingWithDefaultSiteInstances{
-    "ProcessSharingWithDefaultSiteInstances", base::FEATURE_ENABLED_BY_DEFAULT};
+  "ProcessSharingWithDefaultSiteInstances", base::FEATURE_DISABLED_BY_DEFAULT}; //NWJS#7197

 // Whether cross-site frames should get their own SiteInstance even when
 // strict site isolation is disabled. These SiteInstances will still be
@@ -980,7 +983,7 @@ const base::Feature kDisableProcessReuse{"DisableProcessReuse",
 // spare renderer process around for the most recently requested BrowserContext.
 // This feature is only consulted in site-per-process mode.
 const base::Feature kSpareRendererForSitePerProcess{
-    "SpareRendererForSitePerProcess", base::FEATURE_ENABLED_BY_DEFAULT};
+    "SpareRendererForSitePerProcess", base::FEATURE_DISABLED_BY_DEFAULT};

 const base::Feature kStopVideoCaptureOnScreenLock{
     "StopVideoCaptureOnScreenLock", base::FEATURE_ENABLED_BY_DEFAULT};
