diff --git a/content/renderer/worker/embedded_shared_worker_stub.cc b/content/renderer/worker/embedded_shared_worker_stub.cc
index 3d62d44c6f..a59820260e 100644
--- a/content/renderer/worker/embedded_shared_worker_stub.cc
+++ b/content/renderer/worker/embedded_shared_worker_stub.cc
@@ -61,7 +61,8 @@ EmbeddedSharedWorkerStub::EmbeddedSharedWorkerStub(
         browser_interface_broker,
     ukm::SourceId ukm_source_id,
     const std::vector<std::string>& cors_exempt_header_list)
-    : receiver_(this, std::move(receiver)) {
+    : receiver_(this, std::move(receiver)),
+      nodejs_(info->is_node_js), root_path_(info->root_path) {
   DCHECK(main_script_load_params);
   DCHECK(pending_subresource_loader_factory_bundle);

@@ -118,6 +119,7 @@ EmbeddedSharedWorkerStub::EmbeddedSharedWorkerStub(
                                cors_exempt_header_list);

   impl_ = blink::WebSharedWorker::CreateAndStart(
+      nodejs_, root_path_,
       token, info->url, info->options->type, info->options->credentials,
       blink::WebString::FromUTF8(info->options->name),
       blink::WebSecurityOrigin(constructor_origin),
