diff --git a/content/public/browser/render_frame_host.h b/content/public/browser/render_frame_host.h
index c412367886..7017e92a4a 100644
--- a/content/public/browser/render_frame_host.h
+++ b/content/public/browser/render_frame_host.h
@@ -211,6 +211,9 @@ class CONTENT_EXPORT RenderFrameHost : public IPC::Listener,
   // node IDs identified.
   virtual void RequestDistilledAXTree(AXTreeDistillerCallback callback) = 0;

+
+  virtual bool nodejs() = 0;
+  virtual bool context_created() = 0;
   // Returns the SiteInstance grouping all RenderFrameHosts that have script
   // access to this RenderFrameHost, and must therefore live in the same
   // process.
