diff --git a/content/public/renderer/render_thread.h b/content/public/renderer/render_thread.h
index 52af215974..a53af51158 100644
--- a/content/public/renderer/render_thread.h
+++ b/content/public/renderer/render_thread.h
@@ -18,6 +18,8 @@
 #include "third_party/blink/public/platform/web_string.h"
 #include "third_party/perfetto/include/perfetto/tracing/traced_proto.h"

+class GURL;
+
 namespace base {
 class UnguessableToken;
 class WaitableEvent;
@@ -96,6 +98,11 @@ class CONTENT_EXPORT RenderThread : virtual public ChildThread {
   // Post task to all worker threads. Returns number of workers.
   virtual int PostTaskToAllWebWorkers(base::RepeatingClosure closure) = 0;

+  // Resolve the proxy servers to use for a given url. On success true is
+  // returned and |proxy_list| is set to a PAC string containing a list of
+  // proxy servers.
+  virtual bool ResolveProxy(const GURL& url, std::string* proxy_list) = 0;
+
   // Gets the shutdown event for the process.
   virtual base::WaitableEvent* GetShutdownEvent() = 0;

