diff --git a/content/browser/renderer_host/render_view_host_delegate.h b/content/browser/renderer_host/render_view_host_delegate.h
index 1ebbe82d16..50538a0216 100644
--- a/content/browser/renderer_host/render_view_host_delegate.h
+++ b/content/browser/renderer_host/render_view_host_delegate.h
@@ -46,6 +46,7 @@ class RenderViewHostDelegateView;
 // WebContents type (see //renderer_host/DEPS).
 class RenderViewHostDelegate {
  public:
+  RenderViewHostDelegate();
   // Returns the current delegate associated with a feature. May return NULL if
   // there is no corresponding delegate.
   virtual RenderViewHostDelegateView* GetDelegateView();
@@ -80,6 +81,9 @@ class RenderViewHostDelegate {
   // JavaScript window.focus() method).
   virtual void Activate() {}

+  virtual bool GetSkipBlockingParser();
+  virtual void SetSkipBlockingParser(bool);
+
   // Returns true if RenderWidgets under this RenderViewHost will never be
   // user-visible and thus never need to generate pixels for display.
   virtual bool IsNeverComposited();
@@ -127,6 +131,8 @@ class RenderViewHostDelegate {

  protected:
   virtual ~RenderViewHostDelegate() {}
+ private:
+  bool skip_blocking_parser_;
 };

 }  // namespace content
