diff --git a/content/browser/worker_host/shared_worker_service_impl.cc b/content/browser/worker_host/shared_worker_service_impl.cc
index ddbab27d57..0e955c4fcb 100644
--- a/content/browser/worker_host/shared_worker_service_impl.cc
+++ b/content/browser/worker_host/shared_worker_service_impl.cc
@@ -189,7 +189,9 @@ void SharedWorkerServiceImpl::ConnectToWorker(
     return;
   }
   auto partition_domain = site_instance->GetPartitionDomain(storage_partition_);
-  SharedWorkerInstance instance(info->url, info->options->type,
+  SharedWorkerInstance instance(
+                                info->is_node_js, info->root_path,
+                                info->url, info->options->type,
                                 info->options->credentials, info->options->name,
                                 storage_key, creation_context_type);
   host = CreateWorker(
