diff --git a/content/app_shim_remote_cocoa/ns_view_bridge_factory_impl.mm b/content/app_shim_remote_cocoa/ns_view_bridge_factory_impl.mm
index be7b05ee4f..8c6c4983d7 100644
--- a/content/app_shim_remote_cocoa/ns_view_bridge_factory_impl.mm
+++ b/content/app_shim_remote_cocoa/ns_view_bridge_factory_impl.mm
@@ -57,6 +57,8 @@ class RenderWidgetHostNSViewBridgeOwner
  private:
   void OnMojoDisconnect() { delete this; }

+  content::RenderWidgetHostViewMac* GetRenderWidgetHostViewMac() override { return nullptr; }
+
   std::unique_ptr<blink::WebCoalescedInputEvent> TranslateEvent(
       const blink::WebInputEvent& web_event) {
     return std::make_unique<blink::WebCoalescedInputEvent>(
