diff --git a/content/browser/renderer_host/render_widget_host_view_mac.mm b/content/browser/renderer_host/render_widget_host_view_mac.mm
index d460e055c8..b832ab1eb7 100644
--- a/content/browser/renderer_host/render_widget_host_view_mac.mm
+++ b/content/browser/renderer_host/render_widget_host_view_mac.mm
@@ -3,6 +3,7 @@
 // found in the LICENSE file.

 #include "content/browser/renderer_host/render_widget_host_view_mac.h"
+#include "content/public/common/content_switches.h"

 #import <Carbon/Carbon.h>
 #include "third_party/blink/public/mojom/input/input_handler.mojom-forward.h"
@@ -78,6 +79,10 @@ using blink::WebMouseEvent;
 using blink::WebGestureEvent;
 using blink::WebTouchEvent;

+namespace content {
+  extern bool g_force_cpu_draw;
+}
+
 namespace content {

 ////////////////////////////////////////////////////////////////////////////////
@@ -351,6 +356,13 @@ RenderWidgetHostViewCocoa* RenderWidgetHostViewMac::GetInProcessNSView() const {
   return nullptr;
 }

+CALayer* RenderWidgetHostViewMac::background_layer() const {
+  assert(content::g_force_cpu_draw);
+  if (in_process_ns_view_bridge_)
+    return in_process_ns_view_bridge_->GetBackgroundLayer();
+  return nil;
+}
+
 void RenderWidgetHostViewMac::SetDelegate(
     NSObject<RenderWidgetHostViewMacDelegate>* delegate) {
   [GetInProcessNSView() setResponderDelegate:delegate];
@@ -449,6 +461,7 @@ void RenderWidgetHostViewMac::Hide() {
   is_visible_ = false;
   ns_view_->SetVisible(is_visible_);
   browser_compositor_->SetViewVisible(is_visible_);
+  if (!base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kDisableRAFThrottling))
   WasOccluded();
 }

@@ -515,6 +528,7 @@ void RenderWidgetHostViewMac::WasOccluded() {
     return;

   host()->WasHidden();
+  if (!base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kDisableRAFThrottling))
   browser_compositor_->SetRenderWidgetHostIsHidden(true);
 }

@@ -1617,6 +1631,10 @@ bool RenderWidgetHostViewMac::SyncIsWidgetForMainFrame(
   return true;
 }

+RenderWidgetHostViewMac* RenderWidgetHostViewMac::GetRenderWidgetHostViewMac() {
+  return this;
+}
+
 void RenderWidgetHostViewMac::SyncIsWidgetForMainFrame(
     SyncIsWidgetForMainFrameCallback callback) {
   bool is_for_main_frame;
diff --git a/content/browser/resolve_proxy_helper.cc b/content/browser/resolve_proxy_helper.cc
new file mode 100644
index 0000000000..6c0338fb7f
