diff --git a/content/public/browser/web_contents.h b/content/public/browser/web_contents.h
index 024cd37e81..53e477be1c 100644
--- a/content/public/browser/web_contents.h
+++ b/content/public/browser/web_contents.h
@@ -327,6 +327,11 @@ class WebContents : public PageNavigator,

   ~WebContents() override = default;

+  virtual bool is_silent_printing();
+  virtual void set_silent_printing(bool);
+
+  bool silent_printing_ = false;
+
   // Intrinsic tab state -------------------------------------------------------

   // Gets/Sets the delegate.
