diff --git a/content/browser/renderer_host/render_view_host_impl.cc b/content/browser/renderer_host/render_view_host_impl.cc
index 5c03f11946..93a608b8a3 100644
--- a/content/browser/renderer_host/render_view_host_impl.cc
+++ b/content/browser/renderer_host/render_view_host_impl.cc
@@ -492,6 +492,7 @@ bool RenderViewHostImpl::CreateRenderView(
       frame_tree_->controller()
           .GetSessionStorageNamespace(storage_partition_config_)
           ->id();
+  params->skip_blocking_parser = delegate_->GetSkipBlockingParser();
   params->hidden = frame_tree_->delegate()->IsHidden();
   params->never_composited = delegate_->IsNeverComposited();
   params->window_was_opened_by_another_window =
