diff --git a/content/browser/renderer_host/render_widget_host_input_event_router.cc b/content/browser/renderer_host/render_widget_host_input_event_router.cc
index 0ef81ec935..bc9fb7e5ae 100644
--- a/content/browser/renderer_host/render_widget_host_input_event_router.cc
+++ b/content/browser/renderer_host/render_widget_host_input_event_router.cc
@@ -7,6 +7,9 @@
 #include <algorithm>
 #include <deque>
 #include <memory>
+
+#include "content/public/browser/render_frame_host.h"
+
 #include <vector>

 #include "base/containers/cxx20_erase.h"
@@ -1972,8 +1975,13 @@ void RenderWidgetHostInputEventRouter::SetCursor(const WebCursor& cursor) {
     touch_emulator_->SetDeviceScaleFactor(last_device_scale_factor_);
   if (auto* cursor_manager = last_mouse_move_root_view_->GetCursorManager()) {
     for (auto it : owner_map_) {
-      if (it.second)
+      if (it.second) {
+        if (touch_emulator_ && touch_emulator_->rfh_limit()) {
+          if (touch_emulator_->rfh_limit()->GetView() != it.second.get())
+            continue;
+        }
         cursor_manager->UpdateCursor(it.second.get(), cursor);
+      }
     }
   }
 }
