diff --git a/extensions/browser/app_window/app_window_contents.h b/extensions/browser/app_window/app_window_contents.h
index 5ed16ce7ed..cd874060bd 100644
--- a/extensions/browser/app_window/app_window_contents.h
+++ b/extensions/browser/app_window/app_window_contents.h
@@ -17,6 +17,7 @@
 namespace content {
 class BrowserContext;
 class RenderFrameHost;
+class WebContents;
 }

 namespace extensions {
@@ -29,7 +30,7 @@ struct DraggableRegion;
 class AppWindowContentsImpl : public AppWindowContents,
                               public content::WebContentsObserver {
  public:
-  explicit AppWindowContentsImpl(AppWindow* host);
+   explicit AppWindowContentsImpl(AppWindow* host, std::unique_ptr<content::WebContents> web_contents = nullptr);

   AppWindowContentsImpl(const AppWindowContentsImpl&) = delete;
   AppWindowContentsImpl& operator=(const AppWindowContentsImpl&) = delete;
@@ -39,7 +40,9 @@ class AppWindowContentsImpl : public AppWindowContents,
   // AppWindowContents
   void Initialize(content::BrowserContext* context,
                   content::RenderFrameHost* creator_frame,
-                  const GURL& url) override;
+                  const GURL& url,
+                  const Extension* extension,
+                  bool skip_blocking_parser) override;
   void LoadContents(int32_t creator_process_id) override;
   void NativeWindowChanged(NativeAppWindow* native_app_window) override;
   void NativeWindowClosed(bool send_onclosed) override;
@@ -57,6 +60,7 @@ class AppWindowContentsImpl : public AppWindowContents,

   raw_ptr<AppWindow> host_;  // This class is owned by |host_|
   GURL url_;
+
   std::unique_ptr<content::WebContents> web_contents_;
 };

