diff --git a/extensions/renderer/bindings/api_signature.cc b/extensions/renderer/bindings/api_signature.cc
index efd246bf7c..0380641b8c 100644
--- a/extensions/renderer/bindings/api_signature.cc
+++ b/extensions/renderer/bindings/api_signature.cc
@@ -30,6 +30,7 @@ constexpr const char* const kNonCallbackTrailingFunctionAPINames[] = {
     "test.callbackFail",
     "automation.addTreeChangeObserver",
     "automation.removeTreeChangeObserver",
+    "nw.App.on",
 };

 std::vector<std::unique_ptr<ArgumentSpec>> ValueListToArgumentSpecs(
