diff --git a/extensions/browser/process_manager.cc b/extensions/browser/process_manager.cc
index edf33b204d..614480f235 100644
--- a/extensions/browser/process_manager.cc
+++ b/extensions/browser/process_manager.cc
@@ -323,7 +323,7 @@ void ProcessManager::UnregisterRenderFrameHost(

 scoped_refptr<content::SiteInstance> ProcessManager::GetSiteInstanceForURL(
     const GURL& url) {
-  return site_instance_->GetRelatedSiteInstance(url);
+  return site_instance_->GetRelatedSiteInstance(url, false);
 }

 const ProcessManager::FrameSet ProcessManager::GetAllFrames() const {
@@ -831,6 +831,8 @@ void ProcessManager::DecrementServiceWorkerKeepaliveCount(

 void ProcessManager::OnLazyBackgroundPageIdle(const std::string& extension_id,
                                               uint64_t sequence_id) {
+  // bg page will be destroyed if main points to remote page NWJS#5282
+#if 0
   ExtensionHost* host = GetBackgroundHostForExtension(extension_id);
   if (host && !background_page_data_[extension_id].is_closing &&
       sequence_id == background_page_data_[extension_id].close_sequence_id) {
@@ -849,6 +851,7 @@ void ProcessManager::OnLazyBackgroundPageIdle(const std::string& extension_id,
           extension_id, sequence_id));
     }
   }
+#endif
 }

 void ProcessManager::OnLazyBackgroundPageActive(
