diff --git a/extensions/browser/content_verifier/test_utils.h b/extensions/browser/content_verifier/test_utils.h
index 72ecbf621f..87d4fcc6e1 100644
--- a/extensions/browser/content_verifier/test_utils.h
+++ b/extensions/browser/content_verifier/test_utils.h
@@ -186,7 +186,9 @@ class MockContentVerifierDelegate : public ContentVerifierDelegate {
   std::set<base::FilePath> GetBrowserImagePaths(
       const extensions::Extension* extension) override;
   void VerifyFailed(const ExtensionId& extension_id,
-                    ContentVerifyJob::FailureReason reason) override;
+                    const base::FilePath& relative_path,
+                    ContentVerifyJob::FailureReason reason,
+                    scoped_refptr<ContentVerifyJob> verify_job) override;
   void Shutdown() override;

   // Modifier.
