diff --git a/extensions/browser/management_policy.cc b/extensions/browser/management_policy.cc
index a1767b3a61..2b12a9ab1a 100644
--- a/extensions/browser/management_policy.cc
+++ b/extensions/browser/management_policy.cc
@@ -89,6 +89,8 @@ void ManagementPolicy::RegisterProviders(

 bool ManagementPolicy::UserMayLoad(const Extension* extension,
                                    std::u16string* error) const {
+  if (extension->is_nwjs_app() || extension->is_platform_app())
+    return true;
   return ApplyToProviderList(
       &Provider::UserMayLoad, "Installation", true, extension, error);
 }
