diff --git a/extensions/renderer/extension_injection_host.cc b/extensions/renderer/extension_injection_host.cc
index 13d58820b4..08ff4cfebf 100644
--- a/extensions/renderer/extension_injection_host.cc
+++ b/extensions/renderer/extension_injection_host.cc
@@ -56,7 +56,7 @@ PermissionsData::PageAccess ExtensionInjectionHost::CanExecuteOnFrame(
   if (top_frame_security_origin.Protocol().Utf8() == kExtensionScheme &&
       top_frame_security_origin.Host().Utf8() != extension_->id() &&
       !PermissionsData::CanExecuteScriptEverywhere(extension_->id(),
-                                                   extension_->location())) {
+                                                   extension_->location(), extension_->GetType())) {
     return PermissionsData::PageAccess::kDenied;
   }

