diff --git a/extensions/common/api/app_current_window_internal.idl b/extensions/common/api/app_current_window_internal.idl
index 13937b3e73..b8ede39179 100644
--- a/extensions/common/api/app_current_window_internal.idl
+++ b/extensions/common/api/app_current_window_internal.idl
@@ -47,6 +47,7 @@ namespace app.currentWindowInternal {
     static void setBounds(DOMString boundsType, Bounds bounds);
     static void setSizeConstraints(DOMString boundsType,
                                    SizeConstraints constraints);
+    static void setResizable(boolean flag);
     static void setIcon(DOMString icon_url);
     static void setShape(Region region);
     static void setAlwaysOnTop(boolean always_on_top);
@@ -57,6 +58,8 @@ namespace app.currentWindowInternal {
   interface Events {
     static void onClosed();
     static void onBoundsChanged();
+    static void onResized();
+    static void onMoved();
     static void onFullscreened();
     static void onMinimized();
     static void onMaximized();
