diff --git a/extensions/renderer/bindings/event_emitter.h b/extensions/renderer/bindings/event_emitter.h
index ee65780a35..43e6d11b2c 100644
--- a/extensions/renderer/bindings/event_emitter.h
+++ b/extensions/renderer/bindings/event_emitter.h
@@ -73,7 +73,9 @@ class EventEmitter final : public gin::Wrappable<EventEmitter> {
   void RemoveListener(gin::Arguments* arguments);
   bool HasListener(v8::Local<v8::Function> function);
   bool HasListeners();
+  void GetListeners(gin::Arguments* arguments);
   void Dispatch(gin::Arguments* arguments);
+  void DispatchNW(gin::Arguments* arguments);

   // Dispatches an event synchronously to listeners, returning the result.
   v8::Local<v8::Value> DispatchSync(v8::Local<v8::Context> context,
