diff --git a/extensions/browser/guest_view/mime_handler_view/mime_handler_view_guest.cc b/extensions/browser/guest_view/mime_handler_view/mime_handler_view_guest.cc
index 75a2534e3b..868c79540e 100644
--- a/extensions/browser/guest_view/mime_handler_view/mime_handler_view_guest.cc
+++ b/extensions/browser/guest_view/mime_handler_view/mime_handler_view_guest.cc
@@ -264,6 +264,9 @@ bool MimeHandlerViewGuest::ZoomPropagatesFromEmbedderToGuest() const {
 WebContents* MimeHandlerViewGuest::OpenURLFromTab(
     WebContents* source,
     const content::OpenURLParams& params) {
+  if (!embedder_web_contents())
+    return owner_web_contents()->GetDelegate()->OpenURLFromTab(
+      owner_web_contents(), params);
   auto* delegate = embedder_web_contents()->GetDelegate();
   return delegate ? delegate->OpenURLFromTab(embedder_web_contents(), params)
                   : nullptr;
