diff --git a/extensions/browser/content_verifier/test_utils.cc b/extensions/browser/content_verifier/test_utils.cc
index 144c1479e6..9c049eb7be 100644
--- a/extensions/browser/content_verifier/test_utils.cc
+++ b/extensions/browser/content_verifier/test_utils.cc
@@ -236,7 +236,9 @@ std::set<base::FilePath> MockContentVerifierDelegate::GetBrowserImagePaths(

 void MockContentVerifierDelegate::VerifyFailed(
     const ExtensionId& extension_id,
-    ContentVerifyJob::FailureReason reason) {
+    const base::FilePath& relative_path,
+    ContentVerifyJob::FailureReason reason,
+    scoped_refptr<ContentVerifyJob> verify_job) {
   ADD_FAILURE() << "Unexpected call for this test";
 }

