diff --git a/extensions/renderer/user_script_injector.h b/extensions/renderer/user_script_injector.h
index dd7fdebc66..7d4e49fd35 100644
--- a/extensions/renderer/user_script_injector.h
+++ b/extensions/renderer/user_script_injector.h
@@ -46,6 +46,7 @@ class UserScriptInjector : public ScriptInjector,
   // ScriptInjector implementation.
   mojom::InjectionType script_type() const override;
   blink::mojom::UserActivationOption IsUserGesture() const override;
+  bool ShouldExecuteInMainWorld() const override;
   mojom::ExecutionWorld GetExecutionWorld() const override;
   mojom::CSSOrigin GetCssOrigin() const override;
   mojom::CSSInjection::Operation GetCSSInjectionOperation() const override;
