diff --git a/extensions/browser/event_router.h b/extensions/browser/event_router.h
index 2dcb2d63f2..5e5a7e6cf2 100644
--- a/extensions/browser/event_router.h
+++ b/extensions/browser/event_router.h
@@ -281,7 +281,9 @@ class EventRouter : public KeyedService,
   // Returns true if the extension is listening to the given event.
   // (virtual for testing only.)
   virtual bool ExtensionHasEventListener(const std::string& extension_id,
-                                         const std::string& event_name) const;
+                                         const std::string& event_name,
+                                         int instance_id = -1,
+                                         std::string* out_extension_id = nullptr) const;

   // Broadcasts an event to every listener registered for that event.
   virtual void BroadcastEvent(std::unique_ptr<Event> event);
