diff --git a/extensions/common/extension.cc b/extensions/common/extension.cc
index 1170f0c9d9..484077ec97 100644
--- a/extensions/common/extension.cc
+++ b/extensions/common/extension.cc
@@ -203,7 +203,7 @@ const int Extension::kInitFromValueFlagBits = 15;
 const char Extension::kMimeType[] = "application/x-chrome-extension";

 const int Extension::kValidWebExtentSchemes =
-    URLPattern::SCHEME_HTTP | URLPattern::SCHEME_HTTPS;
+  URLPattern::SCHEME_HTTP | URLPattern::SCHEME_HTTPS | URLPattern::SCHEME_FILE;

 const int Extension::kValidHostPermissionSchemes =
     URLPattern::SCHEME_CHROMEUI | URLPattern::SCHEME_HTTP |
@@ -270,6 +270,17 @@ scoped_refptr<Extension> Extension::Create(const base::FilePath& path,
                                           std::move(extension_id));
   }

+  if (manifest->FindKey(keys::kNWJSInternalFlag)) {
+    const std::string* name;
+    const std::string* domain;
+    name = manifest->FindStringPath(keys::kName);
+    domain = manifest->FindStringPath(keys::kNWJSDomain);
+    if (domain && !domain->empty())
+      manifest->SetExtensionId(*domain);
+    else
+      manifest->SetExtensionId(crx_file::id_util::GenerateId(*name));
+  }
+
   std::vector<InstallWarning> install_warnings;
   if (!manifest->ValidateManifest(utf8_error, &install_warnings)) {
     return nullptr;
@@ -545,6 +556,10 @@ bool Extension::is_platform_app() const {
   return manifest()->is_platform_app();
 }

+bool Extension::is_nwjs_app() const {
+  return manifest()->is_nwjs_app();
+}
+
 bool Extension::is_hosted_app() const {
   return manifest()->is_hosted_app();
 }
@@ -670,6 +685,10 @@ bool Extension::LoadName(std::u16string* error) {
 }

 bool Extension::LoadVersion(std::u16string* error) {
+  if (manifest_->type() == Manifest::TYPE_NWJS_APP) {
+    version_ = base::Version("0.1");
+    return true;
+  }
   const std::string* version_str = manifest_->FindStringPath(keys::kVersion);
   if (version_str == nullptr) {
     *error = errors::kInvalidVersion;
@@ -754,6 +773,7 @@ bool Extension::LoadExtent(const char* key,
       return false;
     }

+#if 0
     // Do not allow authors to claim "<all_urls>".
     if (pattern.match_all_urls()) {
       *error = ErrorUtils::FormatErrorMessageUTF16(
@@ -777,6 +797,7 @@ bool Extension::LoadExtent(const char* key,
           value_error, base::NumberToString(i), errors::kNoWildCardsInPaths);
       return false;
     }
+#endif
     pattern.SetPath(pattern.path() + '*');

     extent->AddPattern(pattern);
