diff --git a/extensions/browser/extension_protocols.cc b/extensions/browser/extension_protocols.cc
index fcbc7226df..892fed1123 100644
--- a/extensions/browser/extension_protocols.cc
+++ b/extensions/browser/extension_protocols.cc
@@ -274,7 +274,7 @@ bool AllowExtensionResourceLoad(const network::ResourceRequest& request,
                                 const ProcessMap& process_map) {
   const bool is_main_frame =
       destination == network::mojom::RequestDestination::kDocument;
-  if (is_incognito &&
+  if (is_incognito && !extension->is_nwjs_app() &&
       !ExtensionCanLoadInIncognito(is_main_frame, extension,
                                    extension_enabled_in_incognito)) {
     return false;
