diff --git a/extensions/browser/api/app_current_window_internal/app_current_window_internal_api.h b/extensions/browser/api/app_current_window_internal/app_current_window_internal_api.h
index 00044f0e2a..61574b4332 100644
--- a/extensions/browser/api/app_current_window_internal/app_current_window_internal_api.h
+++ b/extensions/browser/api/app_current_window_internal/app_current_window_internal_api.h
@@ -145,6 +145,16 @@ class AppCurrentWindowInternalSetSizeConstraintsFunction
   ResponseAction Run() override;
 };

+class AppCurrentWindowInternalSetResizableFunction
+    : public AppCurrentWindowInternalExtensionFunction {
+ public:
+  DECLARE_EXTENSION_FUNCTION("app.currentWindowInternal.setResizable",
+                             UNKNOWN)
+ protected:
+  ~AppCurrentWindowInternalSetResizableFunction() override {}
+  ResponseAction Run() override;
+};
+
 class AppCurrentWindowInternalSetIconFunction
     : public AppCurrentWindowInternalExtensionFunction {
  public:
