diff --git a/extensions/renderer/script_injection.cc b/extensions/renderer/script_injection.cc
index 032ef69a5b..49eaa5426c 100644
--- a/extensions/renderer/script_injection.cc
+++ b/extensions/renderer/script_injection.cc
@@ -327,6 +327,8 @@ void ScriptInjection::InjectJs(std::set<std::string>* executing_scripts,
       world_id = blink::kMainDOMWorldId;
       break;
   }
+  if (injector_->ShouldExecuteInMainWorld())
+    world_id = blink::kMainDOMWorldId;
   render_frame_->GetWebFrame()->RequestExecuteScript(
       world_id, sources, injector_->IsUserGesture(), execution_option,
       blink::mojom::LoadEventBlockingOption::kBlock,
