diff --git a/extensions/common/manifest_handlers/content_scripts_handler.cc b/extensions/common/manifest_handlers/content_scripts_handler.cc
index ab24478619..714dabe066 100644
--- a/extensions/common/manifest_handlers/content_scripts_handler.cc
+++ b/extensions/common/manifest_handlers/content_scripts_handler.cc
@@ -105,6 +105,11 @@ std::unique_ptr<UserScript> CreateUserScript(
             : MatchOriginAsFallbackBehavior::kNever;
   }

+  // in main world
+  if (content_script.in_main_world) {
+    result->set_in_main_world(*content_script.in_main_world);
+  }
+
   bool wants_file_access = false;
   if (!script_parsing::ParseMatchPatterns(
           content_script.matches, content_script.exclude_matches.get(),
@@ -215,7 +220,7 @@ bool ContentScriptsHandler::Parse(Extension* extension, std::u16string* error) {

   const bool can_execute_script_everywhere =
       PermissionsData::CanExecuteScriptEverywhere(extension->id(),
-                                                  extension->location());
+                                                  extension->location(), extension->GetType());
   const int valid_schemes =
       UserScript::ValidUserScriptSchemes(can_execute_script_everywhere);
   const bool all_urls_includes_chrome_urls =
