diff --git a/extensions/browser/guest_view/extensions_guest_view_manager_delegate.cc b/extensions/browser/guest_view/extensions_guest_view_manager_delegate.cc
index 11576c1f5a..8acd0f935f 100644
--- a/extensions/browser/guest_view/extensions_guest_view_manager_delegate.cc
+++ b/extensions/browser/guest_view/extensions_guest_view_manager_delegate.cc
@@ -76,9 +76,11 @@ void ExtensionsGuestViewManagerDelegate::DispatchEvent(
   if (!owner)
     return;  // Could happen at tab shutdown.

+  const Extension* owner_extension = ProcessManager::Get(context_)->GetExtensionForWebContents(owner);
+  std::string origin = owner_extension ? owner_extension->id() : guest->owner_host();
   EventRouter::DispatchEventToSender(
       owner->GetPrimaryMainFrame()->GetProcess(), guest->browser_context(),
-      guest->owner_host(), histogram_value, event_name,
+      origin, histogram_value, event_name,
       content::ChildProcessHost::kInvalidUniqueID, extensions::kMainThreadId,
       blink::mojom::kInvalidServiceWorkerVersionId, std::move(event_args),
       std::move(info));
