diff --git a/extensions/BUILD.gn b/extensions/BUILD.gn
index 649622adff..033a8b161b 100644
--- a/extensions/BUILD.gn
+++ b/extensions/BUILD.gn
@@ -181,7 +181,7 @@ static_library("test_support") {
     "//services/network/public/mojom",
     "//testing/gmock",
     "//testing/gtest",
-    "//third_party/cld_3/src/src:cld_3",
+    #"//third_party/cld_3/src/src:cld_3",
     "//third_party/zlib/google:zip",
   ]

