diff --git a/extensions/browser/extension_frame_host.h b/extensions/browser/extension_frame_host.h
index abd83ac66f..473cbfab97 100644
--- a/extensions/browser/extension_frame_host.h
+++ b/extensions/browser/extension_frame_host.h
@@ -46,6 +46,12 @@ class ExtensionFrameHost : public mojom::LocalFrameHost {
                           GetAppInstallStateCallback callback) override;
   void Request(mojom::RequestParamsPtr params,
                RequestCallback callback) override;
+  void RequestSync(mojom::RequestParamsPtr params,
+                   RequestSyncCallback callback) override;
+  bool RequestSync(mojom::RequestParamsPtr params,
+                   bool* success,
+                   base::Value::List* response,
+                   std::string* error) override;
   void WatchedPageChange(
       const std::vector<std::string>& css_selectors) override;

