diff --git a/extensions/browser/extension_function.h b/extensions/browser/extension_function.h
index 23b93ae3bd..498454cdc5 100644
--- a/extensions/browser/extension_function.h
+++ b/extensions/browser/extension_function.h
@@ -206,6 +206,7 @@ class ExtensionFunction : public base::RefCountedThreadSafe<
   //
   // ExtensionFunction implementations are encouraged to just implement Run.
   [[nodiscard]] virtual ResponseAction Run() = 0;
+  virtual bool RunNWSync(base::Value::List* response, std::string* error);

   // Gets whether quota should be applied to this individual function
   // invocation. This is different to GetQuotaLimitHeuristics which is only
@@ -538,8 +539,11 @@ class ExtensionFunction : public base::RefCountedThreadSafe<

   // Any detailed error from the API. This should be populated by the derived
   // class before Run() returns.
+ protected:
   std::string error_;

+ private:
+
   // The callback to run once the function has done execution.
   ResponseCallback response_callback_;

@@ -604,6 +608,7 @@ class ExtensionFunction : public base::RefCountedThreadSafe<

   // Whether this function has responded.
   // TODO(devlin): Replace this with response_type_ != null.
+ public:
   bool did_respond_ = false;

   // If set to true, preserves |results_|, even after SendResponseImpl() was
@@ -641,4 +646,23 @@ class ExtensionFunction : public base::RefCountedThreadSafe<
   int worker_thread_id_ = -1;
 };

+class NWSyncExtensionFunction : public ExtensionFunction {
+ public:
+  NWSyncExtensionFunction();
+  void SetError(const std::string& error);
+
+ protected:
+  ~NWSyncExtensionFunction() override;
+  static bool ValidationFailure(NWSyncExtensionFunction* function);
+
+  void SetResult(std::unique_ptr<base::Value> result);
+  void SetResultList(std::unique_ptr<base::ListValue> results);
+
+  std::unique_ptr<base::ListValue> results_;
+  std::string error_;
+ private:
+  ResponseAction Run() final;
+
+};
+
 #endif  // EXTENSIONS_BROWSER_EXTENSION_FUNCTION_H_
