diff --git a/extensions/browser/url_request_util.cc b/extensions/browser/url_request_util.cc
index 8ccb1c2eda..b9133a980e 100644
--- a/extensions/browser/url_request_util.cc
+++ b/extensions/browser/url_request_util.cc
@@ -127,7 +127,8 @@ bool AllowCrossRendererResourceLoadHelper(bool is_guest,
   if (is_guest) {
     // An extension's resources should only be accessible to WebViews owned by
     // that extension.
-    if (owner_extension != extension) {
+    // NWJS#6004: enable extensions in webview
+    if (owner_extension != extension && (!owner_extension || !owner_extension->is_nwjs_app())) {
       *allowed = false;
       return true;
     }
