diff --git a/extensions/renderer/script_context_set.cc b/extensions/renderer/script_context_set.cc
index 02c46e9660..4eb4430295 100644
--- a/extensions/renderer/script_context_set.cc
+++ b/extensions/renderer/script_context_set.cc
@@ -68,7 +68,7 @@ ScriptContext* ScriptContextSet::Register(
   GURL frame_url = ScriptContext::GetDocumentLoaderURLForFrame(frame);
   Feature::Context context_type = ClassifyJavaScriptContext(
       extension, world_id, frame_url, frame->GetDocument().GetSecurityOrigin(),
-      view_type);
+      view_type, frame);
   Feature::Context effective_context_type = ClassifyJavaScriptContext(
       effective_extension, world_id,
       ScriptContext::GetEffectiveDocumentURLForContext(frame, frame_url, true),
@@ -200,7 +200,9 @@ Feature::Context ScriptContextSet::ClassifyJavaScriptContext(
     int32_t world_id,
     const GURL& url,
     const blink::WebSecurityOrigin& origin,
-    mojom::ViewType view_type) {
+    mojom::ViewType view_type,
+    const blink::WebLocalFrame* frame
+                                                             ) {
   // WARNING: This logic must match ProcessMap::GetContextType, as much as
   // possible.

@@ -222,7 +224,7 @@ Feature::Context ScriptContextSet::ClassifyJavaScriptContext(
   //    before the SecurityContext is updated with the sandbox flags (after
   //    reading the CSP header), so the caller can't check if the context's
   //    security origin is unique yet.
-  if (ScriptContext::IsSandboxedPage(url))
+  if (ScriptContext::IsSandboxedPage(url) || (frame && frame->isNwDisabledChildFrame()))
     return Feature::WEB_PAGE_CONTEXT;

   if (extension && active_extension_ids_->count(extension->id()) > 0) {
