diff --git a/extensions/renderer/dispatcher.h b/extensions/renderer/dispatcher.h
index 675fcc5cf0..d09a4e3d85 100644
--- a/extensions/renderer/dispatcher.h
+++ b/extensions/renderer/dispatcher.h
@@ -164,6 +164,7 @@ class Dispatcher : public content::RenderThreadObserver,

   // This method is not allowed to run JavaScript code in the frame.
   void DidCreateDocumentElement(blink::WebLocalFrame* frame);
+  void DidDispatchDOMContentLoadedEvent(blink::WebLocalFrame* frame);

   // These methods may run (untrusted) JavaScript code in the frame, and
   // cause |render_frame| to become invalid.
