diff --git a/extensions/browser/extension_navigation_throttle.cc b/extensions/browser/extension_navigation_throttle.cc
index a53a9b6817..368e1b7a9b 100644
--- a/extensions/browser/extension_navigation_throttle.cc
+++ b/extensions/browser/extension_navigation_throttle.cc
@@ -262,7 +262,7 @@ ExtensionNavigationThrottle::WillStartOrRedirectRequest() {
     }
   }

-  if (target_extension->is_platform_app() &&
+  if (target_extension->is_platform_app() && !target_extension->is_nwjs_app() &&
       ShouldBlockNavigationToPlatformAppResource(target_extension,
                                                  *navigation_handle())) {
     RecordExtensionResourceAccessResult(
