diff --git a/extensions/renderer/script_injector.h b/extensions/renderer/script_injector.h
index 8abefe478f..6d259adde8 100644
--- a/extensions/renderer/script_injector.h
+++ b/extensions/renderer/script_injector.h
@@ -28,8 +28,8 @@ class WebLocalFrame;
 namespace extensions {

 // The pseudo-delegate class for a ScriptInjection that provides all necessary
-// information about how to inject the script, including what code to inject and
-// when (run location), but without any injection logic.
+// information about how to inject the script, including what code to inject,
+// when (run location), and where (world), but without any injection logic.
 class ScriptInjector {
  public:
   // The possible reasons for not injecting the script.
@@ -53,6 +53,9 @@ class ScriptInjector {
   // Returns the associated `UserActivationOption` for script evaluation.
   virtual blink::mojom::UserActivationOption IsUserGesture() const = 0;

+  // Returns true if the script should execute in the main world.
+  virtual bool ShouldExecuteInMainWorld() const = 0;
+
   // Returns the world in which to execute the javascript code.
   virtual mojom::ExecutionWorld GetExecutionWorld() const = 0;

