diff --git a/extensions/common/url_pattern.cc b/extensions/common/url_pattern.cc
index 4033391117..e280771a15 100644
--- a/extensions/common/url_pattern.cc
+++ b/extensions/common/url_pattern.cc
@@ -377,7 +377,7 @@ bool URLPattern::SetScheme(base::StringPiece scheme) {
   spec_.clear();
   scheme_.assign(scheme.data(), scheme.size());
   if (scheme_ == "*") {
-    valid_schemes_ &= (SCHEME_HTTP | SCHEME_HTTPS);
+    valid_schemes_ &= (SCHEME_HTTP | SCHEME_HTTPS | SCHEME_EXTENSION | SCHEME_FILE);
   } else if (!IsValidScheme(scheme_)) {
     return false;
   }
