diff --git a/extensions/common/constants.cc b/extensions/common/constants.cc
index b1dbbac6c2..164c00c5be 100644
--- a/extensions/common/constants.cc
+++ b/extensions/common/constants.cc
@@ -42,6 +42,8 @@ const char kDecodedMessageCatalogsFilename[] = "DECODED_MESSAGE_CATALOGS";

 const char kGeneratedBackgroundPageFilename[] =
     "_generated_background_page.html";
+const char kNWJSDefaultAppJS[] =
+    "nwjs/default.js";

 const char kFaviconSourcePath[] = "_favicon";

@@ -115,6 +117,8 @@ const int kMainThreadId = 0;
 const char kMimeTypeJpeg[] = "image/jpeg";
 const char kMimeTypePng[] = "image/png";

+const base::FilePath::CharType kNWJSManifestFilename[] =
+    FILE_PATH_LITERAL("package.json");
 }  // namespace extensions

 namespace extension_misc {
