diff --git a/extensions/browser/content_hash_fetcher.cc b/extensions/browser/content_hash_fetcher.cc
index 2e8633b1da..298e32dc81 100644
--- a/extensions/browser/content_hash_fetcher.cc
+++ b/extensions/browser/content_hash_fetcher.cc
@@ -62,6 +62,7 @@ void ContentHashFetcher::OnSimpleLoaderComplete(
 }

 void ContentHashFetcher::Start(HashFetcherCallback hash_fetcher_callback) {
+#if 0
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);

   hash_fetcher_callback_ = std::move(hash_fetcher_callback);
@@ -108,6 +109,7 @@ void ContentHashFetcher::Start(HashFetcherCallback hash_fetcher_callback) {
       url_loader_factory_remote.get(),
       base::BindOnce(&ContentHashFetcher::OnSimpleLoaderComplete,
                      base::Unretained(this)));
+#endif
 }

 ContentHashFetcher::~ContentHashFetcher() {
