diff --git a/extensions/common/permissions/permission_message.cc b/extensions/common/permissions/permission_message.cc
index 624de47f7d..4c6d99070a 100644
--- a/extensions/common/permissions/permission_message.cc
+++ b/extensions/common/permissions/permission_message.cc
@@ -8,7 +8,7 @@ namespace extensions {

 PermissionMessage::PermissionMessage(const std::u16string& message,
                                      const PermissionIDSet& permissions)
-    : message_(message), permissions_(permissions) {}
+    : message_(message), permissions_(permissions), submessages_() {}

 PermissionMessage::PermissionMessage(
     const std::u16string& message,
