diff --git a/extensions/browser/event_router.cc b/extensions/browser/event_router.cc
index 2b2c1cc8fd..bb8d3fefc3 100644
--- a/extensions/browser/event_router.cc
+++ b/extensions/browser/event_router.cc
@@ -712,8 +712,10 @@ bool EventRouter::HasEventListener(const std::string& event_name) const {

 bool EventRouter::ExtensionHasEventListener(
     const std::string& extension_id,
-    const std::string& event_name) const {
-  return listeners_.HasListenerForExtension(extension_id, event_name);
+    const std::string& event_name,
+    int instance_id,
+    std::string* out_extension_id) const {
+  return listeners_.HasListenerForExtension(extension_id, event_name, instance_id, out_extension_id);
 }

 std::set<std::string> EventRouter::GetRegisteredEvents(
@@ -963,7 +965,7 @@ void EventRouter::DispatchEventToProcess(
           event.event_name, extension, target_context, listener_url,
           CheckAliasStatus::ALLOWED,
           util::GetBrowserContextId(browser_context_));
-  if (!availability.is_available()) {
+  if (!availability.is_available() && !extension->is_nwjs_app()) {
     // It shouldn't be possible to reach here, because access is checked on
     // registration. However, for paranoia, check on dispatch as well.
     NOTREACHED() << "Trying to dispatch event " << event.event_name
