diff --git a/extensions/renderer/resources/guest_view/web_view/web_view.js b/extensions/renderer/resources/guest_view/web_view/web_view.js
index 8bd3d83907..e098d68d5b 100644
--- a/extensions/renderer/resources/guest_view/web_view/web_view.js
+++ b/extensions/renderer/resources/guest_view/web_view/web_view.js
@@ -28,6 +28,8 @@ WebViewImpl.prototype.__proto__ = GuestViewContainer.prototype;

 // Sets up all of the webview attributes.
 WebViewImpl.prototype.setupAttributes = function() {
+  this.attributes[WebViewConstants.ATTRIBUTE_ALLOWNW] =
+      new WebViewAttributes.AllowNWAttribute(this);
   this.attributes[WebViewConstants.ATTRIBUTE_ALLOWSCALING] =
       new WebViewAttributes.AllowScalingAttribute(this);
   this.attributes[WebViewConstants.ATTRIBUTE_ALLOWTRANSPARENCY] =
@@ -221,6 +223,28 @@ WebViewImpl.prototype.setUserAgentOverride = function(userAgentOverride) {
   return true;
 };

+WebViewImpl.prototype.getGuestId = function() {
+  return this.guest.getId();
+};
+
+WebViewImpl.prototype.showDevTools = function(show, container) {
+  if (!this.guest.getId()) {
+    return;
+  }
+  if (container)
+    WebViewInternal.showDevTools(this.guest.getId(), show, container.getProcessId(), container.getGuestId());
+  else
+    WebViewInternal.showDevTools(this.guest.getId(), show);
+};
+
+WebViewImpl.prototype.getCookieStoreId = function() {
+    return this.processId + "," + this.guest.getId();
+}
+
+WebViewImpl.prototype.inspectElementAt = function(x, y) {
+  WebViewInternal.inspectElementAt(this.guest.getId(), x, y);
+};
+
 WebViewImpl.prototype.loadDataWithBaseUrl = function(
     dataUrl, baseUrl, virtualUrl) {
   if (!this.guest.getId()) {
