diff --git a/extensions/browser/guest_view/web_view/web_view_constants.h b/extensions/browser/guest_view/web_view/web_view_constants.h
index 1ab358f8eb..1d88efaae1 100644
--- a/extensions/browser/guest_view/web_view/web_view_constants.h
+++ b/extensions/browser/guest_view/web_view/web_view_constants.h
@@ -16,6 +16,7 @@ extern const char kAttributeAllowTransparency[];
 extern const char kAttributeAllowScaling[];
 extern const char kAttributeName[];
 extern const char kAttributeSrc[];
+extern const char kAttributeAllowNW[];

 // API namespace.
 // TODO(kalman): Consolidate this with the other API constants.
