diff --git a/extensions/browser/guest_view/web_view/web_view_constants.cc b/extensions/browser/guest_view/web_view/web_view_constants.cc
index 7bd18dc0f9..137ea47210 100644
--- a/extensions/browser/guest_view/web_view/web_view_constants.cc
+++ b/extensions/browser/guest_view/web_view/web_view_constants.cc
@@ -11,6 +11,7 @@ const char kAttributeAllowTransparency[] = "allowtransparency";
 const char kAttributeAllowScaling[] = "allowscaling";
 const char kAttributeName[] = "name";
 const char kAttributeSrc[] = "src";
+const char kAttributeAllowNW[] = "allownw";

 // API namespace.
 const char kAPINamespace[] = "webViewInternal";
