diff --git a/extensions/renderer/programmatic_script_injector.cc b/extensions/renderer/programmatic_script_injector.cc
index 37c6b12f34..a289364de8 100644
--- a/extensions/renderer/programmatic_script_injector.cc
+++ b/extensions/renderer/programmatic_script_injector.cc
@@ -39,6 +39,10 @@ mojom::InjectionType ProgrammaticScriptInjector::script_type() const {
   return mojom::InjectionType::kProgrammaticScript;
 }

+bool ProgrammaticScriptInjector::ShouldExecuteInMainWorld() const {
+  return params_->in_main_world;
+}
+
 blink::mojom::UserActivationOption ProgrammaticScriptInjector::IsUserGesture()
     const {
   DCHECK(params_->injection->is_js());
