diff --git a/extensions/renderer/module_system.h b/extensions/renderer/module_system.h
index 1830092b15..4c9cea4ed8 100644
--- a/extensions/renderer/module_system.h
+++ b/extensions/renderer/module_system.h
@@ -126,7 +126,7 @@ class ModuleSystem : public ObjectBackedNativeHandler {
   // code to set up various hooks.
   // TODO(devlin): We can get rid of this once we convert all our custom
   // bindings.
-  void OnNativeBindingCreated(const std::string& api_name,
+  v8::Local<v8::Value> OnNativeBindingCreated(const std::string& api_name,
                               v8::Local<v8::Value> api_bridge_value);

   void SetGetInternalAPIHook(v8::Local<v8::FunctionTemplate> get_internal_api);
