diff --git a/extensions/common/manifest_handlers/webview_info.cc b/extensions/common/manifest_handlers/webview_info.cc
index 8a77fc4c7a..ed233c8153 100644
--- a/extensions/common/manifest_handlers/webview_info.cc
+++ b/extensions/common/manifest_handlers/webview_info.cc
@@ -101,6 +101,34 @@ bool WebviewInfo::HasWebviewAccessibleResources(
   }
   return false;
 }
+
+bool WebviewInfo::IsURLWebviewAccessible(const Extension* extension,
+                                         const std::string& partition_id,
+                                         const GURL& url,
+                                         bool* file_scheme) {
+  if (!extension)
+    return false;
+
+  const WebviewInfo* webview_info = static_cast<const WebviewInfo*>(
+      extension->GetManifestData(keys::kWebviewAccessibleResources));
+  if (!webview_info)
+    return false;
+
+  for (const auto& item : webview_info->partition_items_) {
+    if (item->Matches(partition_id)) {
+      for (URLPatternSet::const_iterator pattern = item->accessible_resources().begin();
+           pattern != item->accessible_resources().end(); ++pattern) {
+        if (pattern->MatchesURL(url)) {
+          if (pattern->MatchesScheme("file") && file_scheme)
+            *file_scheme = true;
+          return true;
+        }
+      }
+    }
+  }
+
+  return false;
+}

 void WebviewInfo::AddPartitionItem(std::unique_ptr<PartitionItem> item) {
   partition_items_.push_back(std::move(item));
@@ -174,6 +202,10 @@ bool WebviewHandler::Parse(Extension* extension, std::u16string* error) {
             errors::kInvalidWebviewAccessibleResource, base::NumberToString(i));
         return false;
       }
+      URLPattern try_pattern(URLPattern::SCHEME_ALL);
+      if (try_pattern.Parse(item.GetString()) == URLPattern::ParseResult::kSuccess) {
+        partition_item->AddPattern(try_pattern);
+      } else {

       GURL pattern_url =
           Extension::GetResourceURL(extension->url(), item.GetString());
@@ -205,6 +237,7 @@ bool WebviewHandler::Parse(Extension* extension, std::u16string* error) {
       }

       partition_item->AddPattern(std::move(pattern));
+      } //nwjs
     }
     info->AddPartitionItem(std::move(partition_item));
   }
