diff --git a/google_apis/gaia/gaia_auth_fetcher.cc b/google_apis/gaia/gaia_auth_fetcher.cc
index 8ec8028928..606a8572de 100644
--- a/google_apis/gaia/gaia_auth_fetcher.cc
+++ b/google_apis/gaia/gaia_auth_fetcher.cc
@@ -40,7 +40,7 @@

 namespace {

-const size_t kMaxMessageSize = 1024 * 1024;  // 1MB
+//const size_t kMaxMessageSize = 1024 * 1024;  // 1MB

 constexpr char kBadAuthenticationError[] = "BadAuthentication";
 constexpr char kBadAuthenticationShortError[] = "badauth";
@@ -287,6 +287,7 @@ void GaiaAuthFetcher::CreateAndStartGaiaFetcher(
     const GURL& gaia_gurl,
     network::mojom::CredentialsMode credentials_mode,
     const net::NetworkTrafficAnnotationTag& traffic_annotation) {
+#if 0
   DCHECK(!fetch_pending_) << "Tried to fetch two things at once!";

   auto resource_request = std::make_unique<network::ResourceRequest>();
@@ -346,6 +347,7 @@ void GaiaAuthFetcher::CreateAndStartGaiaFetcher(
       base::BindOnce(&GaiaAuthFetcher::OnURLLoadComplete,
                      base::Unretained(this)),
       kMaxMessageSize);
+#endif
 }

 // static
