diff --git a/headless/BUILD.gn b/headless/BUILD.gn
index a9d686655e..fd2c556571 100644
--- a/headless/BUILD.gn
+++ b/headless/BUILD.gn
@@ -304,6 +304,8 @@ source_set("headless_shared_sources") {
   sources = [
     "app/headless_shell_switches.cc",
     "app/headless_shell_switches.h",
+    "lib/headless_content_main_delegate.cc",
+    "lib/headless_content_main_delegate.h",
     "lib/headless_content_client.cc",
     "lib/headless_content_client.h",
     "public/headless_browser.cc",
@@ -625,10 +627,8 @@ if (!is_component_build) {
   static_library("headless_renderer") {
     defines = []
     sources = [
-      "lib/browser/headless_web_contents_impl.cc",
-      "lib/browser/headless_web_contents_impl.h",
-      "lib/headless_content_main_delegate.cc",
-      "lib/headless_content_main_delegate.h",
+      #"lib/browser/headless_web_contents_impl.cc",
+      #"lib/browser/headless_web_contents_impl.h",
       "lib/renderer/headless_content_renderer_client.cc",
       "lib/renderer/headless_content_renderer_client.h",
       "lib/utility/headless_content_utility_client.cc",
@@ -940,8 +940,8 @@ static_library("headless_shell_lib") {
     sources += [
       "lib/browser/headless_content_browser_client.cc",
       "lib/browser/headless_content_browser_client.h",
-      "lib/utility/headless_content_utility_client.cc",
-      "lib/utility/headless_content_utility_client.h",
+      #"lib/utility/headless_content_utility_client.cc",
+      #"lib/utility/headless_content_utility_client.h",
     ]
   }

@@ -972,6 +972,13 @@ static_library("headless_shell_lib") {
     deps += [ "//components/crash/content/browser" ]
   }

+  if ((is_linux || is_mac) && is_component_build) {
+    sources += [
+      "app/headless_shell_switches.cc",
+      "app/headless_shell_switches.h",
+    ]
+  }
+
   if (enable_basic_printing) {
     deps += [
       "//components/printing/browser",
