diff --git a/ipc/ipc_logging.cc b/ipc/ipc_logging.cc
index 9307bb8042..695e275a02 100644
--- a/ipc/ipc_logging.cc
+++ b/ipc/ipc_logging.cc
@@ -242,7 +242,7 @@ void Logging::Log(const LogData& data) {
       }
     }
   }
-  if (enabled_on_stderr_) {
+  if (enabled_on_stderr_ && !sender_) {
     std::string message_name;
     if (data.message_name.empty()) {
       message_name = base::StringPrintf("[unknown type %d]", data.type);
