diff --git a/media/audio/audio_manager_base.cc b/media/audio/audio_manager_base.cc
index 7090de5b88..fd6463533c 100644
--- a/media/audio/audio_manager_base.cc
+++ b/media/audio/audio_manager_base.cc
@@ -130,9 +130,10 @@ AudioManagerBase::AudioManagerBase(std::unique_ptr<AudioThread> audio_thread,

 AudioManagerBase::~AudioManagerBase() {
   // All the output streams should have been deleted.
-  CHECK_EQ(0, num_output_streams_);
+  //disable the check for NWJS#5318
+  //CHECK_EQ(0, num_output_streams_);
   // All the input streams should have been deleted.
-  CHECK(input_streams_.empty());
+  //CHECK(input_streams_.empty());
 }

 void AudioManagerBase::GetAudioInputDeviceDescriptions(
