diff --git a/mojo/public/cpp/base/message_pump_type_mojom_traits.cc b/mojo/public/cpp/base/message_pump_type_mojom_traits.cc
index f224198d3c..7b24315451 100644
--- a/mojo/public/cpp/base/message_pump_type_mojom_traits.cc
+++ b/mojo/public/cpp/base/message_pump_type_mojom_traits.cc
@@ -20,6 +20,8 @@ EnumTraits<mojo_base::mojom::MessagePumpType, base::MessagePumpType>::ToMojom(
       return mojo_base::mojom::MessagePumpType::kCustom;
     case base::MessagePumpType::IO:
       return mojo_base::mojom::MessagePumpType::kIo;
+    case base::MessagePumpType::NODE:
+      return mojo_base::mojom::MessagePumpType::kNode;
 #if BUILDFLAG(IS_ANDROID)
     case base::MessagePumpType::JAVA:
       return mojo_base::mojom::MessagePumpType::kJava;
@@ -50,6 +52,9 @@ bool EnumTraits<mojo_base::mojom::MessagePumpType, base::MessagePumpType>::
     case mojo_base::mojom::MessagePumpType::kIo:
       *output = base::MessagePumpType::IO;
       return true;
+    case mojo_base::mojom::MessagePumpType::kNode:
+      *output = base::MessagePumpType::NODE;
+      return true;
 #if BUILDFLAG(IS_ANDROID)
     case mojo_base::mojom::MessagePumpType::kJava:
       *output = base::MessagePumpType::JAVA;
