diff --git a/net/proxy_resolution/configured_proxy_resolution_service.h b/net/proxy_resolution/configured_proxy_resolution_service.h
index b640467a1c..2082e7e084 100644
--- a/net/proxy_resolution/configured_proxy_resolution_service.h
+++ b/net/proxy_resolution/configured_proxy_resolution_service.h
@@ -154,6 +154,15 @@ class NET_EXPORT ConfiguredProxyResolutionService
   // ProxyResolutionService
   void OnShutdown() override;

+  // Tells this ProxyResolutionService to start using a new
+  // ProxyConfigService
+  // to retrieve its ProxyConfig from. The new ProxyConfigService
+  // will
+  // immediately be queried for new config info which will be used
+  // for all
+  // subsequent ResolveProxy calls.
+  void ResetConfigService(
+                          std::unique_ptr<ProxyConfigService> new_proxy_config_service);
   // Returns the last configuration fetched from ProxyConfigService.
   const absl::optional<ProxyConfigWithAnnotation>& fetched_config() const {
     return fetched_config_;
