diff --git a/net/cert/cert_verify_proc_mac.cc b/net/cert/cert_verify_proc_mac.cc
index 395c467be7..2e920e8329 100644
--- a/net/cert/cert_verify_proc_mac.cc
+++ b/net/cert/cert_verify_proc_mac.cc
@@ -5,6 +5,7 @@
 #include "net/cert/cert_verify_proc_mac.h"

 #include <CommonCrypto/CommonDigest.h>
+#include <CoreFoundation/CFArray.h>
 #include <CoreServices/CoreServices.h>
 #include <Security/Security.h>

@@ -568,6 +569,7 @@ int BuildAndEvaluateSecTrustRef(CFArrayRef cert_array,
                                 CFArrayRef sct_array_ref,
                                 int flags,
                                 CFArrayRef keychain_search_list,
+                                const CertificateList& additional_trust_anchors,
                                 ScopedCFTypeRef<SecTrustRef>* trust_ref,
                                 SecTrustResultType* trust_result,
                                 ScopedCFTypeRef<CFArrayRef>* verified_chain,
@@ -694,6 +696,7 @@ int VerifyWithGivenFlags(X509Certificate* cert,
                          bool rev_checking_soft_fail,
                          CRLSet* crl_set,
                          CertVerifyResult* verify_result,
+                         const CertificateList& additional_trust_anchors,
                          CRLSetResult* completed_chain_crl_result) {
   ScopedCFTypeRef<CFArrayRef> trust_policies;
   OSStatus status = CreateTrustPolicies(flags, &trust_policies);
@@ -872,7 +875,7 @@ int VerifyWithGivenFlags(X509Certificate* cert,
       int rv = BuildAndEvaluateSecTrustRef(
           cert_array, trust_policies, ocsp_response_ref.get(),
           sct_array_ref.get(), flags,
-          scoped_alternate_keychain_search_list.get(), &temp_ref,
+          scoped_alternate_keychain_search_list.get(), additional_trust_anchors, &temp_ref,
           &temp_trust_result, &temp_chain, &temp_chain_info);
       if (rv != OK)
         return rv;
@@ -1112,7 +1115,7 @@ CertVerifyProcMac::CertVerifyProcMac() = default;
 CertVerifyProcMac::~CertVerifyProcMac() = default;

 bool CertVerifyProcMac::SupportsAdditionalTrustAnchors() const {
-  return false;
+  return true;
 }

 int CertVerifyProcMac::VerifyInternal(
@@ -1136,7 +1139,7 @@ int CertVerifyProcMac::VerifyInternal(
   CRLSetResult completed_chain_crl_result;
   int rv = VerifyWithGivenFlags(cert, hostname, ocsp_response, sct_list, flags,
                                 /*rev_checking_soft_fail=*/true, crl_set,
-                                verify_result, &completed_chain_crl_result);
+                                verify_result, additional_trust_anchors, &completed_chain_crl_result);
   if (rv != OK)
     return rv;

