diff --git a/net/cert/test_root_certs.h b/net/cert/test_root_certs.h
index c299b3c6d3..d717e5a236 100644
--- a/net/cert/test_root_certs.h
+++ b/net/cert/test_root_certs.h
@@ -74,6 +74,7 @@ class NET_EXPORT TestRootCerts {
   TestRootCerts();
   ~TestRootCerts();

+public:
   // Marks |certificate| as trusted in the effective trust store
   // used by CertVerifier::Verify(). Returns false if the
   // certificate could not be marked trusted.
