diff --git a/ppapi/native_client/src/untrusted/pnacl_support_extension/BUILD.gn b/ppapi/native_client/src/untrusted/pnacl_support_extension/BUILD.gn
index b65add1abf..8af87e01fe 100644
--- a/ppapi/native_client/src/untrusted/pnacl_support_extension/BUILD.gn
+++ b/ppapi/native_client/src/untrusted/pnacl_support_extension/BUILD.gn
@@ -35,7 +35,10 @@ action("pnacl_support_extension") {
   shim_cpu = nacl_target_cpu

   output_prefix = "$root_out_dir/pnacl/pnacl_public_"
-  outputs = [ "${output_prefix}pnacl_json" ]
+  outputs = [
+    "$root_out_dir/pnacl",
+    "${output_prefix}pnacl_json",
+  ]

   # Files that will be copied from the toolchain to output.
   outputs_from_toolchain = [
