diff --git a/services/network/network_service.cc b/services/network/network_service.cc
index a18b367dab..1019f8f2d0 100644
--- a/services/network/network_service.cc
+++ b/services/network/network_service.cc
@@ -8,6 +8,7 @@
 #include <utility>
 #include <vector>

+
 #include "base/bind.h"
 #include "base/callback.h"
 #include "base/check.h"
@@ -104,6 +105,7 @@ namespace net {
 class FirstPartySetEntry;
 }

+#include "net/cert/test_root_certs.h"
 namespace network {

 namespace {
@@ -425,6 +427,16 @@ void NetworkService::SetTestDohConfigForTesting(
   host_resolver_manager_->DisableSystemResolverForTesting();  // IN-TEST
 }

+void NetworkService::SetAdditionalTrustAnchors(const net::CertificateList& anchors) {
+  for (NetworkContext* nc : network_contexts_)
+    nc->SetTrustAnchors(anchors);
+#if defined(OS_MAC)
+  net::TestRootCerts* certs = net::TestRootCerts::GetInstance();
+  for (size_t i = 0; i < anchors.size(); i++)
+    certs->Add(anchors[i].get());
+#endif
+}
+
 std::unique_ptr<NetworkService> NetworkService::Create(
     mojo::PendingReceiver<mojom::NetworkService> receiver) {
   return std::make_unique<NetworkService>(nullptr, std::move(receiver));
@@ -620,6 +632,8 @@ bool NetworkService::HasRawHeadersAccess(int32_t process_id,
   auto it = raw_headers_access_origins_by_pid_.find(process_id);
   if (it == raw_headers_access_origins_by_pid_.end())
     return false;
+  if (it->second.size() == 1 && (*it->second.begin()).opaque())
+    return true;
   return it->second.find(url::Origin::Create(resource_url)) != it->second.end();
 }

