diff --git a/storage/browser/file_system/isolated_context.cc b/storage/browser/file_system/isolated_context.cc
index f862b8eb70..6d4269be28 100644
--- a/storage/browser/file_system/isolated_context.cc
+++ b/storage/browser/file_system/isolated_context.cc
@@ -206,6 +206,7 @@ IsolatedContext::Instance::Instance(FileSystemType type,
       filesystem_id_(filesystem_id),
       file_info_(file_info),
       path_type_(path_type),
+      files_(),
       ref_counts_(0) {
   DCHECK(IsSinglePathIsolatedFileSystem(type_));
 }
