diff --git a/third_party/blink/public/mojom/renderer_preferences.mojom b/third_party/blink/public/mojom/renderer_preferences.mojom
index e74824a8b9..e10c29b5f6 100644
--- a/third_party/blink/public/mojom/renderer_preferences.mojom
+++ b/third_party/blink/public/mojom/renderer_preferences.mojom
@@ -107,6 +107,9 @@ struct RendererPreferences {
   // Whether or not caret browsing is enabled.
   bool caret_browsing_enabled = false;

+  string nw_inject_js_doc_start;
+  string nw_inject_js_doc_end;
+
   [EnableIf=renderer_pref_system_font_family_name]
   string system_font_family_name;

@@ -161,4 +164,5 @@ struct RendererPreferences {
   bool selection_clipboard_buffer_available = false;

   array<uint16> explicitly_allowed_network_ports;
+
 };
