diff --git a/third_party/blink/renderer/core/page/drag_controller.cc b/third_party/blink/renderer/core/page/drag_controller.cc
index afcd6148cf..87a83dd221 100644
--- a/third_party/blink/renderer/core/page/drag_controller.cc
+++ b/third_party/blink/renderer/core/page/drag_controller.cc
@@ -230,7 +230,9 @@ void DragController::DragExited(DragData* drag_data, LocalFrame& local_root) {

   LocalFrameView* frame_view(local_root.View());
   if (frame_view) {
-    DataTransferAccessPolicy policy = DataTransferAccessPolicy::kTypesReadable;
+    DataTransferAccessPolicy policy =
+      document_under_mouse_->domWindow()->GetSecurityOrigin()->hasUniversalAccess()
+      ? DataTransferAccessPolicy::kReadable : DataTransferAccessPolicy::kTypesReadable;
     DataTransfer* data_transfer = CreateDraggingDataTransfer(policy, drag_data);
     data_transfer->SetSourceOperation(drag_data->DraggingSourceOperationMask());
     local_root.GetEventHandler().CancelDragAndDrop(CreateMouseEvent(drag_data),
@@ -800,7 +802,10 @@ bool DragController::TryDHTMLDrag(DragData* drag_data,
   if (!local_root.View())
     return false;

-  DataTransferAccessPolicy policy = DataTransferAccessPolicy::kTypesReadable;
+  DataTransferAccessPolicy policy =
+    document_under_mouse_->domWindow()->GetSecurityOrigin()->hasUniversalAccess()
+    ? DataTransferAccessPolicy::kReadable
+    : DataTransferAccessPolicy::kTypesReadable;
   DataTransfer* data_transfer = CreateDraggingDataTransfer(policy, drag_data);
   DragOperationsMask src_op_mask = drag_data->DraggingSourceOperationMask();
   data_transfer->SetSourceOperation(src_op_mask);
