diff --git a/third_party/blink/renderer/core/loader/empty_clients.h b/third_party/blink/renderer/core/loader/empty_clients.h
index a2d7e97958..520ad5a50e 100644
--- a/third_party/blink/renderer/core/loader/empty_clients.h
+++ b/third_party/blink/renderer/core/loader/empty_clients.h
@@ -105,7 +105,7 @@ class CORE_EXPORT EmptyChromeClient : public ChromeClient {
             LocalFrame& opener_frame,
             NavigationPolicy navigation_policy,
             const gfx::Rect& initial_rect,
-            bool consumed_user_gesture) override {}
+            bool consumed_user_gesture, WebString* manifest = nullptr) override {}
   void DidOverscroll(const gfx::Vector2dF&,
                      const gfx::Vector2dF&,
                      const gfx::PointF&,
@@ -155,7 +155,7 @@ class CORE_EXPORT EmptyChromeClient : public ChromeClient {
                              const WebWindowFeatures&,
                              network::mojom::blink::WebSandboxFlags,
                              const SessionStorageNamespaceId&,
-                             bool& consumed_user_gesture) override {
+                             bool& consumed_user_gesture, WebString*) override {
     return nullptr;
   }
   bool OpenJavaScriptAlertDelegate(LocalFrame*, const String&) override {
