diff --git a/third_party/blink/renderer/core/html/forms/file_chooser.cc b/third_party/blink/renderer/core/html/forms/file_chooser.cc
index b0b75f7dce..acb475082b 100644
--- a/third_party/blink/renderer/core/html/forms/file_chooser.cc
+++ b/third_party/blink/renderer/core/html/forms/file_chooser.cc
@@ -110,6 +110,7 @@ void FileChooser::EnumerateChosenDirectory() {
 }

 void FileChooser::DidChooseFiles(mojom::blink::FileChooserResultPtr result) {
+  bool canceled = false;
   // TODO(tkent): If |result| is nullptr, we should not clear the
   // already-selected files in <input type=file> like other browsers.
   FileChooserFileInfoList files;
@@ -135,14 +136,16 @@ void FileChooser::DidChooseFiles(mojom::blink::FileChooserResultPtr result) {
     }
     if (!was_changed) {
       DidCloseChooser();
-      return;
+      canceled = true;
     }
   }

   if (client_) {
     client_->FilesChosen(std::move(files),
-                         result ? result->base_directory : base::FilePath());
+                         result ? result->base_directory : base::FilePath(), canceled);
   }
+  if (canceled)
+    return;
   DidCloseChooser();
 }

