diff --git a/third_party/blink/public/web/web_frame.h b/third_party/blink/public/web/web_frame.h
index 4c01b39235..181d93d344 100644
--- a/third_party/blink/public/web/web_frame.h
+++ b/third_party/blink/public/web/web_frame.h
@@ -90,6 +90,12 @@ class BLINK_EXPORT WebFrame {
       CrossVariantMojoAssociatedReceiver<mojom::RemoteFrameInterfaceBase>
           receiver,
       mojom::FrameReplicationStatePtr replicated_state);
+  virtual bool isNwFakeTop() const {return false;}
+  virtual void setNodeJS(bool) {}
+  virtual bool isNodeJS() const {return false;}
+  virtual bool isNwDisabledChildFrame() const {return false;}
+  virtual void setDevtoolsJail(WebFrame*) {}
+  virtual WebFrame* getDevtoolsJail() {return nullptr;}

   // This method closes and deletes the WebFrame. This is typically called by
   // the embedder in response to a frame detached callback to the WebFrame
@@ -187,8 +193,8 @@ class BLINK_EXPORT WebFrame {
   // RemoteFrame <--> RenderFrameProxyHost in the browser process.
   const FrameToken& GetFrameToken() const { return frame_token_; }

-#if INSIDE_BLINK
   static WebFrame* FromCoreFrame(Frame*);
+#if INSIDE_BLINK
   static Frame* ToCoreFrame(const WebFrame&);
 #endif

