diff --git a/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.h b/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.h
index 178568cec1..5bb2427863 100644
--- a/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.h
+++ b/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.h
@@ -62,7 +62,7 @@ class CORE_EXPORT ArrayBufferContents {
                       size_t element_byte_size,
                       SharingType is_shared,
                       InitializationPolicy);
-  ArrayBufferContents(void* data, size_t length, DataDeleter deleter);
+  ArrayBufferContents(void* data, size_t length, bool nodejs, DataDeleter deleter);
   ArrayBufferContents(
       const base::subtle::PlatformSharedMemoryRegion& shared_memory_region,
       uint64_t offset,
@@ -114,6 +114,7 @@ class CORE_EXPORT ArrayBufferContents {
   static void FreeMemory(void*);

  private:
+  bool nodejs_;
   static void* AllocateMemoryWithFlags(size_t,
                                        InitializationPolicy,
                                        unsigned int);
