diff --git a/third_party/blink/public/mojom/frame/frame_owner_properties.mojom b/third_party/blink/public/mojom/frame/frame_owner_properties.mojom
index 479863ca47..1a81ff4475 100644
--- a/third_party/blink/public/mojom/frame/frame_owner_properties.mojom
+++ b/third_party/blink/public/mojom/frame/frame_owner_properties.mojom
@@ -18,8 +18,10 @@ struct FrameOwnerProperties {
   int32 margin_height;
   // True if the frame allows fullscreen.
   bool allow_fullscreen;
+  bool nwfaketop;
   // True if the frame allows payment request.
   bool allow_payment_request;
+  string nwuseragent;
   // True if the frame is display:none.
   bool is_display_none;
   // The used value of color-scheme for the frame element.
