diff --git a/third_party/blink/public/mojom/frame/frame.mojom b/third_party/blink/public/mojom/frame/frame.mojom
index cc48c69754..a8ab7bfe09 100644
--- a/third_party/blink/public/mojom/frame/frame.mojom
+++ b/third_party/blink/public/mojom/frame/frame.mojom
@@ -996,7 +996,8 @@ interface LocalMainFrameHost {
   // which is always set. This is called on the main frame of the window to be shown.
   ShowCreatedWindow(blink.mojom.LocalFrameToken opener_frame_token,
                     ui.mojom.WindowOpenDisposition disposition,
-                    gfx.mojom.Rect rect, bool opened_by_user_gesture) => ();
+                    gfx.mojom.Rect rect, bool opened_by_user_gesture,
+                    mojo_base.mojom.String16 manifest) => ();

   // Request that the browser change the bounds of the window.
   // This corresponds to the window.resizeTo() and window.moveTo() APIs, and the browser
