diff --git a/third_party/blink/common/privacy_budget/identifiability_study_settings.cc b/third_party/blink/common/privacy_budget/identifiability_study_settings.cc
index 7b493bec66..0e28d00390 100644
--- a/third_party/blink/common/privacy_budget/identifiability_study_settings.cc
+++ b/third_party/blink/common/privacy_budget/identifiability_study_settings.cc
@@ -60,7 +60,8 @@ class ThreadsafeSettingsWrapper {
   // Same restrictions as IdentifiabilityStudySettings::SetGlobalProvider().
   void SetProvider(
       std::unique_ptr<IdentifiabilityStudySettingsProvider> provider) {
-    DCHECK(!initialized_.IsSet());
+    if (initialized_.IsSet())
+      return;
     initialized_settings_.emplace(std::move(provider));
     initialized_.Set();
   }
