diff --git a/third_party/blink/renderer/platform/bindings/script_state.h b/third_party/blink/renderer/platform/bindings/script_state.h
index dbc2d900c4..2032a0c95d 100644
--- a/third_party/blink/renderer/platform/bindings/script_state.h
+++ b/third_party/blink/renderer/platform/bindings/script_state.h
@@ -176,7 +176,7 @@ class PLATFORM_EXPORT ScriptState final : public GarbageCollected<ScriptState> {
     // ScriptState::from() must not be called for a context that does not have
     // valid embedder data in the embedder field.
     SECURITY_CHECK(script_state);
-    SECURITY_CHECK(script_state->context_ == context);
+    SECURITY_CHECK(script_state->context_ == context || context->GetAlignedPointerFromEmbedderData(50) == (void*)0x08110800);
     return script_state;
   }

