diff --git a/third_party/blink/renderer/core/exported/web_shared_worker_impl.h b/third_party/blink/renderer/core/exported/web_shared_worker_impl.h
index ae80372de1..016c25186a 100644
--- a/third_party/blink/renderer/core/exported/web_shared_worker_impl.h
+++ b/third_party/blink/renderer/core/exported/web_shared_worker_impl.h
@@ -35,6 +35,7 @@

 #include <memory>

+#include "base/files/file_path.h"
 #include "base/memory/scoped_refptr.h"
 #include "base/memory/weak_ptr.h"
 #include "mojo/public/cpp/bindings/remote.h"
@@ -96,6 +97,7 @@ class CORE_EXPORT WebSharedWorkerImpl final : public WebSharedWorker {
       WebSharedWorkerClient*);

   void StartWorkerContext(
+                          bool, const base::FilePath&,
       const WebURL&,
       mojom::blink::ScriptType,
       network::mojom::CredentialsMode,
@@ -147,6 +149,8 @@ class CORE_EXPORT WebSharedWorkerImpl final : public WebSharedWorker {

   bool running_ = false;
   bool asked_to_terminate_ = false;
+  bool nodejs_;
+  base::FilePath root_path_;

   base::WeakPtrFactory<WebSharedWorkerImpl> weak_ptr_factory_{this};
 };
