diff --git a/third_party/blink/renderer/platform/bindings/script_state.cc b/third_party/blink/renderer/platform/bindings/script_state.cc
index 2e12ccfd4e..6a941a1a3c 100644
--- a/third_party/blink/renderer/platform/bindings/script_state.cc
+++ b/third_party/blink/renderer/platform/bindings/script_state.cc
@@ -23,6 +23,10 @@ ScriptState::ScriptState(v8::Local<v8::Context> context,
   context->SetAlignedPointerInEmbedderData(kV8ContextPerContextDataIndex, this);
   RendererResourceCoordinator::Get()->OnScriptStateCreated(this,
                                                            execution_context);
+  for (int i = 32; i <= 36; i++) { //node_context_data.h
+    context->SetAlignedPointerInEmbedderData(i, nullptr);
+  }
+  context->SetAlignedPointerInEmbedderData(50, nullptr);
 }

 ScriptState::~ScriptState() {
