diff --git a/third_party/blink/renderer/core/script/module_map.h b/third_party/blink/renderer/core/script/module_map.h
index bd67e46dc1..bc3904e304 100644
--- a/third_party/blink/renderer/core/script/module_map.h
+++ b/third_party/blink/renderer/core/script/module_map.h
@@ -51,6 +51,7 @@ class CORE_EXPORT ModuleMap final : public GarbageCollected<ModuleMap>,
   // If the URL wasn't fetched, or is currently being fetched, this returns a
   // nullptr.
   ModuleScript* GetFetchedModuleScript(const KURL&, ModuleType) const;
+  void AddToMap(const KURL&, ModuleScript*);

   Modulator* GetModulator() { return modulator_; }

