diff --git a/third_party/blink/public/devtools_protocol/browser_protocol.pdl b/third_party/blink/public/devtools_protocol/browser_protocol.pdl
index bf0a4f2201..4ba233ac83 100644
--- a/third_party/blink/public/devtools_protocol/browser_protocol.pdl
+++ b/third_party/blink/public/devtools_protocol/browser_protocol.pdl
@@ -7489,6 +7489,7 @@ domain Page
       newTab
       newWindow
       download
+      nwignore

   experimental type InstallabilityErrorArgument extends object
     properties
