diff --git a/third_party/blink/renderer/modules/gamepad/navigator_gamepad.h b/third_party/blink/renderer/modules/gamepad/navigator_gamepad.h
index d21d0d6487..1f0d3d7bdb 100644
--- a/third_party/blink/renderer/modules/gamepad/navigator_gamepad.h
+++ b/third_party/blink/renderer/modules/gamepad/navigator_gamepad.h
@@ -47,6 +47,9 @@ namespace blink {
 class GamepadDispatcher;
 class GamepadHapticActuator;
 class Navigator;
+class WebLocalFrame;
+
+MODULES_EXPORT void fix_gamepad_nw(WebLocalFrame*);

 class MODULES_EXPORT NavigatorGamepad final
     : public GarbageCollected<NavigatorGamepad>,
@@ -68,7 +71,7 @@ class MODULES_EXPORT NavigatorGamepad final

   void Trace(Visitor*) const override;

- private:
+ public:
   void SampleGamepads();

   void DidRemoveGamepadEventListeners();
