diff --git a/third_party/blink/renderer/bindings/scripts/bind_gen/interface.py b/third_party/blink/renderer/bindings/scripts/bind_gen/interface.py
index 2b18b6a590..ca84f80aba 100644
--- a/third_party/blink/renderer/bindings/scripts/bind_gen/interface.py
+++ b/third_party/blink/renderer/bindings/scripts/bind_gen/interface.py
@@ -2100,7 +2100,12 @@ def make_constructor_function_def(cg_context, function_name):
         EmptyNode(),
     ])

-    if "HTMLConstructor" in cg_context.constructor.extended_attributes:
+    if "Custom" in cg_context.constructor.extended_attributes:
+        body.append(T("// [Custom Constructor]"))
+        text = _format(
+            "${class_name}::ConstructorCustom(${info});")
+        body.append(T(text))
+    elif "HTMLConstructor" in cg_context.constructor.extended_attributes:
         body.append(T("// [HTMLConstructor]"))
         text = _format(
             "V8HTMLConstructor::HtmlConstructor("
@@ -7356,6 +7361,12 @@ def generate_class_like(class_like):
             add_custom_callback_impl_decl(
                 operation_group=operation_group,
                 arg_decls=["const v8::FunctionCallbackInfo<v8::Value>&"])
+    for constructor_group in class_like.constructor_groups:
+        if "Custom" in constructor_group.extended_attributes:
+            add_custom_callback_impl_decl(
+                name="ConstructorCustom",
+                constructor_group=constructor_group,
+                arg_decls=["const v8::FunctionCallbackInfo<v8::Value>&"])
     if interface and interface.indexed_and_named_properties:
         props = interface.indexed_and_named_properties
         operation = props.own_named_getter
