diff --git a/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.cc b/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.cc
index 7fd865ab97..e7dc9f755c 100644
--- a/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.cc
+++ b/third_party/blink/renderer/core/typed_arrays/array_buffer/array_buffer_contents.cc
@@ -39,9 +39,11 @@ namespace blink {

 ArrayBufferContents::ArrayBufferContents(void* data,
                                          size_t length,
+                                         bool nodejs,
                                          DataDeleter deleter) {
   DCHECK(data || length == 0);

+  nodejs_ = nodejs;
   backing_store_ =
       v8::ArrayBuffer::NewBackingStore(data, length, deleter, nullptr);
 }
