diff --git a/third_party/blink/renderer/core/workers/shared_worker.cc b/third_party/blink/renderer/core/workers/shared_worker.cc
index 1e19049d4d..e8ef8b684c 100644
--- a/third_party/blink/renderer/core/workers/shared_worker.cc
+++ b/third_party/blink/renderer/core/workers/shared_worker.cc
@@ -30,6 +30,7 @@
  */

 #include "third_party/blink/renderer/core/workers/shared_worker.h"
+#include "base/command_line.h"

 #include "mojo/public/cpp/bindings/pending_remote.h"
 #include "third_party/abseil-cpp/absl/types/optional.h"
@@ -118,6 +119,8 @@ SharedWorker* SharedWorker::Create(
         script_url, blob_url_token.InitWithNewPipeAndPassReceiver());
   }

+  const base::CommandLine& command_line = *base::CommandLine::ForCurrentProcess();
+  bool isNodeJS = window->GetFrame()->isNodeJS() && command_line.HasSwitch("enable-node-worker");
   auto options = mojom::blink::WorkerOptions::New();
   switch (name_or_options->GetContentType()) {
     case V8UnionStringOrWorkerOptions::ContentType::kString:
@@ -145,7 +148,7 @@ SharedWorker* SharedWorker::Create(

   SharedWorkerClientHolder::From(*window)->Connect(
       worker, std::move(remote_port), script_url, std::move(blob_url_token),
-      std::move(options), context->UkmSourceID());
+      std::move(options), context->UkmSourceID(), isNodeJS);

   return worker;
 }
