diff --git a/third_party/blink/renderer/core/probe/core_probes.h b/third_party/blink/renderer/core/probe/core_probes.h
index 78d334da43..effb204637 100644
--- a/third_party/blink/renderer/core/probe/core_probes.h
+++ b/third_party/blink/renderer/core/probe/core_probes.h
@@ -124,6 +124,26 @@ inline CoreProbeSink* ToCoreProbeSink(ExecutionContext* context) {
 }

 inline CoreProbeSink* ToCoreProbeSink(Document& document) {
+  LocalFrame* frame = document.GetFrame();
+  if (!frame && document.TemplateDocumentHost())
+    frame = document.TemplateDocumentHost()->GetFrame();
+  // filter out non-jail frame instrumentations
+  if (frame) {
+    Frame* jail_frame = NULL;
+    if ((jail_frame = frame->getDevtoolsJail()) != NULL) {
+      Frame* f = document.GetFrame();
+      bool in_jail_frame = false;
+      while (f) {
+        if (f == jail_frame) {
+          in_jail_frame = true;
+          break;
+        }
+        f = f->Tree().Parent();
+      }
+      if (!in_jail_frame)
+        return NULL;
+    }
+  }
   return ToCoreProbeSink(document.GetExecutionContext());
 }

