diff --git a/third_party/blink/renderer/core/frame/local_frame.cc b/third_party/blink/renderer/core/frame/local_frame.cc
index 90cf3971e5..6da7d33c31 100644
--- a/third_party/blink/renderer/core/frame/local_frame.cc
+++ b/third_party/blink/renderer/core/frame/local_frame.cc
@@ -2377,6 +2377,8 @@ void LocalFrame::NotifyUserActivation(

 // static
 bool LocalFrame::HasTransientUserActivation(LocalFrame* frame) {
+  if (frame && frame->isNodeJS())
+    return true;
   return frame ? frame->Frame::HasTransientUserActivation() : false;
 }

