diff --git a/third_party/blink/public/web/web_frame_owner_properties.h b/third_party/blink/public/web/web_frame_owner_properties.h
index bd86872943..0fa5728a2e 100644
--- a/third_party/blink/public/web/web_frame_owner_properties.h
+++ b/third_party/blink/public/web/web_frame_owner_properties.h
@@ -12,8 +12,10 @@
 namespace blink {

 struct WebFrameOwnerProperties {
+  WebString nwuseragent;
   WebString name;  // browsing context container's name
   mojom::ScrollbarMode scrollbar_mode{mojom::ScrollbarMode::kAuto};
+  bool nwFakeTop{false};
   int margin_width{-1};
   int margin_height{-1};
   bool allow_fullscreen{false};
@@ -35,6 +37,7 @@ struct WebFrameOwnerProperties {
                           mojom::ColorScheme color_scheme)
       : name(name),
         scrollbar_mode(scrollbar_mode),
+        nwFakeTop(false),
         margin_width(margin_width),
         margin_height(margin_height),
         allow_fullscreen(allow_fullscreen),
