diff --git a/tools/grit/grit_args.gni b/tools/grit/grit_args.gni
index 18792c9891..0f4b434e0a 100644
--- a/tools/grit/grit_args.gni
+++ b/tools/grit/grit_args.gni
@@ -35,6 +35,7 @@ _grit_defines = [

   # Mac and iOS want Title Case strings.
   "use_titlecase=${is_apple}",
+  "nwjs_sdk=${nwjs_sdk}",
 ]

 # Must match `enable_hidpi` in ui/base/ui_features.gni.
@@ -42,6 +43,11 @@ if (!is_android) {
   _grit_defines += [ "scale_factors=2x" ]
 }

+if (nwjs_sdk) {
+  _grit_defines += [ "nwjs_sdk" ]
+}
+
+
 # Environment variables to be used by grit and its subproceses.

 _grit_environment_vars = [
