diff --git a/tools/v8_context_snapshot/BUILD.gn b/tools/v8_context_snapshot/BUILD.gn
index a06eae4ab3..bc49308ed6 100644
--- a/tools/v8_context_snapshot/BUILD.gn
+++ b/tools/v8_context_snapshot/BUILD.gn
@@ -53,7 +53,7 @@ if (use_v8_context_snapshot) {

   if (v8_snapshot_toolchain == current_toolchain) {
     action("generate_v8_context_snapshot") {
-      script = "//build/gn_run_binary.py"
+      script = "run.py"

       # When cross compiling this target runs in a toolchain other than the
       # default_toolchain. Code depending on this expect the snapshot to be
@@ -85,10 +85,10 @@ if (use_v8_context_snapshot) {
         # that it's possible that some of the shared libraries are updated
         # without having the executable updated.
         inputs = [
-          "$root_out_dir/${shlib_prefix}blink_core${shlib_extension}",
-          "$root_out_dir/${shlib_prefix}blink_modules${shlib_extension}",
-          "$root_out_dir/${shlib_prefix}blink_platform${shlib_extension}",
-          "$root_out_dir/${shlib_prefix}v8${shlib_extension}",
+          "$root_out_dir/${shlib_subdir}/${shlib_prefix}blink_core${shlib_extension}",
+          "$root_out_dir/${shlib_subdir}/${shlib_prefix}blink_modules${shlib_extension}",
+          "$root_out_dir/${shlib_subdir}/${shlib_prefix}blink_platform${shlib_extension}",
+          "$root_out_dir/${shlib_subdir}/${shlib_prefix}v8${shlib_extension}",
         ]

         deps += [
@@ -111,6 +111,21 @@ if (use_v8_context_snapshot) {
         "//v8",
       ]

+      if ((is_win || is_linux) && is_component_build) {
+      } else {
+        deps += [
+          "//v8:v8_initializers",
+        ]
+      }
+
+      if (is_linux && !is_component_build) {
+        configs += [ "//build/config/gcc:rpath_for_built_shared_libraries" ]
+      }
+      if (is_mac && !is_component_build) {
+        # linking with ffmpeg lib
+        ldflags = [ "-rpath", "@executable_path/." ]
+      }
+
       configs += [
         "//v8:external_startup_data",
         ":disable_icf",
diff --git a/tools/v8_context_snapshot/run.py b/tools/v8_context_snapshot/run.py
new file mode 100644
index 0000000000..5febd93d08
