diff --git a/ui/base/resource/resource_bundle.cc b/ui/base/resource/resource_bundle.cc
index 44d0390848..2ddebd16d5 100644
--- a/ui/base/resource/resource_bundle.cc
+++ b/ui/base/resource/resource_bundle.cc
@@ -4,6 +4,8 @@

 #include "ui/base/resource/resource_bundle.h"

+#include "base/strings/string_util.h"
+
 #include <stdint.h>

 #include <limits>
@@ -99,6 +101,8 @@ ResourceBundle::LottieThemedImageParseFunction
 ResourceBundle* g_shared_instance_ = nullptr;

 #if !BUILDFLAG(IS_CHROMEOS_LACROS)
+std::u16string *chromium_name, *nwjs_name;
+
 base::FilePath GetResourcesPakFilePath(const std::string& pak_name) {
   base::FilePath path;
   if (base::PathService::Get(base::DIR_ASSETS, &path))
@@ -922,6 +926,8 @@ void ResourceBundle::InitSharedInstance(Delegate* delegate) {
   DCHECK(g_shared_instance_ == nullptr) << "ResourceBundle initialized twice";
   g_shared_instance_ = new ResourceBundle(delegate);
   std::vector<ResourceScaleFactor> supported_scale_factors;
+  chromium_name = new std::u16string(u"Chromium");
+  nwjs_name     = new std::u16string(u"NW.js");
 #if BUILDFLAG(IS_IOS)
   display::Display display = display::Screen::GetScreen()->GetPrimaryDisplay();
   if (display.device_scale_factor() > 2.0) {
@@ -957,13 +963,13 @@ void ResourceBundle::LoadChromeResources() {
   // 2x images. The 1x data pack only has 1x images, thus passes in an accurate
   // scale factor to gfx::ImageSkia::AddRepresentation.
   if (IsScaleFactorSupported(k100Percent)) {
-    AddDataPackFromPath(GetResourcesPakFilePath("chrome_100_percent.pak"),
+    AddDataPackFromPath(GetResourcesPakFilePath("nw_100_percent.pak"),
                         k100Percent);
   }

   if (IsScaleFactorSupported(k200Percent)) {
     AddOptionalDataPackFromPath(
-        GetResourcesPakFilePath("chrome_200_percent.pak"), k200Percent);
+        GetResourcesPakFilePath("nw_200_percent.pak"), k200Percent);
   }
 }
 #endif  // !BUILDFLAG(IS_CHROMEOS_LACROS)
@@ -1136,8 +1142,10 @@ const ui::ImageModel& ResourceBundle::GetEmptyImageModel() {

 std::u16string ResourceBundle::GetLocalizedStringImpl(int resource_id) const {
   std::u16string string;
-  if (delegate_ && delegate_->GetLocalizedString(resource_id, &string))
+  if (delegate_ && delegate_->GetLocalizedString(resource_id, &string)) {
+    base::ReplaceSubstringsAfterOffset(&string, 0, *chromium_name, *nwjs_name);
     return MaybeMangleLocalizedString(string);
+  }

   // Ensure that ReloadLocaleResources() doesn't drop the resources while
   // we're using them.
@@ -1191,6 +1199,7 @@ std::u16string ResourceBundle::GetLocalizedStringImpl(int resource_id) const {
   } else if (encoding == ResourceHandle::UTF8) {
     msg = base::UTF8ToUTF16(data);
   }
+  base::ReplaceSubstringsAfterOffset(&msg, 0, *chromium_name, *nwjs_name);
   return MaybeMangleLocalizedString(msg);
 }

