diff --git a/ui/base/ui_base_types.h b/ui/base/ui_base_types.h
index de4a1a4990..18353dbac8 100644
--- a/ui/base/ui_base_types.h
+++ b/ui/base/ui_base_types.h
@@ -32,7 +32,8 @@ enum WindowShowState {
   SHOW_STATE_MAXIMIZED = 3,
   SHOW_STATE_INACTIVE = 4,  // Views only, not persisted.
   SHOW_STATE_FULLSCREEN = 5,
-  SHOW_STATE_END = 6  // The end of show state enum.
+  SHOW_STATE_HIDDEN = 6,
+  SHOW_STATE_END = 7  // The end of show state enum.
 };

 // Dialog button identifiers used to specify which buttons to show the user.
