diff --git a/ui/base/x/visual_picker_glx.cc b/ui/base/x/visual_picker_glx.cc
index 0aef2ab88a..f6fddb6c8b 100644
--- a/ui/base/x/visual_picker_glx.cc
+++ b/ui/base/x/visual_picker_glx.cc
@@ -296,6 +296,9 @@ VisualPickerGlx::VisualPickerGlx() : connection_(x11::Connection::Get()) {
   if (configs) {
     system_visual_ = PickBestSystemVisual(*configs.reply);
     rgba_visual_ = PickBestRgbaVisual(*configs.reply);
+  } else {
+    system_visual_ = x11::VisualId{};
+    rgba_visual_   = x11::VisualId{};
   }
 }

