diff --git a/ui/base/clipboard/clipboard.h b/ui/base/clipboard/clipboard.h
index d7e8f89973..f2d36957dc 100644
--- a/ui/base/clipboard/clipboard.h
+++ b/ui/base/clipboard/clipboard.h
@@ -273,7 +273,7 @@ class COMPONENT_EXPORT(UI_BASE_CLIPBOARD) Clipboard
       ClipboardBuffer buffer,
       const DataTransferEndpoint* data_dst) const;

- protected:
+ public:
   // PortableFormat designates the type of data to be stored in the clipboard.
   // This designation is shared across all OSes. The system-specific designation
   // is defined by ClipboardFormatType. A single PortableFormat might be
