diff --git a/ui/display/display.cc b/ui/display/display.cc
index 4a9b5d7a50..c5ec73a861 100644
--- a/ui/display/display.cc
+++ b/ui/display/display.cc
@@ -21,6 +21,16 @@
 #include "ui/gfx/geometry/transform.h"
 #include "ui/gfx/icc_profile.h"

+namespace content {
+
+// Switch to enable / disable code for window's transparency
+DISPLAY_EXPORT bool g_support_transparency = true;
+
+// Switch to force cpu drawing, is used to enable click through on alpha pixels
+extern bool g_force_cpu_draw;
+bool g_force_cpu_draw = false;
+}
+
 namespace display {
 namespace {

