diff --git a/ui/display/display.h b/ui/display/display.h
index ba01a3c70b..c45e5ac3d2 100644
--- a/ui/display/display.h
+++ b/ui/display/display.h
@@ -14,6 +14,11 @@
 #include "ui/gfx/display_color_spaces.h"
 #include "ui/gfx/geometry/rect.h"

+namespace content {
+  DISPLAY_EXPORT extern bool g_support_transparency;
+  DISPLAY_EXPORT extern bool g_force_cpu_draw;
+}
+
 namespace display {

 namespace mojom {
