diff --git a/ui/gfx/mac/nswindow_frame_controls.h b/ui/gfx/mac/nswindow_frame_controls.h
index d58eecae3b..f6bae3aae5 100644
--- a/ui/gfx/mac/nswindow_frame_controls.h
+++ b/ui/gfx/mac/nswindow_frame_controls.h
@@ -29,6 +29,7 @@ GFX_EXPORT void ApplyNSWindowSizeConstraints(NSWindow* window,
                                              const gfx::Size& max_size,
                                              bool can_resize,
                                              bool can_fullscreen);
+GFX_EXPORT void SetNSWindowShowInTaskbar(NSWindow* window, bool show);

 }  // namespace gfx

