diff --git a/ui/gfx/icon_util.h b/ui/gfx/icon_util.h
index 4496cb963e..cfd9aca09e 100644
--- a/ui/gfx/icon_util.h
+++ b/ui/gfx/icon_util.h
@@ -87,6 +87,10 @@ class GFX_EXPORT IconUtil {
   // icon and returns the corresponding HICON handle if conversion succeeds.
   static base::win::ScopedHICON CreateHICONFromSkBitmap(const SkBitmap& bitmap);

+  static base::win::ScopedHICON CreateHICONFromSkBitmapSizedTo(const SkBitmap& bitmap,
+    int width,
+    int height);
+
   // Given a valid HICON handle representing an icon, this function converts
   // the icon into an SkBitmap object containing an ARGB bitmap using the
   // dimensions specified in |s|. |s| must specify valid dimensions (both
