diff --git a/ui/gl/gl_features.cc b/ui/gl/gl_features.cc
index 7c5c84f5fb..d138a0d62b 100644
--- a/ui/gl/gl_features.cc
+++ b/ui/gl/gl_features.cc
@@ -83,9 +83,7 @@ const base::Feature kAndroidFrameDeadline{"AndroidFrameDeadline",
 // Launched on Windows, still experimental on other platforms.
 const base::Feature kDefaultPassthroughCommandDecoder {
   "DefaultPassthroughCommandDecoder",
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_FUCHSIA) ||     \
-    (BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_CASTOS)) || \
-    BUILDFLAG(IS_CHROMEOS_LACROS) || BUILDFLAG(IS_MAC)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC)
       base::FEATURE_ENABLED_BY_DEFAULT
 #else
       base::FEATURE_DISABLED_BY_DEFAULT
