diff --git a/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc b/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
index 269c7a6fc8..acf90e2a5e 100644
--- a/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
+++ b/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
@@ -850,6 +850,14 @@ void DesktopNativeWidgetAura::SetBounds(const gfx::Rect& bounds) {
   desktop_window_tree_host_->SetBoundsInDIP(bounds);
 }

+#if defined(OS_WIN)
+void DesktopNativeWidgetAura::SetPosition(const gfx::Point& pos) {
+  if (!content_window_)
+    return;
+  desktop_window_tree_host_->SetPositionInDIP(pos);
+}
+#endif
+
 void DesktopNativeWidgetAura::SetBoundsConstrained(const gfx::Rect& bounds) {
   if (!content_window_)
     return;
