diff --git a/ui/views/win/hwnd_message_handler_delegate.h b/ui/views/win/hwnd_message_handler_delegate.h
index d8e0f1d313..d56bf79387 100644
--- a/ui/views/win/hwnd_message_handler_delegate.h
+++ b/ui/views/win/hwnd_message_handler_delegate.h
@@ -113,6 +113,10 @@ class VIEWS_EXPORT HWNDMessageHandlerDelegate {

   virtual gfx::NativeViewAccessible GetNativeViewAccessible() = 0;

+  // on windows, maximizing sometime is sent through WM_SIZE, not
+  // WM_SYSCOMMAND, see node-webkit#753
+  virtual bool ShouldHandleOnSize() const = 0;
+
   // TODO(beng): Investigate migrating these methods to On* prefixes once
   // HWNDMessageHandler is the WindowImpl.

@@ -137,6 +141,8 @@ class VIEWS_EXPORT HWNDMessageHandlerDelegate {
   // true if the command was handled.
   virtual bool HandleCommand(int command) = 0;

+  virtual bool HandleSize(UINT param, const gfx::Size& size) = 0;
+
   // Called when an accelerator is invoked.
   virtual void HandleAccelerator(const ui::Accelerator& accelerator) = 0;

