diff --git a/ui/views/accessibility/view_ax_platform_node_delegate_auralinux.cc b/ui/views/accessibility/view_ax_platform_node_delegate_auralinux.cc
index c97c118478..53c063f4e9 100644
--- a/ui/views/accessibility/view_ax_platform_node_delegate_auralinux.cc
+++ b/ui/views/accessibility/view_ax_platform_node_delegate_auralinux.cc
@@ -4,6 +4,8 @@

 #include "ui/views/accessibility/view_ax_platform_node_delegate_auralinux.h"

+#include "content/public/browser/browser_thread.h"
+
 #include <algorithm>
 #include <memory>
 #include <vector>
